<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 6 Biogeography and Macroevolution in R | BIO595 The Tree of Life</title>
  <meta name="description" content="Test writing the book for the course bookdown::gitbook." />
  <meta name="generator" content="bookdown 0.28 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 6 Biogeography and Macroevolution in R | BIO595 The Tree of Life" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="Test writing the book for the course bookdown::gitbook." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 6 Biogeography and Macroevolution in R | BIO595 The Tree of Life" />
  
  <meta name="twitter:description" content="Test writing the book for the course bookdown::gitbook." />
  

<meta name="author" content="Qixin He, Daniel Park" />


<meta name="date" content="2023-04-10" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="tree-and-traits-in-r.html"/>
<link rel="next" href="phylogenetic-diversity-in-r.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Tree of Life: Phylogenetic inferences and applications</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Overview</a></li>
<li class="chapter" data-level="2" data-path="unix.html"><a href="unix.html"><i class="fa fa-check"></i><b>2</b> Unix</a>
<ul>
<li class="chapter" data-level="2.1" data-path="unix.html"><a href="unix.html#what-is-unix-and-why-do-we-use-it"><i class="fa fa-check"></i><b>2.1</b> What is Unix and why do we use it?</a></li>
<li class="chapter" data-level="2.2" data-path="unix.html"><a href="unix.html#installation"><i class="fa fa-check"></i><b>2.2</b> Installation</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="unix.html"><a href="unix.html#mac-os-and-linux"><i class="fa fa-check"></i><b>2.2.1</b> Mac OS and Linux</a></li>
<li class="chapter" data-level="2.2.2" data-path="unix.html"><a href="unix.html#windows"><i class="fa fa-check"></i><b>2.2.2</b> Windows</a></li>
<li class="chapter" data-level="2.2.3" data-path="unix.html"><a href="unix.html#use-scholar-remote-desktop-during-the-class"><i class="fa fa-check"></i><b>2.2.3</b> Use Scholar Remote Desktop During the Class</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="unix.html"><a href="unix.html#operation"><i class="fa fa-check"></i><b>2.3</b> Operation</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="unix.html"><a href="unix.html#starting-the-shell"><i class="fa fa-check"></i><b>2.3.1</b> Starting the Shell</a></li>
<li class="chapter" data-level="2.3.2" data-path="unix.html"><a href="unix.html#directoryfolder-sturcture"><i class="fa fa-check"></i><b>2.3.2</b> Directory/folder sturcture</a></li>
<li class="chapter" data-level="2.3.3" data-path="unix.html"><a href="unix.html#absolute-and-relative-directory"><i class="fa fa-check"></i><b>2.3.3</b> Absolute and Relative directory</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="unix.html"><a href="unix.html#handling-directories-and-files"><i class="fa fa-check"></i><b>2.4</b> Handling directories and files</a></li>
<li class="chapter" data-level="2.5" data-path="unix.html"><a href="unix.html#excercises-i"><i class="fa fa-check"></i><b>2.5</b> Excercises I</a></li>
<li class="chapter" data-level="2.6" data-path="unix.html"><a href="unix.html#viewing-and-processing-text-files"><i class="fa fa-check"></i><b>2.6</b> Viewing and processing text files</a></li>
<li class="chapter" data-level="2.7" data-path="unix.html"><a href="unix.html#redirecting-printouts-and-piping"><i class="fa fa-check"></i><b>2.7</b> Redirecting printouts and piping</a></li>
<li class="chapter" data-level="2.8" data-path="unix.html"><a href="unix.html#exercises-ii"><i class="fa fa-check"></i><b>2.8</b> Exercises II</a></li>
<li class="chapter" data-level="2.9" data-path="unix.html"><a href="unix.html#useful-wildcards"><i class="fa fa-check"></i><b>2.9</b> Useful wildcards</a></li>
<li class="chapter" data-level="2.10" data-path="unix.html"><a href="unix.html#exercises-iii"><i class="fa fa-check"></i><b>2.10</b> Exercises III</a></li>
<li class="chapter" data-level="2.11" data-path="unix.html"><a href="unix.html#search-and-select-lines-using-grep"><i class="fa fa-check"></i><b>2.11</b> Search and select lines using <em>grep</em></a></li>
<li class="chapter" data-level="2.12" data-path="unix.html"><a href="unix.html#substituting-characters-using-tr"><i class="fa fa-check"></i><b>2.12</b> Substituting characters using <em>tr</em></a></li>
<li class="chapter" data-level="2.13" data-path="unix.html"><a href="unix.html#cut-sort-and-unique"><i class="fa fa-check"></i><b>2.13</b> cut, sort and unique</a></li>
<li class="chapter" data-level="2.14" data-path="unix.html"><a href="unix.html#exercises-iv"><i class="fa fa-check"></i><b>2.14</b> Exercises IV</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="r-basics-i.html"><a href="r-basics-i.html"><i class="fa fa-check"></i><b>3</b> R Basics I</a>
<ul>
<li class="chapter" data-level="3.1" data-path="r-basics-i.html"><a href="r-basics-i.html#what-is-programming"><i class="fa fa-check"></i><b>3.1</b> What is programming?</a></li>
<li class="chapter" data-level="3.2" data-path="r-basics-i.html"><a href="r-basics-i.html#r"><i class="fa fa-check"></i><b>3.2</b> <code>R</code></a></li>
<li class="chapter" data-level="3.3" data-path="r-basics-i.html"><a href="r-basics-i.html#installation-1"><i class="fa fa-check"></i><b>3.3</b> Installation</a></li>
<li class="chapter" data-level="3.4" data-path="r-basics-i.html"><a href="r-basics-i.html#variables"><i class="fa fa-check"></i><b>3.4</b> Variables</a>
<ul>
<li class="chapter" data-level="3.4.1" data-path="r-basics-i.html"><a href="r-basics-i.html#exercise-1"><i class="fa fa-check"></i><b>3.4.1</b> Exercise (1):</a></li>
<li class="chapter" data-level="3.4.2" data-path="r-basics-i.html"><a href="r-basics-i.html#variable-naming-conventions"><i class="fa fa-check"></i><b>3.4.2</b> Variable naming conventions</a></li>
<li class="chapter" data-level="3.4.3" data-path="r-basics-i.html"><a href="r-basics-i.html#arithmetic-operators"><i class="fa fa-check"></i><b>3.4.3</b> Arithmetic Operators</a></li>
<li class="chapter" data-level="3.4.4" data-path="r-basics-i.html"><a href="r-basics-i.html#logical-operators"><i class="fa fa-check"></i><b>3.4.4</b> Logical Operators</a></li>
<li class="chapter" data-level="3.4.5" data-path="r-basics-i.html"><a href="r-basics-i.html#basic-built-in-mathematical-functions"><i class="fa fa-check"></i><b>3.4.5</b> Basic built-in mathematical functions</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="r-basics-i.html"><a href="r-basics-i.html#subsetting-data"><i class="fa fa-check"></i><b>3.5</b> Subsetting data</a></li>
<li class="chapter" data-level="3.6" data-path="r-basics-i.html"><a href="r-basics-i.html#logical-values"><i class="fa fa-check"></i><b>3.6</b> Logical values</a>
<ul>
<li class="chapter" data-level="3.6.1" data-path="r-basics-i.html"><a href="r-basics-i.html#missing-dealing-with-na-values"><i class="fa fa-check"></i><b>3.6.1</b> Missing: dealing with <code>NA</code> values</a></li>
</ul></li>
<li class="chapter" data-level="3.7" data-path="r-basics-i.html"><a href="r-basics-i.html#generating-data"><i class="fa fa-check"></i><b>3.7</b> Generating Data</a></li>
<li class="chapter" data-level="3.8" data-path="r-basics-i.html"><a href="r-basics-i.html#data-structures-and-types"><i class="fa fa-check"></i><b>3.8</b> Data Structures and Types</a>
<ul>
<li class="chapter" data-level="3.8.1" data-path="r-basics-i.html"><a href="r-basics-i.html#vectors"><i class="fa fa-check"></i><b>3.8.1</b> Vectors</a></li>
</ul></li>
<li class="chapter" data-level="3.9" data-path="r-basics-i.html"><a href="r-basics-i.html#matrices"><i class="fa fa-check"></i><b>3.9</b> Matrices</a>
<ul>
<li class="chapter" data-level="3.9.1" data-path="r-basics-i.html"><a href="r-basics-i.html#exercise-2"><i class="fa fa-check"></i><b>3.9.1</b> Exercise (2):</a></li>
</ul></li>
<li class="chapter" data-level="3.10" data-path="r-basics-i.html"><a href="r-basics-i.html#data-frames"><i class="fa fa-check"></i><b>3.10</b> Data Frames</a>
<ul>
<li class="chapter" data-level="3.10.1" data-path="r-basics-i.html"><a href="r-basics-i.html#exercise-3"><i class="fa fa-check"></i><b>3.10.1</b> Exercise (3):</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="r-basics-ii.html"><a href="r-basics-ii.html"><i class="fa fa-check"></i><b>4</b> R Basics II</a>
<ul>
<li class="chapter" data-level="4.1" data-path="r-basics-ii.html"><a href="r-basics-ii.html#setting-up-the-environment"><i class="fa fa-check"></i><b>4.1</b> Setting up the environment</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="r-basics-ii.html"><a href="r-basics-ii.html#exercise-1-1"><i class="fa fa-check"></i><b>4.1.1</b> Exercise (1)</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="r-basics-ii.html"><a href="r-basics-ii.html#reading-in-data-from-files"><i class="fa fa-check"></i><b>4.2</b> Reading in data from files</a></li>
<li class="chapter" data-level="4.3" data-path="r-basics-ii.html"><a href="r-basics-ii.html#wrangling-data"><i class="fa fa-check"></i><b>4.3</b> Wrangling Data</a></li>
<li class="chapter" data-level="4.4" data-path="r-basics-ii.html"><a href="r-basics-ii.html#subsetting-by-columns-and-rows"><i class="fa fa-check"></i><b>4.4</b> Subsetting by columns and rows</a></li>
<li class="chapter" data-level="4.5" data-path="r-basics-ii.html"><a href="r-basics-ii.html#subsetting-with-conditions"><i class="fa fa-check"></i><b>4.5</b> Subsetting with conditions</a>
<ul>
<li class="chapter" data-level="4.5.1" data-path="r-basics-ii.html"><a href="r-basics-ii.html#exercise-2-1"><i class="fa fa-check"></i><b>4.5.1</b> Exercise (2)</a></li>
<li class="chapter" data-level="4.5.2" data-path="r-basics-ii.html"><a href="r-basics-ii.html#bonus-1"><i class="fa fa-check"></i><b>4.5.2</b> Bonus [1]</a></li>
</ul></li>
<li class="chapter" data-level="4.6" data-path="r-basics-ii.html"><a href="r-basics-ii.html#merging-data"><i class="fa fa-check"></i><b>4.6</b> Merging data</a></li>
<li class="chapter" data-level="4.7" data-path="r-basics-ii.html"><a href="r-basics-ii.html#saving-files-and-sessions"><i class="fa fa-check"></i><b>4.7</b> Saving files and sessions</a>
<ul>
<li class="chapter" data-level="4.7.1" data-path="r-basics-ii.html"><a href="r-basics-ii.html#exercise-3-1"><i class="fa fa-check"></i><b>4.7.1</b> Exercise (3)</a></li>
</ul></li>
<li class="chapter" data-level="4.8" data-path="r-basics-ii.html"><a href="r-basics-ii.html#the-for-loop"><i class="fa fa-check"></i><b>4.8</b> The <span class="smallcaps">for</span>-loop</a></li>
<li class="chapter" data-level="4.9" data-path="r-basics-ii.html"><a href="r-basics-ii.html#lists"><i class="fa fa-check"></i><b>4.9</b> Lists</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="tree-and-traits-in-r.html"><a href="tree-and-traits-in-r.html"><i class="fa fa-check"></i><b>5</b> Tree and Traits in R</a>
<ul>
<li class="chapter" data-level="5.1" data-path="tree-and-traits-in-r.html"><a href="tree-and-traits-in-r.html#setting-up-the-environment-1"><i class="fa fa-check"></i><b>5.1</b> Setting up the environment</a></li>
<li class="chapter" data-level="5.2" data-path="tree-and-traits-in-r.html"><a href="tree-and-traits-in-r.html#examining-phylogenetic-trees"><i class="fa fa-check"></i><b>5.2</b> Examining phylogenetic trees</a></li>
<li class="chapter" data-level="5.3" data-path="tree-and-traits-in-r.html"><a href="tree-and-traits-in-r.html#plotting-phylogenetic-trees"><i class="fa fa-check"></i><b>5.3</b> Plotting phylogenetic trees</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="tree-and-traits-in-r.html"><a href="tree-and-traits-in-r.html#exercise-1-2"><i class="fa fa-check"></i><b>5.3.1</b> Exercise (1)</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="tree-and-traits-in-r.html"><a href="tree-and-traits-in-r.html#associating-phylogenies-and-traits"><i class="fa fa-check"></i><b>5.4</b> Associating phylogenies and traits</a>
<ul>
<li class="chapter" data-level="5.4.1" data-path="tree-and-traits-in-r.html"><a href="tree-and-traits-in-r.html#saving-plots"><i class="fa fa-check"></i><b>5.4.1</b> Saving plots</a></li>
<li class="chapter" data-level="5.4.2" data-path="tree-and-traits-in-r.html"><a href="tree-and-traits-in-r.html#exercise-2-2"><i class="fa fa-check"></i><b>5.4.2</b> Exercise (2)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html"><i class="fa fa-check"></i><b>6</b> Biogeography and Macroevolution in R</a>
<ul>
<li class="chapter" data-level="6.1" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#setting-up-the-environment-2"><i class="fa fa-check"></i><b>6.1</b> Setting up the environment</a></li>
<li class="chapter" data-level="6.2" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#ancestral-state-reconstructions"><i class="fa fa-check"></i><b>6.2</b> Ancestral State Reconstructions</a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#bonus-2"><i class="fa fa-check"></i><b>6.2.1</b> Bonus [2]</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#continuous-traits"><i class="fa fa-check"></i><b>6.3</b> Continuous traits</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#maximum-likelihood"><i class="fa fa-check"></i><b>6.3.1</b> Maximum likelihood</a></li>
<li class="chapter" data-level="6.3.2" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#phylogenetically-independent-contrasts"><i class="fa fa-check"></i><b>6.3.2</b> Phylogenetically independent contrasts</a></li>
<li class="chapter" data-level="6.3.3" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#plotting-ancestral-states"><i class="fa fa-check"></i><b>6.3.3</b> Plotting ancestral states</a></li>
<li class="chapter" data-level="6.3.4" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#exercise-1-3"><i class="fa fa-check"></i><b>6.3.4</b> Exercise (1)</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#discrete-traits"><i class="fa fa-check"></i><b>6.4</b> Discrete traits</a>
<ul>
<li class="chapter" data-level="6.4.1" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#exercise-2-3"><i class="fa fa-check"></i><b>6.4.1</b> Exercise (2)</a></li>
<li class="chapter" data-level="6.4.2" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#maximum-likelihood-1"><i class="fa fa-check"></i><b>6.4.2</b> Maximum likelihood</a></li>
<li class="chapter" data-level="6.4.3" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#choosing-models"><i class="fa fa-check"></i><b>6.4.3</b> Choosing models</a></li>
</ul></li>
<li class="chapter" data-level="6.5" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#saving-multiple-plots-together"><i class="fa fa-check"></i><b>6.5</b> Saving multiple plots together</a>
<ul>
<li class="chapter" data-level="6.5.1" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#exercise-3-2"><i class="fa fa-check"></i><b>6.5.1</b> Exercise (3)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="phylogenetic-diversity-in-r.html"><a href="phylogenetic-diversity-in-r.html"><i class="fa fa-check"></i><b>7</b> Phylogenetic Diversity in R</a>
<ul>
<li class="chapter" data-level="7.1" data-path="phylogenetic-diversity-in-r.html"><a href="phylogenetic-diversity-in-r.html#setting-up-the-environment-3"><i class="fa fa-check"></i><b>7.1</b> Setting up the environment</a></li>
<li class="chapter" data-level="7.2" data-path="phylogenetic-diversity-in-r.html"><a href="phylogenetic-diversity-in-r.html#phylogenetic-diversity"><i class="fa fa-check"></i><b>7.2</b> Phylogenetic Diversity</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="phylogenetic-diversity-in-r.html"><a href="phylogenetic-diversity-in-r.html#preprocessing"><i class="fa fa-check"></i><b>7.2.1</b> Preprocessing</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="phylogenetic-diversity-in-r.html"><a href="phylogenetic-diversity-in-r.html#estimating-phylogenetic-diversity-metrics"><i class="fa fa-check"></i><b>7.3</b> Estimating phylogenetic diversity metrics</a>
<ul>
<li class="chapter" data-level="7.3.1" data-path="phylogenetic-diversity-in-r.html"><a href="phylogenetic-diversity-in-r.html#exercise-1-4"><i class="fa fa-check"></i><b>7.3.1</b> Exercise (1)</a></li>
<li class="chapter" data-level="7.3.2" data-path="phylogenetic-diversity-in-r.html"><a href="phylogenetic-diversity-in-r.html#exercise-2-4"><i class="fa fa-check"></i><b>7.3.2</b> Exercise (2)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="community-phylogenetics.html"><a href="community-phylogenetics.html"><i class="fa fa-check"></i><b>8</b> Community Phylogenetics</a>
<ul>
<li class="chapter" data-level="8.1" data-path="community-phylogenetics.html"><a href="community-phylogenetics.html#setting-up-the-environment-4"><i class="fa fa-check"></i><b>8.1</b> Setting up the environment</a></li>
<li class="chapter" data-level="8.2" data-path="community-phylogenetics.html"><a href="community-phylogenetics.html#definition-of-community-phylogenetics"><i class="fa fa-check"></i><b>8.2</b> Definition of Community Phylogenetics</a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="community-phylogenetics.html"><a href="community-phylogenetics.html#alpha-diversity"><i class="fa fa-check"></i><b>8.2.1</b> Alpha diversity</a></li>
<li class="chapter" data-level="8.2.2" data-path="community-phylogenetics.html"><a href="community-phylogenetics.html#beta-diversity"><i class="fa fa-check"></i><b>8.2.2</b> Beta diversity</a></li>
<li class="chapter" data-level="8.2.3" data-path="community-phylogenetics.html"><a href="community-phylogenetics.html#exercise-1-5"><i class="fa fa-check"></i><b>8.2.3</b> Exercise (1)</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="community-phylogenetics.html"><a href="community-phylogenetics.html#standardized-effect-sizes"><i class="fa fa-check"></i><b>8.3</b> Standardized effect sizes</a>
<ul>
<li class="chapter" data-level="8.3.1" data-path="community-phylogenetics.html"><a href="community-phylogenetics.html#exercise-2-5"><i class="fa fa-check"></i><b>8.3.1</b> Exercise (2)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="coevolution-and-phylogenetic-signals-in-r.html"><a href="coevolution-and-phylogenetic-signals-in-r.html"><i class="fa fa-check"></i><b>9</b> Coevolution and Phylogenetic Signals in R</a>
<ul>
<li class="chapter" data-level="9.1" data-path="coevolution-and-phylogenetic-signals-in-r.html"><a href="coevolution-and-phylogenetic-signals-in-r.html#setting-up-the-environment-5"><i class="fa fa-check"></i><b>9.1</b> Setting up the environment</a></li>
<li class="chapter" data-level="9.2" data-path="coevolution-and-phylogenetic-signals-in-r.html"><a href="coevolution-and-phylogenetic-signals-in-r.html#phylogenetic-signal"><i class="fa fa-check"></i><b>9.2</b> Phylogenetic Signal</a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="coevolution-and-phylogenetic-signals-in-r.html"><a href="coevolution-and-phylogenetic-signals-in-r.html#blombergs-k"><i class="fa fa-check"></i><b>9.2.1</b> Blomberg’s K</a></li>
<li class="chapter" data-level="9.2.2" data-path="coevolution-and-phylogenetic-signals-in-r.html"><a href="coevolution-and-phylogenetic-signals-in-r.html#pagels-lambda-λ"><i class="fa fa-check"></i><b>9.2.2</b> Pagel’s lambda (λ)</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="coevolution-and-phylogenetic-signals-in-r.html"><a href="coevolution-and-phylogenetic-signals-in-r.html#phylogenetically-independent-contrasts-1"><i class="fa fa-check"></i><b>9.3</b> Phylogenetically Independent Contrasts</a></li>
<li class="chapter" data-level="9.4" data-path="coevolution-and-phylogenetic-signals-in-r.html"><a href="coevolution-and-phylogenetic-signals-in-r.html#coevolution"><i class="fa fa-check"></i><b>9.4</b> Coevolution</a>
<ul>
<li class="chapter" data-level="9.4.1" data-path="coevolution-and-phylogenetic-signals-in-r.html"><a href="coevolution-and-phylogenetic-signals-in-r.html#plotting-tanglegrams"><i class="fa fa-check"></i><b>9.4.1</b> Plotting tanglegrams</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="phylogenetic-visualization-in-r.html"><a href="phylogenetic-visualization-in-r.html"><i class="fa fa-check"></i><b>10</b> Phylogenetic Visualization in R</a>
<ul>
<li class="chapter" data-level="10.1" data-path="phylogenetic-visualization-in-r.html"><a href="phylogenetic-visualization-in-r.html#setting-up-the-environment-6"><i class="fa fa-check"></i><b>10.1</b> Setting up the environment</a></li>
<li class="chapter" data-level="10.2" data-path="phylogenetic-visualization-in-r.html"><a href="phylogenetic-visualization-in-r.html#using-phytools"><i class="fa fa-check"></i><b>10.2</b> Using <code>phytools</code></a>
<ul>
<li class="chapter" data-level="10.2.1" data-path="phylogenetic-visualization-in-r.html"><a href="phylogenetic-visualization-in-r.html#phenograms"><i class="fa fa-check"></i><b>10.2.1</b> Phenograms</a></li>
<li class="chapter" data-level="10.2.2" data-path="phylogenetic-visualization-in-r.html"><a href="phylogenetic-visualization-in-r.html#mapping-continuous-characters-on-branches"><i class="fa fa-check"></i><b>10.2.2</b> Mapping continuous characters on branches</a></li>
<li class="chapter" data-level="10.2.3" data-path="phylogenetic-visualization-in-r.html"><a href="phylogenetic-visualization-in-r.html#plotting-images-on-phylogenies"><i class="fa fa-check"></i><b>10.2.3</b> Plotting images on phylogenies</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="phylogenetic-visualization-in-r.html"><a href="phylogenetic-visualization-in-r.html#using-ggtree"><i class="fa fa-check"></i><b>10.3</b> Using <code>ggtree</code></a>
<ul>
<li class="chapter" data-level="10.3.1" data-path="phylogenetic-visualization-in-r.html"><a href="phylogenetic-visualization-in-r.html#basic-plotting"><i class="fa fa-check"></i><b>10.3.1</b> Basic plotting</a></li>
<li class="chapter" data-level="10.3.2" data-path="phylogenetic-visualization-in-r.html"><a href="phylogenetic-visualization-in-r.html#labeling-tips-nodes-and-clades"><i class="fa fa-check"></i><b>10.3.2</b> Labeling tips, nodes, and clades</a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="phylogenetic-visualization-in-r.html"><a href="phylogenetic-visualization-in-r.html#plotting-maps"><i class="fa fa-check"></i><b>10.4</b> Plotting maps</a>
<ul>
<li class="chapter" data-level="10.4.1" data-path="phylogenetic-visualization-in-r.html"><a href="phylogenetic-visualization-in-r.html#choropleth-maps"><i class="fa fa-check"></i><b>10.4.1</b> Choropleth maps</a></li>
<li class="chapter" data-level="10.4.2" data-path="phylogenetic-visualization-in-r.html"><a href="phylogenetic-visualization-in-r.html#working-with-rasters"><i class="fa fa-check"></i><b>10.4.2</b> Working with rasters</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html"><i class="fa fa-check"></i><b>11</b> Gene tree vs Species tree</a>
<ul>
<li class="chapter" data-level="11.1" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#learning-goals"><i class="fa fa-check"></i><b>11.1</b> Learning goals:</a></li>
<li class="chapter" data-level="11.2" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#tracing-ancestry-within-one-population"><i class="fa fa-check"></i><b>11.2</b> Tracing ancestry within one population</a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#vary-pop-size-and-generation"><i class="fa fa-check"></i><b>11.2.1</b> Vary pop size and generation</a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#speciation-and-coalescence"><i class="fa fa-check"></i><b>11.3</b> Speciation and coalescence</a>
<ul>
<li class="chapter" data-level="11.3.1" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#two-species-coalescent-history"><i class="fa fa-check"></i><b>11.3.1</b> Two-species coalescent history</a></li>
</ul></li>
<li class="chapter" data-level="11.4" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#exercise-i"><i class="fa fa-check"></i><b>11.4</b> Exercise I</a></li>
<li class="chapter" data-level="11.5" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#gene-tree-locus-tree-and-species-tree"><i class="fa fa-check"></i><b>11.5</b> Gene tree, Locus tree and Species tree</a>
<ul>
<li class="chapter" data-level="11.5.1" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#loading-data-and-packages"><i class="fa fa-check"></i><b>11.5.1</b> loading data and packages</a></li>
</ul></li>
<li class="chapter" data-level="11.6" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#effects-of-incomplete-lineage-sorting-ils-alone"><i class="fa fa-check"></i><b>11.6</b> Effects of Incomplete Lineage Sorting (ILS) alone</a>
<ul>
<li class="chapter" data-level="11.6.1" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#generate-species-tree-with-no-extinction"><i class="fa fa-check"></i><b>11.6.1</b> generate species tree with no extinction</a></li>
<li class="chapter" data-level="11.6.2" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#simulate-gene-trees-using-the-multispecies-coalescent-process"><i class="fa fa-check"></i><b>11.6.2</b> simulate gene trees using the multispecies coalescent process</a></li>
</ul></li>
<li class="chapter" data-level="11.7" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#calculate-tree-distances"><i class="fa fa-check"></i><b>11.7</b> Calculate tree distances</a>
<ul>
<li class="chapter" data-level="11.7.1" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#rf-distances"><i class="fa fa-check"></i><b>11.7.1</b> RF distances</a></li>
<li class="chapter" data-level="11.7.2" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#inspect-pairwise-rf-distances-among-all-the-gene-trees-generated-under-one-guiding-species-tree"><i class="fa fa-check"></i><b>11.7.2</b> Inspect pairwise RF distances among all the gene trees generated under one guiding species tree</a></li>
</ul></li>
<li class="chapter" data-level="11.8" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#inspect-node-support-i.e.-subtrees-defined-by-a-split"><i class="fa fa-check"></i><b>11.8</b> Inspect node support (i.e., subtrees defined by a split)</a></li>
<li class="chapter" data-level="11.9" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#combined-effects-of-gene-duplication-loss-and-horizontal-gene-transfer-events"><i class="fa fa-check"></i><b>11.9</b> Combined effects of gene duplication, loss and horizontal gene transfer events</a></li>
<li class="chapter" data-level="11.10" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#exercise-ii"><i class="fa fa-check"></i><b>11.10</b> Exercise II</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="overview-of-phylogenetic-inferences.html"><a href="overview-of-phylogenetic-inferences.html"><i class="fa fa-check"></i><b>12</b> Overview of Phylogenetic Inferences</a>
<ul>
<li class="chapter" data-level="12.1" data-path="overview-of-phylogenetic-inferences.html"><a href="overview-of-phylogenetic-inferences.html#mega"><i class="fa fa-check"></i><b>12.1</b> <code>MEGA</code></a></li>
<li class="chapter" data-level="12.2" data-path="overview-of-phylogenetic-inferences.html"><a href="overview-of-phylogenetic-inferences.html#view-and-align-sequences"><i class="fa fa-check"></i><b>12.2</b> View and Align Sequences</a></li>
<li class="chapter" data-level="12.3" data-path="overview-of-phylogenetic-inferences.html"><a href="overview-of-phylogenetic-inferences.html#dnaprotein-evolution-model"><i class="fa fa-check"></i><b>12.3</b> DNA/protein evolution model</a></li>
<li class="chapter" data-level="12.4" data-path="overview-of-phylogenetic-inferences.html"><a href="overview-of-phylogenetic-inferences.html#phylogenetic-inference"><i class="fa fa-check"></i><b>12.4</b> Phylogenetic inference</a></li>
<li class="chapter" data-level="12.5" data-path="overview-of-phylogenetic-inferences.html"><a href="overview-of-phylogenetic-inferences.html#estimate-divergence-times"><i class="fa fa-check"></i><b>12.5</b> Estimate divergence times</a></li>
<li class="chapter" data-level="12.6" data-path="overview-of-phylogenetic-inferences.html"><a href="overview-of-phylogenetic-inferences.html#exercises"><i class="fa fa-check"></i><b>12.6</b> Exercises</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html"><i class="fa fa-check"></i><b>13</b> Resources and Guidelines for the Term Project</a>
<ul>
<li class="chapter" data-level="13.1" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#bioinformatics-databases"><i class="fa fa-check"></i><b>13.1</b> Bioinformatics Databases</a>
<ul>
<li class="chapter" data-level="13.1.1" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#ncbi-national-center-for-biotechnology-information"><i class="fa fa-check"></i><b>13.1.1</b> NCBI (National Center for Biotechnology Information)</a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#retrieving-genetic-resources-for-your-taxa"><i class="fa fa-check"></i><b>13.2</b> Retrieving genetic resources for your taxa</a>
<ul>
<li class="chapter" data-level="13.2.1" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#ncbi-taxonomy-database"><i class="fa fa-check"></i><b>13.2.1</b> NCBI taxonomy database</a></li>
<li class="chapter" data-level="13.2.2" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#individual-genes-from-the-whole-genome-data"><i class="fa fa-check"></i><b>13.2.2</b> Individual genes from the whole genome data</a></li>
<li class="chapter" data-level="13.2.3" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#get-homologous-sequences-from-related-species"><i class="fa fa-check"></i><b>13.2.3</b> Get homologous sequences from related species</a></li>
<li class="chapter" data-level="13.2.4" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#european-bioinformatics-institute-ebi"><i class="fa fa-check"></i><b>13.2.4</b> European Bioinformatics Institute (EBI)</a></li>
<li class="chapter" data-level="13.2.5" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#protein-data-bank"><i class="fa fa-check"></i><b>13.2.5</b> Protein Data Bank</a></li>
</ul></li>
<li class="chapter" data-level="13.3" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#biogeographic-databases"><i class="fa fa-check"></i><b>13.3</b> Biogeographic Databases</a>
<ul>
<li class="chapter" data-level="13.3.1" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#climate-data"><i class="fa fa-check"></i><b>13.3.1</b> Climate Data</a></li>
<li class="chapter" data-level="13.3.2" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#occurrence-data"><i class="fa fa-check"></i><b>13.3.2</b> Occurrence Data</a></li>
<li class="chapter" data-level="13.3.3" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#topographic-data"><i class="fa fa-check"></i><b>13.3.3</b> TOPOGRAPHIC DATA</a></li>
<li class="chapter" data-level="13.3.4" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#trait-data"><i class="fa fa-check"></i><b>13.3.4</b> TRAIT DATA</a></li>
</ul></li>
<li class="chapter" data-level="13.4" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#publication-databases"><i class="fa fa-check"></i><b>13.4</b> Publication Databases</a></li>
<li class="chapter" data-level="13.5" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#example-research-questions"><i class="fa fa-check"></i><b>13.5</b> Example research questions:</a>
<ul>
<li class="chapter" data-level="13.5.1" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#biogeography-and-phylogenetic-diversity"><i class="fa fa-check"></i><b>13.5.1</b> Biogeography and Phylogenetic Diversity:</a></li>
<li class="chapter" data-level="13.5.2" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#community-ecology"><i class="fa fa-check"></i><b>13.5.2</b> Community Ecology:</a></li>
<li class="chapter" data-level="13.5.3" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#coevolution-1"><i class="fa fa-check"></i><b>13.5.3</b> Coevolution:</a></li>
<li class="chapter" data-level="13.5.4" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#species-tree"><i class="fa fa-check"></i><b>13.5.4</b> Species tree:</a></li>
</ul></li>
<li class="chapter" data-level="13.6" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#how-to-submit-jobs-on-scholar-cluster-and-run-remotely"><i class="fa fa-check"></i><b>13.6</b> How to submit jobs on Scholar Cluster and run remotely</a></li>
<li class="chapter" data-level="13.7" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#term-project-proposal"><i class="fa fa-check"></i><b>13.7</b> Term project proposal</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><i class="fa fa-check"></i><b>14</b> Systematic Retrieval of orthologous DNA sequences form GenBank</a>
<ul>
<li class="chapter" data-level="14.1" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#introduction"><i class="fa fa-check"></i><b>14.1</b> Introduction</a></li>
<li class="chapter" data-level="14.2" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#installing-tools"><i class="fa fa-check"></i><b>14.2</b> Installing Tools</a>
<ul>
<li class="chapter" data-level="14.2.1" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#windows-1"><i class="fa fa-check"></i><b>14.2.1</b> Windows</a></li>
<li class="chapter" data-level="14.2.2" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#mac"><i class="fa fa-check"></i><b>14.2.2</b> Mac</a></li>
<li class="chapter" data-level="14.2.3" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#ncbi-blast-tools"><i class="fa fa-check"></i><b>14.2.3</b> NCBI BLAST+ Tools</a></li>
</ul></li>
<li class="chapter" data-level="14.3" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#using-phylotar"><i class="fa fa-check"></i><b>14.3</b> Using <code>phylotaR</code></a>
<ul>
<li class="chapter" data-level="14.3.1" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#pipeline"><i class="fa fa-check"></i><b>14.3.1</b> Pipeline</a></li>
<li class="chapter" data-level="14.3.2" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#running"><i class="fa fa-check"></i><b>14.3.2</b> Running</a></li>
<li class="chapter" data-level="14.3.3" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#restarting"><i class="fa fa-check"></i><b>14.3.3</b> Restarting</a></li>
<li class="chapter" data-level="14.3.4" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#cluster-i.e.-orthologous-genes-selection"><i class="fa fa-check"></i><b>14.3.4</b> Cluster (i.e., orthologous genes) selection</a></li>
</ul></li>
<li class="chapter" data-level="14.4" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#exercise-i-1"><i class="fa fa-check"></i><b>14.4</b> Exercise I</a></li>
<li class="chapter" data-level="14.5" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#using-annotationbustr"><i class="fa fa-check"></i><b>14.5</b> Using <code>AnnotationBustR</code></a>
<ul>
<li class="chapter" data-level="14.5.1" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#introduction-1"><i class="fa fa-check"></i><b>14.5.1</b> Introduction</a></li>
<li class="chapter" data-level="14.5.2" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#install-the-tools"><i class="fa fa-check"></i><b>14.5.2</b> Install the tools</a></li>
<li class="chapter" data-level="14.5.3" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#specify-search-terms-of-genomes"><i class="fa fa-check"></i><b>14.5.3</b> Specify search terms of genomes</a></li>
<li class="chapter" data-level="14.5.4" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#load-gene-search-terms-of-interest"><i class="fa fa-check"></i><b>14.5.4</b> Load Gene Search Terms of Interest</a></li>
<li class="chapter" data-level="14.5.5" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#extract-sequences-with-annotationbust"><i class="fa fa-check"></i><b>14.5.5</b> Extract sequences with AnnotationBust</a></li>
</ul></li>
<li class="chapter" data-level="14.6" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#exercise-ii-1"><i class="fa fa-check"></i><b>14.6</b> Exercise II</a></li>
<li class="chapter" data-level="14.7" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#combine-gene-information-from-the-two-analyses"><i class="fa fa-check"></i><b>14.7</b> Combine gene information from the two analyses</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="sequence-alignment.html"><a href="sequence-alignment.html"><i class="fa fa-check"></i><b>15</b> Sequence alignment</a>
<ul>
<li class="chapter" data-level="15.1" data-path="sequence-alignment.html"><a href="sequence-alignment.html#load-libraries"><i class="fa fa-check"></i><b>15.1</b> Load libraries</a></li>
<li class="chapter" data-level="15.2" data-path="sequence-alignment.html"><a href="sequence-alignment.html#perform-alignments"><i class="fa fa-check"></i><b>15.2</b> Perform Alignments</a></li>
<li class="chapter" data-level="15.3" data-path="sequence-alignment.html"><a href="sequence-alignment.html#processing-multiple-alignments"><i class="fa fa-check"></i><b>15.3</b> Processing Multiple Alignments</a></li>
<li class="chapter" data-level="15.4" data-path="sequence-alignment.html"><a href="sequence-alignment.html#build-your-first-nj-tree"><i class="fa fa-check"></i><b>15.4</b> Build your first NJ tree</a></li>
<li class="chapter" data-level="15.5" data-path="sequence-alignment.html"><a href="sequence-alignment.html#exercise"><i class="fa fa-check"></i><b>15.5</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="maximum-likelihood-model-tests-tree-and-branch-tests.html"><a href="maximum-likelihood-model-tests-tree-and-branch-tests.html"><i class="fa fa-check"></i><b>16</b> Maximum likelihood, model tests, tree and branch tests</a>
<ul>
<li class="chapter" data-level="16.1" data-path="maximum-likelihood-model-tests-tree-and-branch-tests.html"><a href="maximum-likelihood-model-tests-tree-and-branch-tests.html#learning-goals-1"><i class="fa fa-check"></i><b>16.1</b> Learning Goals</a></li>
<li class="chapter" data-level="16.2" data-path="maximum-likelihood-model-tests-tree-and-branch-tests.html"><a href="maximum-likelihood-model-tests-tree-and-branch-tests.html#iqtree"><i class="fa fa-check"></i><b>16.2</b> IQtree</a>
<ul>
<li class="chapter" data-level="16.2.1" data-path="maximum-likelihood-model-tests-tree-and-branch-tests.html"><a href="maximum-likelihood-model-tests-tree-and-branch-tests.html#version"><i class="fa fa-check"></i><b>16.2.1</b> Version</a></li>
<li class="chapter" data-level="16.2.2" data-path="maximum-likelihood-model-tests-tree-and-branch-tests.html"><a href="maximum-likelihood-model-tests-tree-and-branch-tests.html#find-the-best-explanatory-sequence-evolution-model"><i class="fa fa-check"></i><b>16.2.2</b> find the best explanatory sequence evolution model</a></li>
<li class="chapter" data-level="16.2.3" data-path="maximum-likelihood-model-tests-tree-and-branch-tests.html"><a href="maximum-likelihood-model-tests-tree-and-branch-tests.html#using-the-best-sequence-evolution-model"><i class="fa fa-check"></i><b>16.2.3</b> using the best sequence evolution model</a></li>
<li class="chapter" data-level="16.2.4" data-path="maximum-likelihood-model-tests-tree-and-branch-tests.html"><a href="maximum-likelihood-model-tests-tree-and-branch-tests.html#different-partitions-for-the-data-for-multiple-genes-or-different-codons"><i class="fa fa-check"></i><b>16.2.4</b> Different partitions for the data for multiple genes, or different codons</a></li>
</ul></li>
<li class="chapter" data-level="16.3" data-path="maximum-likelihood-model-tests-tree-and-branch-tests.html"><a href="maximum-likelihood-model-tests-tree-and-branch-tests.html#run-an-ml-tree-with-multiple-gene-files-in-a-folder"><i class="fa fa-check"></i><b>16.3</b> Run an ML tree with multiple gene files in a folder</a></li>
<li class="chapter" data-level="16.4" data-path="maximum-likelihood-model-tests-tree-and-branch-tests.html"><a href="maximum-likelihood-model-tests-tree-and-branch-tests.html#gene-concordance-factor"><i class="fa fa-check"></i><b>16.4</b> Gene concordance factor</a></li>
<li class="chapter" data-level="16.5" data-path="maximum-likelihood-model-tests-tree-and-branch-tests.html"><a href="maximum-likelihood-model-tests-tree-and-branch-tests.html#tree-topology-tests"><i class="fa fa-check"></i><b>16.5</b> Tree topology tests</a></li>
<li class="chapter" data-level="16.6" data-path="maximum-likelihood-model-tests-tree-and-branch-tests.html"><a href="maximum-likelihood-model-tests-tree-and-branch-tests.html#tree-branches-tests"><i class="fa fa-check"></i><b>16.6</b> Tree branches tests</a></li>
<li class="chapter" data-level="16.7" data-path="maximum-likelihood-model-tests-tree-and-branch-tests.html"><a href="maximum-likelihood-model-tests-tree-and-branch-tests.html#assignments"><i class="fa fa-check"></i><b>16.7</b> Assignments</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="molecular-clock-and-phylogenetic-dating.html"><a href="molecular-clock-and-phylogenetic-dating.html"><i class="fa fa-check"></i><b>17</b> Molecular Clock and Phylogenetic Dating</a>
<ul>
<li class="chapter" data-level="17.1" data-path="molecular-clock-and-phylogenetic-dating.html"><a href="molecular-clock-and-phylogenetic-dating.html#inferring-time-tree-with-tip-dates"><i class="fa fa-check"></i><b>17.1</b> Inferring time tree with tip dates</a></li>
<li class="chapter" data-level="17.2" data-path="molecular-clock-and-phylogenetic-dating.html"><a href="molecular-clock-and-phylogenetic-dating.html#calibrating-tree-using-ancestral-dates"><i class="fa fa-check"></i><b>17.2</b> Calibrating tree using ancestral dates</a>
<ul>
<li class="chapter" data-level="17.2.1" data-path="molecular-clock-and-phylogenetic-dating.html"><a href="molecular-clock-and-phylogenetic-dating.html#search-for-an-ml-tree"><i class="fa fa-check"></i><b>17.2.1</b> search for an ML tree</a></li>
<li class="chapter" data-level="17.2.2" data-path="molecular-clock-and-phylogenetic-dating.html"><a href="molecular-clock-and-phylogenetic-dating.html#root-calibration"><i class="fa fa-check"></i><b>17.2.2</b> Root calibration</a></li>
<li class="chapter" data-level="17.2.3" data-path="molecular-clock-and-phylogenetic-dating.html"><a href="molecular-clock-and-phylogenetic-dating.html#internal-nodes"><i class="fa fa-check"></i><b>17.2.3</b> Internal nodes</a></li>
<li class="chapter" data-level="17.2.4" data-path="molecular-clock-and-phylogenetic-dating.html"><a href="molecular-clock-and-phylogenetic-dating.html#confidence-interval"><i class="fa fa-check"></i><b>17.2.4</b> Confidence interval</a></li>
</ul></li>
<li class="chapter" data-level="17.3" data-path="molecular-clock-and-phylogenetic-dating.html"><a href="molecular-clock-and-phylogenetic-dating.html#exercise-4"><i class="fa fa-check"></i><b>17.3</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="character-evolution-and-ancestral-state-reconstruction.html"><a href="character-evolution-and-ancestral-state-reconstruction.html"><i class="fa fa-check"></i><b>18</b> Character evolution and Ancestral state reconstruction</a>
<ul>
<li class="chapter" data-level="18.1" data-path="character-evolution-and-ancestral-state-reconstruction.html"><a href="character-evolution-and-ancestral-state-reconstruction.html#continuous-characters"><i class="fa fa-check"></i><b>18.1</b> Continuous characters</a>
<ul>
<li class="chapter" data-level="18.1.1" data-path="character-evolution-and-ancestral-state-reconstruction.html"><a href="character-evolution-and-ancestral-state-reconstruction.html#load-libraries-1"><i class="fa fa-check"></i><b>18.1.1</b> load libraries</a></li>
<li class="chapter" data-level="18.1.2" data-path="character-evolution-and-ancestral-state-reconstruction.html"><a href="character-evolution-and-ancestral-state-reconstruction.html#tree-file-and-the-character-file"><i class="fa fa-check"></i><b>18.1.2</b> Tree file and the character file</a></li>
<li class="chapter" data-level="18.1.3" data-path="character-evolution-and-ancestral-state-reconstruction.html"><a href="character-evolution-and-ancestral-state-reconstruction.html#visualization-of-the-reconstructed-nodes"><i class="fa fa-check"></i><b>18.1.3</b> Visualization of the reconstructed nodes</a></li>
<li class="chapter" data-level="18.1.4" data-path="character-evolution-and-ancestral-state-reconstruction.html"><a href="character-evolution-and-ancestral-state-reconstruction.html#visualization-of-phenotypic-evolution"><i class="fa fa-check"></i><b>18.1.4</b> Visualization of phenotypic evolution</a></li>
</ul></li>
<li class="chapter" data-level="18.2" data-path="character-evolution-and-ancestral-state-reconstruction.html"><a href="character-evolution-and-ancestral-state-reconstruction.html#brownian-motion-vs-ornstein-uhlenbeck-process"><i class="fa fa-check"></i><b>18.2</b> Brownian Motion vs Ornstein-Uhlenbeck Process</a></li>
<li class="chapter" data-level="18.3" data-path="character-evolution-and-ancestral-state-reconstruction.html"><a href="character-evolution-and-ancestral-state-reconstruction.html#brownian-motion"><i class="fa fa-check"></i><b>18.3</b> Brownian Motion</a>
<ul>
<li class="chapter" data-level="18.3.1" data-path="character-evolution-and-ancestral-state-reconstruction.html"><a href="character-evolution-and-ancestral-state-reconstruction.html#ornsteinuhlenbeck-process"><i class="fa fa-check"></i><b>18.3.1</b> Ornstein–Uhlenbeck process</a></li>
<li class="chapter" data-level="18.3.2" data-path="character-evolution-and-ancestral-state-reconstruction.html"><a href="character-evolution-and-ancestral-state-reconstruction.html#incorporate-prior-knowledge-using-bayesian-estimates"><i class="fa fa-check"></i><b>18.3.2</b> Incorporate prior knowledge using Bayesian estimates</a></li>
</ul></li>
<li class="chapter" data-level="18.4" data-path="character-evolution-and-ancestral-state-reconstruction.html"><a href="character-evolution-and-ancestral-state-reconstruction.html#discrete-characters"><i class="fa fa-check"></i><b>18.4</b> Discrete characters</a></li>
<li class="chapter" data-level="18.5" data-path="character-evolution-and-ancestral-state-reconstruction.html"><a href="character-evolution-and-ancestral-state-reconstruction.html#exercises-1"><i class="fa fa-check"></i><b>18.5</b> Exercises:</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="model-selection-of-character-evolution.html"><a href="model-selection-of-character-evolution.html"><i class="fa fa-check"></i><b>19</b> Model Selection of Character Evolution</a>
<ul>
<li class="chapter" data-level="19.1" data-path="model-selection-of-character-evolution.html"><a href="model-selection-of-character-evolution.html#load-libraries-2"><i class="fa fa-check"></i><b>19.1</b> Load libraries</a></li>
<li class="chapter" data-level="19.2" data-path="model-selection-of-character-evolution.html"><a href="model-selection-of-character-evolution.html#discrete-traits-model-fitting"><i class="fa fa-check"></i><b>19.2</b> Discrete traits model fitting</a>
<ul>
<li class="chapter" data-level="19.2.1" data-path="model-selection-of-character-evolution.html"><a href="model-selection-of-character-evolution.html#choosing-a-character-model"><i class="fa fa-check"></i><b>19.2.1</b> Choosing a character model</a></li>
<li class="chapter" data-level="19.2.2" data-path="model-selection-of-character-evolution.html"><a href="model-selection-of-character-evolution.html#marginal-ancestral-state-reconstruction"><i class="fa fa-check"></i><b>19.2.2</b> Marginal ancestral state reconstruction</a></li>
<li class="chapter" data-level="19.2.3" data-path="model-selection-of-character-evolution.html"><a href="model-selection-of-character-evolution.html#more-than-two-states"><i class="fa fa-check"></i><b>19.2.3</b> More than two states</a></li>
</ul></li>
<li class="chapter" data-level="19.3" data-path="model-selection-of-character-evolution.html"><a href="model-selection-of-character-evolution.html#exercise-i-2"><i class="fa fa-check"></i><b>19.3</b> Exercise I</a></li>
<li class="chapter" data-level="19.4" data-path="model-selection-of-character-evolution.html"><a href="model-selection-of-character-evolution.html#continuous-traits-model-fitting"><i class="fa fa-check"></i><b>19.4</b> Continuous traits model fitting</a>
<ul>
<li class="chapter" data-level="19.4.1" data-path="model-selection-of-character-evolution.html"><a href="model-selection-of-character-evolution.html#model-selection"><i class="fa fa-check"></i><b>19.4.1</b> Model selection</a></li>
<li class="chapter" data-level="19.4.2" data-path="model-selection-of-character-evolution.html"><a href="model-selection-of-character-evolution.html#ancestral-states"><i class="fa fa-check"></i><b>19.4.2</b> Ancestral states</a></li>
</ul></li>
<li class="chapter" data-level="19.5" data-path="model-selection-of-character-evolution.html"><a href="model-selection-of-character-evolution.html#exercise-ii-2"><i class="fa fa-check"></i><b>19.5</b> Exercise II</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">BIO595 The Tree of Life</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="biogeography-and-macroevolution-in-r" class="section level1 hasAnchor" number="6">
<h1><span class="header-section-number">Chapter 6</span> Biogeography and Macroevolution in R<a href="biogeography-and-macroevolution-in-r.html#biogeography-and-macroevolution-in-r" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="setting-up-the-environment-2" class="section level2 hasAnchor" number="6.1">
<h2><span class="header-section-number">6.1</span> Setting up the environment<a href="biogeography-and-macroevolution-in-r.html#setting-up-the-environment-2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Download the files for the lab (<code>"Daniel_tree.tre"</code> and <code>"Daniel_data.csv"</code>) and save them in a folder you will be working in this session.
Set your working directory to this folder.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="biogeography-and-macroevolution-in-r.html#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;ape&quot;</span>)</span>
<span id="cb1-2"><a href="biogeography-and-macroevolution-in-r.html#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;phytools&quot;</span>)</span>
<span id="cb1-3"><a href="biogeography-and-macroevolution-in-r.html#cb1-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-4"><a href="biogeography-and-macroevolution-in-r.html#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;ape&quot;</span>)</span>
<span id="cb1-5"><a href="biogeography-and-macroevolution-in-r.html#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;phytools&quot;</span>)</span>
<span id="cb1-6"><a href="biogeography-and-macroevolution-in-r.html#cb1-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-7"><a href="biogeography-and-macroevolution-in-r.html#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="co"># set working directory</span></span>
<span id="cb1-8"><a href="biogeography-and-macroevolution-in-r.html#cb1-8" aria-hidden="true" tabindex="-1"></a>wd <span class="ot">&lt;-</span> <span class="st">&quot;~/Dropbox/teaching/bookdown-demo-main/data2023/&quot;</span> <span class="co"># absolute address of the folder where you have the data you need and plan to work in.</span></span>
<span id="cb1-9"><a href="biogeography-and-macroevolution-in-r.html#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(wd)</span>
<span id="cb1-10"><a href="biogeography-and-macroevolution-in-r.html#cb1-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-11"><a href="biogeography-and-macroevolution-in-r.html#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="co"># load data</span></span>
<span id="cb1-12"><a href="biogeography-and-macroevolution-in-r.html#cb1-12" aria-hidden="true" tabindex="-1"></a>tree <span class="ot">&lt;-</span> <span class="fu">read.tree</span>(<span class="st">&quot;Daniel_tree.tre&quot;</span>) <span class="co"># reads newick files</span></span>
<span id="cb1-13"><a href="biogeography-and-macroevolution-in-r.html#cb1-13" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;Daniel_data.csv&quot;</span>, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>, <span class="at">row.names =</span> <span class="dv">1</span>) <span class="co"># designates the values in the first column as row names </span></span></code></pre></div>
<p><br></p>
</div>
<div id="ancestral-state-reconstructions" class="section level2 hasAnchor" number="6.2">
<h2><span class="header-section-number">6.2</span> Ancestral State Reconstructions<a href="biogeography-and-macroevolution-in-r.html#ancestral-state-reconstructions" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Ancestral state reconstruction is a fundamental tool for exploring evolution because it provides estimates of otherwise unobservable processes.
Given a character – any heritable trait for which there is a reasonable hypothesis that all states are homologous – ancestral state reconstruction estimates character states for both recent and deep time ancestors (i.e., nodes on a phylogeny).
Ancestral state reconstruction methods combine information about evolutionary relationships from a phylogenetic tree with the observed state of a character for each tip (each terminal node, often an extant species).
Many ancestral state reconstruction methods have been developed, including parsimony-based methods, which minimize the total number of changes of state across the tree, and a range of likelihood-based approaches that use stochastic Markov models of character change.
Generally, ancestral state reconstruction is performed after a phylogeny of the group of interest has been reconstructed.</p>
<p></br></p>
<p>The <code>ace</code> function in the <code>ape</code> package estimates ancestral character states and the associated uncertainty.
Let’s take a look at what kind of information it requires.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="biogeography-and-macroevolution-in-r.html#cb2-1" aria-hidden="true" tabindex="-1"></a>?ace</span></code></pre></div>
<p></br></p>
<div id="bonus-2" class="section level3 hasAnchor" number="6.2.1">
<h3><span class="header-section-number">6.2.1</span> Bonus [2]<a href="biogeography-and-macroevolution-in-r.html#bonus-2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Examine the example data referenced in the documentation for the <code>ace</code> function.
Provide short answers to the following questions on BrightSpace:</p>
<ol style="list-style-type: decimal">
<li><p>What are the minimum data inputs (not arguments or conditions) required to run this function?
How are they different from the data inputs for the <code>plotTree.barplot</code> function?</p></li>
<li><p>How do you think trait values are associated with the correct taxon (i.e., tip label) in the <code>ace</code> function?</p></li>
</ol>
<p>Write your answers below:</p>
<hr />
<hr />
<hr />
<hr />
<p></br></p>
</div>
</div>
<div id="continuous-traits" class="section level2 hasAnchor" number="6.3">
<h2><span class="header-section-number">6.3</span> Continuous traits<a href="biogeography-and-macroevolution-in-r.html#continuous-traits" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Also referred to as quantitative traits, continuous traits refer to traits (or characters) that express continuous variation.
Common examples include traits such as weight and height.
Characters can exhibit continuous character states which can be homologous.
Sometimes continuous traits are binned or categorized into discrete traits (e.g., size: large, medium, small) which we will discuss below.
For now, let’s examine an example of a continuous trait.
To conduct ancestral state reconstructions we will need to format the trait data in a certain way.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="biogeography-and-macroevolution-in-r.html#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co"># we want the input to be a numeric vector with names matching tree tip labels</span></span>
<span id="cb3-2"><a href="biogeography-and-macroevolution-in-r.html#cb3-2" aria-hidden="true" tabindex="-1"></a>weight <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(df[<span class="st">&quot;weight&quot;</span>, ])</span>
<span id="cb3-3"><a href="biogeography-and-macroevolution-in-r.html#cb3-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-4"><a href="biogeography-and-macroevolution-in-r.html#cb3-4" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(weight) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(df) </span>
<span id="cb3-5"><a href="biogeography-and-macroevolution-in-r.html#cb3-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-6"><a href="biogeography-and-macroevolution-in-r.html#cb3-6" aria-hidden="true" tabindex="-1"></a>weight <span class="co">#examine the object</span></span>
<span id="cb3-7"><a href="biogeography-and-macroevolution-in-r.html#cb3-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-8"><a href="biogeography-and-macroevolution-in-r.html#cb3-8" aria-hidden="true" tabindex="-1"></a>weight <span class="ot">&lt;-</span> weight[tree<span class="sc">$</span>tip.label] </span>
<span id="cb3-9"><a href="biogeography-and-macroevolution-in-r.html#cb3-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-10"><a href="biogeography-and-macroevolution-in-r.html#cb3-10" aria-hidden="true" tabindex="-1"></a>weight <span class="co">#examine the object again - what changed?</span></span></code></pre></div>
<p></br></p>
<div id="maximum-likelihood" class="section level3 hasAnchor" number="6.3.1">
<h3><span class="header-section-number">6.3.1</span> Maximum likelihood<a href="biogeography-and-macroevolution-in-r.html#maximum-likelihood" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>There are multiple approaches to reconstructing ancestral states of continuous traits.
One is to use a maximum likelihood approach.
Maximum likelihood estimation is a method of estimating the parameters of an assumed probability distribution (in this case, based on Brownian motion), given some observed data.
This is achieved by maximizing a likelihood function so that, under the assumed statistical model, the observed data is most probable.
In other words, here we are estimating the character states of ancestral taxa that best explain the current distribution of character states among extant taxa.
This approach requires branch lengths.
We will revisit the concept of likelihood later in the semester when we learn different approaches to phylogenetic reconstruction.</p>
<p>For more information see: Schluter, D., Price, T., Mooers, A. O.
and Ludwig, D.
(1997) Likelihood of ancestor states in adaptive radiation.
Evolution, 51, 1699–1711.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="biogeography-and-macroevolution-in-r.html#cb4-1" aria-hidden="true" tabindex="-1"></a>MLreconstruction <span class="ot">&lt;-</span> <span class="fu">ace</span>(weight, tree, <span class="at">type =</span> <span class="st">&quot;continuous&quot;</span>, <span class="at">method =</span> <span class="st">&quot;ML&quot;</span>)</span>
<span id="cb4-2"><a href="biogeography-and-macroevolution-in-r.html#cb4-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-3"><a href="biogeography-and-macroevolution-in-r.html#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(MLreconstruction)</span>
<span id="cb4-4"><a href="biogeography-and-macroevolution-in-r.html#cb4-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-5"><a href="biogeography-and-macroevolution-in-r.html#cb4-5" aria-hidden="true" tabindex="-1"></a>MLreconstruction</span></code></pre></div>
<p>**ML reconstruction using ace may generate a large number of not-a-number error messages.
These result when the program calculates the likelihood of particularly poor fits.
You can safely ignore these messages for now.</p>
<p></br></p>
<p>The output is essentially a list with multiple elements.
Thus we can refer to these elements using their names (e.g., <code>"$se"</code>) or double square brackets (<code>"[[]]"</code>).</p>
<table>
<colgroup>
<col width="27%" />
<col width="72%" />
</colgroup>
<tbody>
<tr class="odd">
<td>ace</td>
<td>if type = “continuous”, the estimates of the ancestral character values.</td>
</tr>
<tr class="even">
<td>CI95</td>
<td>if type = “continuous”, the estimated 95% confidence intervals.</td>
</tr>
<tr class="odd">
<td>sigma2</td>
<td>if type = “continuous”, model = “BM”, and method = “ML”, the maximum likelihood estimate of the Brownian parameter.</td>
</tr>
<tr class="even">
<td>rates</td>
<td>if type = “discrete”, the maximum likelihood estimates of the transition rates.</td>
</tr>
<tr class="odd">
<td>se</td>
<td>if type = “discrete”, the standard-errors of estimated rates.</td>
</tr>
<tr class="even">
<td>index.matrix</td>
<td>if type = “discrete”, gives the indices of the rates in the rate matrix.</td>
</tr>
<tr class="odd">
<td>loglik</td>
<td>if method = “ML”, the maximum log-likelihood.</td>
</tr>
<tr class="even">
<td>lik.anc</td>
<td>if type = “discrete”, the scaled likelihoods of each ancestral state.</td>
</tr>
<tr class="odd">
<td>call</td>
<td>the function call.</td>
</tr>
</tbody>
</table>
<p><em>*The higher the log likelihood, the better a model fits the observed data.</em></p>
</div>
<div id="phylogenetically-independent-contrasts" class="section level3 hasAnchor" number="6.3.2">
<h3><span class="header-section-number">6.3.2</span> Phylogenetically independent contrasts<a href="biogeography-and-macroevolution-in-r.html#phylogenetically-independent-contrasts" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Another approach is to reconstruct ancestral states for a continuous (numeric) trait using phylogenetic independent contrasts (PIC; Felsenstein, 1985).
PICs are calculated as differences in trait values (i.e., character states) between adjacent pairs of nodes or terminal taxa in a phylogenetic tree.
This is also a Brownian-motion based estimator, but it only takes descendants of each node into account in reconstructing the state at that node.
More basal nodes are ignored.
Branch lengths are not necessary.</p>
<p>For more information see: Ricklefs, R. E., &amp; Starck, J. M.
(1996).
Applications of phylogenetically independent contrasts: a mixed progress report.
Oikos, 167-172.
Felsenstein, J.
(1985).
Phylogenies and the comparative method.
The American Naturalist, 125(1), 1-15.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="biogeography-and-macroevolution-in-r.html#cb5-1" aria-hidden="true" tabindex="-1"></a>picreconstruction <span class="ot">&lt;-</span> <span class="fu">ace</span>(weight, tree, <span class="at">type=</span><span class="st">&quot;continuous&quot;</span>, <span class="at">method =</span> <span class="st">&quot;pic&quot;</span>) </span>
<span id="cb5-2"><a href="biogeography-and-macroevolution-in-r.html#cb5-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-3"><a href="biogeography-and-macroevolution-in-r.html#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(picreconstruction)</span>
<span id="cb5-4"><a href="biogeography-and-macroevolution-in-r.html#cb5-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-5"><a href="biogeography-and-macroevolution-in-r.html#cb5-5" aria-hidden="true" tabindex="-1"></a>picreconstruction</span></code></pre></div>
<p><br></p>
</div>
<div id="plotting-ancestral-states" class="section level3 hasAnchor" number="6.3.3">
<h3><span class="header-section-number">6.3.3</span> Plotting ancestral states<a href="biogeography-and-macroevolution-in-r.html#plotting-ancestral-states" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Each of the objects (e.g., <code>"MLreconstruction"</code>) that we reconstructed above is a list of several elements.
These include <code>$loglik</code> (the log-likelihood of the most likely reconstruction), <code>$ace</code> (the vector of reconstructed node values), <code>$sigma2</code> (a two element vector of the rate estimate and its standard error) and <code>$CI95</code>, the 95% confidence intervals around the vector of reconstructed node values.
Using the information in this list, we can visualize the reconstructed states on a phylogeny.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="biogeography-and-macroevolution-in-r.html#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co"># plot tree without tip labels</span></span>
<span id="cb6-2"><a href="biogeography-and-macroevolution-in-r.html#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(tree, <span class="at">show.tip.label =</span> <span class="cn">FALSE</span>, <span class="at">type =</span> <span class="st">&quot;phylogram&quot;</span>)</span>
<span id="cb6-3"><a href="biogeography-and-macroevolution-in-r.html#cb6-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-4"><a href="biogeography-and-macroevolution-in-r.html#cb6-4" aria-hidden="true" tabindex="-1"></a><span class="co"># add circles to tips proportionate to the character states of extant taxa - pch numbers refer to different shapes</span></span>
<span id="cb6-5"><a href="biogeography-and-macroevolution-in-r.html#cb6-5" aria-hidden="true" tabindex="-1"></a><span class="fu">tiplabels</span>(<span class="at">pch =</span> <span class="dv">21</span>, <span class="at">cex =</span> weight<span class="sc">/</span><span class="dv">20</span>) </span>
<span id="cb6-6"><a href="biogeography-and-macroevolution-in-r.html#cb6-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-7"><a href="biogeography-and-macroevolution-in-r.html#cb6-7" aria-hidden="true" tabindex="-1"></a><span class="co"># add circles to nodes proportionate to the character states of ancestral taxa</span></span>
<span id="cb6-8"><a href="biogeography-and-macroevolution-in-r.html#cb6-8" aria-hidden="true" tabindex="-1"></a><span class="fu">nodelabels</span>(<span class="at">pch =</span> <span class="dv">21</span>, <span class="at">cex =</span> MLreconstruction<span class="sc">$</span>ace<span class="sc">/</span><span class="dv">20</span>) </span>
<span id="cb6-9"><a href="biogeography-and-macroevolution-in-r.html#cb6-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-10"><a href="biogeography-and-macroevolution-in-r.html#cb6-10" aria-hidden="true" tabindex="-1"></a><span class="co"># add character state values to tips</span></span>
<span id="cb6-11"><a href="biogeography-and-macroevolution-in-r.html#cb6-11" aria-hidden="true" tabindex="-1"></a><span class="fu">tiplabels</span>(weight, <span class="at">cex =</span> <span class="dv">1</span>, <span class="at">bg =</span> <span class="st">&quot;#00000000&quot;</span>, <span class="at">frame =</span> <span class="st">&quot;none&quot;</span>, <span class="at">adj =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>, <span class="dv">0</span>)) </span>
<span id="cb6-12"><a href="biogeography-and-macroevolution-in-r.html#cb6-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-13"><a href="biogeography-and-macroevolution-in-r.html#cb6-13" aria-hidden="true" tabindex="-1"></a><span class="co"># add character state values to nodes</span></span>
<span id="cb6-14"><a href="biogeography-and-macroevolution-in-r.html#cb6-14" aria-hidden="true" tabindex="-1"></a><span class="fu">nodelabels</span>(<span class="fu">round</span>(MLreconstruction<span class="sc">$</span>ace, <span class="dv">1</span>), <span class="at">cex =</span> <span class="dv">1</span>, <span class="at">bg =</span> <span class="st">&quot;#00000000&quot;</span>, <span class="at">frame =</span> <span class="st">&quot;none&quot;</span>, <span class="at">adj =</span> <span class="fu">c</span>(<span class="fl">1.5</span>, <span class="fl">1.5</span>)) </span></code></pre></div>
<p><br></p>
</div>
<div id="exercise-1-3" class="section level3 hasAnchor" number="6.3.4">
<h3><span class="header-section-number">6.3.4</span> Exercise (1)<a href="biogeography-and-macroevolution-in-r.html#exercise-1-3" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Plot the results of the phylogenetically independent contrasts (i.e., <code>"picreconstruction"</code>) in a similar manner to above.
You do not need to plot the actual estimated trait values.
Make sure you:</p>
<ul>
<li><p>Plot the tree as a cladogram where all the tips are aligned.</p></li>
<li><p>Use squares instead of circles.</p></li>
<li><p>Use different colors for the squares on the internal nodes and the squares on the tips, but not the default colors we used above.
You can refer to this <a href="http://www.stat.columbia.edu/~tzheng/files/Rcolor.pdf">link</a> for color names.</p></li>
<li><p>Show the names of the taxa on the tips of the phylogeny (e.g., “t1”, “t2”…).</p></li>
</ul>
<p><em>Hint:</em> “?pch” “?nodelabels” “?tiplabels”</p>
<p>Write your answers below:</p>
<hr />
<hr />
<hr />
<hr />
<p></br></p>
</div>
</div>
<div id="discrete-traits" class="section level2 hasAnchor" number="6.4">
<h2><span class="header-section-number">6.4</span> Discrete traits<a href="biogeography-and-macroevolution-in-r.html#discrete-traits" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Also referred to as qualitative traits, continuous traits refer to traits (or characters) that express distinct phenotypes.
Think of characters that have a set of categorized states, such as the presence/absence of a certain feature or organ.
Many taxonomic groups are defined by discrete character states (e.g., mammals - presence of hair, mammary glands).
As mentioned above, sometimes continuous traits are binned or categorized into discrete traits (e.g., size: large, medium, small).
The states of characters such as color can be expressed in both continuous and discrete values.</p>
<p></br></p>
<div id="exercise-2-3" class="section level3 hasAnchor" number="6.4.1">
<h3><span class="header-section-number">6.4.1</span> Exercise (2)<a href="biogeography-and-macroevolution-in-r.html#exercise-2-3" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Describe, in your own words, what a synapomorphy is.
Feel free to use examples but keep your answer short.</p>
<p>Write your answers below:</p>
<hr />
<hr />
<hr />
<hr />
<p></br></p>
<p>Let’s generate some discrete trait data.
We are going to create an object of the class factor.
Factors are used to represent categorical data.
Factors can be ordered (i.e., ranked) or unordered (i.e., unranked) and are an important for statistical analyses and plotting.
Factors are stored as integers, and have labels associated with these unique integers.
While factors look (and often behave) like character vectors, they are actually integers under the hood, and you need to be careful when treating them like strings.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="biogeography-and-macroevolution-in-r.html#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co"># generate a factor with discrete character states</span></span>
<span id="cb7-2"><a href="biogeography-and-macroevolution-in-r.html#cb7-2" aria-hidden="true" tabindex="-1"></a>continent <span class="ot">&lt;-</span> <span class="fu">factor</span>(<span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">&quot;Africa&quot;</span>, <span class="dv">5</span>), <span class="fu">rep</span>(<span class="st">&quot;Eurasia&quot;</span>, <span class="dv">5</span>))) </span>
<span id="cb7-3"><a href="biogeography-and-macroevolution-in-r.html#cb7-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-4"><a href="biogeography-and-macroevolution-in-r.html#cb7-4" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(continent)</span>
<span id="cb7-5"><a href="biogeography-and-macroevolution-in-r.html#cb7-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-6"><a href="biogeography-and-macroevolution-in-r.html#cb7-6" aria-hidden="true" tabindex="-1"></a>continent</span></code></pre></div>
<p></br></p>
</div>
<div id="maximum-likelihood-1" class="section level3 hasAnchor" number="6.4.2">
<h3><span class="header-section-number">6.4.2</span> Maximum likelihood<a href="biogeography-and-macroevolution-in-r.html#maximum-likelihood-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The <code>ace</code> function can reconstruct ancestral states for discrete characters using maximum likelihood approaches.
However there are several different options for specifying the transition probabilities between the states of discrete characters (e.g., think of nucleotide substitutions - transitions are more likely to happen than transversions) In our example, range transitions from Africa to Eurasia could not be as likely to happen as range transistions from Eurasia to Africa.</p>
<p>The following three models are built in the function, but you can also specify your own rates of transition (see <code>?ace</code> for more details).</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="biogeography-and-macroevolution-in-r.html#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co"># ER: one-parameter equal rates model</span></span>
<span id="cb8-2"><a href="biogeography-and-macroevolution-in-r.html#cb8-2" aria-hidden="true" tabindex="-1"></a>ERreconstruction <span class="ot">&lt;-</span> <span class="fu">ace</span>(continent, tree, <span class="at">type =</span> <span class="st">&quot;discrete&quot;</span>, <span class="at">model =</span> <span class="st">&quot;ER&quot;</span>)</span>
<span id="cb8-3"><a href="biogeography-and-macroevolution-in-r.html#cb8-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-4"><a href="biogeography-and-macroevolution-in-r.html#cb8-4" aria-hidden="true" tabindex="-1"></a><span class="co"># SYM: symmetric model in which forwards and reverse transitions between states are constrained to be equal</span></span>
<span id="cb8-5"><a href="biogeography-and-macroevolution-in-r.html#cb8-5" aria-hidden="true" tabindex="-1"></a>SYMreconstruction <span class="ot">&lt;-</span> <span class="fu">ace</span>(continent, tree, <span class="at">type =</span> <span class="st">&quot;discrete&quot;</span>, <span class="at">model =</span> <span class="st">&quot;SYM&quot;</span>) </span>
<span id="cb8-6"><a href="biogeography-and-macroevolution-in-r.html#cb8-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-7"><a href="biogeography-and-macroevolution-in-r.html#cb8-7" aria-hidden="true" tabindex="-1"></a><span class="co"># ARD: all rates different matrix where all possible transitions between states receive distinct parameters</span></span>
<span id="cb8-8"><a href="biogeography-and-macroevolution-in-r.html#cb8-8" aria-hidden="true" tabindex="-1"></a>ARDreconstruction <span class="ot">&lt;-</span> <span class="fu">ace</span>(continent, tree, <span class="at">type =</span> <span class="st">&quot;discrete&quot;</span>, <span class="at">model =</span> <span class="st">&quot;ARD&quot;</span>) </span>
<span id="cb8-9"><a href="biogeography-and-macroevolution-in-r.html#cb8-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-10"><a href="biogeography-and-macroevolution-in-r.html#cb8-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Let&#39;s look at the different estimated rates of transition</span></span>
<span id="cb8-11"><a href="biogeography-and-macroevolution-in-r.html#cb8-11" aria-hidden="true" tabindex="-1"></a>ERreconstruction<span class="sc">$</span>rates</span>
<span id="cb8-12"><a href="biogeography-and-macroevolution-in-r.html#cb8-12" aria-hidden="true" tabindex="-1"></a>SYMreconstruction<span class="sc">$</span>rates</span>
<span id="cb8-13"><a href="biogeography-and-macroevolution-in-r.html#cb8-13" aria-hidden="true" tabindex="-1"></a>ARDreconstruction<span class="sc">$</span>rates</span>
<span id="cb8-14"><a href="biogeography-and-macroevolution-in-r.html#cb8-14" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb8-15"><a href="biogeography-and-macroevolution-in-r.html#cb8-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Let&#39;s look at the scaled likelihoods of each ancestral state </span></span>
<span id="cb8-16"><a href="biogeography-and-macroevolution-in-r.html#cb8-16" aria-hidden="true" tabindex="-1"></a>ERreconstruction<span class="sc">$</span>lik.anc</span>
<span id="cb8-17"><a href="biogeography-and-macroevolution-in-r.html#cb8-17" aria-hidden="true" tabindex="-1"></a>SYMreconstruction<span class="sc">$</span>lik.anc</span>
<span id="cb8-18"><a href="biogeography-and-macroevolution-in-r.html#cb8-18" aria-hidden="true" tabindex="-1"></a>ARDreconstruction<span class="sc">$</span>lik.anc</span></code></pre></div>
<p></br></p>
</div>
<div id="choosing-models" class="section level3 hasAnchor" number="6.4.3">
<h3><span class="header-section-number">6.4.3</span> Choosing models<a href="biogeography-and-macroevolution-in-r.html#choosing-models" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The log-likelihood value for a given model can range from negative infinity to positive infinity.
The actual log-likelihood value for a given model is mostly meaningless, but it’s useful for comparing two or more models.
Again, the higher the value of the log-likelihood, the better a model fits a dataset.
One of the models above has a higher log-likelihood value than the others.
However, do we know if it is significantly better?</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="biogeography-and-macroevolution-in-r.html#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co"># log likelihood is a measure of model fit: higher values are better</span></span>
<span id="cb9-2"><a href="biogeography-and-macroevolution-in-r.html#cb9-2" aria-hidden="true" tabindex="-1"></a>ERreconstruction<span class="sc">$</span>loglik</span>
<span id="cb9-3"><a href="biogeography-and-macroevolution-in-r.html#cb9-3" aria-hidden="true" tabindex="-1"></a>SYMreconstruction<span class="sc">$</span>loglik</span>
<span id="cb9-4"><a href="biogeography-and-macroevolution-in-r.html#cb9-4" aria-hidden="true" tabindex="-1"></a>ARDreconstruction<span class="sc">$</span>loglik</span>
<span id="cb9-5"><a href="biogeography-and-macroevolution-in-r.html#cb9-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-6"><a href="biogeography-and-macroevolution-in-r.html#cb9-6" aria-hidden="true" tabindex="-1"></a><span class="co"># testing if one model is significantly better than the other</span></span>
<span id="cb9-7"><a href="biogeography-and-macroevolution-in-r.html#cb9-7" aria-hidden="true" tabindex="-1"></a><span class="co"># the following command gives the percentage of the chisquare distribution that lies to the right of the observed likelihood difference for the ER and ARD models (which differ by one parameter)</span></span>
<span id="cb9-8"><a href="biogeography-and-macroevolution-in-r.html#cb9-8" aria-hidden="true" tabindex="-1"></a><span class="co"># generally we consider percentages below 5% (0.05) to be significant</span></span>
<span id="cb9-9"><a href="biogeography-and-macroevolution-in-r.html#cb9-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-10"><a href="biogeography-and-macroevolution-in-r.html#cb9-10" aria-hidden="true" tabindex="-1"></a><span class="dv">1</span> <span class="sc">-</span> <span class="fu">pchisq</span>(<span class="dv">2</span><span class="sc">*</span><span class="fu">abs</span>(ERreconstruction<span class="sc">$</span>loglik <span class="sc">-</span> ARDreconstruction<span class="sc">$</span>loglik), <span class="dv">1</span>)</span></code></pre></div>
<p>When calculating log-likelihood values, it’s important to note that adding more predictor variables to a model will almost always increase the log-likelihood value even if the additional predictor variables aren’t statistically significant.
This means you should only compare the log-likelihood values between two regression models if each model has the same number of predictor variables.
In this case, since the log-likelihood values are not significantly different, let’s go with the model with less parameters - all else equal, simpler models are better!</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="biogeography-and-macroevolution-in-r.html#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="co"># plotting the results of our selected model</span></span>
<span id="cb10-2"><a href="biogeography-and-macroevolution-in-r.html#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(tree, <span class="at">type =</span> <span class="st">&quot;p&quot;</span>, <span class="cn">FALSE</span>, <span class="at">label.offset =</span> <span class="dv">1</span>)</span>
<span id="cb10-3"><a href="biogeography-and-macroevolution-in-r.html#cb10-3" aria-hidden="true" tabindex="-1"></a>co <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;blue&quot;</span>, <span class="st">&quot;yellow&quot;</span>)</span>
<span id="cb10-4"><a href="biogeography-and-macroevolution-in-r.html#cb10-4" aria-hidden="true" tabindex="-1"></a><span class="fu">tiplabels</span>(<span class="at">pch =</span> <span class="dv">21</span>, <span class="at">bg =</span> co[<span class="fu">as.numeric</span>(continent)], <span class="at">cex =</span> <span class="dv">2</span>, <span class="at">adj =</span> <span class="dv">1</span>)</span>
<span id="cb10-5"><a href="biogeography-and-macroevolution-in-r.html#cb10-5" aria-hidden="true" tabindex="-1"></a><span class="fu">nodelabels</span>(<span class="at">pie =</span> ERreconstruction<span class="sc">$</span>lik.anc, <span class="at">piecol =</span> co, <span class="at">cex =</span> <span class="fl">0.75</span>)</span></code></pre></div>
</div>
</div>
<div id="saving-multiple-plots-together" class="section level2 hasAnchor" number="6.5">
<h2><span class="header-section-number">6.5</span> Saving multiple plots together<a href="biogeography-and-macroevolution-in-r.html#saving-multiple-plots-together" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We can save separately created plots in the same file - you may have noticed that many figures in scientific papers and textbooks are multi-paneled.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="biogeography-and-macroevolution-in-r.html#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="st">&quot;anc.pdf&quot;</span>, <span class="at">width =</span> <span class="dv">12</span>, <span class="at">height =</span> <span class="dv">5</span>)</span>
<span id="cb11-2"><a href="biogeography-and-macroevolution-in-r.html#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="co"># &#39;mfrow&#39; specifies the number of rows and columns; &#39;mar&#39; sets the width of the margins in the order: &#39;bottom&#39;, &#39;left&#39;, &#39;top&#39;, &#39;right&#39;. </span></span>
<span id="cb11-3"><a href="biogeography-and-macroevolution-in-r.html#cb11-3" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>), <span class="at">mar =</span> <span class="fu">c</span>(<span class="dv">2</span>, <span class="dv">2</span>, <span class="dv">2</span>, <span class="dv">2</span>) <span class="sc">+</span> .<span class="dv">1</span>) </span>
<span id="cb11-4"><a href="biogeography-and-macroevolution-in-r.html#cb11-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-5"><a href="biogeography-and-macroevolution-in-r.html#cb11-5" aria-hidden="true" tabindex="-1"></a>  <span class="co"># ancestral weight reconstruction plot</span></span>
<span id="cb11-6"><a href="biogeography-and-macroevolution-in-r.html#cb11-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot</span>(tree, <span class="at">show.tip.label =</span> <span class="cn">FALSE</span>, <span class="at">type =</span> <span class="st">&quot;phylogram&quot;</span>, <span class="at">main =</span> <span class="st">&quot;weight&quot;</span>)</span>
<span id="cb11-7"><a href="biogeography-and-macroevolution-in-r.html#cb11-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tiplabels</span>(<span class="at">pch =</span> <span class="dv">21</span>, <span class="at">cex =</span> weight<span class="sc">/</span><span class="dv">20</span>) </span>
<span id="cb11-8"><a href="biogeography-and-macroevolution-in-r.html#cb11-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">nodelabels</span>(<span class="at">pch =</span> <span class="dv">21</span>, <span class="at">cex =</span> MLreconstruction<span class="sc">$</span>ace<span class="sc">/</span><span class="dv">20</span>) </span>
<span id="cb11-9"><a href="biogeography-and-macroevolution-in-r.html#cb11-9" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb11-10"><a href="biogeography-and-macroevolution-in-r.html#cb11-10" aria-hidden="true" tabindex="-1"></a>  <span class="co"># ancestral range reconstruction plot</span></span>
<span id="cb11-11"><a href="biogeography-and-macroevolution-in-r.html#cb11-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot</span>(tree, <span class="at">type =</span> <span class="st">&quot;p&quot;</span>, <span class="cn">FALSE</span>, <span class="at">label.offset =</span> <span class="dv">1</span>, <span class="at">main =</span> <span class="st">&quot;range&quot;</span>)</span>
<span id="cb11-12"><a href="biogeography-and-macroevolution-in-r.html#cb11-12" aria-hidden="true" tabindex="-1"></a>  co <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;blue&quot;</span>, <span class="st">&quot;yellow&quot;</span>)</span>
<span id="cb11-13"><a href="biogeography-and-macroevolution-in-r.html#cb11-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tiplabels</span>(<span class="at">pch =</span> <span class="dv">21</span>, <span class="at">bg =</span> co[<span class="fu">as.numeric</span>(continent)], <span class="at">cex =</span> <span class="dv">2</span>, <span class="at">adj =</span> <span class="dv">1</span>)</span>
<span id="cb11-14"><a href="biogeography-and-macroevolution-in-r.html#cb11-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">nodelabels</span>(<span class="at">pie =</span> ERreconstruction<span class="sc">$</span>lik.anc, <span class="at">piecol =</span> co, <span class="at">cex =</span> <span class="fl">0.75</span>)</span>
<span id="cb11-15"><a href="biogeography-and-macroevolution-in-r.html#cb11-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-16"><a href="biogeography-and-macroevolution-in-r.html#cb11-16" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code></pre></div>
<p></br></p>
<div id="exercise-3-2" class="section level3 hasAnchor" number="6.5.1">
<h3><span class="header-section-number">6.5.1</span> Exercise (3)<a href="biogeography-and-macroevolution-in-r.html#exercise-3-2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Download the tree and data files with your first name (e.g., “Daniel_tree.nex” and “Daniel_data.csv”) from BrightSpace.
Using these data:</p>
<ul>
<li><p>Calculate the mean and standard deviation of species’ (t1, t2… t10) weight separately for:<br />
</p>
<ol style="list-style-type: lower-alpha">
<li>all species;<br />
</li>
<li>native species; and<br />
</li>
<li>introduced species<br />
</li>
</ol>
<p>*If you want an extra challenge, merge your answers into a dataframe and save as a csv file (see example_csv.csv in the lab 7 folder).
If not, just copy and paste your answers into the template for assignments and comment them out.</p></li>
<li><p>Reconstruct the ancestral ranges of the species on your tree after assigning 3 different continents to your data: Africa, Asia, Europe.
Plot your results on the tree as we did above and save the plot as a pdf file named “firstname_lastname_anc.pdf”.</p></li>
</ul>
<p>Write your code and answers below:</p>
<hr />
<hr />
<hr />
<hr />
<p></br></p>
<p><strong>Assignment [7]: Upload the code you wrote for the two exercises and their results to BrightSpace (lab7).</strong></p>
<p></br></p>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="tree-and-traits-in-r.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="phylogenetic-diversity-in-r.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/05_R_lab4_biogeo.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["bookdown-demo.pdf", "bookdown-demo.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
