<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 18 Character evolution and Ancestral state reconstruction | BIO595 The Tree of Life</title>
  <meta name="description" content="Test writing the book for the course bookdown::gitbook." />
  <meta name="generator" content="bookdown 0.28 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 18 Character evolution and Ancestral state reconstruction | BIO595 The Tree of Life" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="Test writing the book for the course bookdown::gitbook." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 18 Character evolution and Ancestral state reconstruction | BIO595 The Tree of Life" />
  
  <meta name="twitter:description" content="Test writing the book for the course bookdown::gitbook." />
  

<meta name="author" content="Qixin He, Daniel Park" />


<meta name="date" content="2023-04-10" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="molecular-clock-and-phylogenetic-dating.html"/>
<link rel="next" href="model-selection-of-character-evolution.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Tree of Life: Phylogenetic inferences and applications</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Overview</a></li>
<li class="chapter" data-level="2" data-path="unix.html"><a href="unix.html"><i class="fa fa-check"></i><b>2</b> Unix</a>
<ul>
<li class="chapter" data-level="2.1" data-path="unix.html"><a href="unix.html#what-is-unix-and-why-do-we-use-it"><i class="fa fa-check"></i><b>2.1</b> What is Unix and why do we use it?</a></li>
<li class="chapter" data-level="2.2" data-path="unix.html"><a href="unix.html#installation"><i class="fa fa-check"></i><b>2.2</b> Installation</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="unix.html"><a href="unix.html#mac-os-and-linux"><i class="fa fa-check"></i><b>2.2.1</b> Mac OS and Linux</a></li>
<li class="chapter" data-level="2.2.2" data-path="unix.html"><a href="unix.html#windows"><i class="fa fa-check"></i><b>2.2.2</b> Windows</a></li>
<li class="chapter" data-level="2.2.3" data-path="unix.html"><a href="unix.html#use-scholar-remote-desktop-during-the-class"><i class="fa fa-check"></i><b>2.2.3</b> Use Scholar Remote Desktop During the Class</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="unix.html"><a href="unix.html#operation"><i class="fa fa-check"></i><b>2.3</b> Operation</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="unix.html"><a href="unix.html#starting-the-shell"><i class="fa fa-check"></i><b>2.3.1</b> Starting the Shell</a></li>
<li class="chapter" data-level="2.3.2" data-path="unix.html"><a href="unix.html#directoryfolder-sturcture"><i class="fa fa-check"></i><b>2.3.2</b> Directory/folder sturcture</a></li>
<li class="chapter" data-level="2.3.3" data-path="unix.html"><a href="unix.html#absolute-and-relative-directory"><i class="fa fa-check"></i><b>2.3.3</b> Absolute and Relative directory</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="unix.html"><a href="unix.html#handling-directories-and-files"><i class="fa fa-check"></i><b>2.4</b> Handling directories and files</a></li>
<li class="chapter" data-level="2.5" data-path="unix.html"><a href="unix.html#excercises-i"><i class="fa fa-check"></i><b>2.5</b> Excercises I</a></li>
<li class="chapter" data-level="2.6" data-path="unix.html"><a href="unix.html#viewing-and-processing-text-files"><i class="fa fa-check"></i><b>2.6</b> Viewing and processing text files</a></li>
<li class="chapter" data-level="2.7" data-path="unix.html"><a href="unix.html#redirecting-printouts-and-piping"><i class="fa fa-check"></i><b>2.7</b> Redirecting printouts and piping</a></li>
<li class="chapter" data-level="2.8" data-path="unix.html"><a href="unix.html#exercises-ii"><i class="fa fa-check"></i><b>2.8</b> Exercises II</a></li>
<li class="chapter" data-level="2.9" data-path="unix.html"><a href="unix.html#useful-wildcards"><i class="fa fa-check"></i><b>2.9</b> Useful wildcards</a></li>
<li class="chapter" data-level="2.10" data-path="unix.html"><a href="unix.html#exercises-iii"><i class="fa fa-check"></i><b>2.10</b> Exercises III</a></li>
<li class="chapter" data-level="2.11" data-path="unix.html"><a href="unix.html#search-and-select-lines-using-grep"><i class="fa fa-check"></i><b>2.11</b> Search and select lines using <em>grep</em></a></li>
<li class="chapter" data-level="2.12" data-path="unix.html"><a href="unix.html#substituting-characters-using-tr"><i class="fa fa-check"></i><b>2.12</b> Substituting characters using <em>tr</em></a></li>
<li class="chapter" data-level="2.13" data-path="unix.html"><a href="unix.html#cut-sort-and-unique"><i class="fa fa-check"></i><b>2.13</b> cut, sort and unique</a></li>
<li class="chapter" data-level="2.14" data-path="unix.html"><a href="unix.html#exercises-iv"><i class="fa fa-check"></i><b>2.14</b> Exercises IV</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="r-basics-i.html"><a href="r-basics-i.html"><i class="fa fa-check"></i><b>3</b> R Basics I</a>
<ul>
<li class="chapter" data-level="3.1" data-path="r-basics-i.html"><a href="r-basics-i.html#what-is-programming"><i class="fa fa-check"></i><b>3.1</b> What is programming?</a></li>
<li class="chapter" data-level="3.2" data-path="r-basics-i.html"><a href="r-basics-i.html#r"><i class="fa fa-check"></i><b>3.2</b> <code>R</code></a></li>
<li class="chapter" data-level="3.3" data-path="r-basics-i.html"><a href="r-basics-i.html#installation-1"><i class="fa fa-check"></i><b>3.3</b> Installation</a></li>
<li class="chapter" data-level="3.4" data-path="r-basics-i.html"><a href="r-basics-i.html#variables"><i class="fa fa-check"></i><b>3.4</b> Variables</a>
<ul>
<li class="chapter" data-level="3.4.1" data-path="r-basics-i.html"><a href="r-basics-i.html#exercise-1"><i class="fa fa-check"></i><b>3.4.1</b> Exercise (1):</a></li>
<li class="chapter" data-level="3.4.2" data-path="r-basics-i.html"><a href="r-basics-i.html#variable-naming-conventions"><i class="fa fa-check"></i><b>3.4.2</b> Variable naming conventions</a></li>
<li class="chapter" data-level="3.4.3" data-path="r-basics-i.html"><a href="r-basics-i.html#arithmetic-operators"><i class="fa fa-check"></i><b>3.4.3</b> Arithmetic Operators</a></li>
<li class="chapter" data-level="3.4.4" data-path="r-basics-i.html"><a href="r-basics-i.html#logical-operators"><i class="fa fa-check"></i><b>3.4.4</b> Logical Operators</a></li>
<li class="chapter" data-level="3.4.5" data-path="r-basics-i.html"><a href="r-basics-i.html#basic-built-in-mathematical-functions"><i class="fa fa-check"></i><b>3.4.5</b> Basic built-in mathematical functions</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="r-basics-i.html"><a href="r-basics-i.html#subsetting-data"><i class="fa fa-check"></i><b>3.5</b> Subsetting data</a></li>
<li class="chapter" data-level="3.6" data-path="r-basics-i.html"><a href="r-basics-i.html#logical-values"><i class="fa fa-check"></i><b>3.6</b> Logical values</a>
<ul>
<li class="chapter" data-level="3.6.1" data-path="r-basics-i.html"><a href="r-basics-i.html#missing-dealing-with-na-values"><i class="fa fa-check"></i><b>3.6.1</b> Missing: dealing with <code>NA</code> values</a></li>
</ul></li>
<li class="chapter" data-level="3.7" data-path="r-basics-i.html"><a href="r-basics-i.html#generating-data"><i class="fa fa-check"></i><b>3.7</b> Generating Data</a></li>
<li class="chapter" data-level="3.8" data-path="r-basics-i.html"><a href="r-basics-i.html#data-structures-and-types"><i class="fa fa-check"></i><b>3.8</b> Data Structures and Types</a>
<ul>
<li class="chapter" data-level="3.8.1" data-path="r-basics-i.html"><a href="r-basics-i.html#vectors"><i class="fa fa-check"></i><b>3.8.1</b> Vectors</a></li>
</ul></li>
<li class="chapter" data-level="3.9" data-path="r-basics-i.html"><a href="r-basics-i.html#matrices"><i class="fa fa-check"></i><b>3.9</b> Matrices</a>
<ul>
<li class="chapter" data-level="3.9.1" data-path="r-basics-i.html"><a href="r-basics-i.html#exercise-2"><i class="fa fa-check"></i><b>3.9.1</b> Exercise (2):</a></li>
</ul></li>
<li class="chapter" data-level="3.10" data-path="r-basics-i.html"><a href="r-basics-i.html#data-frames"><i class="fa fa-check"></i><b>3.10</b> Data Frames</a>
<ul>
<li class="chapter" data-level="3.10.1" data-path="r-basics-i.html"><a href="r-basics-i.html#exercise-3"><i class="fa fa-check"></i><b>3.10.1</b> Exercise (3):</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="r-basics-ii.html"><a href="r-basics-ii.html"><i class="fa fa-check"></i><b>4</b> R Basics II</a>
<ul>
<li class="chapter" data-level="4.1" data-path="r-basics-ii.html"><a href="r-basics-ii.html#setting-up-the-environment"><i class="fa fa-check"></i><b>4.1</b> Setting up the environment</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="r-basics-ii.html"><a href="r-basics-ii.html#exercise-1-1"><i class="fa fa-check"></i><b>4.1.1</b> Exercise (1)</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="r-basics-ii.html"><a href="r-basics-ii.html#reading-in-data-from-files"><i class="fa fa-check"></i><b>4.2</b> Reading in data from files</a></li>
<li class="chapter" data-level="4.3" data-path="r-basics-ii.html"><a href="r-basics-ii.html#wrangling-data"><i class="fa fa-check"></i><b>4.3</b> Wrangling Data</a></li>
<li class="chapter" data-level="4.4" data-path="r-basics-ii.html"><a href="r-basics-ii.html#subsetting-by-columns-and-rows"><i class="fa fa-check"></i><b>4.4</b> Subsetting by columns and rows</a></li>
<li class="chapter" data-level="4.5" data-path="r-basics-ii.html"><a href="r-basics-ii.html#subsetting-with-conditions"><i class="fa fa-check"></i><b>4.5</b> Subsetting with conditions</a>
<ul>
<li class="chapter" data-level="4.5.1" data-path="r-basics-ii.html"><a href="r-basics-ii.html#exercise-2-1"><i class="fa fa-check"></i><b>4.5.1</b> Exercise (2)</a></li>
<li class="chapter" data-level="4.5.2" data-path="r-basics-ii.html"><a href="r-basics-ii.html#bonus-1"><i class="fa fa-check"></i><b>4.5.2</b> Bonus [1]</a></li>
</ul></li>
<li class="chapter" data-level="4.6" data-path="r-basics-ii.html"><a href="r-basics-ii.html#merging-data"><i class="fa fa-check"></i><b>4.6</b> Merging data</a></li>
<li class="chapter" data-level="4.7" data-path="r-basics-ii.html"><a href="r-basics-ii.html#saving-files-and-sessions"><i class="fa fa-check"></i><b>4.7</b> Saving files and sessions</a>
<ul>
<li class="chapter" data-level="4.7.1" data-path="r-basics-ii.html"><a href="r-basics-ii.html#exercise-3-1"><i class="fa fa-check"></i><b>4.7.1</b> Exercise (3)</a></li>
</ul></li>
<li class="chapter" data-level="4.8" data-path="r-basics-ii.html"><a href="r-basics-ii.html#the-for-loop"><i class="fa fa-check"></i><b>4.8</b> The <span class="smallcaps">for</span>-loop</a></li>
<li class="chapter" data-level="4.9" data-path="r-basics-ii.html"><a href="r-basics-ii.html#lists"><i class="fa fa-check"></i><b>4.9</b> Lists</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="tree-and-traits-in-r.html"><a href="tree-and-traits-in-r.html"><i class="fa fa-check"></i><b>5</b> Tree and Traits in R</a>
<ul>
<li class="chapter" data-level="5.1" data-path="tree-and-traits-in-r.html"><a href="tree-and-traits-in-r.html#setting-up-the-environment-1"><i class="fa fa-check"></i><b>5.1</b> Setting up the environment</a></li>
<li class="chapter" data-level="5.2" data-path="tree-and-traits-in-r.html"><a href="tree-and-traits-in-r.html#examining-phylogenetic-trees"><i class="fa fa-check"></i><b>5.2</b> Examining phylogenetic trees</a></li>
<li class="chapter" data-level="5.3" data-path="tree-and-traits-in-r.html"><a href="tree-and-traits-in-r.html#plotting-phylogenetic-trees"><i class="fa fa-check"></i><b>5.3</b> Plotting phylogenetic trees</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="tree-and-traits-in-r.html"><a href="tree-and-traits-in-r.html#exercise-1-2"><i class="fa fa-check"></i><b>5.3.1</b> Exercise (1)</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="tree-and-traits-in-r.html"><a href="tree-and-traits-in-r.html#associating-phylogenies-and-traits"><i class="fa fa-check"></i><b>5.4</b> Associating phylogenies and traits</a>
<ul>
<li class="chapter" data-level="5.4.1" data-path="tree-and-traits-in-r.html"><a href="tree-and-traits-in-r.html#saving-plots"><i class="fa fa-check"></i><b>5.4.1</b> Saving plots</a></li>
<li class="chapter" data-level="5.4.2" data-path="tree-and-traits-in-r.html"><a href="tree-and-traits-in-r.html#exercise-2-2"><i class="fa fa-check"></i><b>5.4.2</b> Exercise (2)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html"><i class="fa fa-check"></i><b>6</b> Biogeography and Macroevolution in R</a>
<ul>
<li class="chapter" data-level="6.1" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#setting-up-the-environment-2"><i class="fa fa-check"></i><b>6.1</b> Setting up the environment</a></li>
<li class="chapter" data-level="6.2" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#ancestral-state-reconstructions"><i class="fa fa-check"></i><b>6.2</b> Ancestral State Reconstructions</a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#bonus-2"><i class="fa fa-check"></i><b>6.2.1</b> Bonus [2]</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#continuous-traits"><i class="fa fa-check"></i><b>6.3</b> Continuous traits</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#maximum-likelihood"><i class="fa fa-check"></i><b>6.3.1</b> Maximum likelihood</a></li>
<li class="chapter" data-level="6.3.2" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#phylogenetically-independent-contrasts"><i class="fa fa-check"></i><b>6.3.2</b> Phylogenetically independent contrasts</a></li>
<li class="chapter" data-level="6.3.3" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#plotting-ancestral-states"><i class="fa fa-check"></i><b>6.3.3</b> Plotting ancestral states</a></li>
<li class="chapter" data-level="6.3.4" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#exercise-1-3"><i class="fa fa-check"></i><b>6.3.4</b> Exercise (1)</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#discrete-traits"><i class="fa fa-check"></i><b>6.4</b> Discrete traits</a>
<ul>
<li class="chapter" data-level="6.4.1" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#exercise-2-3"><i class="fa fa-check"></i><b>6.4.1</b> Exercise (2)</a></li>
<li class="chapter" data-level="6.4.2" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#maximum-likelihood-1"><i class="fa fa-check"></i><b>6.4.2</b> Maximum likelihood</a></li>
<li class="chapter" data-level="6.4.3" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#choosing-models"><i class="fa fa-check"></i><b>6.4.3</b> Choosing models</a></li>
</ul></li>
<li class="chapter" data-level="6.5" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#saving-multiple-plots-together"><i class="fa fa-check"></i><b>6.5</b> Saving multiple plots together</a>
<ul>
<li class="chapter" data-level="6.5.1" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#exercise-3-2"><i class="fa fa-check"></i><b>6.5.1</b> Exercise (3)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="phylogenetic-diversity-in-r.html"><a href="phylogenetic-diversity-in-r.html"><i class="fa fa-check"></i><b>7</b> Phylogenetic Diversity in R</a>
<ul>
<li class="chapter" data-level="7.1" data-path="phylogenetic-diversity-in-r.html"><a href="phylogenetic-diversity-in-r.html#setting-up-the-environment-3"><i class="fa fa-check"></i><b>7.1</b> Setting up the environment</a></li>
<li class="chapter" data-level="7.2" data-path="phylogenetic-diversity-in-r.html"><a href="phylogenetic-diversity-in-r.html#phylogenetic-diversity"><i class="fa fa-check"></i><b>7.2</b> Phylogenetic Diversity</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="phylogenetic-diversity-in-r.html"><a href="phylogenetic-diversity-in-r.html#preprocessing"><i class="fa fa-check"></i><b>7.2.1</b> Preprocessing</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="phylogenetic-diversity-in-r.html"><a href="phylogenetic-diversity-in-r.html#estimating-phylogenetic-diversity-metrics"><i class="fa fa-check"></i><b>7.3</b> Estimating phylogenetic diversity metrics</a>
<ul>
<li class="chapter" data-level="7.3.1" data-path="phylogenetic-diversity-in-r.html"><a href="phylogenetic-diversity-in-r.html#exercise-1-4"><i class="fa fa-check"></i><b>7.3.1</b> Exercise (1)</a></li>
<li class="chapter" data-level="7.3.2" data-path="phylogenetic-diversity-in-r.html"><a href="phylogenetic-diversity-in-r.html#exercise-2-4"><i class="fa fa-check"></i><b>7.3.2</b> Exercise (2)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="community-phylogenetics.html"><a href="community-phylogenetics.html"><i class="fa fa-check"></i><b>8</b> Community Phylogenetics</a>
<ul>
<li class="chapter" data-level="8.1" data-path="community-phylogenetics.html"><a href="community-phylogenetics.html#setting-up-the-environment-4"><i class="fa fa-check"></i><b>8.1</b> Setting up the environment</a></li>
<li class="chapter" data-level="8.2" data-path="community-phylogenetics.html"><a href="community-phylogenetics.html#definition-of-community-phylogenetics"><i class="fa fa-check"></i><b>8.2</b> Definition of Community Phylogenetics</a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="community-phylogenetics.html"><a href="community-phylogenetics.html#alpha-diversity"><i class="fa fa-check"></i><b>8.2.1</b> Alpha diversity</a></li>
<li class="chapter" data-level="8.2.2" data-path="community-phylogenetics.html"><a href="community-phylogenetics.html#beta-diversity"><i class="fa fa-check"></i><b>8.2.2</b> Beta diversity</a></li>
<li class="chapter" data-level="8.2.3" data-path="community-phylogenetics.html"><a href="community-phylogenetics.html#exercise-1-5"><i class="fa fa-check"></i><b>8.2.3</b> Exercise (1)</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="community-phylogenetics.html"><a href="community-phylogenetics.html#standardized-effect-sizes"><i class="fa fa-check"></i><b>8.3</b> Standardized effect sizes</a>
<ul>
<li class="chapter" data-level="8.3.1" data-path="community-phylogenetics.html"><a href="community-phylogenetics.html#exercise-2-5"><i class="fa fa-check"></i><b>8.3.1</b> Exercise (2)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="coevolution-and-phylogenetic-signals-in-r.html"><a href="coevolution-and-phylogenetic-signals-in-r.html"><i class="fa fa-check"></i><b>9</b> Coevolution and Phylogenetic Signals in R</a>
<ul>
<li class="chapter" data-level="9.1" data-path="coevolution-and-phylogenetic-signals-in-r.html"><a href="coevolution-and-phylogenetic-signals-in-r.html#setting-up-the-environment-5"><i class="fa fa-check"></i><b>9.1</b> Setting up the environment</a></li>
<li class="chapter" data-level="9.2" data-path="coevolution-and-phylogenetic-signals-in-r.html"><a href="coevolution-and-phylogenetic-signals-in-r.html#phylogenetic-signal"><i class="fa fa-check"></i><b>9.2</b> Phylogenetic Signal</a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="coevolution-and-phylogenetic-signals-in-r.html"><a href="coevolution-and-phylogenetic-signals-in-r.html#blombergs-k"><i class="fa fa-check"></i><b>9.2.1</b> Blomberg’s K</a></li>
<li class="chapter" data-level="9.2.2" data-path="coevolution-and-phylogenetic-signals-in-r.html"><a href="coevolution-and-phylogenetic-signals-in-r.html#pagels-lambda-λ"><i class="fa fa-check"></i><b>9.2.2</b> Pagel’s lambda (λ)</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="coevolution-and-phylogenetic-signals-in-r.html"><a href="coevolution-and-phylogenetic-signals-in-r.html#phylogenetically-independent-contrasts-1"><i class="fa fa-check"></i><b>9.3</b> Phylogenetically Independent Contrasts</a></li>
<li class="chapter" data-level="9.4" data-path="coevolution-and-phylogenetic-signals-in-r.html"><a href="coevolution-and-phylogenetic-signals-in-r.html#coevolution"><i class="fa fa-check"></i><b>9.4</b> Coevolution</a>
<ul>
<li class="chapter" data-level="9.4.1" data-path="coevolution-and-phylogenetic-signals-in-r.html"><a href="coevolution-and-phylogenetic-signals-in-r.html#plotting-tanglegrams"><i class="fa fa-check"></i><b>9.4.1</b> Plotting tanglegrams</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="phylogenetic-visualization-in-r.html"><a href="phylogenetic-visualization-in-r.html"><i class="fa fa-check"></i><b>10</b> Phylogenetic Visualization in R</a>
<ul>
<li class="chapter" data-level="10.1" data-path="phylogenetic-visualization-in-r.html"><a href="phylogenetic-visualization-in-r.html#setting-up-the-environment-6"><i class="fa fa-check"></i><b>10.1</b> Setting up the environment</a></li>
<li class="chapter" data-level="10.2" data-path="phylogenetic-visualization-in-r.html"><a href="phylogenetic-visualization-in-r.html#using-phytools"><i class="fa fa-check"></i><b>10.2</b> Using <code>phytools</code></a>
<ul>
<li class="chapter" data-level="10.2.1" data-path="phylogenetic-visualization-in-r.html"><a href="phylogenetic-visualization-in-r.html#phenograms"><i class="fa fa-check"></i><b>10.2.1</b> Phenograms</a></li>
<li class="chapter" data-level="10.2.2" data-path="phylogenetic-visualization-in-r.html"><a href="phylogenetic-visualization-in-r.html#mapping-continuous-characters-on-branches"><i class="fa fa-check"></i><b>10.2.2</b> Mapping continuous characters on branches</a></li>
<li class="chapter" data-level="10.2.3" data-path="phylogenetic-visualization-in-r.html"><a href="phylogenetic-visualization-in-r.html#plotting-images-on-phylogenies"><i class="fa fa-check"></i><b>10.2.3</b> Plotting images on phylogenies</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="phylogenetic-visualization-in-r.html"><a href="phylogenetic-visualization-in-r.html#using-ggtree"><i class="fa fa-check"></i><b>10.3</b> Using <code>ggtree</code></a>
<ul>
<li class="chapter" data-level="10.3.1" data-path="phylogenetic-visualization-in-r.html"><a href="phylogenetic-visualization-in-r.html#basic-plotting"><i class="fa fa-check"></i><b>10.3.1</b> Basic plotting</a></li>
<li class="chapter" data-level="10.3.2" data-path="phylogenetic-visualization-in-r.html"><a href="phylogenetic-visualization-in-r.html#labeling-tips-nodes-and-clades"><i class="fa fa-check"></i><b>10.3.2</b> Labeling tips, nodes, and clades</a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="phylogenetic-visualization-in-r.html"><a href="phylogenetic-visualization-in-r.html#plotting-maps"><i class="fa fa-check"></i><b>10.4</b> Plotting maps</a>
<ul>
<li class="chapter" data-level="10.4.1" data-path="phylogenetic-visualization-in-r.html"><a href="phylogenetic-visualization-in-r.html#choropleth-maps"><i class="fa fa-check"></i><b>10.4.1</b> Choropleth maps</a></li>
<li class="chapter" data-level="10.4.2" data-path="phylogenetic-visualization-in-r.html"><a href="phylogenetic-visualization-in-r.html#working-with-rasters"><i class="fa fa-check"></i><b>10.4.2</b> Working with rasters</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html"><i class="fa fa-check"></i><b>11</b> Gene tree vs Species tree</a>
<ul>
<li class="chapter" data-level="11.1" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#learning-goals"><i class="fa fa-check"></i><b>11.1</b> Learning goals:</a></li>
<li class="chapter" data-level="11.2" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#tracing-ancestry-within-one-population"><i class="fa fa-check"></i><b>11.2</b> Tracing ancestry within one population</a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#vary-pop-size-and-generation"><i class="fa fa-check"></i><b>11.2.1</b> Vary pop size and generation</a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#speciation-and-coalescence"><i class="fa fa-check"></i><b>11.3</b> Speciation and coalescence</a>
<ul>
<li class="chapter" data-level="11.3.1" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#two-species-coalescent-history"><i class="fa fa-check"></i><b>11.3.1</b> Two-species coalescent history</a></li>
</ul></li>
<li class="chapter" data-level="11.4" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#exercise-i"><i class="fa fa-check"></i><b>11.4</b> Exercise I</a></li>
<li class="chapter" data-level="11.5" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#gene-tree-locus-tree-and-species-tree"><i class="fa fa-check"></i><b>11.5</b> Gene tree, Locus tree and Species tree</a>
<ul>
<li class="chapter" data-level="11.5.1" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#loading-data-and-packages"><i class="fa fa-check"></i><b>11.5.1</b> loading data and packages</a></li>
</ul></li>
<li class="chapter" data-level="11.6" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#effects-of-incomplete-lineage-sorting-ils-alone"><i class="fa fa-check"></i><b>11.6</b> Effects of Incomplete Lineage Sorting (ILS) alone</a>
<ul>
<li class="chapter" data-level="11.6.1" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#generate-species-tree-with-no-extinction"><i class="fa fa-check"></i><b>11.6.1</b> generate species tree with no extinction</a></li>
<li class="chapter" data-level="11.6.2" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#simulate-gene-trees-using-the-multispecies-coalescent-process"><i class="fa fa-check"></i><b>11.6.2</b> simulate gene trees using the multispecies coalescent process</a></li>
</ul></li>
<li class="chapter" data-level="11.7" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#calculate-tree-distances"><i class="fa fa-check"></i><b>11.7</b> Calculate tree distances</a>
<ul>
<li class="chapter" data-level="11.7.1" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#rf-distances"><i class="fa fa-check"></i><b>11.7.1</b> RF distances</a></li>
<li class="chapter" data-level="11.7.2" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#inspect-pairwise-rf-distances-among-all-the-gene-trees-generated-under-one-guiding-species-tree"><i class="fa fa-check"></i><b>11.7.2</b> Inspect pairwise RF distances among all the gene trees generated under one guiding species tree</a></li>
</ul></li>
<li class="chapter" data-level="11.8" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#inspect-node-support-i.e.-subtrees-defined-by-a-split"><i class="fa fa-check"></i><b>11.8</b> Inspect node support (i.e., subtrees defined by a split)</a></li>
<li class="chapter" data-level="11.9" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#combined-effects-of-gene-duplication-loss-and-horizontal-gene-transfer-events"><i class="fa fa-check"></i><b>11.9</b> Combined effects of gene duplication, loss and horizontal gene transfer events</a></li>
<li class="chapter" data-level="11.10" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#exercise-ii"><i class="fa fa-check"></i><b>11.10</b> Exercise II</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="overview-of-phylogenetic-inferences.html"><a href="overview-of-phylogenetic-inferences.html"><i class="fa fa-check"></i><b>12</b> Overview of Phylogenetic Inferences</a>
<ul>
<li class="chapter" data-level="12.1" data-path="overview-of-phylogenetic-inferences.html"><a href="overview-of-phylogenetic-inferences.html#mega"><i class="fa fa-check"></i><b>12.1</b> <code>MEGA</code></a></li>
<li class="chapter" data-level="12.2" data-path="overview-of-phylogenetic-inferences.html"><a href="overview-of-phylogenetic-inferences.html#view-and-align-sequences"><i class="fa fa-check"></i><b>12.2</b> View and Align Sequences</a></li>
<li class="chapter" data-level="12.3" data-path="overview-of-phylogenetic-inferences.html"><a href="overview-of-phylogenetic-inferences.html#dnaprotein-evolution-model"><i class="fa fa-check"></i><b>12.3</b> DNA/protein evolution model</a></li>
<li class="chapter" data-level="12.4" data-path="overview-of-phylogenetic-inferences.html"><a href="overview-of-phylogenetic-inferences.html#phylogenetic-inference"><i class="fa fa-check"></i><b>12.4</b> Phylogenetic inference</a></li>
<li class="chapter" data-level="12.5" data-path="overview-of-phylogenetic-inferences.html"><a href="overview-of-phylogenetic-inferences.html#estimate-divergence-times"><i class="fa fa-check"></i><b>12.5</b> Estimate divergence times</a></li>
<li class="chapter" data-level="12.6" data-path="overview-of-phylogenetic-inferences.html"><a href="overview-of-phylogenetic-inferences.html#exercises"><i class="fa fa-check"></i><b>12.6</b> Exercises</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html"><i class="fa fa-check"></i><b>13</b> Resources and Guidelines for the Term Project</a>
<ul>
<li class="chapter" data-level="13.1" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#bioinformatics-databases"><i class="fa fa-check"></i><b>13.1</b> Bioinformatics Databases</a>
<ul>
<li class="chapter" data-level="13.1.1" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#ncbi-national-center-for-biotechnology-information"><i class="fa fa-check"></i><b>13.1.1</b> NCBI (National Center for Biotechnology Information)</a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#retrieving-genetic-resources-for-your-taxa"><i class="fa fa-check"></i><b>13.2</b> Retrieving genetic resources for your taxa</a>
<ul>
<li class="chapter" data-level="13.2.1" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#ncbi-taxonomy-database"><i class="fa fa-check"></i><b>13.2.1</b> NCBI taxonomy database</a></li>
<li class="chapter" data-level="13.2.2" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#individual-genes-from-the-whole-genome-data"><i class="fa fa-check"></i><b>13.2.2</b> Individual genes from the whole genome data</a></li>
<li class="chapter" data-level="13.2.3" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#get-homologous-sequences-from-related-species"><i class="fa fa-check"></i><b>13.2.3</b> Get homologous sequences from related species</a></li>
<li class="chapter" data-level="13.2.4" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#european-bioinformatics-institute-ebi"><i class="fa fa-check"></i><b>13.2.4</b> European Bioinformatics Institute (EBI)</a></li>
<li class="chapter" data-level="13.2.5" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#protein-data-bank"><i class="fa fa-check"></i><b>13.2.5</b> Protein Data Bank</a></li>
</ul></li>
<li class="chapter" data-level="13.3" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#biogeographic-databases"><i class="fa fa-check"></i><b>13.3</b> Biogeographic Databases</a>
<ul>
<li class="chapter" data-level="13.3.1" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#climate-data"><i class="fa fa-check"></i><b>13.3.1</b> Climate Data</a></li>
<li class="chapter" data-level="13.3.2" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#occurrence-data"><i class="fa fa-check"></i><b>13.3.2</b> Occurrence Data</a></li>
<li class="chapter" data-level="13.3.3" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#topographic-data"><i class="fa fa-check"></i><b>13.3.3</b> TOPOGRAPHIC DATA</a></li>
<li class="chapter" data-level="13.3.4" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#trait-data"><i class="fa fa-check"></i><b>13.3.4</b> TRAIT DATA</a></li>
</ul></li>
<li class="chapter" data-level="13.4" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#publication-databases"><i class="fa fa-check"></i><b>13.4</b> Publication Databases</a></li>
<li class="chapter" data-level="13.5" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#example-research-questions"><i class="fa fa-check"></i><b>13.5</b> Example research questions:</a>
<ul>
<li class="chapter" data-level="13.5.1" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#biogeography-and-phylogenetic-diversity"><i class="fa fa-check"></i><b>13.5.1</b> Biogeography and Phylogenetic Diversity:</a></li>
<li class="chapter" data-level="13.5.2" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#community-ecology"><i class="fa fa-check"></i><b>13.5.2</b> Community Ecology:</a></li>
<li class="chapter" data-level="13.5.3" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#coevolution-1"><i class="fa fa-check"></i><b>13.5.3</b> Coevolution:</a></li>
<li class="chapter" data-level="13.5.4" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#species-tree"><i class="fa fa-check"></i><b>13.5.4</b> Species tree:</a></li>
</ul></li>
<li class="chapter" data-level="13.6" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#how-to-submit-jobs-on-scholar-cluster-and-run-remotely"><i class="fa fa-check"></i><b>13.6</b> How to submit jobs on Scholar Cluster and run remotely</a></li>
<li class="chapter" data-level="13.7" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#term-project-proposal"><i class="fa fa-check"></i><b>13.7</b> Term project proposal</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><i class="fa fa-check"></i><b>14</b> Systematic Retrieval of orthologous DNA sequences form GenBank</a>
<ul>
<li class="chapter" data-level="14.1" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#introduction"><i class="fa fa-check"></i><b>14.1</b> Introduction</a></li>
<li class="chapter" data-level="14.2" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#installing-tools"><i class="fa fa-check"></i><b>14.2</b> Installing Tools</a>
<ul>
<li class="chapter" data-level="14.2.1" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#windows-1"><i class="fa fa-check"></i><b>14.2.1</b> Windows</a></li>
<li class="chapter" data-level="14.2.2" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#mac"><i class="fa fa-check"></i><b>14.2.2</b> Mac</a></li>
<li class="chapter" data-level="14.2.3" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#ncbi-blast-tools"><i class="fa fa-check"></i><b>14.2.3</b> NCBI BLAST+ Tools</a></li>
</ul></li>
<li class="chapter" data-level="14.3" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#using-phylotar"><i class="fa fa-check"></i><b>14.3</b> Using <code>phylotaR</code></a>
<ul>
<li class="chapter" data-level="14.3.1" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#pipeline"><i class="fa fa-check"></i><b>14.3.1</b> Pipeline</a></li>
<li class="chapter" data-level="14.3.2" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#running"><i class="fa fa-check"></i><b>14.3.2</b> Running</a></li>
<li class="chapter" data-level="14.3.3" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#restarting"><i class="fa fa-check"></i><b>14.3.3</b> Restarting</a></li>
<li class="chapter" data-level="14.3.4" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#cluster-i.e.-orthologous-genes-selection"><i class="fa fa-check"></i><b>14.3.4</b> Cluster (i.e., orthologous genes) selection</a></li>
</ul></li>
<li class="chapter" data-level="14.4" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#exercise-i-1"><i class="fa fa-check"></i><b>14.4</b> Exercise I</a></li>
<li class="chapter" data-level="14.5" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#using-annotationbustr"><i class="fa fa-check"></i><b>14.5</b> Using <code>AnnotationBustR</code></a>
<ul>
<li class="chapter" data-level="14.5.1" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#introduction-1"><i class="fa fa-check"></i><b>14.5.1</b> Introduction</a></li>
<li class="chapter" data-level="14.5.2" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#install-the-tools"><i class="fa fa-check"></i><b>14.5.2</b> Install the tools</a></li>
<li class="chapter" data-level="14.5.3" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#specify-search-terms-of-genomes"><i class="fa fa-check"></i><b>14.5.3</b> Specify search terms of genomes</a></li>
<li class="chapter" data-level="14.5.4" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#load-gene-search-terms-of-interest"><i class="fa fa-check"></i><b>14.5.4</b> Load Gene Search Terms of Interest</a></li>
<li class="chapter" data-level="14.5.5" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#extract-sequences-with-annotationbust"><i class="fa fa-check"></i><b>14.5.5</b> Extract sequences with AnnotationBust</a></li>
</ul></li>
<li class="chapter" data-level="14.6" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#exercise-ii-1"><i class="fa fa-check"></i><b>14.6</b> Exercise II</a></li>
<li class="chapter" data-level="14.7" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#combine-gene-information-from-the-two-analyses"><i class="fa fa-check"></i><b>14.7</b> Combine gene information from the two analyses</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="sequence-alignment.html"><a href="sequence-alignment.html"><i class="fa fa-check"></i><b>15</b> Sequence alignment</a>
<ul>
<li class="chapter" data-level="15.1" data-path="sequence-alignment.html"><a href="sequence-alignment.html#load-libraries"><i class="fa fa-check"></i><b>15.1</b> Load libraries</a></li>
<li class="chapter" data-level="15.2" data-path="sequence-alignment.html"><a href="sequence-alignment.html#perform-alignments"><i class="fa fa-check"></i><b>15.2</b> Perform Alignments</a></li>
<li class="chapter" data-level="15.3" data-path="sequence-alignment.html"><a href="sequence-alignment.html#processing-multiple-alignments"><i class="fa fa-check"></i><b>15.3</b> Processing Multiple Alignments</a></li>
<li class="chapter" data-level="15.4" data-path="sequence-alignment.html"><a href="sequence-alignment.html#build-your-first-nj-tree"><i class="fa fa-check"></i><b>15.4</b> Build your first NJ tree</a></li>
<li class="chapter" data-level="15.5" data-path="sequence-alignment.html"><a href="sequence-alignment.html#exercise"><i class="fa fa-check"></i><b>15.5</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="maximum-likelihood-model-tests-tree-and-branch-tests.html"><a href="maximum-likelihood-model-tests-tree-and-branch-tests.html"><i class="fa fa-check"></i><b>16</b> Maximum likelihood, model tests, tree and branch tests</a>
<ul>
<li class="chapter" data-level="16.1" data-path="maximum-likelihood-model-tests-tree-and-branch-tests.html"><a href="maximum-likelihood-model-tests-tree-and-branch-tests.html#learning-goals-1"><i class="fa fa-check"></i><b>16.1</b> Learning Goals</a></li>
<li class="chapter" data-level="16.2" data-path="maximum-likelihood-model-tests-tree-and-branch-tests.html"><a href="maximum-likelihood-model-tests-tree-and-branch-tests.html#iqtree"><i class="fa fa-check"></i><b>16.2</b> IQtree</a>
<ul>
<li class="chapter" data-level="16.2.1" data-path="maximum-likelihood-model-tests-tree-and-branch-tests.html"><a href="maximum-likelihood-model-tests-tree-and-branch-tests.html#version"><i class="fa fa-check"></i><b>16.2.1</b> Version</a></li>
<li class="chapter" data-level="16.2.2" data-path="maximum-likelihood-model-tests-tree-and-branch-tests.html"><a href="maximum-likelihood-model-tests-tree-and-branch-tests.html#find-the-best-explanatory-sequence-evolution-model"><i class="fa fa-check"></i><b>16.2.2</b> find the best explanatory sequence evolution model</a></li>
<li class="chapter" data-level="16.2.3" data-path="maximum-likelihood-model-tests-tree-and-branch-tests.html"><a href="maximum-likelihood-model-tests-tree-and-branch-tests.html#using-the-best-sequence-evolution-model"><i class="fa fa-check"></i><b>16.2.3</b> using the best sequence evolution model</a></li>
<li class="chapter" data-level="16.2.4" data-path="maximum-likelihood-model-tests-tree-and-branch-tests.html"><a href="maximum-likelihood-model-tests-tree-and-branch-tests.html#different-partitions-for-the-data-for-multiple-genes-or-different-codons"><i class="fa fa-check"></i><b>16.2.4</b> Different partitions for the data for multiple genes, or different codons</a></li>
</ul></li>
<li class="chapter" data-level="16.3" data-path="maximum-likelihood-model-tests-tree-and-branch-tests.html"><a href="maximum-likelihood-model-tests-tree-and-branch-tests.html#run-an-ml-tree-with-multiple-gene-files-in-a-folder"><i class="fa fa-check"></i><b>16.3</b> Run an ML tree with multiple gene files in a folder</a></li>
<li class="chapter" data-level="16.4" data-path="maximum-likelihood-model-tests-tree-and-branch-tests.html"><a href="maximum-likelihood-model-tests-tree-and-branch-tests.html#gene-concordance-factor"><i class="fa fa-check"></i><b>16.4</b> Gene concordance factor</a></li>
<li class="chapter" data-level="16.5" data-path="maximum-likelihood-model-tests-tree-and-branch-tests.html"><a href="maximum-likelihood-model-tests-tree-and-branch-tests.html#tree-topology-tests"><i class="fa fa-check"></i><b>16.5</b> Tree topology tests</a></li>
<li class="chapter" data-level="16.6" data-path="maximum-likelihood-model-tests-tree-and-branch-tests.html"><a href="maximum-likelihood-model-tests-tree-and-branch-tests.html#tree-branches-tests"><i class="fa fa-check"></i><b>16.6</b> Tree branches tests</a></li>
<li class="chapter" data-level="16.7" data-path="maximum-likelihood-model-tests-tree-and-branch-tests.html"><a href="maximum-likelihood-model-tests-tree-and-branch-tests.html#assignments"><i class="fa fa-check"></i><b>16.7</b> Assignments</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="molecular-clock-and-phylogenetic-dating.html"><a href="molecular-clock-and-phylogenetic-dating.html"><i class="fa fa-check"></i><b>17</b> Molecular Clock and Phylogenetic Dating</a>
<ul>
<li class="chapter" data-level="17.1" data-path="molecular-clock-and-phylogenetic-dating.html"><a href="molecular-clock-and-phylogenetic-dating.html#inferring-time-tree-with-tip-dates"><i class="fa fa-check"></i><b>17.1</b> Inferring time tree with tip dates</a></li>
<li class="chapter" data-level="17.2" data-path="molecular-clock-and-phylogenetic-dating.html"><a href="molecular-clock-and-phylogenetic-dating.html#calibrating-tree-using-ancestral-dates"><i class="fa fa-check"></i><b>17.2</b> Calibrating tree using ancestral dates</a>
<ul>
<li class="chapter" data-level="17.2.1" data-path="molecular-clock-and-phylogenetic-dating.html"><a href="molecular-clock-and-phylogenetic-dating.html#search-for-an-ml-tree"><i class="fa fa-check"></i><b>17.2.1</b> search for an ML tree</a></li>
<li class="chapter" data-level="17.2.2" data-path="molecular-clock-and-phylogenetic-dating.html"><a href="molecular-clock-and-phylogenetic-dating.html#root-calibration"><i class="fa fa-check"></i><b>17.2.2</b> Root calibration</a></li>
<li class="chapter" data-level="17.2.3" data-path="molecular-clock-and-phylogenetic-dating.html"><a href="molecular-clock-and-phylogenetic-dating.html#internal-nodes"><i class="fa fa-check"></i><b>17.2.3</b> Internal nodes</a></li>
<li class="chapter" data-level="17.2.4" data-path="molecular-clock-and-phylogenetic-dating.html"><a href="molecular-clock-and-phylogenetic-dating.html#confidence-interval"><i class="fa fa-check"></i><b>17.2.4</b> Confidence interval</a></li>
</ul></li>
<li class="chapter" data-level="17.3" data-path="molecular-clock-and-phylogenetic-dating.html"><a href="molecular-clock-and-phylogenetic-dating.html#exercise-4"><i class="fa fa-check"></i><b>17.3</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="character-evolution-and-ancestral-state-reconstruction.html"><a href="character-evolution-and-ancestral-state-reconstruction.html"><i class="fa fa-check"></i><b>18</b> Character evolution and Ancestral state reconstruction</a>
<ul>
<li class="chapter" data-level="18.1" data-path="character-evolution-and-ancestral-state-reconstruction.html"><a href="character-evolution-and-ancestral-state-reconstruction.html#continuous-characters"><i class="fa fa-check"></i><b>18.1</b> Continuous characters</a>
<ul>
<li class="chapter" data-level="18.1.1" data-path="character-evolution-and-ancestral-state-reconstruction.html"><a href="character-evolution-and-ancestral-state-reconstruction.html#load-libraries-1"><i class="fa fa-check"></i><b>18.1.1</b> load libraries</a></li>
<li class="chapter" data-level="18.1.2" data-path="character-evolution-and-ancestral-state-reconstruction.html"><a href="character-evolution-and-ancestral-state-reconstruction.html#tree-file-and-the-character-file"><i class="fa fa-check"></i><b>18.1.2</b> Tree file and the character file</a></li>
<li class="chapter" data-level="18.1.3" data-path="character-evolution-and-ancestral-state-reconstruction.html"><a href="character-evolution-and-ancestral-state-reconstruction.html#visualization-of-the-reconstructed-nodes"><i class="fa fa-check"></i><b>18.1.3</b> Visualization of the reconstructed nodes</a></li>
<li class="chapter" data-level="18.1.4" data-path="character-evolution-and-ancestral-state-reconstruction.html"><a href="character-evolution-and-ancestral-state-reconstruction.html#visualization-of-phenotypic-evolution"><i class="fa fa-check"></i><b>18.1.4</b> Visualization of phenotypic evolution</a></li>
</ul></li>
<li class="chapter" data-level="18.2" data-path="character-evolution-and-ancestral-state-reconstruction.html"><a href="character-evolution-and-ancestral-state-reconstruction.html#brownian-motion-vs-ornstein-uhlenbeck-process"><i class="fa fa-check"></i><b>18.2</b> Brownian Motion vs Ornstein-Uhlenbeck Process</a></li>
<li class="chapter" data-level="18.3" data-path="character-evolution-and-ancestral-state-reconstruction.html"><a href="character-evolution-and-ancestral-state-reconstruction.html#brownian-motion"><i class="fa fa-check"></i><b>18.3</b> Brownian Motion</a>
<ul>
<li class="chapter" data-level="18.3.1" data-path="character-evolution-and-ancestral-state-reconstruction.html"><a href="character-evolution-and-ancestral-state-reconstruction.html#ornsteinuhlenbeck-process"><i class="fa fa-check"></i><b>18.3.1</b> Ornstein–Uhlenbeck process</a></li>
<li class="chapter" data-level="18.3.2" data-path="character-evolution-and-ancestral-state-reconstruction.html"><a href="character-evolution-and-ancestral-state-reconstruction.html#incorporate-prior-knowledge-using-bayesian-estimates"><i class="fa fa-check"></i><b>18.3.2</b> Incorporate prior knowledge using Bayesian estimates</a></li>
</ul></li>
<li class="chapter" data-level="18.4" data-path="character-evolution-and-ancestral-state-reconstruction.html"><a href="character-evolution-and-ancestral-state-reconstruction.html#discrete-characters"><i class="fa fa-check"></i><b>18.4</b> Discrete characters</a></li>
<li class="chapter" data-level="18.5" data-path="character-evolution-and-ancestral-state-reconstruction.html"><a href="character-evolution-and-ancestral-state-reconstruction.html#exercises-1"><i class="fa fa-check"></i><b>18.5</b> Exercises:</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="model-selection-of-character-evolution.html"><a href="model-selection-of-character-evolution.html"><i class="fa fa-check"></i><b>19</b> Model Selection of Character Evolution</a>
<ul>
<li class="chapter" data-level="19.1" data-path="model-selection-of-character-evolution.html"><a href="model-selection-of-character-evolution.html#load-libraries-2"><i class="fa fa-check"></i><b>19.1</b> Load libraries</a></li>
<li class="chapter" data-level="19.2" data-path="model-selection-of-character-evolution.html"><a href="model-selection-of-character-evolution.html#discrete-traits-model-fitting"><i class="fa fa-check"></i><b>19.2</b> Discrete traits model fitting</a>
<ul>
<li class="chapter" data-level="19.2.1" data-path="model-selection-of-character-evolution.html"><a href="model-selection-of-character-evolution.html#choosing-a-character-model"><i class="fa fa-check"></i><b>19.2.1</b> Choosing a character model</a></li>
<li class="chapter" data-level="19.2.2" data-path="model-selection-of-character-evolution.html"><a href="model-selection-of-character-evolution.html#marginal-ancestral-state-reconstruction"><i class="fa fa-check"></i><b>19.2.2</b> Marginal ancestral state reconstruction</a></li>
<li class="chapter" data-level="19.2.3" data-path="model-selection-of-character-evolution.html"><a href="model-selection-of-character-evolution.html#more-than-two-states"><i class="fa fa-check"></i><b>19.2.3</b> More than two states</a></li>
</ul></li>
<li class="chapter" data-level="19.3" data-path="model-selection-of-character-evolution.html"><a href="model-selection-of-character-evolution.html#exercise-i-2"><i class="fa fa-check"></i><b>19.3</b> Exercise I</a></li>
<li class="chapter" data-level="19.4" data-path="model-selection-of-character-evolution.html"><a href="model-selection-of-character-evolution.html#continuous-traits-model-fitting"><i class="fa fa-check"></i><b>19.4</b> Continuous traits model fitting</a>
<ul>
<li class="chapter" data-level="19.4.1" data-path="model-selection-of-character-evolution.html"><a href="model-selection-of-character-evolution.html#model-selection"><i class="fa fa-check"></i><b>19.4.1</b> Model selection</a></li>
<li class="chapter" data-level="19.4.2" data-path="model-selection-of-character-evolution.html"><a href="model-selection-of-character-evolution.html#ancestral-states"><i class="fa fa-check"></i><b>19.4.2</b> Ancestral states</a></li>
</ul></li>
<li class="chapter" data-level="19.5" data-path="model-selection-of-character-evolution.html"><a href="model-selection-of-character-evolution.html#exercise-ii-2"><i class="fa fa-check"></i><b>19.5</b> Exercise II</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">BIO595 The Tree of Life</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="character-evolution-and-ancestral-state-reconstruction" class="section level1 hasAnchor" number="18">
<h1><span class="header-section-number">Chapter 18</span> Character evolution and Ancestral state reconstruction<a href="character-evolution-and-ancestral-state-reconstruction.html#character-evolution-and-ancestral-state-reconstruction" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Adapted from <code>phytools</code> tutorial by Liam J. Revell.</p>
<p>The estimation of phenotypic trait values for ancestral nodes in the tree continues to be an important application of a time-tree from phylogenetic inference.</p>
<p>We are going to explore the methodologies of the reconstruction of continuous and discrete characters, and the potential bias in estimation and quantify the uncertainties using the packages <code>phytools</code> and <code>ape</code>.</p>
<div id="continuous-characters" class="section level2 hasAnchor" number="18.1">
<h2><span class="header-section-number">18.1</span> Continuous characters<a href="character-evolution-and-ancestral-state-reconstruction.html#continuous-characters" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Let’s start by estimating ancestral states for a a continuous character, in this case overall body size, on a phylogeny of Anolis lizards from the Caribbean.</p>
<p>To estimate the states for a continuously valued character at ancestral nodes, we need find the states that have the highest probability of having arisen under our model informed from Maximum Likelihood estimates.</p>
<p>In Chapter 6, we explored the function <code>ace</code> from the <code>ape</code> package to perform ancestral characters. Here we will use <code>fastAnc</code> from <code>phytools</code> as it improves on the running time.</p>
<p>If the package <code>phytools</code> hasn’t been installed:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;phytools&quot;</span>)</span></code></pre></div>
<div id="load-libraries-1" class="section level3 hasAnchor" number="18.1.1">
<h3><span class="header-section-number">18.1.1</span> load libraries<a href="character-evolution-and-ancestral-state-reconstruction.html#load-libraries-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co"># phytools will load ape and maps</span></span>
<span id="cb2-2"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(phytools)</span></code></pre></div>
<pre><code>## Loading required package: ape</code></pre>
<pre><code>## Loading required package: maps</code></pre>
</div>
<div id="tree-file-and-the-character-file" class="section level3 hasAnchor" number="18.1.2">
<h3><span class="header-section-number">18.1.2</span> Tree file and the character file<a href="character-evolution-and-ancestral-state-reconstruction.html#tree-file-and-the-character-file" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb5-1" aria-hidden="true" tabindex="-1"></a>anole.tree<span class="ot">&lt;-</span><span class="fu">read.tree</span>(<span class="st">&quot;http://www.phytools.org/eqg2015/data/anole.tre&quot;</span>)</span>
<span id="cb5-2"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plotTree</span>(anole.tree,<span class="at">type=</span><span class="st">&quot;fan&quot;</span>,<span class="at">ftype=</span><span class="st">&quot;i&quot;</span>)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/asr1-1.png" width="672" /></p>
<p>We then read the character states <strong>snout-vent lengths</strong> (SVL) from a csv file.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb6-1" aria-hidden="true" tabindex="-1"></a>svl<span class="ot">&lt;-</span><span class="fu">read.csv</span>(<span class="st">&quot;http://www.phytools.org/eqg2015/data/svl.csv&quot;</span>,<span class="at">row.names=</span><span class="dv">1</span>)</span>
<span id="cb6-2"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="co"># convert to a vector</span></span>
<span id="cb6-3"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb6-3" aria-hidden="true" tabindex="-1"></a>svl<span class="ot">&lt;-</span><span class="fu">as.matrix</span>(svl)[,<span class="dv">1</span>]</span>
<span id="cb6-4"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb6-4" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(svl)</span></code></pre></div>
<pre><code>##     ahli  alayoni  alfaroi aliniger allisoni  allogus 
## 4.039125 3.815705 3.526655 4.036557 4.375390 4.040138</code></pre>
<p>Now we can estimate ancestral states using <code>fastAnc</code> function using Felsentein’s 1985 PIC approach. We will also compute variances &amp; 95% confidence intervals for each node:</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb8-1" aria-hidden="true" tabindex="-1"></a>?fastAnc</span>
<span id="cb8-2"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb8-2" aria-hidden="true" tabindex="-1"></a>fit<span class="ot">&lt;-</span><span class="fu">fastAnc</span>(anole.tree,svl,<span class="at">vars=</span><span class="cn">TRUE</span>,<span class="at">CI=</span><span class="cn">TRUE</span>)</span>
<span id="cb8-3"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb8-3" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(fit)</span></code></pre></div>
<pre><code>##      Length Class  Mode   
## ace   99    -none- numeric
## var   99    -none- numeric
## CI95 198    -none- numeric</code></pre>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(fit<span class="sc">$</span>ace)</span>
<span id="cb10-2"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(fit<span class="sc">$</span>var)</span>
<span id="cb10-3"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb10-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(fit<span class="sc">$</span>CI95)</span></code></pre></div>
<p>95% CIs can be quite broad:</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb11-1" aria-hidden="true" tabindex="-1"></a>fit<span class="sc">$</span>CI[<span class="dv">1</span>,]</span>
<span id="cb11-2"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="fu">range</span>(svl)</span></code></pre></div>
</div>
<div id="visualization-of-the-reconstructed-nodes" class="section level3 hasAnchor" number="18.1.3">
<h3><span class="header-section-number">18.1.3</span> Visualization of the reconstructed nodes<a href="character-evolution-and-ancestral-state-reconstruction.html#visualization-of-the-reconstructed-nodes" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><code>phytools</code> has several different methods for visualizing reconstructed ancestral states for a continuous trait on the tree. One is a color gradient projection onto the tree:</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="co"># projection of continuous trait on the tree</span></span>
<span id="cb12-2"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb12-2" aria-hidden="true" tabindex="-1"></a>?contMap</span>
<span id="cb12-3"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb12-3" aria-hidden="true" tabindex="-1"></a>obj<span class="ot">&lt;-</span><span class="fu">contMap</span>(<span class="at">tree =</span> anole.tree,<span class="at">x =</span> svl,<span class="at">plot=</span><span class="cn">FALSE</span>)</span>
<span id="cb12-4"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb12-4" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(obj,<span class="at">type=</span><span class="st">&quot;fan&quot;</span>,<span class="at">legend=</span><span class="fl">0.7</span><span class="sc">*</span><span class="fu">max</span>(<span class="fu">nodeHeights</span>(anole.tree)),<span class="at">fsize=</span><span class="fu">c</span>(<span class="fl">0.7</span>,<span class="fl">0.9</span>))</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/asr2-1.png" width="672" /></p>
</div>
<div id="visualization-of-phenotypic-evolution" class="section level3 hasAnchor" number="18.1.4">
<h3><span class="header-section-number">18.1.4</span> Visualization of phenotypic evolution<a href="character-evolution-and-ancestral-state-reconstruction.html#visualization-of-phenotypic-evolution" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Time is represented on the horizontal axis, and phenotypic evolution on the vertical dimension.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb13-1" aria-hidden="true" tabindex="-1"></a>?phenogram</span>
<span id="cb13-2"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb13-2" aria-hidden="true" tabindex="-1"></a><span class="fu">phenogram</span>(anole.tree,svl,<span class="at">fsize=</span><span class="fl">0.6</span>,<span class="at">spread.costs=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">0</span>))</span></code></pre></div>
<pre><code>## Optimizing the positions of the tip labels...</code></pre>
<p><img src="bookdown-demo_files/figure-html/asr3-1.png" width="672" /></p>
<p>Incorporating the uncertainties by displaying the CI of reconstructed states:</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="fu">phenogram95</span>(anole.tree,svl,<span class="at">fsize=</span><span class="fl">0.6</span>,<span class="at">spread.costs=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">0</span>))</span></code></pre></div>
<pre><code>## Computing density traitgram...
## Optimizing the positions of the tip labels...</code></pre>
<p><img src="bookdown-demo_files/figure-html/asr21-1.png" width="672" /></p>
<p>Then the important next question would be: How accurate is the reconstruction?</p>
</div>
</div>
<div id="brownian-motion-vs-ornstein-uhlenbeck-process" class="section level2 hasAnchor" number="18.2">
<h2><span class="header-section-number">18.2</span> Brownian Motion vs Ornstein-Uhlenbeck Process<a href="character-evolution-and-ancestral-state-reconstruction.html#brownian-motion-vs-ornstein-uhlenbeck-process" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>To do this, we will use some simulation functions for character evolution in the <code>phytools</code> package. We inspect two scenarios: Brownian Motion (random walk) of character evolution over time and Ornstein–Uhlenbeck process (random walk + directed selection towards a value)</p>
</div>
<div id="brownian-motion" class="section level2 hasAnchor" number="18.3">
<h2><span class="header-section-number">18.3</span> Brownian Motion<a href="character-evolution-and-ancestral-state-reconstruction.html#brownian-motion" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb17-1" aria-hidden="true" tabindex="-1"></a>?pbtree</span>
<span id="cb17-2"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb17-2" aria-hidden="true" tabindex="-1"></a>tree<span class="ot">&lt;-</span><span class="fu">pbtree</span>(<span class="at">n=</span><span class="dv">26</span>,<span class="at">scale=</span><span class="dv">1</span>,<span class="at">tip.label=</span>LETTERS)</span>
<span id="cb17-3"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb17-3" aria-hidden="true" tabindex="-1"></a>?fastBM</span>
<span id="cb17-4"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb17-4" aria-hidden="true" tabindex="-1"></a>ars_bm<span class="ot">&lt;-</span><span class="fu">fastBM</span>(tree,<span class="at">internal=</span><span class="cn">TRUE</span>)</span>
<span id="cb17-5"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb17-5" aria-hidden="true" tabindex="-1"></a>ars_bm</span></code></pre></div>
<pre><code>##            A            B            C            D            E            F 
## -0.933521138 -1.239651739 -0.879686731 -0.394631502 -0.529359049 -1.605159793 
##            G            H            I            J            K            L 
## -2.263027823 -1.690122365 -0.083137641 -0.082284493  0.240420612  0.339641972 
##            M            N            O            P            Q            R 
##  0.176788020  0.360521127  0.278645260 -0.405896204 -1.605876255 -0.110699963 
##            S            T            U            V            W            X 
## -1.773122107  0.113078642  0.080417577  0.135907805  0.073426219 -0.778555110 
##            Y            Z           27           28           29           30 
## -1.124301058 -0.815110218  0.000000000  0.146340903 -0.202449874 -0.307705281 
##           31           32           33           34           35           36 
## -0.789856039 -1.154582130 -0.139602880 -0.194976472 -1.596380054 -0.622620539 
##           37           38           39           40           41           42 
## -0.136786202 -0.269356603  0.168088674  0.207985368 -0.185453950  0.323712157 
##           43           44           45           46           47           48 
## -0.331681984 -0.520407602 -0.821697971 -0.960912663 -0.075393808  0.006356082 
##           49           50           51 
## -0.061993536  0.151145058 -0.255399546</code></pre>
<p>The <code>fastBM</code> simulates the characters along the phylogenetic tree, and assign each node and tips some values. We then want to use <code>fastAnc</code> to reconstruc the internal nodes pretending that we only observed the states of the tips. We therefore need to separate the tips from the ancestral nodes:</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="do">## ancestral states, labeled as Arabic numbers</span></span>
<span id="cb19-2"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb19-2" aria-hidden="true" tabindex="-1"></a>a<span class="ot">&lt;-</span>ars_bm[<span class="fu">as.character</span>(<span class="dv">1</span><span class="sc">:</span>tree<span class="sc">$</span>Nnode<span class="sc">+</span><span class="fu">Ntip</span>(tree))]</span>
<span id="cb19-3"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb19-3" aria-hidden="true" tabindex="-1"></a><span class="do">## extract tip data</span></span>
<span id="cb19-4"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb19-4" aria-hidden="true" tabindex="-1"></a>x<span class="ot">&lt;-</span>ars_bm[tree<span class="sc">$</span>tip.label]</span></code></pre></div>
<p>Now, let’s estimate ancestral states using <code>fastAnc</code> which uses the PIC algorithm:</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb20-1" aria-hidden="true" tabindex="-1"></a>fit<span class="ot">&lt;-</span><span class="fu">fastAnc</span>(tree,x,<span class="at">CI=</span><span class="cn">TRUE</span>)</span>
<span id="cb20-2"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb20-2" aria-hidden="true" tabindex="-1"></a>fit</span></code></pre></div>
<pre><code>## Ancestral character estimates using fastAnc:
##        27        28        29        30        31        32        33        34 
## -0.912102 -0.937441 -0.817802 -0.804493 -0.990523 -1.038332 -0.887994 -0.872737 
##        35        36        37        38        39        40        41        42 
## -1.833347 -0.291044  0.021252  0.025754  0.137296  0.152362  0.088978  0.269271 
##        43        44        45        46        47        48        49        50 
## -0.888075 -0.893963 -0.965820 -0.955744 -0.588914 -0.077958  0.043817  0.093634 
##        51 
## -0.679270 
## 
## Lower &amp; upper 95% CIs:
##        lower     upper
## 27 -1.713774 -0.110429
## 28 -1.618742 -0.256140
## 29 -1.428256 -0.207348
## 30 -1.412964 -0.196023
## 31 -1.278183 -0.702862
## 32 -1.277226 -0.799438
## 33 -1.424763 -0.351224
## 34 -1.452044 -0.293430
## 35 -2.222588 -1.444107
## 36 -0.765932  0.183845
## 37 -0.305576  0.348079
## 38 -0.301242  0.352749
## 39 -0.097564  0.372155
## 40 -0.078646  0.383371
## 41 -0.241205  0.419161
## 42  0.015688  0.522855
## 43 -1.417952 -0.358198
## 44 -1.413557 -0.374369
## 45 -1.502522 -0.429118
## 46 -1.509452 -0.402036
## 47 -1.066782 -0.111046
## 48 -0.449961  0.294046
## 49 -0.232430  0.320064
## 50 -0.053043  0.240311
## 51 -1.149941 -0.208600</code></pre>
<p>Let’s compare these estimates to the (normally unknown) actual simulated internal character states:</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(a,fit<span class="sc">$</span>ace,<span class="at">xlab=</span><span class="st">&quot;true states&quot;</span>,<span class="at">ylab=</span><span class="st">&quot;estimated states&quot;</span>)</span>
<span id="cb22-2"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb22-2" aria-hidden="true" tabindex="-1"></a><span class="fu">lines</span>(<span class="fu">range</span>(<span class="fu">c</span>(x,a)),<span class="fu">range</span>(<span class="fu">c</span>(x,a)),<span class="at">lty=</span><span class="st">&quot;dashed&quot;</span>,<span class="at">col=</span><span class="st">&quot;red&quot;</span>) <span class="do">## 1:1 line</span></span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/asr5-1.png" width="672" /></p>
<p>Generally speaking, they agree pretty well. Let’s formally inspect this by calculating whether the 95% CI encompass the true value of the states:</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="fu">mean</span>((a<span class="sc">&gt;=</span>fit<span class="sc">$</span>CI95[,<span class="dv">1</span>])<span class="sc">+</span>(a<span class="sc">&lt;=</span>fit<span class="sc">$</span>CI95[,<span class="dv">2</span>]))<span class="sc">/</span><span class="dv">2</span></span></code></pre></div>
<p>We will get different values for one realization, but generally it’s close to the 95% we used to calculate CI.</p>
<div id="ornsteinuhlenbeck-process" class="section level3 hasAnchor" number="18.3.1">
<h3><span class="header-section-number">18.3.1</span> Ornstein–Uhlenbeck process<a href="character-evolution-and-ancestral-state-reconstruction.html#ornsteinuhlenbeck-process" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb24-1" aria-hidden="true" tabindex="-1"></a>tree<span class="ot">&lt;-</span><span class="fu">pbtree</span>(<span class="at">n=</span><span class="dv">100</span>,<span class="at">scale=</span><span class="dv">1</span>)</span>
<span id="cb24-2"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb24-2" aria-hidden="true" tabindex="-1"></a><span class="co"># fastBM can also simulate an OU process if we set mu to nonzero</span></span>
<span id="cb24-3"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb24-3" aria-hidden="true" tabindex="-1"></a>ars_OU<span class="ot">&lt;-</span><span class="fu">fastBM</span>(tree,<span class="at">internal=</span><span class="cn">TRUE</span>,<span class="at">mu=</span><span class="dv">3</span>)</span>
<span id="cb24-4"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb24-4" aria-hidden="true" tabindex="-1"></a><span class="fu">phenogram</span>(tree,ars_OU,<span class="at">ftype=</span><span class="st">&quot;off&quot;</span>,<span class="at">spread.labels=</span><span class="cn">FALSE</span>)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/asr7-1.png" width="672" /></p>
<p>Separate ancestral nodes from the tip values</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb25-1" aria-hidden="true" tabindex="-1"></a>a<span class="ot">&lt;-</span>ars_OU[<span class="fu">as.character</span>(<span class="dv">1</span><span class="sc">:</span>tree<span class="sc">$</span>Nnode<span class="sc">+</span><span class="fu">Ntip</span>(tree))]</span>
<span id="cb25-2"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb25-2" aria-hidden="true" tabindex="-1"></a>x<span class="ot">&lt;-</span>ars_OU[tree<span class="sc">$</span>tip.label]</span></code></pre></div>
<p>Let’s see how bad the reconstruction would be if we ignore the trend</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(a,<span class="fu">fastAnc</span>(tree,x),<span class="at">xlab=</span><span class="st">&quot;true values&quot;</span>,</span>
<span id="cb26-2"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb26-2" aria-hidden="true" tabindex="-1"></a>    <span class="at">ylab=</span><span class="st">&quot;estimated states under BM&quot;</span>)</span>
<span id="cb26-3"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb26-3" aria-hidden="true" tabindex="-1"></a><span class="fu">lines</span>(<span class="fu">range</span>(ars_OU),<span class="fu">range</span>(ars_OU),<span class="at">lty=</span><span class="st">&quot;dashed&quot;</span>,<span class="at">col=</span><span class="st">&quot;red&quot;</span>)</span>
<span id="cb26-4"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb26-4" aria-hidden="true" tabindex="-1"></a><span class="fu">title</span>(<span class="st">&quot;estimated without prior information&quot;</span>)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/asr10-1.png" width="672" /></p>
</div>
<div id="incorporate-prior-knowledge-using-bayesian-estimates" class="section level3 hasAnchor" number="18.3.2">
<h3><span class="header-section-number">18.3.2</span> Incorporate prior knowledge using Bayesian estimates<a href="character-evolution-and-ancestral-state-reconstruction.html#incorporate-prior-knowledge-using-bayesian-estimates" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Set prior knowledge of the mean of states</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Here sig2 indicates the brownian motion parameter (evolutionary rates)</span></span>
<span id="cb27-2"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb27-2" aria-hidden="true" tabindex="-1"></a>pm<span class="ot">&lt;-</span><span class="fu">setNames</span>(<span class="fu">c</span>(<span class="dv">1000</span>,<span class="fu">rep</span>(<span class="dv">0</span>,tree<span class="sc">$</span>Nnode)),</span>
<span id="cb27-3"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb27-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">c</span>(<span class="st">&quot;sig2&quot;</span>,<span class="dv">1</span><span class="sc">:</span>tree<span class="sc">$</span>Nnode<span class="sc">+</span><span class="fu">length</span>(tree<span class="sc">$</span>tip.label)))</span>
<span id="cb27-4"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb27-4" aria-hidden="true" tabindex="-1"></a><span class="co"># we assume that we acquired some knowledge of the root state and two randomly selected internal nodes</span></span>
<span id="cb27-5"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb27-5" aria-hidden="true" tabindex="-1"></a><span class="co"># root node is the one that&#39;s right after all the tips</span></span>
<span id="cb27-6"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb27-6" aria-hidden="true" tabindex="-1"></a>nn<span class="ot">&lt;-</span><span class="fu">as.character</span>(<span class="fu">c</span>(<span class="fu">length</span>(tree<span class="sc">$</span>tip.label)<span class="sc">+</span><span class="dv">1</span>,</span>
<span id="cb27-7"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb27-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">sample</span>(<span class="fu">length</span>(tree<span class="sc">$</span>tip.label) <span class="sc">+</span> (<span class="dv">2</span><span class="sc">:</span>tree<span class="sc">$</span>Nnode),<span class="dv">2</span>)))</span>
<span id="cb27-8"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb27-8" aria-hidden="true" tabindex="-1"></a>pm[nn]<span class="ot">&lt;-</span>a[<span class="fu">as.character</span>(nn)]</span>
<span id="cb27-9"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb27-9" aria-hidden="true" tabindex="-1"></a>pm</span></code></pre></div>
<pre><code>##        sig2         101         102         103         104         105 
## 1000.000000    0.000000    0.000000    0.000000    0.677838    0.000000 
##         106         107         108         109         110         111 
##    0.000000    0.000000    0.000000    0.000000    0.000000    0.000000 
##         112         113         114         115         116         117 
##    0.000000    0.000000    0.000000    0.000000    0.000000    0.000000 
##         118         119         120         121         122         123 
##    0.000000    0.000000    0.000000    0.000000    0.000000    0.000000 
##         124         125         126         127         128         129 
##    0.000000    0.000000    0.000000    0.000000    0.000000    0.000000 
##         130         131         132         133         134         135 
##    0.000000    0.000000    0.000000    0.000000    0.000000    0.000000 
##         136         137         138         139         140         141 
##    0.000000    0.000000    0.000000    0.000000    0.000000    0.000000 
##         142         143         144         145         146         147 
##    0.000000    0.000000    0.000000    0.000000    0.000000    0.000000 
##         148         149         150         151         152         153 
##    0.000000    0.000000    0.000000    0.000000    0.000000    0.000000 
##         154         155         156         157         158         159 
##    0.000000    0.000000    0.000000    0.000000    0.000000    0.000000 
##         160         161         162         163         164         165 
##    0.000000    0.000000    0.000000    0.000000    0.000000    0.000000 
##         166         167         168         169         170         171 
##    0.000000    0.000000    0.000000    0.000000    0.000000    0.000000 
##         172         173         174         175         176         177 
##    0.000000    0.000000    2.213532    0.000000    0.000000    0.000000 
##         178         179         180         181         182         183 
##    0.000000    0.000000    0.000000    0.000000    0.000000    0.000000 
##         184         185         186         187         188         189 
##    0.000000    0.000000    0.000000    0.000000    0.000000    0.000000 
##         190         191         192         193         194         195 
##    0.000000    0.000000    0.000000    0.000000    0.000000    0.000000 
##         196         197         198         199 
##    0.000000    0.000000    0.000000    0.000000</code></pre>
<p>Set prior knowledge of the variance</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb29-1" aria-hidden="true" tabindex="-1"></a>pv<span class="ot">&lt;-</span><span class="fu">setNames</span>(<span class="fu">c</span>(<span class="dv">1000</span><span class="sc">^</span><span class="dv">2</span>,<span class="fu">rep</span>(<span class="dv">1000</span>,<span class="fu">length</span>(pm)<span class="sc">-</span><span class="dv">1</span>)),<span class="fu">names</span>(pm))</span>
<span id="cb29-2"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb29-2" aria-hidden="true" tabindex="-1"></a>pv[<span class="fu">as.character</span>(nn)]<span class="ot">&lt;-</span><span class="fl">1e-100</span></span></code></pre></div>
<p>Use Bayesian MCMC to estimate the posterior distribution of the internal nodes states</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb30-1" aria-hidden="true" tabindex="-1"></a>?anc.Bayes</span>
<span id="cb30-2"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb30-2" aria-hidden="true" tabindex="-1"></a>mcmc<span class="ot">&lt;-</span><span class="fu">anc.Bayes</span>(tree,x,<span class="at">ngen=</span><span class="dv">100000</span>,</span>
<span id="cb30-3"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb30-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">control=</span><span class="fu">list</span>(<span class="at">pr.mean=</span>pm,<span class="at">pr.var=</span>pv,</span>
<span id="cb30-4"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb30-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">a=</span>pm[<span class="fu">as.character</span>(<span class="fu">length</span>(tree<span class="sc">$</span>tip.label)<span class="sc">+</span><span class="dv">1</span>)],</span>
<span id="cb30-5"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb30-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">y=</span>pm[<span class="fu">as.character</span>(<span class="dv">2</span><span class="sc">:</span>tree<span class="sc">$</span>Nnode<span class="sc">+</span><span class="fu">length</span>(tree<span class="sc">$</span>tip.label))]))</span></code></pre></div>
<pre><code>Control parameters (set by user or default):
List of 7
 $ sig2   : num 1.01
 $ a      : Named num 0
  ..- attr(*, &quot;names&quot;)= chr &quot;101&quot;
 $ y      : Named num [1:98] 0 0 0 0 0 0 0 0 0 0 ...
  ..- attr(*, &quot;names&quot;)= chr [1:98] &quot;102&quot; &quot;103&quot; &quot;104&quot; &quot;105&quot; ...
 $ pr.mean: Named num [1:100] 1000 0 0 0 0 0 0 0 0 0 ...
  ..- attr(*, &quot;names&quot;)= chr [1:100] &quot;sig2&quot; &quot;101&quot; &quot;102&quot; &quot;103&quot; ...
 $ pr.var : Named num [1:100] 1e+06 1e-100 1e+03 1e+03 1e+03 ...
  ..- attr(*, &quot;names&quot;)= chr [1:100] &quot;sig2&quot; &quot;101&quot; &quot;102&quot; &quot;103&quot; ...
 $ prop   : num [1:100] 0.0101 0.0101 0.0101 0.0101 0.0101 ...
 $ sample : num 100
Starting MCMC...
Done MCMC.</code></pre>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(mcmc<span class="sc">$</span>mcmc)</span>
<span id="cb32-2"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb32-2" aria-hidden="true" tabindex="-1"></a><span class="co"># we get the ancestral states by taking the mean of mcmc outputs over time</span></span>
<span id="cb32-3"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb32-3" aria-hidden="true" tabindex="-1"></a><span class="co"># We set the burn-in to 20,000 generations</span></span>
<span id="cb32-4"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb32-4" aria-hidden="true" tabindex="-1"></a>anc.est<span class="ot">&lt;-</span><span class="fu">colMeans</span>(mcmc<span class="sc">$</span>mcmc[<span class="dv">201</span><span class="sc">:</span><span class="dv">1001</span>,</span>
<span id="cb32-5"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb32-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as.character</span>(<span class="dv">1</span><span class="sc">:</span>tree<span class="sc">$</span>Nnode<span class="sc">+</span><span class="fu">length</span>(tree<span class="sc">$</span>tip.label))])</span>
<span id="cb32-6"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb32-6" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(a[<span class="fu">names</span>(anc.est)],anc.est,<span class="at">xlab=</span><span class="st">&quot;true values&quot;</span>,</span>
<span id="cb32-7"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb32-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">ylab=</span><span class="st">&quot;estimated states using informative prior&quot;</span>)</span>
<span id="cb32-8"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb32-8" aria-hidden="true" tabindex="-1"></a><span class="fu">lines</span>(<span class="fu">range</span>(<span class="fu">c</span>(x,a)),<span class="fu">range</span>(<span class="fu">c</span>(x,a)),<span class="at">lty=</span><span class="st">&quot;dashed&quot;</span>,<span class="at">col=</span><span class="st">&quot;red&quot;</span>)</span>
<span id="cb32-9"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb32-9" aria-hidden="true" tabindex="-1"></a><span class="fu">title</span>(<span class="st">&quot;estimated using informative prior&quot;</span>)</span></code></pre></div>
<p><img src="pics/anc_mcmc.png" /></p>
</div>
</div>
<div id="discrete-characters" class="section level2 hasAnchor" number="18.4">
<h2><span class="header-section-number">18.4</span> Discrete characters<a href="character-evolution-and-ancestral-state-reconstruction.html#discrete-characters" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>This part of the tutorial focuses on the estimation of ancestral character states for discretely valued traits using a continuous-time Markov chain model commonly known as the Mk model. We are going to apply the <code>ace</code> function in ape as in Chapter 6 first. For other models of evolutionary change not covered by this function, we can use Rich Fitzjohn’s <code>diversitree</code> package.</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="co"># read in the built-in dataset of phytools</span></span>
<span id="cb33-2"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb33-2" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(anoletree)</span>
<span id="cb33-3"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb33-3" aria-hidden="true" tabindex="-1"></a><span class="co"># get the character states</span></span>
<span id="cb33-4"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb33-4" aria-hidden="true" tabindex="-1"></a>x<span class="ot">&lt;-</span><span class="fu">getStates</span>(anoletree,<span class="st">&quot;tips&quot;</span>)</span>
<span id="cb33-5"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb33-5" aria-hidden="true" tabindex="-1"></a>tree<span class="ot">&lt;-</span>anoletree</span>
<span id="cb33-6"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb33-6" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(anoletree)</span>
<span id="cb33-7"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb33-7" aria-hidden="true" tabindex="-1"></a><span class="co"># inspect the tree</span></span>
<span id="cb33-8"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb33-8" aria-hidden="true" tabindex="-1"></a>tree</span></code></pre></div>
<pre><code>## 
## Phylogenetic tree with 82 tips and 81 internal nodes.
## 
## Tip labels:
##  ahli, allogus, rubribarbus, imias, sagrei, bremeri, ...
## 
## The tree includes a mapped, 6-state discrete character
## with states:
##  CG, GB, TC, TG, Tr, Tw
## 
## Rooted; includes branch lengths.</code></pre>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(x)</span></code></pre></div>
<pre><code>##        ahli     allogus rubribarbus       imias      sagrei     bremeri 
##        &quot;TG&quot;        &quot;TG&quot;        &quot;TG&quot;        &quot;TG&quot;        &quot;TG&quot;        &quot;TG&quot;</code></pre>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotTree</span>(tree,<span class="at">type=</span><span class="st">&quot;fan&quot;</span>,<span class="at">fsize=</span><span class="fl">0.8</span>,<span class="at">ftype=</span><span class="st">&quot;i&quot;</span>)</span>
<span id="cb37-2"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb37-2" aria-hidden="true" tabindex="-1"></a>cols<span class="ot">&lt;-</span><span class="fu">setNames</span>(<span class="fu">palette</span>()[<span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(<span class="fu">unique</span>(x))],<span class="fu">sort</span>(<span class="fu">unique</span>(x)))</span>
<span id="cb37-3"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb37-3" aria-hidden="true" tabindex="-1"></a><span class="fu">tiplabels</span>(<span class="at">pie=</span><span class="fu">to.matrix</span>(x,<span class="fu">sort</span>(<span class="fu">unique</span>(x))),<span class="at">piecol=</span>cols,<span class="at">cex=</span><span class="fl">0.3</span>)</span>
<span id="cb37-4"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb37-4" aria-hidden="true" tabindex="-1"></a><span class="fu">add.simmap.legend</span>(<span class="at">colors=</span>cols,<span class="at">prompt=</span><span class="cn">FALSE</span>,<span class="at">x=</span><span class="fl">0.9</span><span class="sc">*</span><span class="fu">par</span>()<span class="sc">$</span>usr[<span class="dv">1</span>],</span>
<span id="cb37-5"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb37-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">y=</span><span class="sc">-</span><span class="fu">max</span>(<span class="fu">nodeHeights</span>(tree)),<span class="at">fsize=</span><span class="fl">0.8</span>)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/asr20-1.png" width="672" /></p>
<p>Next, let’s fit a single-rate ER model &amp; reconstruct ancestral states at internal nodes in the tree.</p>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb38-1" aria-hidden="true" tabindex="-1"></a>fitER<span class="ot">&lt;-</span><span class="fu">ace</span>(x,tree,<span class="at">model=</span><span class="st">&quot;ER&quot;</span>,<span class="at">type=</span><span class="st">&quot;discrete&quot;</span>)</span>
<span id="cb38-2"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb38-2" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(fitER)</span></code></pre></div>
<pre><code>##              Length Class  Mode   
## loglik         1    -none- numeric
## rates          1    -none- numeric
## se             1    -none- numeric
## index.matrix  36    -none- numeric
## lik.anc      486    -none- numeric
## call           5    -none- call</code></pre>
<p>The element <code>lik.anc</code> gives us the marginal ancestral states by summarizing all the uncertainties in other nodes.</p>
<p>It is fairly straightforward to overlay these posterior probabilities on the tree:</p>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotTree</span>(tree,<span class="at">type=</span><span class="st">&quot;fan&quot;</span>,<span class="at">fsize=</span><span class="fl">0.8</span>,<span class="at">ftype=</span><span class="st">&quot;i&quot;</span>)</span>
<span id="cb40-2"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb40-2" aria-hidden="true" tabindex="-1"></a><span class="fu">nodelabels</span>(<span class="at">node=</span><span class="dv">1</span><span class="sc">:</span>tree<span class="sc">$</span>Nnode<span class="sc">+</span><span class="fu">Ntip</span>(tree),</span>
<span id="cb40-3"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb40-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">pie=</span>fitER<span class="sc">$</span>lik.anc,<span class="at">piecol=</span>cols,<span class="at">cex=</span><span class="fl">0.5</span>)</span>
<span id="cb40-4"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb40-4" aria-hidden="true" tabindex="-1"></a><span class="fu">tiplabels</span>(<span class="at">pie=</span><span class="fu">to.matrix</span>(x,<span class="fu">sort</span>(<span class="fu">unique</span>(x))),<span class="at">piecol=</span>cols,<span class="at">cex=</span><span class="fl">0.3</span>)</span>
<span id="cb40-5"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb40-5" aria-hidden="true" tabindex="-1"></a><span class="fu">add.simmap.legend</span>(<span class="at">colors=</span>cols,<span class="at">prompt=</span><span class="cn">FALSE</span>,<span class="at">x=</span><span class="fl">0.9</span><span class="sc">*</span><span class="fu">par</span>()<span class="sc">$</span>usr[<span class="dv">1</span>],</span>
<span id="cb40-6"><a href="character-evolution-and-ancestral-state-reconstruction.html#cb40-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">y=</span><span class="sc">-</span><span class="fu">max</span>(<span class="fu">nodeHeights</span>(tree)),<span class="at">fsize=</span><span class="fl">0.8</span>)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/asr50-1.png" width="672" /></p>
</div>
<div id="exercises-1" class="section level2 hasAnchor" number="18.5">
<h2><span class="header-section-number">18.5</span> Exercises:<a href="character-evolution-and-ancestral-state-reconstruction.html#exercises-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ol style="list-style-type: decimal">
<li><p>Simulate a species tree with 30 taxa, simulate character evolution along the tree with 1) random walk + mu = 4; 2) random walk + mu = -3;</p></li>
<li><p>Estimate the internal node states using a PIC approach, and a Bayesian MCMC approach with prior knowledge of the direction of evolution.</p></li>
<li><p>Compare the estimated traits and the true simulate traits of the internal nodes and the likelihood from the two methodologies, and explain which one performed better and why.</p></li>
</ol>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="molecular-clock-and-phylogenetic-dating.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="model-selection-of-character-evolution.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/17_asr.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["bookdown-demo.pdf", "bookdown-demo.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
