<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 19 Model Selection of Character Evolution | BIO595 The Tree of Life</title>
  <meta name="description" content="Test writing the book for the course bookdown::gitbook." />
  <meta name="generator" content="bookdown 0.28 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 19 Model Selection of Character Evolution | BIO595 The Tree of Life" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="Test writing the book for the course bookdown::gitbook." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 19 Model Selection of Character Evolution | BIO595 The Tree of Life" />
  
  <meta name="twitter:description" content="Test writing the book for the course bookdown::gitbook." />
  

<meta name="author" content="Qixin He, Daniel Park" />


<meta name="date" content="2023-04-10" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="character-evolution-and-ancestral-state-reconstruction.html"/>

<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Tree of Life: Phylogenetic inferences and applications</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Overview</a></li>
<li class="chapter" data-level="2" data-path="unix.html"><a href="unix.html"><i class="fa fa-check"></i><b>2</b> Unix</a>
<ul>
<li class="chapter" data-level="2.1" data-path="unix.html"><a href="unix.html#what-is-unix-and-why-do-we-use-it"><i class="fa fa-check"></i><b>2.1</b> What is Unix and why do we use it?</a></li>
<li class="chapter" data-level="2.2" data-path="unix.html"><a href="unix.html#installation"><i class="fa fa-check"></i><b>2.2</b> Installation</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="unix.html"><a href="unix.html#mac-os-and-linux"><i class="fa fa-check"></i><b>2.2.1</b> Mac OS and Linux</a></li>
<li class="chapter" data-level="2.2.2" data-path="unix.html"><a href="unix.html#windows"><i class="fa fa-check"></i><b>2.2.2</b> Windows</a></li>
<li class="chapter" data-level="2.2.3" data-path="unix.html"><a href="unix.html#use-scholar-remote-desktop-during-the-class"><i class="fa fa-check"></i><b>2.2.3</b> Use Scholar Remote Desktop During the Class</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="unix.html"><a href="unix.html#operation"><i class="fa fa-check"></i><b>2.3</b> Operation</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="unix.html"><a href="unix.html#starting-the-shell"><i class="fa fa-check"></i><b>2.3.1</b> Starting the Shell</a></li>
<li class="chapter" data-level="2.3.2" data-path="unix.html"><a href="unix.html#directoryfolder-sturcture"><i class="fa fa-check"></i><b>2.3.2</b> Directory/folder sturcture</a></li>
<li class="chapter" data-level="2.3.3" data-path="unix.html"><a href="unix.html#absolute-and-relative-directory"><i class="fa fa-check"></i><b>2.3.3</b> Absolute and Relative directory</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="unix.html"><a href="unix.html#handling-directories-and-files"><i class="fa fa-check"></i><b>2.4</b> Handling directories and files</a></li>
<li class="chapter" data-level="2.5" data-path="unix.html"><a href="unix.html#excercises-i"><i class="fa fa-check"></i><b>2.5</b> Excercises I</a></li>
<li class="chapter" data-level="2.6" data-path="unix.html"><a href="unix.html#viewing-and-processing-text-files"><i class="fa fa-check"></i><b>2.6</b> Viewing and processing text files</a></li>
<li class="chapter" data-level="2.7" data-path="unix.html"><a href="unix.html#redirecting-printouts-and-piping"><i class="fa fa-check"></i><b>2.7</b> Redirecting printouts and piping</a></li>
<li class="chapter" data-level="2.8" data-path="unix.html"><a href="unix.html#exercises-ii"><i class="fa fa-check"></i><b>2.8</b> Exercises II</a></li>
<li class="chapter" data-level="2.9" data-path="unix.html"><a href="unix.html#useful-wildcards"><i class="fa fa-check"></i><b>2.9</b> Useful wildcards</a></li>
<li class="chapter" data-level="2.10" data-path="unix.html"><a href="unix.html#exercises-iii"><i class="fa fa-check"></i><b>2.10</b> Exercises III</a></li>
<li class="chapter" data-level="2.11" data-path="unix.html"><a href="unix.html#search-and-select-lines-using-grep"><i class="fa fa-check"></i><b>2.11</b> Search and select lines using <em>grep</em></a></li>
<li class="chapter" data-level="2.12" data-path="unix.html"><a href="unix.html#substituting-characters-using-tr"><i class="fa fa-check"></i><b>2.12</b> Substituting characters using <em>tr</em></a></li>
<li class="chapter" data-level="2.13" data-path="unix.html"><a href="unix.html#cut-sort-and-unique"><i class="fa fa-check"></i><b>2.13</b> cut, sort and unique</a></li>
<li class="chapter" data-level="2.14" data-path="unix.html"><a href="unix.html#exercises-iv"><i class="fa fa-check"></i><b>2.14</b> Exercises IV</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="r-basics-i.html"><a href="r-basics-i.html"><i class="fa fa-check"></i><b>3</b> R Basics I</a>
<ul>
<li class="chapter" data-level="3.1" data-path="r-basics-i.html"><a href="r-basics-i.html#what-is-programming"><i class="fa fa-check"></i><b>3.1</b> What is programming?</a></li>
<li class="chapter" data-level="3.2" data-path="r-basics-i.html"><a href="r-basics-i.html#r"><i class="fa fa-check"></i><b>3.2</b> <code>R</code></a></li>
<li class="chapter" data-level="3.3" data-path="r-basics-i.html"><a href="r-basics-i.html#installation-1"><i class="fa fa-check"></i><b>3.3</b> Installation</a></li>
<li class="chapter" data-level="3.4" data-path="r-basics-i.html"><a href="r-basics-i.html#variables"><i class="fa fa-check"></i><b>3.4</b> Variables</a>
<ul>
<li class="chapter" data-level="3.4.1" data-path="r-basics-i.html"><a href="r-basics-i.html#exercise-1"><i class="fa fa-check"></i><b>3.4.1</b> Exercise (1):</a></li>
<li class="chapter" data-level="3.4.2" data-path="r-basics-i.html"><a href="r-basics-i.html#variable-naming-conventions"><i class="fa fa-check"></i><b>3.4.2</b> Variable naming conventions</a></li>
<li class="chapter" data-level="3.4.3" data-path="r-basics-i.html"><a href="r-basics-i.html#arithmetic-operators"><i class="fa fa-check"></i><b>3.4.3</b> Arithmetic Operators</a></li>
<li class="chapter" data-level="3.4.4" data-path="r-basics-i.html"><a href="r-basics-i.html#logical-operators"><i class="fa fa-check"></i><b>3.4.4</b> Logical Operators</a></li>
<li class="chapter" data-level="3.4.5" data-path="r-basics-i.html"><a href="r-basics-i.html#basic-built-in-mathematical-functions"><i class="fa fa-check"></i><b>3.4.5</b> Basic built-in mathematical functions</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="r-basics-i.html"><a href="r-basics-i.html#subsetting-data"><i class="fa fa-check"></i><b>3.5</b> Subsetting data</a></li>
<li class="chapter" data-level="3.6" data-path="r-basics-i.html"><a href="r-basics-i.html#logical-values"><i class="fa fa-check"></i><b>3.6</b> Logical values</a>
<ul>
<li class="chapter" data-level="3.6.1" data-path="r-basics-i.html"><a href="r-basics-i.html#missing-dealing-with-na-values"><i class="fa fa-check"></i><b>3.6.1</b> Missing: dealing with <code>NA</code> values</a></li>
</ul></li>
<li class="chapter" data-level="3.7" data-path="r-basics-i.html"><a href="r-basics-i.html#generating-data"><i class="fa fa-check"></i><b>3.7</b> Generating Data</a></li>
<li class="chapter" data-level="3.8" data-path="r-basics-i.html"><a href="r-basics-i.html#data-structures-and-types"><i class="fa fa-check"></i><b>3.8</b> Data Structures and Types</a>
<ul>
<li class="chapter" data-level="3.8.1" data-path="r-basics-i.html"><a href="r-basics-i.html#vectors"><i class="fa fa-check"></i><b>3.8.1</b> Vectors</a></li>
</ul></li>
<li class="chapter" data-level="3.9" data-path="r-basics-i.html"><a href="r-basics-i.html#matrices"><i class="fa fa-check"></i><b>3.9</b> Matrices</a>
<ul>
<li class="chapter" data-level="3.9.1" data-path="r-basics-i.html"><a href="r-basics-i.html#exercise-2"><i class="fa fa-check"></i><b>3.9.1</b> Exercise (2):</a></li>
</ul></li>
<li class="chapter" data-level="3.10" data-path="r-basics-i.html"><a href="r-basics-i.html#data-frames"><i class="fa fa-check"></i><b>3.10</b> Data Frames</a>
<ul>
<li class="chapter" data-level="3.10.1" data-path="r-basics-i.html"><a href="r-basics-i.html#exercise-3"><i class="fa fa-check"></i><b>3.10.1</b> Exercise (3):</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="r-basics-ii.html"><a href="r-basics-ii.html"><i class="fa fa-check"></i><b>4</b> R Basics II</a>
<ul>
<li class="chapter" data-level="4.1" data-path="r-basics-ii.html"><a href="r-basics-ii.html#setting-up-the-environment"><i class="fa fa-check"></i><b>4.1</b> Setting up the environment</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="r-basics-ii.html"><a href="r-basics-ii.html#exercise-1-1"><i class="fa fa-check"></i><b>4.1.1</b> Exercise (1)</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="r-basics-ii.html"><a href="r-basics-ii.html#reading-in-data-from-files"><i class="fa fa-check"></i><b>4.2</b> Reading in data from files</a></li>
<li class="chapter" data-level="4.3" data-path="r-basics-ii.html"><a href="r-basics-ii.html#wrangling-data"><i class="fa fa-check"></i><b>4.3</b> Wrangling Data</a></li>
<li class="chapter" data-level="4.4" data-path="r-basics-ii.html"><a href="r-basics-ii.html#subsetting-by-columns-and-rows"><i class="fa fa-check"></i><b>4.4</b> Subsetting by columns and rows</a></li>
<li class="chapter" data-level="4.5" data-path="r-basics-ii.html"><a href="r-basics-ii.html#subsetting-with-conditions"><i class="fa fa-check"></i><b>4.5</b> Subsetting with conditions</a>
<ul>
<li class="chapter" data-level="4.5.1" data-path="r-basics-ii.html"><a href="r-basics-ii.html#exercise-2-1"><i class="fa fa-check"></i><b>4.5.1</b> Exercise (2)</a></li>
<li class="chapter" data-level="4.5.2" data-path="r-basics-ii.html"><a href="r-basics-ii.html#bonus-1"><i class="fa fa-check"></i><b>4.5.2</b> Bonus [1]</a></li>
</ul></li>
<li class="chapter" data-level="4.6" data-path="r-basics-ii.html"><a href="r-basics-ii.html#merging-data"><i class="fa fa-check"></i><b>4.6</b> Merging data</a></li>
<li class="chapter" data-level="4.7" data-path="r-basics-ii.html"><a href="r-basics-ii.html#saving-files-and-sessions"><i class="fa fa-check"></i><b>4.7</b> Saving files and sessions</a>
<ul>
<li class="chapter" data-level="4.7.1" data-path="r-basics-ii.html"><a href="r-basics-ii.html#exercise-3-1"><i class="fa fa-check"></i><b>4.7.1</b> Exercise (3)</a></li>
</ul></li>
<li class="chapter" data-level="4.8" data-path="r-basics-ii.html"><a href="r-basics-ii.html#the-for-loop"><i class="fa fa-check"></i><b>4.8</b> The <span class="smallcaps">for</span>-loop</a></li>
<li class="chapter" data-level="4.9" data-path="r-basics-ii.html"><a href="r-basics-ii.html#lists"><i class="fa fa-check"></i><b>4.9</b> Lists</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="tree-and-traits-in-r.html"><a href="tree-and-traits-in-r.html"><i class="fa fa-check"></i><b>5</b> Tree and Traits in R</a>
<ul>
<li class="chapter" data-level="5.1" data-path="tree-and-traits-in-r.html"><a href="tree-and-traits-in-r.html#setting-up-the-environment-1"><i class="fa fa-check"></i><b>5.1</b> Setting up the environment</a></li>
<li class="chapter" data-level="5.2" data-path="tree-and-traits-in-r.html"><a href="tree-and-traits-in-r.html#examining-phylogenetic-trees"><i class="fa fa-check"></i><b>5.2</b> Examining phylogenetic trees</a></li>
<li class="chapter" data-level="5.3" data-path="tree-and-traits-in-r.html"><a href="tree-and-traits-in-r.html#plotting-phylogenetic-trees"><i class="fa fa-check"></i><b>5.3</b> Plotting phylogenetic trees</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="tree-and-traits-in-r.html"><a href="tree-and-traits-in-r.html#exercise-1-2"><i class="fa fa-check"></i><b>5.3.1</b> Exercise (1)</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="tree-and-traits-in-r.html"><a href="tree-and-traits-in-r.html#associating-phylogenies-and-traits"><i class="fa fa-check"></i><b>5.4</b> Associating phylogenies and traits</a>
<ul>
<li class="chapter" data-level="5.4.1" data-path="tree-and-traits-in-r.html"><a href="tree-and-traits-in-r.html#saving-plots"><i class="fa fa-check"></i><b>5.4.1</b> Saving plots</a></li>
<li class="chapter" data-level="5.4.2" data-path="tree-and-traits-in-r.html"><a href="tree-and-traits-in-r.html#exercise-2-2"><i class="fa fa-check"></i><b>5.4.2</b> Exercise (2)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html"><i class="fa fa-check"></i><b>6</b> Biogeography and Macroevolution in R</a>
<ul>
<li class="chapter" data-level="6.1" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#setting-up-the-environment-2"><i class="fa fa-check"></i><b>6.1</b> Setting up the environment</a></li>
<li class="chapter" data-level="6.2" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#ancestral-state-reconstructions"><i class="fa fa-check"></i><b>6.2</b> Ancestral State Reconstructions</a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#bonus-2"><i class="fa fa-check"></i><b>6.2.1</b> Bonus [2]</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#continuous-traits"><i class="fa fa-check"></i><b>6.3</b> Continuous traits</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#maximum-likelihood"><i class="fa fa-check"></i><b>6.3.1</b> Maximum likelihood</a></li>
<li class="chapter" data-level="6.3.2" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#phylogenetically-independent-contrasts"><i class="fa fa-check"></i><b>6.3.2</b> Phylogenetically independent contrasts</a></li>
<li class="chapter" data-level="6.3.3" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#plotting-ancestral-states"><i class="fa fa-check"></i><b>6.3.3</b> Plotting ancestral states</a></li>
<li class="chapter" data-level="6.3.4" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#exercise-1-3"><i class="fa fa-check"></i><b>6.3.4</b> Exercise (1)</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#discrete-traits"><i class="fa fa-check"></i><b>6.4</b> Discrete traits</a>
<ul>
<li class="chapter" data-level="6.4.1" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#exercise-2-3"><i class="fa fa-check"></i><b>6.4.1</b> Exercise (2)</a></li>
<li class="chapter" data-level="6.4.2" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#maximum-likelihood-1"><i class="fa fa-check"></i><b>6.4.2</b> Maximum likelihood</a></li>
<li class="chapter" data-level="6.4.3" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#choosing-models"><i class="fa fa-check"></i><b>6.4.3</b> Choosing models</a></li>
</ul></li>
<li class="chapter" data-level="6.5" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#saving-multiple-plots-together"><i class="fa fa-check"></i><b>6.5</b> Saving multiple plots together</a>
<ul>
<li class="chapter" data-level="6.5.1" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#exercise-3-2"><i class="fa fa-check"></i><b>6.5.1</b> Exercise (3)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="phylogenetic-diversity-in-r.html"><a href="phylogenetic-diversity-in-r.html"><i class="fa fa-check"></i><b>7</b> Phylogenetic Diversity in R</a>
<ul>
<li class="chapter" data-level="7.1" data-path="phylogenetic-diversity-in-r.html"><a href="phylogenetic-diversity-in-r.html#setting-up-the-environment-3"><i class="fa fa-check"></i><b>7.1</b> Setting up the environment</a></li>
<li class="chapter" data-level="7.2" data-path="phylogenetic-diversity-in-r.html"><a href="phylogenetic-diversity-in-r.html#phylogenetic-diversity"><i class="fa fa-check"></i><b>7.2</b> Phylogenetic Diversity</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="phylogenetic-diversity-in-r.html"><a href="phylogenetic-diversity-in-r.html#preprocessing"><i class="fa fa-check"></i><b>7.2.1</b> Preprocessing</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="phylogenetic-diversity-in-r.html"><a href="phylogenetic-diversity-in-r.html#estimating-phylogenetic-diversity-metrics"><i class="fa fa-check"></i><b>7.3</b> Estimating phylogenetic diversity metrics</a>
<ul>
<li class="chapter" data-level="7.3.1" data-path="phylogenetic-diversity-in-r.html"><a href="phylogenetic-diversity-in-r.html#exercise-1-4"><i class="fa fa-check"></i><b>7.3.1</b> Exercise (1)</a></li>
<li class="chapter" data-level="7.3.2" data-path="phylogenetic-diversity-in-r.html"><a href="phylogenetic-diversity-in-r.html#exercise-2-4"><i class="fa fa-check"></i><b>7.3.2</b> Exercise (2)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="community-phylogenetics.html"><a href="community-phylogenetics.html"><i class="fa fa-check"></i><b>8</b> Community Phylogenetics</a>
<ul>
<li class="chapter" data-level="8.1" data-path="community-phylogenetics.html"><a href="community-phylogenetics.html#setting-up-the-environment-4"><i class="fa fa-check"></i><b>8.1</b> Setting up the environment</a></li>
<li class="chapter" data-level="8.2" data-path="community-phylogenetics.html"><a href="community-phylogenetics.html#definition-of-community-phylogenetics"><i class="fa fa-check"></i><b>8.2</b> Definition of Community Phylogenetics</a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="community-phylogenetics.html"><a href="community-phylogenetics.html#alpha-diversity"><i class="fa fa-check"></i><b>8.2.1</b> Alpha diversity</a></li>
<li class="chapter" data-level="8.2.2" data-path="community-phylogenetics.html"><a href="community-phylogenetics.html#beta-diversity"><i class="fa fa-check"></i><b>8.2.2</b> Beta diversity</a></li>
<li class="chapter" data-level="8.2.3" data-path="community-phylogenetics.html"><a href="community-phylogenetics.html#exercise-1-5"><i class="fa fa-check"></i><b>8.2.3</b> Exercise (1)</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="community-phylogenetics.html"><a href="community-phylogenetics.html#standardized-effect-sizes"><i class="fa fa-check"></i><b>8.3</b> Standardized effect sizes</a>
<ul>
<li class="chapter" data-level="8.3.1" data-path="community-phylogenetics.html"><a href="community-phylogenetics.html#exercise-2-5"><i class="fa fa-check"></i><b>8.3.1</b> Exercise (2)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="coevolution-and-phylogenetic-signals-in-r.html"><a href="coevolution-and-phylogenetic-signals-in-r.html"><i class="fa fa-check"></i><b>9</b> Coevolution and Phylogenetic Signals in R</a>
<ul>
<li class="chapter" data-level="9.1" data-path="coevolution-and-phylogenetic-signals-in-r.html"><a href="coevolution-and-phylogenetic-signals-in-r.html#setting-up-the-environment-5"><i class="fa fa-check"></i><b>9.1</b> Setting up the environment</a></li>
<li class="chapter" data-level="9.2" data-path="coevolution-and-phylogenetic-signals-in-r.html"><a href="coevolution-and-phylogenetic-signals-in-r.html#phylogenetic-signal"><i class="fa fa-check"></i><b>9.2</b> Phylogenetic Signal</a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="coevolution-and-phylogenetic-signals-in-r.html"><a href="coevolution-and-phylogenetic-signals-in-r.html#blombergs-k"><i class="fa fa-check"></i><b>9.2.1</b> Blomberg’s K</a></li>
<li class="chapter" data-level="9.2.2" data-path="coevolution-and-phylogenetic-signals-in-r.html"><a href="coevolution-and-phylogenetic-signals-in-r.html#pagels-lambda-λ"><i class="fa fa-check"></i><b>9.2.2</b> Pagel’s lambda (λ)</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="coevolution-and-phylogenetic-signals-in-r.html"><a href="coevolution-and-phylogenetic-signals-in-r.html#phylogenetically-independent-contrasts-1"><i class="fa fa-check"></i><b>9.3</b> Phylogenetically Independent Contrasts</a></li>
<li class="chapter" data-level="9.4" data-path="coevolution-and-phylogenetic-signals-in-r.html"><a href="coevolution-and-phylogenetic-signals-in-r.html#coevolution"><i class="fa fa-check"></i><b>9.4</b> Coevolution</a>
<ul>
<li class="chapter" data-level="9.4.1" data-path="coevolution-and-phylogenetic-signals-in-r.html"><a href="coevolution-and-phylogenetic-signals-in-r.html#plotting-tanglegrams"><i class="fa fa-check"></i><b>9.4.1</b> Plotting tanglegrams</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="phylogenetic-visualization-in-r.html"><a href="phylogenetic-visualization-in-r.html"><i class="fa fa-check"></i><b>10</b> Phylogenetic Visualization in R</a>
<ul>
<li class="chapter" data-level="10.1" data-path="phylogenetic-visualization-in-r.html"><a href="phylogenetic-visualization-in-r.html#setting-up-the-environment-6"><i class="fa fa-check"></i><b>10.1</b> Setting up the environment</a></li>
<li class="chapter" data-level="10.2" data-path="phylogenetic-visualization-in-r.html"><a href="phylogenetic-visualization-in-r.html#using-phytools"><i class="fa fa-check"></i><b>10.2</b> Using <code>phytools</code></a>
<ul>
<li class="chapter" data-level="10.2.1" data-path="phylogenetic-visualization-in-r.html"><a href="phylogenetic-visualization-in-r.html#phenograms"><i class="fa fa-check"></i><b>10.2.1</b> Phenograms</a></li>
<li class="chapter" data-level="10.2.2" data-path="phylogenetic-visualization-in-r.html"><a href="phylogenetic-visualization-in-r.html#mapping-continuous-characters-on-branches"><i class="fa fa-check"></i><b>10.2.2</b> Mapping continuous characters on branches</a></li>
<li class="chapter" data-level="10.2.3" data-path="phylogenetic-visualization-in-r.html"><a href="phylogenetic-visualization-in-r.html#plotting-images-on-phylogenies"><i class="fa fa-check"></i><b>10.2.3</b> Plotting images on phylogenies</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="phylogenetic-visualization-in-r.html"><a href="phylogenetic-visualization-in-r.html#using-ggtree"><i class="fa fa-check"></i><b>10.3</b> Using <code>ggtree</code></a>
<ul>
<li class="chapter" data-level="10.3.1" data-path="phylogenetic-visualization-in-r.html"><a href="phylogenetic-visualization-in-r.html#basic-plotting"><i class="fa fa-check"></i><b>10.3.1</b> Basic plotting</a></li>
<li class="chapter" data-level="10.3.2" data-path="phylogenetic-visualization-in-r.html"><a href="phylogenetic-visualization-in-r.html#labeling-tips-nodes-and-clades"><i class="fa fa-check"></i><b>10.3.2</b> Labeling tips, nodes, and clades</a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="phylogenetic-visualization-in-r.html"><a href="phylogenetic-visualization-in-r.html#plotting-maps"><i class="fa fa-check"></i><b>10.4</b> Plotting maps</a>
<ul>
<li class="chapter" data-level="10.4.1" data-path="phylogenetic-visualization-in-r.html"><a href="phylogenetic-visualization-in-r.html#choropleth-maps"><i class="fa fa-check"></i><b>10.4.1</b> Choropleth maps</a></li>
<li class="chapter" data-level="10.4.2" data-path="phylogenetic-visualization-in-r.html"><a href="phylogenetic-visualization-in-r.html#working-with-rasters"><i class="fa fa-check"></i><b>10.4.2</b> Working with rasters</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html"><i class="fa fa-check"></i><b>11</b> Gene tree vs Species tree</a>
<ul>
<li class="chapter" data-level="11.1" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#learning-goals"><i class="fa fa-check"></i><b>11.1</b> Learning goals:</a></li>
<li class="chapter" data-level="11.2" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#tracing-ancestry-within-one-population"><i class="fa fa-check"></i><b>11.2</b> Tracing ancestry within one population</a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#vary-pop-size-and-generation"><i class="fa fa-check"></i><b>11.2.1</b> Vary pop size and generation</a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#speciation-and-coalescence"><i class="fa fa-check"></i><b>11.3</b> Speciation and coalescence</a>
<ul>
<li class="chapter" data-level="11.3.1" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#two-species-coalescent-history"><i class="fa fa-check"></i><b>11.3.1</b> Two-species coalescent history</a></li>
</ul></li>
<li class="chapter" data-level="11.4" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#exercise-i"><i class="fa fa-check"></i><b>11.4</b> Exercise I</a></li>
<li class="chapter" data-level="11.5" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#gene-tree-locus-tree-and-species-tree"><i class="fa fa-check"></i><b>11.5</b> Gene tree, Locus tree and Species tree</a>
<ul>
<li class="chapter" data-level="11.5.1" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#loading-data-and-packages"><i class="fa fa-check"></i><b>11.5.1</b> loading data and packages</a></li>
</ul></li>
<li class="chapter" data-level="11.6" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#effects-of-incomplete-lineage-sorting-ils-alone"><i class="fa fa-check"></i><b>11.6</b> Effects of Incomplete Lineage Sorting (ILS) alone</a>
<ul>
<li class="chapter" data-level="11.6.1" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#generate-species-tree-with-no-extinction"><i class="fa fa-check"></i><b>11.6.1</b> generate species tree with no extinction</a></li>
<li class="chapter" data-level="11.6.2" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#simulate-gene-trees-using-the-multispecies-coalescent-process"><i class="fa fa-check"></i><b>11.6.2</b> simulate gene trees using the multispecies coalescent process</a></li>
</ul></li>
<li class="chapter" data-level="11.7" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#calculate-tree-distances"><i class="fa fa-check"></i><b>11.7</b> Calculate tree distances</a>
<ul>
<li class="chapter" data-level="11.7.1" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#rf-distances"><i class="fa fa-check"></i><b>11.7.1</b> RF distances</a></li>
<li class="chapter" data-level="11.7.2" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#inspect-pairwise-rf-distances-among-all-the-gene-trees-generated-under-one-guiding-species-tree"><i class="fa fa-check"></i><b>11.7.2</b> Inspect pairwise RF distances among all the gene trees generated under one guiding species tree</a></li>
</ul></li>
<li class="chapter" data-level="11.8" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#inspect-node-support-i.e.-subtrees-defined-by-a-split"><i class="fa fa-check"></i><b>11.8</b> Inspect node support (i.e., subtrees defined by a split)</a></li>
<li class="chapter" data-level="11.9" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#combined-effects-of-gene-duplication-loss-and-horizontal-gene-transfer-events"><i class="fa fa-check"></i><b>11.9</b> Combined effects of gene duplication, loss and horizontal gene transfer events</a></li>
<li class="chapter" data-level="11.10" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#exercise-ii"><i class="fa fa-check"></i><b>11.10</b> Exercise II</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="overview-of-phylogenetic-inferences.html"><a href="overview-of-phylogenetic-inferences.html"><i class="fa fa-check"></i><b>12</b> Overview of Phylogenetic Inferences</a>
<ul>
<li class="chapter" data-level="12.1" data-path="overview-of-phylogenetic-inferences.html"><a href="overview-of-phylogenetic-inferences.html#mega"><i class="fa fa-check"></i><b>12.1</b> <code>MEGA</code></a></li>
<li class="chapter" data-level="12.2" data-path="overview-of-phylogenetic-inferences.html"><a href="overview-of-phylogenetic-inferences.html#view-and-align-sequences"><i class="fa fa-check"></i><b>12.2</b> View and Align Sequences</a></li>
<li class="chapter" data-level="12.3" data-path="overview-of-phylogenetic-inferences.html"><a href="overview-of-phylogenetic-inferences.html#dnaprotein-evolution-model"><i class="fa fa-check"></i><b>12.3</b> DNA/protein evolution model</a></li>
<li class="chapter" data-level="12.4" data-path="overview-of-phylogenetic-inferences.html"><a href="overview-of-phylogenetic-inferences.html#phylogenetic-inference"><i class="fa fa-check"></i><b>12.4</b> Phylogenetic inference</a></li>
<li class="chapter" data-level="12.5" data-path="overview-of-phylogenetic-inferences.html"><a href="overview-of-phylogenetic-inferences.html#estimate-divergence-times"><i class="fa fa-check"></i><b>12.5</b> Estimate divergence times</a></li>
<li class="chapter" data-level="12.6" data-path="overview-of-phylogenetic-inferences.html"><a href="overview-of-phylogenetic-inferences.html#exercises"><i class="fa fa-check"></i><b>12.6</b> Exercises</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html"><i class="fa fa-check"></i><b>13</b> Resources and Guidelines for the Term Project</a>
<ul>
<li class="chapter" data-level="13.1" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#bioinformatics-databases"><i class="fa fa-check"></i><b>13.1</b> Bioinformatics Databases</a>
<ul>
<li class="chapter" data-level="13.1.1" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#ncbi-national-center-for-biotechnology-information"><i class="fa fa-check"></i><b>13.1.1</b> NCBI (National Center for Biotechnology Information)</a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#retrieving-genetic-resources-for-your-taxa"><i class="fa fa-check"></i><b>13.2</b> Retrieving genetic resources for your taxa</a>
<ul>
<li class="chapter" data-level="13.2.1" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#ncbi-taxonomy-database"><i class="fa fa-check"></i><b>13.2.1</b> NCBI taxonomy database</a></li>
<li class="chapter" data-level="13.2.2" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#individual-genes-from-the-whole-genome-data"><i class="fa fa-check"></i><b>13.2.2</b> Individual genes from the whole genome data</a></li>
<li class="chapter" data-level="13.2.3" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#get-homologous-sequences-from-related-species"><i class="fa fa-check"></i><b>13.2.3</b> Get homologous sequences from related species</a></li>
<li class="chapter" data-level="13.2.4" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#european-bioinformatics-institute-ebi"><i class="fa fa-check"></i><b>13.2.4</b> European Bioinformatics Institute (EBI)</a></li>
<li class="chapter" data-level="13.2.5" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#protein-data-bank"><i class="fa fa-check"></i><b>13.2.5</b> Protein Data Bank</a></li>
</ul></li>
<li class="chapter" data-level="13.3" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#biogeographic-databases"><i class="fa fa-check"></i><b>13.3</b> Biogeographic Databases</a>
<ul>
<li class="chapter" data-level="13.3.1" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#climate-data"><i class="fa fa-check"></i><b>13.3.1</b> Climate Data</a></li>
<li class="chapter" data-level="13.3.2" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#occurrence-data"><i class="fa fa-check"></i><b>13.3.2</b> Occurrence Data</a></li>
<li class="chapter" data-level="13.3.3" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#topographic-data"><i class="fa fa-check"></i><b>13.3.3</b> TOPOGRAPHIC DATA</a></li>
<li class="chapter" data-level="13.3.4" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#trait-data"><i class="fa fa-check"></i><b>13.3.4</b> TRAIT DATA</a></li>
</ul></li>
<li class="chapter" data-level="13.4" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#publication-databases"><i class="fa fa-check"></i><b>13.4</b> Publication Databases</a></li>
<li class="chapter" data-level="13.5" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#example-research-questions"><i class="fa fa-check"></i><b>13.5</b> Example research questions:</a>
<ul>
<li class="chapter" data-level="13.5.1" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#biogeography-and-phylogenetic-diversity"><i class="fa fa-check"></i><b>13.5.1</b> Biogeography and Phylogenetic Diversity:</a></li>
<li class="chapter" data-level="13.5.2" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#community-ecology"><i class="fa fa-check"></i><b>13.5.2</b> Community Ecology:</a></li>
<li class="chapter" data-level="13.5.3" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#coevolution-1"><i class="fa fa-check"></i><b>13.5.3</b> Coevolution:</a></li>
<li class="chapter" data-level="13.5.4" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#species-tree"><i class="fa fa-check"></i><b>13.5.4</b> Species tree:</a></li>
</ul></li>
<li class="chapter" data-level="13.6" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#how-to-submit-jobs-on-scholar-cluster-and-run-remotely"><i class="fa fa-check"></i><b>13.6</b> How to submit jobs on Scholar Cluster and run remotely</a></li>
<li class="chapter" data-level="13.7" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#term-project-proposal"><i class="fa fa-check"></i><b>13.7</b> Term project proposal</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><i class="fa fa-check"></i><b>14</b> Systematic Retrieval of orthologous DNA sequences form GenBank</a>
<ul>
<li class="chapter" data-level="14.1" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#introduction"><i class="fa fa-check"></i><b>14.1</b> Introduction</a></li>
<li class="chapter" data-level="14.2" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#installing-tools"><i class="fa fa-check"></i><b>14.2</b> Installing Tools</a>
<ul>
<li class="chapter" data-level="14.2.1" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#windows-1"><i class="fa fa-check"></i><b>14.2.1</b> Windows</a></li>
<li class="chapter" data-level="14.2.2" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#mac"><i class="fa fa-check"></i><b>14.2.2</b> Mac</a></li>
<li class="chapter" data-level="14.2.3" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#ncbi-blast-tools"><i class="fa fa-check"></i><b>14.2.3</b> NCBI BLAST+ Tools</a></li>
</ul></li>
<li class="chapter" data-level="14.3" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#using-phylotar"><i class="fa fa-check"></i><b>14.3</b> Using <code>phylotaR</code></a>
<ul>
<li class="chapter" data-level="14.3.1" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#pipeline"><i class="fa fa-check"></i><b>14.3.1</b> Pipeline</a></li>
<li class="chapter" data-level="14.3.2" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#running"><i class="fa fa-check"></i><b>14.3.2</b> Running</a></li>
<li class="chapter" data-level="14.3.3" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#restarting"><i class="fa fa-check"></i><b>14.3.3</b> Restarting</a></li>
<li class="chapter" data-level="14.3.4" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#cluster-i.e.-orthologous-genes-selection"><i class="fa fa-check"></i><b>14.3.4</b> Cluster (i.e., orthologous genes) selection</a></li>
</ul></li>
<li class="chapter" data-level="14.4" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#exercise-i-1"><i class="fa fa-check"></i><b>14.4</b> Exercise I</a></li>
<li class="chapter" data-level="14.5" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#using-annotationbustr"><i class="fa fa-check"></i><b>14.5</b> Using <code>AnnotationBustR</code></a>
<ul>
<li class="chapter" data-level="14.5.1" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#introduction-1"><i class="fa fa-check"></i><b>14.5.1</b> Introduction</a></li>
<li class="chapter" data-level="14.5.2" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#install-the-tools"><i class="fa fa-check"></i><b>14.5.2</b> Install the tools</a></li>
<li class="chapter" data-level="14.5.3" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#specify-search-terms-of-genomes"><i class="fa fa-check"></i><b>14.5.3</b> Specify search terms of genomes</a></li>
<li class="chapter" data-level="14.5.4" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#load-gene-search-terms-of-interest"><i class="fa fa-check"></i><b>14.5.4</b> Load Gene Search Terms of Interest</a></li>
<li class="chapter" data-level="14.5.5" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#extract-sequences-with-annotationbust"><i class="fa fa-check"></i><b>14.5.5</b> Extract sequences with AnnotationBust</a></li>
</ul></li>
<li class="chapter" data-level="14.6" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#exercise-ii-1"><i class="fa fa-check"></i><b>14.6</b> Exercise II</a></li>
<li class="chapter" data-level="14.7" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#combine-gene-information-from-the-two-analyses"><i class="fa fa-check"></i><b>14.7</b> Combine gene information from the two analyses</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="sequence-alignment.html"><a href="sequence-alignment.html"><i class="fa fa-check"></i><b>15</b> Sequence alignment</a>
<ul>
<li class="chapter" data-level="15.1" data-path="sequence-alignment.html"><a href="sequence-alignment.html#load-libraries"><i class="fa fa-check"></i><b>15.1</b> Load libraries</a></li>
<li class="chapter" data-level="15.2" data-path="sequence-alignment.html"><a href="sequence-alignment.html#perform-alignments"><i class="fa fa-check"></i><b>15.2</b> Perform Alignments</a></li>
<li class="chapter" data-level="15.3" data-path="sequence-alignment.html"><a href="sequence-alignment.html#processing-multiple-alignments"><i class="fa fa-check"></i><b>15.3</b> Processing Multiple Alignments</a></li>
<li class="chapter" data-level="15.4" data-path="sequence-alignment.html"><a href="sequence-alignment.html#build-your-first-nj-tree"><i class="fa fa-check"></i><b>15.4</b> Build your first NJ tree</a></li>
<li class="chapter" data-level="15.5" data-path="sequence-alignment.html"><a href="sequence-alignment.html#exercise"><i class="fa fa-check"></i><b>15.5</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="maximum-likelihood-model-tests-tree-and-branch-tests.html"><a href="maximum-likelihood-model-tests-tree-and-branch-tests.html"><i class="fa fa-check"></i><b>16</b> Maximum likelihood, model tests, tree and branch tests</a>
<ul>
<li class="chapter" data-level="16.1" data-path="maximum-likelihood-model-tests-tree-and-branch-tests.html"><a href="maximum-likelihood-model-tests-tree-and-branch-tests.html#learning-goals-1"><i class="fa fa-check"></i><b>16.1</b> Learning Goals</a></li>
<li class="chapter" data-level="16.2" data-path="maximum-likelihood-model-tests-tree-and-branch-tests.html"><a href="maximum-likelihood-model-tests-tree-and-branch-tests.html#iqtree"><i class="fa fa-check"></i><b>16.2</b> IQtree</a>
<ul>
<li class="chapter" data-level="16.2.1" data-path="maximum-likelihood-model-tests-tree-and-branch-tests.html"><a href="maximum-likelihood-model-tests-tree-and-branch-tests.html#version"><i class="fa fa-check"></i><b>16.2.1</b> Version</a></li>
<li class="chapter" data-level="16.2.2" data-path="maximum-likelihood-model-tests-tree-and-branch-tests.html"><a href="maximum-likelihood-model-tests-tree-and-branch-tests.html#find-the-best-explanatory-sequence-evolution-model"><i class="fa fa-check"></i><b>16.2.2</b> find the best explanatory sequence evolution model</a></li>
<li class="chapter" data-level="16.2.3" data-path="maximum-likelihood-model-tests-tree-and-branch-tests.html"><a href="maximum-likelihood-model-tests-tree-and-branch-tests.html#using-the-best-sequence-evolution-model"><i class="fa fa-check"></i><b>16.2.3</b> using the best sequence evolution model</a></li>
<li class="chapter" data-level="16.2.4" data-path="maximum-likelihood-model-tests-tree-and-branch-tests.html"><a href="maximum-likelihood-model-tests-tree-and-branch-tests.html#different-partitions-for-the-data-for-multiple-genes-or-different-codons"><i class="fa fa-check"></i><b>16.2.4</b> Different partitions for the data for multiple genes, or different codons</a></li>
</ul></li>
<li class="chapter" data-level="16.3" data-path="maximum-likelihood-model-tests-tree-and-branch-tests.html"><a href="maximum-likelihood-model-tests-tree-and-branch-tests.html#run-an-ml-tree-with-multiple-gene-files-in-a-folder"><i class="fa fa-check"></i><b>16.3</b> Run an ML tree with multiple gene files in a folder</a></li>
<li class="chapter" data-level="16.4" data-path="maximum-likelihood-model-tests-tree-and-branch-tests.html"><a href="maximum-likelihood-model-tests-tree-and-branch-tests.html#gene-concordance-factor"><i class="fa fa-check"></i><b>16.4</b> Gene concordance factor</a></li>
<li class="chapter" data-level="16.5" data-path="maximum-likelihood-model-tests-tree-and-branch-tests.html"><a href="maximum-likelihood-model-tests-tree-and-branch-tests.html#tree-topology-tests"><i class="fa fa-check"></i><b>16.5</b> Tree topology tests</a></li>
<li class="chapter" data-level="16.6" data-path="maximum-likelihood-model-tests-tree-and-branch-tests.html"><a href="maximum-likelihood-model-tests-tree-and-branch-tests.html#tree-branches-tests"><i class="fa fa-check"></i><b>16.6</b> Tree branches tests</a></li>
<li class="chapter" data-level="16.7" data-path="maximum-likelihood-model-tests-tree-and-branch-tests.html"><a href="maximum-likelihood-model-tests-tree-and-branch-tests.html#assignments"><i class="fa fa-check"></i><b>16.7</b> Assignments</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="molecular-clock-and-phylogenetic-dating.html"><a href="molecular-clock-and-phylogenetic-dating.html"><i class="fa fa-check"></i><b>17</b> Molecular Clock and Phylogenetic Dating</a>
<ul>
<li class="chapter" data-level="17.1" data-path="molecular-clock-and-phylogenetic-dating.html"><a href="molecular-clock-and-phylogenetic-dating.html#inferring-time-tree-with-tip-dates"><i class="fa fa-check"></i><b>17.1</b> Inferring time tree with tip dates</a></li>
<li class="chapter" data-level="17.2" data-path="molecular-clock-and-phylogenetic-dating.html"><a href="molecular-clock-and-phylogenetic-dating.html#calibrating-tree-using-ancestral-dates"><i class="fa fa-check"></i><b>17.2</b> Calibrating tree using ancestral dates</a>
<ul>
<li class="chapter" data-level="17.2.1" data-path="molecular-clock-and-phylogenetic-dating.html"><a href="molecular-clock-and-phylogenetic-dating.html#search-for-an-ml-tree"><i class="fa fa-check"></i><b>17.2.1</b> search for an ML tree</a></li>
<li class="chapter" data-level="17.2.2" data-path="molecular-clock-and-phylogenetic-dating.html"><a href="molecular-clock-and-phylogenetic-dating.html#root-calibration"><i class="fa fa-check"></i><b>17.2.2</b> Root calibration</a></li>
<li class="chapter" data-level="17.2.3" data-path="molecular-clock-and-phylogenetic-dating.html"><a href="molecular-clock-and-phylogenetic-dating.html#internal-nodes"><i class="fa fa-check"></i><b>17.2.3</b> Internal nodes</a></li>
<li class="chapter" data-level="17.2.4" data-path="molecular-clock-and-phylogenetic-dating.html"><a href="molecular-clock-and-phylogenetic-dating.html#confidence-interval"><i class="fa fa-check"></i><b>17.2.4</b> Confidence interval</a></li>
</ul></li>
<li class="chapter" data-level="17.3" data-path="molecular-clock-and-phylogenetic-dating.html"><a href="molecular-clock-and-phylogenetic-dating.html#exercise-4"><i class="fa fa-check"></i><b>17.3</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="character-evolution-and-ancestral-state-reconstruction.html"><a href="character-evolution-and-ancestral-state-reconstruction.html"><i class="fa fa-check"></i><b>18</b> Character evolution and Ancestral state reconstruction</a>
<ul>
<li class="chapter" data-level="18.1" data-path="character-evolution-and-ancestral-state-reconstruction.html"><a href="character-evolution-and-ancestral-state-reconstruction.html#continuous-characters"><i class="fa fa-check"></i><b>18.1</b> Continuous characters</a>
<ul>
<li class="chapter" data-level="18.1.1" data-path="character-evolution-and-ancestral-state-reconstruction.html"><a href="character-evolution-and-ancestral-state-reconstruction.html#load-libraries-1"><i class="fa fa-check"></i><b>18.1.1</b> load libraries</a></li>
<li class="chapter" data-level="18.1.2" data-path="character-evolution-and-ancestral-state-reconstruction.html"><a href="character-evolution-and-ancestral-state-reconstruction.html#tree-file-and-the-character-file"><i class="fa fa-check"></i><b>18.1.2</b> Tree file and the character file</a></li>
<li class="chapter" data-level="18.1.3" data-path="character-evolution-and-ancestral-state-reconstruction.html"><a href="character-evolution-and-ancestral-state-reconstruction.html#visualization-of-the-reconstructed-nodes"><i class="fa fa-check"></i><b>18.1.3</b> Visualization of the reconstructed nodes</a></li>
<li class="chapter" data-level="18.1.4" data-path="character-evolution-and-ancestral-state-reconstruction.html"><a href="character-evolution-and-ancestral-state-reconstruction.html#visualization-of-phenotypic-evolution"><i class="fa fa-check"></i><b>18.1.4</b> Visualization of phenotypic evolution</a></li>
</ul></li>
<li class="chapter" data-level="18.2" data-path="character-evolution-and-ancestral-state-reconstruction.html"><a href="character-evolution-and-ancestral-state-reconstruction.html#brownian-motion-vs-ornstein-uhlenbeck-process"><i class="fa fa-check"></i><b>18.2</b> Brownian Motion vs Ornstein-Uhlenbeck Process</a></li>
<li class="chapter" data-level="18.3" data-path="character-evolution-and-ancestral-state-reconstruction.html"><a href="character-evolution-and-ancestral-state-reconstruction.html#brownian-motion"><i class="fa fa-check"></i><b>18.3</b> Brownian Motion</a>
<ul>
<li class="chapter" data-level="18.3.1" data-path="character-evolution-and-ancestral-state-reconstruction.html"><a href="character-evolution-and-ancestral-state-reconstruction.html#ornsteinuhlenbeck-process"><i class="fa fa-check"></i><b>18.3.1</b> Ornstein–Uhlenbeck process</a></li>
<li class="chapter" data-level="18.3.2" data-path="character-evolution-and-ancestral-state-reconstruction.html"><a href="character-evolution-and-ancestral-state-reconstruction.html#incorporate-prior-knowledge-using-bayesian-estimates"><i class="fa fa-check"></i><b>18.3.2</b> Incorporate prior knowledge using Bayesian estimates</a></li>
</ul></li>
<li class="chapter" data-level="18.4" data-path="character-evolution-and-ancestral-state-reconstruction.html"><a href="character-evolution-and-ancestral-state-reconstruction.html#discrete-characters"><i class="fa fa-check"></i><b>18.4</b> Discrete characters</a></li>
<li class="chapter" data-level="18.5" data-path="character-evolution-and-ancestral-state-reconstruction.html"><a href="character-evolution-and-ancestral-state-reconstruction.html#exercises-1"><i class="fa fa-check"></i><b>18.5</b> Exercises:</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="model-selection-of-character-evolution.html"><a href="model-selection-of-character-evolution.html"><i class="fa fa-check"></i><b>19</b> Model Selection of Character Evolution</a>
<ul>
<li class="chapter" data-level="19.1" data-path="model-selection-of-character-evolution.html"><a href="model-selection-of-character-evolution.html#load-libraries-2"><i class="fa fa-check"></i><b>19.1</b> Load libraries</a></li>
<li class="chapter" data-level="19.2" data-path="model-selection-of-character-evolution.html"><a href="model-selection-of-character-evolution.html#discrete-traits-model-fitting"><i class="fa fa-check"></i><b>19.2</b> Discrete traits model fitting</a>
<ul>
<li class="chapter" data-level="19.2.1" data-path="model-selection-of-character-evolution.html"><a href="model-selection-of-character-evolution.html#choosing-a-character-model"><i class="fa fa-check"></i><b>19.2.1</b> Choosing a character model</a></li>
<li class="chapter" data-level="19.2.2" data-path="model-selection-of-character-evolution.html"><a href="model-selection-of-character-evolution.html#marginal-ancestral-state-reconstruction"><i class="fa fa-check"></i><b>19.2.2</b> Marginal ancestral state reconstruction</a></li>
<li class="chapter" data-level="19.2.3" data-path="model-selection-of-character-evolution.html"><a href="model-selection-of-character-evolution.html#more-than-two-states"><i class="fa fa-check"></i><b>19.2.3</b> More than two states</a></li>
</ul></li>
<li class="chapter" data-level="19.3" data-path="model-selection-of-character-evolution.html"><a href="model-selection-of-character-evolution.html#exercise-i-2"><i class="fa fa-check"></i><b>19.3</b> Exercise I</a></li>
<li class="chapter" data-level="19.4" data-path="model-selection-of-character-evolution.html"><a href="model-selection-of-character-evolution.html#continuous-traits-model-fitting"><i class="fa fa-check"></i><b>19.4</b> Continuous traits model fitting</a>
<ul>
<li class="chapter" data-level="19.4.1" data-path="model-selection-of-character-evolution.html"><a href="model-selection-of-character-evolution.html#model-selection"><i class="fa fa-check"></i><b>19.4.1</b> Model selection</a></li>
<li class="chapter" data-level="19.4.2" data-path="model-selection-of-character-evolution.html"><a href="model-selection-of-character-evolution.html#ancestral-states"><i class="fa fa-check"></i><b>19.4.2</b> Ancestral states</a></li>
</ul></li>
<li class="chapter" data-level="19.5" data-path="model-selection-of-character-evolution.html"><a href="model-selection-of-character-evolution.html#exercise-ii-2"><i class="fa fa-check"></i><b>19.5</b> Exercise II</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">BIO595 The Tree of Life</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="model-selection-of-character-evolution" class="section level1 hasAnchor" number="19">
<h1><span class="header-section-number">Chapter 19</span> Model Selection of Character Evolution<a href="model-selection-of-character-evolution.html#model-selection-of-character-evolution" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>In the previous chapter, we’ve explored ancestral state reconstruction for continuous and discrete characters, and we discovered that the validity of the reconstructed states rely on credible models of character evolution. In this lab, we will compare the likelihood of different models given the data (i.e., the phylogeny and the characters), choose the best model, and then estimate the ancestral states.</p>
<div id="load-libraries-2" class="section level2 hasAnchor" number="19.1">
<h2><span class="header-section-number">19.1</span> Load libraries<a href="model-selection-of-character-evolution.html#load-libraries-2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="model-selection-of-character-evolution.html#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># install.packages(&quot;corHMM&quot;)</span></span>
<span id="cb1-2"><a href="model-selection-of-character-evolution.html#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(corHMM)</span>
<span id="cb1-3"><a href="model-selection-of-character-evolution.html#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(geiger)</span>
<span id="cb1-4"><a href="model-selection-of-character-evolution.html#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(phytools)</span></code></pre></div>
</div>
<div id="discrete-traits-model-fitting" class="section level2 hasAnchor" number="19.2">
<h2><span class="header-section-number">19.2</span> Discrete traits model fitting<a href="model-selection-of-character-evolution.html#discrete-traits-model-fitting" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We’d like to first fit the evolutionary model of discrete traits by testing which transition matrix Q fits the data the best. The example file comes from fhenotypic data for feeding mode for 61 eel species. Source: Collar et al. (<a href="http://dx.doi.org/10.1038/ncomms6505">2014</a>).</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="model-selection-of-character-evolution.html#cb2-1" aria-hidden="true" tabindex="-1"></a>eel.data<span class="ot">&lt;-</span><span class="fu">read.csv</span>(<span class="st">&quot;http://www.phytools.org/Rbook/8/elopomorph.csv&quot;</span>,<span class="at">row.names=</span><span class="dv">1</span>)</span>
<span id="cb2-2"><a href="model-selection-of-character-evolution.html#cb2-2" aria-hidden="true" tabindex="-1"></a>eel.tree<span class="ot">&lt;-</span><span class="fu">read.tree</span>(<span class="st">&quot;http://www.phytools.org/Rbook/8/elopomorph.tre&quot;</span>)</span></code></pre></div>
<p>Let’s inspect the data and extract the feeding mode column, and convert it into a factor vector.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="model-selection-of-character-evolution.html#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(eel.data)</span></code></pre></div>
<pre><code>##                   feed_mode Max_TL_cm
## Albula_vulpes       suction       104
## Anguilla_anguilla   suction        50
## Anguilla_bicolor    suction       120
## Anguilla_japonica   suction       150
## Anguilla_rostrata   suction       152
## Ariosoma_anago      suction        60</code></pre>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="model-selection-of-character-evolution.html#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="do">## extract feeding mode as a vector with species name associated with each character</span></span>
<span id="cb5-2"><a href="model-selection-of-character-evolution.html#cb5-2" aria-hidden="true" tabindex="-1"></a>feed.mode <span class="ot">&lt;-</span> <span class="fu">setNames</span>(eel.data<span class="sc">$</span>feed_mode, <span class="fu">rownames</span>(eel.data))</span>
<span id="cb5-3"><a href="model-selection-of-character-evolution.html#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="do">## convert to a factor</span></span>
<span id="cb5-4"><a href="model-selection-of-character-evolution.html#cb5-4" aria-hidden="true" tabindex="-1"></a>feed.mode<span class="ot">&lt;-</span><span class="fu">as.factor</span>(feed.mode)</span>
<span id="cb5-5"><a href="model-selection-of-character-evolution.html#cb5-5" aria-hidden="true" tabindex="-1"></a>feed.mode[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>]</span></code></pre></div>
<pre><code>##               Albula_vulpes           Anguilla_anguilla 
##                     suction                     suction 
##            Anguilla_bicolor           Anguilla_japonica 
##                     suction                     suction 
##           Anguilla_rostrata              Ariosoma_anago 
##                     suction                     suction 
##         Ariosoma_balearicum         Ariosoma_shiroanago 
##                     suction                     suction 
##      Bathyuroconger_vicinus Brachysomophis_crocodilinus 
##                        bite                        bite 
## Levels: bite suction</code></pre>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="model-selection-of-character-evolution.html#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="do">## &quot;bite is trait 1, suction is trait 2&quot;</span></span>
<span id="cb7-2"><a href="model-selection-of-character-evolution.html#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="fu">levels</span>(feed.mode)</span></code></pre></div>
<pre><code>## [1] &quot;bite&quot;    &quot;suction&quot;</code></pre>
<div id="choosing-a-character-model" class="section level3 hasAnchor" number="19.2.1">
<h3><span class="header-section-number">19.2.1</span> Choosing a character model<a href="model-selection-of-character-evolution.html#choosing-a-character-model" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>For a two-state character, there are four possible models of Q: an equal-rates (ER) model, an all-rates-different (ARD) model, and two irreversible models – one that only allows state 1 to state 2 changes and the other that only allows state 2 to state 1 changes. We will begin by comparing which one fits the data the best.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="model-selection-of-character-evolution.html#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Fit ER model</span></span>
<span id="cb9-2"><a href="model-selection-of-character-evolution.html#cb9-2" aria-hidden="true" tabindex="-1"></a>fitER<span class="ot">&lt;-</span> <span class="fu">fitMk</span>(eel.tree, feed.mode,<span class="at">model=</span><span class="st">&quot;ER&quot;</span>)</span>
<span id="cb9-3"><a href="model-selection-of-character-evolution.html#cb9-3" aria-hidden="true" tabindex="-1"></a><span class="do">## Fit ARD model</span></span>
<span id="cb9-4"><a href="model-selection-of-character-evolution.html#cb9-4" aria-hidden="true" tabindex="-1"></a>fitARD<span class="ot">&lt;-</span> <span class="fu">fitMk</span>(eel.tree, feed.mode,<span class="at">model=</span><span class="st">&quot;ARD&quot;</span>)</span>
<span id="cb9-5"><a href="model-selection-of-character-evolution.html#cb9-5" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(fitARD)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/asr200-1.png" width="672" /></p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="model-selection-of-character-evolution.html#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="do">## We can also see the color version of the transition matrix by specifying:</span></span>
<span id="cb10-2"><a href="model-selection-of-character-evolution.html#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(fitARD, <span class="at">color=</span>T, <span class="at">width=</span>T)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/asr200-2.png" width="672" /></p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="model-selection-of-character-evolution.html#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Fit bite-&gt;suction only model</span></span>
<span id="cb11-2"><a href="model-selection-of-character-evolution.html#cb11-2" aria-hidden="true" tabindex="-1"></a>BStransMat <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">0</span>,<span class="dv">0</span>),<span class="dv">2</span>,<span class="dv">2</span>,<span class="at">byrow=</span>T)</span>
<span id="cb11-3"><a href="model-selection-of-character-evolution.html#cb11-3" aria-hidden="true" tabindex="-1"></a>fitBS<span class="ot">&lt;-</span> <span class="fu">fitMk</span>(eel.tree, feed.mode,<span class="at">model=</span>BStransMat)</span>
<span id="cb11-4"><a href="model-selection-of-character-evolution.html#cb11-4" aria-hidden="true" tabindex="-1"></a><span class="do">## Fit suction-&gt;bite only model</span></span>
<span id="cb11-5"><a href="model-selection-of-character-evolution.html#cb11-5" aria-hidden="true" tabindex="-1"></a>SBtransMat <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">0</span>),<span class="dv">2</span>,<span class="dv">2</span>,<span class="at">byrow=</span>T)</span>
<span id="cb11-6"><a href="model-selection-of-character-evolution.html#cb11-6" aria-hidden="true" tabindex="-1"></a>fitSB<span class="ot">&lt;-</span> <span class="fu">fitMk</span>(eel.tree, feed.mode,<span class="at">model=</span>SBtransMat)</span></code></pre></div>
<p>After model fitting, we would compare the models by pulling out their AIC and likelihood information:</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="model-selection-of-character-evolution.html#cb12-1" aria-hidden="true" tabindex="-1"></a>aicVec <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">AIC</span>(fitER),<span class="fu">AIC</span>(fitARD),<span class="fu">AIC</span>(fitBS),<span class="fu">AIC</span>(fitSB))</span>
<span id="cb12-2"><a href="model-selection-of-character-evolution.html#cb12-2" aria-hidden="true" tabindex="-1"></a>modelSummary <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">model=</span><span class="fu">c</span>(<span class="st">&quot;ER&quot;</span>,<span class="st">&quot;ARD&quot;</span>,<span class="st">&quot;bite-&gt;suction&quot;</span>,<span class="st">&quot;suction-&gt;bite&quot;</span>),</span>
<span id="cb12-3"><a href="model-selection-of-character-evolution.html#cb12-3" aria-hidden="true" tabindex="-1"></a>                           <span class="at">logL =</span> <span class="fu">c</span>(<span class="fu">logLik</span>(fitER),<span class="fu">logLik</span>(fitARD),<span class="fu">logLik</span>(fitBS),<span class="fu">logLik</span>(fitSB)),</span>
<span id="cb12-4"><a href="model-selection-of-character-evolution.html#cb12-4" aria-hidden="true" tabindex="-1"></a>                           <span class="at">AIC =</span> aicVec, </span>
<span id="cb12-5"><a href="model-selection-of-character-evolution.html#cb12-5" aria-hidden="true" tabindex="-1"></a>                           <span class="at">delta.AIC =</span> aicVec <span class="sc">-</span> <span class="fu">min</span>(aicVec))</span>
<span id="cb12-6"><a href="model-selection-of-character-evolution.html#cb12-6" aria-hidden="true" tabindex="-1"></a>modelSummary</span></code></pre></div>
<pre><code>##           model      logL      AIC delta.AIC
## 1            ER -37.03307 76.06614  0.000000
## 2           ARD -37.00365 78.00730  1.941156
## 3 bite-&gt;suction -38.64057 79.28114  3.214999
## 4 suction-&gt;bite -40.50138 83.00276  6.936613</code></pre>
<p>The four models do not differ that much, and since ER has the best AIC, we go with the simplest model.</p>
</div>
<div id="marginal-ancestral-state-reconstruction" class="section level3 hasAnchor" number="19.2.2">
<h3><span class="header-section-number">19.2.2</span> Marginal ancestral state reconstruction<a href="model-selection-of-character-evolution.html#marginal-ancestral-state-reconstruction" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>For discrete traits, we mentioned in the lecture that we can either calculate joint ancestral state reconstruction or marginal ancestral state reconstruction (Pagel 1999l; Yang 2006). Marginal ancestral state reconstruction is much more popular than joint one because it is the type of reconstruction we should do fi want to make specific, probabilistic statements about the values at particular nodes of the tree.</p>
<p>Here we use the package <code>corHMM</code> (Beaulieu et al. 2020) to estimate marginal ancestral states of the internal nodes using the best supported character evolution we found in the previous section</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="model-selection-of-character-evolution.html#cb14-1" aria-hidden="true" tabindex="-1"></a>ERMat <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="fu">c</span>(<span class="cn">NA</span>,<span class="dv">1</span>,<span class="dv">1</span>,<span class="cn">NA</span>),<span class="dv">2</span>,<span class="dv">2</span>)</span>
<span id="cb14-2"><a href="model-selection-of-character-evolution.html#cb14-2" aria-hidden="true" tabindex="-1"></a><span class="co"># corHMM</span></span>
<span id="cb14-3"><a href="model-selection-of-character-evolution.html#cb14-3" aria-hidden="true" tabindex="-1"></a>feedDat <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Genus_sp =</span> <span class="fu">names</span>(feed.mode), <span class="at">feed.mode =</span> feed.mode)</span>
<span id="cb14-4"><a href="model-selection-of-character-evolution.html#cb14-4" aria-hidden="true" tabindex="-1"></a>?corHMM</span>
<span id="cb14-5"><a href="model-selection-of-character-evolution.html#cb14-5" aria-hidden="true" tabindex="-1"></a>eel.anc.fit <span class="ot">&lt;-</span> <span class="fu">corHMM</span>(eel.tree, feedDat, <span class="at">node.states =</span> <span class="st">&quot;marginal&quot;</span>,</span>
<span id="cb14-6"><a href="model-selection-of-character-evolution.html#cb14-6" aria-hidden="true" tabindex="-1"></a>                      <span class="at">rate.cat =</span> <span class="dv">1</span>, <span class="at">rate.mat =</span> ERMat)</span></code></pre></div>
<pre><code>## State distribution in data:
## States:  1   2   
## Counts:  31  30  
## Beginning thorough optimization search -- performing 0 random restarts 
## Finished. Inferring ancestral states using marginal reconstruction.</code></pre>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="model-selection-of-character-evolution.html#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(eel.anc.fit<span class="sc">$</span>states)</span></code></pre></div>
<pre><code>##         (1,R1)    (2,R1)
## [1,] 0.5480196 0.4519804
## [2,] 0.6242662 0.3757338
## [3,] 0.6561270 0.3438730
## [4,] 0.6904540 0.3095460
## [5,] 0.7066905 0.2933095
## [6,] 0.7158532 0.2841468</code></pre>
<p>Let’s plot the estimated states for the internal nodes</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="model-selection-of-character-evolution.html#cb18-1" aria-hidden="true" tabindex="-1"></a>colPattern <span class="ot">&lt;-</span> <span class="fu">setNames</span>(<span class="fu">c</span>(<span class="st">&quot;red&quot;</span>,<span class="st">&quot;lightblue&quot;</span>),<span class="fu">levels</span>(feed.mode))</span>
<span id="cb18-2"><a href="model-selection-of-character-evolution.html#cb18-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plotTree.datamatrix</span>(eel.tree, <span class="fu">as.data.frame</span>(feed.mode), </span>
<span id="cb18-3"><a href="model-selection-of-character-evolution.html#cb18-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">colors=</span><span class="fu">list</span>(colPattern), <span class="at">header=</span>F,<span class="at">fsize=</span><span class="fl">0.45</span>)</span>
<span id="cb18-4"><a href="model-selection-of-character-evolution.html#cb18-4" aria-hidden="true" tabindex="-1"></a><span class="co"># add legend</span></span>
<span id="cb18-5"><a href="model-selection-of-character-evolution.html#cb18-5" aria-hidden="true" tabindex="-1"></a><span class="fu">legend</span>(<span class="st">&quot;topright&quot;</span>,<span class="at">legend=</span><span class="fu">levels</span>(feed.mode),<span class="at">pch=</span><span class="dv">22</span>,<span class="at">pt.cex=</span><span class="fl">1.5</span>, <span class="at">pt.bg=</span>colPattern,<span class="at">bty=</span><span class="st">&quot;n&quot;</span>,<span class="at">cex=</span><span class="fl">0.8</span>)</span>
<span id="cb18-6"><a href="model-selection-of-character-evolution.html#cb18-6" aria-hidden="true" tabindex="-1"></a><span class="co"># add marginal ancestral states</span></span>
<span id="cb18-7"><a href="model-selection-of-character-evolution.html#cb18-7" aria-hidden="true" tabindex="-1"></a><span class="fu">nodelabels</span>(<span class="at">pie=</span>eel.anc.fit<span class="sc">$</span>states,<span class="at">piecol=</span>colPattern,<span class="at">cex=</span><span class="fl">0.5</span>)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/asr101-1.png" width="672" />
We can also generate a random evolutionary history based on the internal node reconstruction from a corHMM object:</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="model-selection-of-character-evolution.html#cb19-1" aria-hidden="true" tabindex="-1"></a>eel.model<span class="ot">&lt;-</span>eel.anc.fit<span class="sc">$</span>solution</span>
<span id="cb19-2"><a href="model-selection-of-character-evolution.html#cb19-2" aria-hidden="true" tabindex="-1"></a>eelFeedMapped <span class="ot">&lt;-</span> <span class="fu">makeSimmap</span>(<span class="at">tree=</span>eel.tree, <span class="at">data =</span> feedDat, <span class="at">mode=</span>eel.model,<span class="at">rate.cat=</span><span class="dv">1</span>,</span>
<span id="cb19-3"><a href="model-selection-of-character-evolution.html#cb19-3" aria-hidden="true" tabindex="-1"></a>                            <span class="at">nSim=</span><span class="dv">1</span>, <span class="at">nCores=</span><span class="dv">1</span>)</span>
<span id="cb19-4"><a href="model-selection-of-character-evolution.html#cb19-4" aria-hidden="true" tabindex="-1"></a><span class="fu">plotSimmap</span>(eelFeedMapped[[<span class="dv">1</span>]],<span class="at">col=</span>colPattern,<span class="at">fsize=</span><span class="fl">0.45</span>)</span></code></pre></div>
</div>
<div id="more-than-two-states" class="section level3 hasAnchor" number="19.2.3">
<h3><span class="header-section-number">19.2.3</span> More than two states<a href="model-selection-of-character-evolution.html#more-than-two-states" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Let’s revisit the ecomorphs of anoles lizards from the previous lab. There are six different ecomorphs present in the group.</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="model-selection-of-character-evolution.html#cb20-1" aria-hidden="true" tabindex="-1"></a>anole.ecomorph <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;http://www.phytools.org/Rbook/1/ecomorph.csv&quot;</span>,<span class="at">row.names=</span><span class="dv">1</span>,</span>
<span id="cb20-2"><a href="model-selection-of-character-evolution.html#cb20-2" aria-hidden="true" tabindex="-1"></a>                           <span class="at">stringsAsFactors =</span> T)</span>
<span id="cb20-3"><a href="model-selection-of-character-evolution.html#cb20-3" aria-hidden="true" tabindex="-1"></a>anole.tree <span class="ot">&lt;-</span> <span class="fu">read.tree</span>(<span class="st">&quot;http://www.phytools.org/Rbook/1/Anolis.tre&quot;</span>)</span>
<span id="cb20-4"><a href="model-selection-of-character-evolution.html#cb20-4" aria-hidden="true" tabindex="-1"></a>anoleNames <span class="ot">&lt;-</span> <span class="fu">name.check</span>(anole.tree, anole.ecomorph)</span>
<span id="cb20-5"><a href="model-selection-of-character-evolution.html#cb20-5" aria-hidden="true" tabindex="-1"></a>anoleNames</span></code></pre></div>
<pre><code>## $tree_not_data
##  [1] &quot;argenteolus&quot;    &quot;argillaceus&quot;    &quot;barbatus&quot;       &quot;barbouri&quot;      
##  [5] &quot;bartschi&quot;       &quot;centralis&quot;      &quot;chamaeleonides&quot; &quot;christophei&quot;   
##  [9] &quot;etheridgei&quot;     &quot;eugenegrahami&quot;  &quot;fowleri&quot;        &quot;guamuhaya&quot;     
## [13] &quot;lucius&quot;         &quot;monticola&quot;      &quot;porcus&quot;         &quot;pumilis&quot;       
## [17] &quot;reconditus&quot;     &quot;vermiculatus&quot;  
## 
## $data_not_tree
## character(0)</code></pre>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="model-selection-of-character-evolution.html#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="co"># There are taxa in tree, but not in data. We need to remove these tips</span></span>
<span id="cb22-2"><a href="model-selection-of-character-evolution.html#cb22-2" aria-hidden="true" tabindex="-1"></a>anole.tree.pruned <span class="ot">&lt;-</span> <span class="fu">drop.tip</span>(anole.tree, anoleNames<span class="sc">$</span>tree_not_data)</span>
<span id="cb22-3"><a href="model-selection-of-character-evolution.html#cb22-3" aria-hidden="true" tabindex="-1"></a><span class="fu">name.check</span>(anole.tree.pruned, anole.ecomorph)</span></code></pre></div>
<pre><code>## [1] &quot;OK&quot;</code></pre>
<p>Again, we first test which model fits the best. With six states, there are a lot more than four models that we enumerated in the previous example. Here we are just going to test three, but you can test more of them.</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="model-selection-of-character-evolution.html#cb24-1" aria-hidden="true" tabindex="-1"></a>ecomorphFactor <span class="ot">&lt;-</span> <span class="fu">setNames</span>(anole.ecomorph<span class="sc">$</span>ecomorph, <span class="fu">rownames</span>(anole.ecomorph))</span>
<span id="cb24-2"><a href="model-selection-of-character-evolution.html#cb24-2" aria-hidden="true" tabindex="-1"></a>fitER<span class="ot">&lt;-</span><span class="fu">fitMk</span>(anole.tree.pruned, ecomorphFactor,<span class="at">model=</span><span class="st">&quot;ER&quot;</span>)</span>
<span id="cb24-3"><a href="model-selection-of-character-evolution.html#cb24-3" aria-hidden="true" tabindex="-1"></a><span class="do">## Fit symmetric rate model</span></span>
<span id="cb24-4"><a href="model-selection-of-character-evolution.html#cb24-4" aria-hidden="true" tabindex="-1"></a>fitSYM<span class="ot">&lt;-</span> <span class="fu">fitMk</span>(anole.tree.pruned, ecomorphFactor,<span class="at">model=</span><span class="st">&quot;SYM&quot;</span>)</span>
<span id="cb24-5"><a href="model-selection-of-character-evolution.html#cb24-5" aria-hidden="true" tabindex="-1"></a><span class="do">## Fit ARD model</span></span>
<span id="cb24-6"><a href="model-selection-of-character-evolution.html#cb24-6" aria-hidden="true" tabindex="-1"></a>fitARD<span class="ot">&lt;-</span> <span class="fu">fitMk</span>(anole.tree.pruned, ecomorphFactor,<span class="at">model=</span><span class="st">&quot;ARD&quot;</span>)</span></code></pre></div>
</div>
</div>
<div id="exercise-i-2" class="section level2 hasAnchor" number="19.3">
<h2><span class="header-section-number">19.3</span> Exercise I<a href="model-selection-of-character-evolution.html#exercise-i-2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ol style="list-style-type: decimal">
<li><p>Compare the three models of ecomorph evolution in Anoles lizards: which model is best supported by AICs?</p></li>
<li><p>Use the best supported model and reconstruct marginal ancestral states. (Hint: always set <code>rate.cat=1</code> if you think your extant tips included all the possible states of the character. If you know there are unobserved hidden states, you can increase <code>rate.cat</code>. If you are using the standard models, ER, SYM, ARD, you can directly supply <code>model = "SYM" or model = "ER"</code> instead of the <code>rate.mat</code>; Otherwise, if you test other types of transition rates matrix, then you need to provide a rate indexing matrix)</p></li>
<li><p>Plot the marginal ancestral states as pies for internal nodes on the Anoles phylogeny.</p></li>
</ol>
</div>
<div id="continuous-traits-model-fitting" class="section level2 hasAnchor" number="19.4">
<h2><span class="header-section-number">19.4</span> Continuous traits model fitting<a href="model-selection-of-character-evolution.html#continuous-traits-model-fitting" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>For continuous traits, we are going to use <code>fitContinuous</code> function from the package <code>geiger</code> to test the fit of different evolutionary models of characters. Here we are inspecting the length data of eel species.</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="model-selection-of-character-evolution.html#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(eel.data<span class="sc">$</span>Max_TL_cm)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/asr300-1.png" width="672" /></p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="model-selection-of-character-evolution.html#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(<span class="fu">log</span>(eel.data<span class="sc">$</span>Max_TL_cm))</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/asr300-2.png" width="672" /></p>
<p>The original length data is very skewed. We therefore log-transformed the original length data as the input of model fit.</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="model-selection-of-character-evolution.html#cb27-1" aria-hidden="true" tabindex="-1"></a>lnTL <span class="ot">&lt;-</span> <span class="fu">setNames</span>(<span class="fu">log</span>(eel.data<span class="sc">$</span>Max_TL_cm), <span class="fu">rownames</span>(eel.data))</span></code></pre></div>
<div id="model-selection" class="section level3 hasAnchor" number="19.4.1">
<h3><span class="header-section-number">19.4.1</span> Model selection<a href="model-selection-of-character-evolution.html#model-selection" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>We then compare the fits of the brownian motion model against other ones, such as OU model or Early-burst model.</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="model-selection-of-character-evolution.html#cb28-1" aria-hidden="true" tabindex="-1"></a>?fitContinuous</span></code></pre></div>
<p>Read the section of possible models and choose the ones appropriate for your system of interests for comparison. Here, let’s test <code>BM</code>, <code>OU</code> and <code>EB</code>.</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="model-selection-of-character-evolution.html#cb29-1" aria-hidden="true" tabindex="-1"></a>tl_BM<span class="ot">&lt;-</span><span class="fu">fitContinuous</span>(eel.tree, lnTL,<span class="at">model=</span><span class="st">&quot;BM&quot;</span>)</span>
<span id="cb29-2"><a href="model-selection-of-character-evolution.html#cb29-2" aria-hidden="true" tabindex="-1"></a>tl_OU<span class="ot">&lt;-</span><span class="fu">fitContinuous</span>(eel.tree, lnTL,<span class="at">model=</span><span class="st">&quot;OU&quot;</span>)</span>
<span id="cb29-3"><a href="model-selection-of-character-evolution.html#cb29-3" aria-hidden="true" tabindex="-1"></a>tl_EB<span class="ot">&lt;-</span><span class="fu">fitContinuous</span>(eel.tree, lnTL,<span class="at">model=</span><span class="st">&quot;EB&quot;</span>)</span></code></pre></div>
</div>
<div id="ancestral-states" class="section level3 hasAnchor" number="19.4.2">
<h3><span class="header-section-number">19.4.2</span> Ancestral states<a href="model-selection-of-character-evolution.html#ancestral-states" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>From the AIC comparisons, it seems that the OU model is best supported. We then estimate the ancestral states by the MCMC searches. The advantage of this function is that we can provide priors of internal nodes if we know some of the ancestral nodes from fossil records. In the function of <code>fitContinuousMCMC</code>, There’s no model called “OU”. We can specify “Trend” to represent BM with a directional trend or “SSP”, a type of OU in which the characters are evolving towards a single optimum.</p>
<p>Default number of generation is 10^6. If you’d like to have a shorter MCMC chain, you can add <code>Ngens=1e+05</code> or less.</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="model-selection-of-character-evolution.html#cb30-1" aria-hidden="true" tabindex="-1"></a>tl_OUMC<span class="ot">&lt;-</span><span class="fu">fitContinuousMCMC</span>(eel.tree, lnTL,<span class="at">model=</span><span class="st">&quot;Trend&quot;</span>, <span class="at">root.prior=</span><span class="fu">c</span>(tl_OU<span class="sc">$</span>opt<span class="sc">$</span>alpha<span class="sc">-</span>(<span class="fu">abs</span>(tl_OU<span class="sc">$</span>opt<span class="sc">$</span>alpha)<span class="sc">/</span><span class="dv">10</span>),tl_OU<span class="sc">$</span>opt<span class="sc">$</span>alpha<span class="sc">+</span>(<span class="fu">abs</span>(tl_OU<span class="sc">$</span>opt<span class="sc">$</span>alpha)<span class="sc">/</span><span class="dv">10</span>),<span class="st">&quot;uniform&quot;</span>), <span class="at">outputName =</span> <span class="st">&quot;tl_OU_mcmc.output&quot;</span>)</span></code></pre></div>
<p>The model writes out two files, one with model parameters and one with estimated node states. Similar to <code>anc.Bayes</code>, we will get the ancestral state estimation by taking the average of the MCMC output:</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="model-selection-of-character-evolution.html#cb31-1" aria-hidden="true" tabindex="-1"></a>mcTLlikelihood <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;tl_OU_mcmc.output_model_params.txt&quot;</span>,<span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span>
<span id="cb31-2"><a href="model-selection-of-character-evolution.html#cb31-2" aria-hidden="true" tabindex="-1"></a>nodeState<span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;tl_OU_mcmc.output_nodestates.txt&quot;</span>,<span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span>
<span id="cb31-3"><a href="model-selection-of-character-evolution.html#cb31-3" aria-hidden="true" tabindex="-1"></a>eel.anc.est<span class="ot">&lt;-</span><span class="fu">colMeans</span>(nodeState[<span class="dv">201</span><span class="sc">:</span><span class="dv">1001</span>,])</span>
<span id="cb31-4"><a href="model-selection-of-character-evolution.html#cb31-4" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(eel.anc.est)<span class="ot">&lt;-</span><span class="fu">as.character</span>(<span class="dv">1</span><span class="sc">:</span>eel.tree<span class="sc">$</span>Nnode<span class="sc">+</span><span class="fu">length</span>(eel.tree<span class="sc">$</span>tip.label))</span></code></pre></div>
<p>We then use the <code>contMap</code> in phytools to generate an interpolated character evolution history</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="model-selection-of-character-evolution.html#cb32-1" aria-hidden="true" tabindex="-1"></a>eel_mcmc_mapped<span class="ot">&lt;-</span><span class="fu">contMap</span>(eel.tree, lnTL,<span class="at">anc.states=</span>eel.anc.est)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/ars400-1.png" width="672" /></p>
<p>We can export this mapped phylogeny by <code>writeNexus</code>, which can read into various tree manipulation programs for later analyses.</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="model-selection-of-character-evolution.html#cb33-1" aria-hidden="true" tabindex="-1"></a>?writeNexus</span>
<span id="cb33-2"><a href="model-selection-of-character-evolution.html#cb33-2" aria-hidden="true" tabindex="-1"></a><span class="fu">writeNexus</span>(eel_mcmc_mapped<span class="sc">$</span>tree, <span class="at">file=</span><span class="st">&quot;eel_OU_mapped.nexus&quot;</span>)</span></code></pre></div>
</div>
</div>
<div id="exercise-ii-2" class="section level2 hasAnchor" number="19.5">
<h2><span class="header-section-number">19.5</span> Exercise II<a href="model-selection-of-character-evolution.html#exercise-ii-2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Inspect the <code>nodeState</code> for the reconstructed states above, how many of the ancestral reconstructed nodes show that the ancestral eel length is longer than 80cm? (Hint: for an ancestral node x, we try to test if most of the reconstructed ancestral state (let’s say 2.5 percentile of the distribution) is longer than 80cm from the mcmc samples: quantile(nodeState[x, 201:1001], 0.025) &gt; log(80) )</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="character-evolution-and-ancestral-state-reconstruction.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/18_asr2.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["bookdown-demo.pdf", "bookdown-demo.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
