<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 10 Phylogenetic Visualization in R | BIO595 The Tree of Life</title>
  <meta name="description" content="Test writing the book for the course bookdown::gitbook." />
  <meta name="generator" content="bookdown 0.28 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 10 Phylogenetic Visualization in R | BIO595 The Tree of Life" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="Test writing the book for the course bookdown::gitbook." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 10 Phylogenetic Visualization in R | BIO595 The Tree of Life" />
  
  <meta name="twitter:description" content="Test writing the book for the course bookdown::gitbook." />
  

<meta name="author" content="Qixin He, Daniel Park" />


<meta name="date" content="2023-04-10" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="coevolution-and-phylogenetic-signals-in-r.html"/>
<link rel="next" href="gene-tree-vs-species-tree.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Tree of Life: Phylogenetic inferences and applications</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Overview</a></li>
<li class="chapter" data-level="2" data-path="unix.html"><a href="unix.html"><i class="fa fa-check"></i><b>2</b> Unix</a>
<ul>
<li class="chapter" data-level="2.1" data-path="unix.html"><a href="unix.html#what-is-unix-and-why-do-we-use-it"><i class="fa fa-check"></i><b>2.1</b> What is Unix and why do we use it?</a></li>
<li class="chapter" data-level="2.2" data-path="unix.html"><a href="unix.html#installation"><i class="fa fa-check"></i><b>2.2</b> Installation</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="unix.html"><a href="unix.html#mac-os-and-linux"><i class="fa fa-check"></i><b>2.2.1</b> Mac OS and Linux</a></li>
<li class="chapter" data-level="2.2.2" data-path="unix.html"><a href="unix.html#windows"><i class="fa fa-check"></i><b>2.2.2</b> Windows</a></li>
<li class="chapter" data-level="2.2.3" data-path="unix.html"><a href="unix.html#use-scholar-remote-desktop-during-the-class"><i class="fa fa-check"></i><b>2.2.3</b> Use Scholar Remote Desktop During the Class</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="unix.html"><a href="unix.html#operation"><i class="fa fa-check"></i><b>2.3</b> Operation</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="unix.html"><a href="unix.html#starting-the-shell"><i class="fa fa-check"></i><b>2.3.1</b> Starting the Shell</a></li>
<li class="chapter" data-level="2.3.2" data-path="unix.html"><a href="unix.html#directoryfolder-sturcture"><i class="fa fa-check"></i><b>2.3.2</b> Directory/folder sturcture</a></li>
<li class="chapter" data-level="2.3.3" data-path="unix.html"><a href="unix.html#absolute-and-relative-directory"><i class="fa fa-check"></i><b>2.3.3</b> Absolute and Relative directory</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="unix.html"><a href="unix.html#handling-directories-and-files"><i class="fa fa-check"></i><b>2.4</b> Handling directories and files</a></li>
<li class="chapter" data-level="2.5" data-path="unix.html"><a href="unix.html#excercises-i"><i class="fa fa-check"></i><b>2.5</b> Excercises I</a></li>
<li class="chapter" data-level="2.6" data-path="unix.html"><a href="unix.html#viewing-and-processing-text-files"><i class="fa fa-check"></i><b>2.6</b> Viewing and processing text files</a></li>
<li class="chapter" data-level="2.7" data-path="unix.html"><a href="unix.html#redirecting-printouts-and-piping"><i class="fa fa-check"></i><b>2.7</b> Redirecting printouts and piping</a></li>
<li class="chapter" data-level="2.8" data-path="unix.html"><a href="unix.html#exercises-ii"><i class="fa fa-check"></i><b>2.8</b> Exercises II</a></li>
<li class="chapter" data-level="2.9" data-path="unix.html"><a href="unix.html#useful-wildcards"><i class="fa fa-check"></i><b>2.9</b> Useful wildcards</a></li>
<li class="chapter" data-level="2.10" data-path="unix.html"><a href="unix.html#exercises-iii"><i class="fa fa-check"></i><b>2.10</b> Exercises III</a></li>
<li class="chapter" data-level="2.11" data-path="unix.html"><a href="unix.html#search-and-select-lines-using-grep"><i class="fa fa-check"></i><b>2.11</b> Search and select lines using <em>grep</em></a></li>
<li class="chapter" data-level="2.12" data-path="unix.html"><a href="unix.html#substituting-characters-using-tr"><i class="fa fa-check"></i><b>2.12</b> Substituting characters using <em>tr</em></a></li>
<li class="chapter" data-level="2.13" data-path="unix.html"><a href="unix.html#cut-sort-and-unique"><i class="fa fa-check"></i><b>2.13</b> cut, sort and unique</a></li>
<li class="chapter" data-level="2.14" data-path="unix.html"><a href="unix.html#exercises-iv"><i class="fa fa-check"></i><b>2.14</b> Exercises IV</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="r-basics-i.html"><a href="r-basics-i.html"><i class="fa fa-check"></i><b>3</b> R Basics I</a>
<ul>
<li class="chapter" data-level="3.1" data-path="r-basics-i.html"><a href="r-basics-i.html#what-is-programming"><i class="fa fa-check"></i><b>3.1</b> What is programming?</a></li>
<li class="chapter" data-level="3.2" data-path="r-basics-i.html"><a href="r-basics-i.html#r"><i class="fa fa-check"></i><b>3.2</b> <code>R</code></a></li>
<li class="chapter" data-level="3.3" data-path="r-basics-i.html"><a href="r-basics-i.html#installation-1"><i class="fa fa-check"></i><b>3.3</b> Installation</a></li>
<li class="chapter" data-level="3.4" data-path="r-basics-i.html"><a href="r-basics-i.html#variables"><i class="fa fa-check"></i><b>3.4</b> Variables</a>
<ul>
<li class="chapter" data-level="3.4.1" data-path="r-basics-i.html"><a href="r-basics-i.html#exercise-1"><i class="fa fa-check"></i><b>3.4.1</b> Exercise (1):</a></li>
<li class="chapter" data-level="3.4.2" data-path="r-basics-i.html"><a href="r-basics-i.html#variable-naming-conventions"><i class="fa fa-check"></i><b>3.4.2</b> Variable naming conventions</a></li>
<li class="chapter" data-level="3.4.3" data-path="r-basics-i.html"><a href="r-basics-i.html#arithmetic-operators"><i class="fa fa-check"></i><b>3.4.3</b> Arithmetic Operators</a></li>
<li class="chapter" data-level="3.4.4" data-path="r-basics-i.html"><a href="r-basics-i.html#logical-operators"><i class="fa fa-check"></i><b>3.4.4</b> Logical Operators</a></li>
<li class="chapter" data-level="3.4.5" data-path="r-basics-i.html"><a href="r-basics-i.html#basic-built-in-mathematical-functions"><i class="fa fa-check"></i><b>3.4.5</b> Basic built-in mathematical functions</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="r-basics-i.html"><a href="r-basics-i.html#subsetting-data"><i class="fa fa-check"></i><b>3.5</b> Subsetting data</a></li>
<li class="chapter" data-level="3.6" data-path="r-basics-i.html"><a href="r-basics-i.html#logical-values"><i class="fa fa-check"></i><b>3.6</b> Logical values</a>
<ul>
<li class="chapter" data-level="3.6.1" data-path="r-basics-i.html"><a href="r-basics-i.html#missing-dealing-with-na-values"><i class="fa fa-check"></i><b>3.6.1</b> Missing: dealing with <code>NA</code> values</a></li>
</ul></li>
<li class="chapter" data-level="3.7" data-path="r-basics-i.html"><a href="r-basics-i.html#generating-data"><i class="fa fa-check"></i><b>3.7</b> Generating Data</a></li>
<li class="chapter" data-level="3.8" data-path="r-basics-i.html"><a href="r-basics-i.html#data-structures-and-types"><i class="fa fa-check"></i><b>3.8</b> Data Structures and Types</a>
<ul>
<li class="chapter" data-level="3.8.1" data-path="r-basics-i.html"><a href="r-basics-i.html#vectors"><i class="fa fa-check"></i><b>3.8.1</b> Vectors</a></li>
</ul></li>
<li class="chapter" data-level="3.9" data-path="r-basics-i.html"><a href="r-basics-i.html#matrices"><i class="fa fa-check"></i><b>3.9</b> Matrices</a>
<ul>
<li class="chapter" data-level="3.9.1" data-path="r-basics-i.html"><a href="r-basics-i.html#exercise-2"><i class="fa fa-check"></i><b>3.9.1</b> Exercise (2):</a></li>
</ul></li>
<li class="chapter" data-level="3.10" data-path="r-basics-i.html"><a href="r-basics-i.html#data-frames"><i class="fa fa-check"></i><b>3.10</b> Data Frames</a>
<ul>
<li class="chapter" data-level="3.10.1" data-path="r-basics-i.html"><a href="r-basics-i.html#exercise-3"><i class="fa fa-check"></i><b>3.10.1</b> Exercise (3):</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="r-basics-ii.html"><a href="r-basics-ii.html"><i class="fa fa-check"></i><b>4</b> R Basics II</a>
<ul>
<li class="chapter" data-level="4.1" data-path="r-basics-ii.html"><a href="r-basics-ii.html#setting-up-the-environment"><i class="fa fa-check"></i><b>4.1</b> Setting up the environment</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="r-basics-ii.html"><a href="r-basics-ii.html#exercise-1-1"><i class="fa fa-check"></i><b>4.1.1</b> Exercise (1)</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="r-basics-ii.html"><a href="r-basics-ii.html#reading-in-data-from-files"><i class="fa fa-check"></i><b>4.2</b> Reading in data from files</a></li>
<li class="chapter" data-level="4.3" data-path="r-basics-ii.html"><a href="r-basics-ii.html#wrangling-data"><i class="fa fa-check"></i><b>4.3</b> Wrangling Data</a></li>
<li class="chapter" data-level="4.4" data-path="r-basics-ii.html"><a href="r-basics-ii.html#subsetting-by-columns-and-rows"><i class="fa fa-check"></i><b>4.4</b> Subsetting by columns and rows</a></li>
<li class="chapter" data-level="4.5" data-path="r-basics-ii.html"><a href="r-basics-ii.html#subsetting-with-conditions"><i class="fa fa-check"></i><b>4.5</b> Subsetting with conditions</a>
<ul>
<li class="chapter" data-level="4.5.1" data-path="r-basics-ii.html"><a href="r-basics-ii.html#exercise-2-1"><i class="fa fa-check"></i><b>4.5.1</b> Exercise (2)</a></li>
<li class="chapter" data-level="4.5.2" data-path="r-basics-ii.html"><a href="r-basics-ii.html#bonus-1"><i class="fa fa-check"></i><b>4.5.2</b> Bonus [1]</a></li>
</ul></li>
<li class="chapter" data-level="4.6" data-path="r-basics-ii.html"><a href="r-basics-ii.html#merging-data"><i class="fa fa-check"></i><b>4.6</b> Merging data</a></li>
<li class="chapter" data-level="4.7" data-path="r-basics-ii.html"><a href="r-basics-ii.html#saving-files-and-sessions"><i class="fa fa-check"></i><b>4.7</b> Saving files and sessions</a>
<ul>
<li class="chapter" data-level="4.7.1" data-path="r-basics-ii.html"><a href="r-basics-ii.html#exercise-3-1"><i class="fa fa-check"></i><b>4.7.1</b> Exercise (3)</a></li>
</ul></li>
<li class="chapter" data-level="4.8" data-path="r-basics-ii.html"><a href="r-basics-ii.html#the-for-loop"><i class="fa fa-check"></i><b>4.8</b> The <span class="smallcaps">for</span>-loop</a></li>
<li class="chapter" data-level="4.9" data-path="r-basics-ii.html"><a href="r-basics-ii.html#lists"><i class="fa fa-check"></i><b>4.9</b> Lists</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="tree-and-traits-in-r.html"><a href="tree-and-traits-in-r.html"><i class="fa fa-check"></i><b>5</b> Tree and Traits in R</a>
<ul>
<li class="chapter" data-level="5.1" data-path="tree-and-traits-in-r.html"><a href="tree-and-traits-in-r.html#setting-up-the-environment-1"><i class="fa fa-check"></i><b>5.1</b> Setting up the environment</a></li>
<li class="chapter" data-level="5.2" data-path="tree-and-traits-in-r.html"><a href="tree-and-traits-in-r.html#examining-phylogenetic-trees"><i class="fa fa-check"></i><b>5.2</b> Examining phylogenetic trees</a></li>
<li class="chapter" data-level="5.3" data-path="tree-and-traits-in-r.html"><a href="tree-and-traits-in-r.html#plotting-phylogenetic-trees"><i class="fa fa-check"></i><b>5.3</b> Plotting phylogenetic trees</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="tree-and-traits-in-r.html"><a href="tree-and-traits-in-r.html#exercise-1-2"><i class="fa fa-check"></i><b>5.3.1</b> Exercise (1)</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="tree-and-traits-in-r.html"><a href="tree-and-traits-in-r.html#associating-phylogenies-and-traits"><i class="fa fa-check"></i><b>5.4</b> Associating phylogenies and traits</a>
<ul>
<li class="chapter" data-level="5.4.1" data-path="tree-and-traits-in-r.html"><a href="tree-and-traits-in-r.html#saving-plots"><i class="fa fa-check"></i><b>5.4.1</b> Saving plots</a></li>
<li class="chapter" data-level="5.4.2" data-path="tree-and-traits-in-r.html"><a href="tree-and-traits-in-r.html#exercise-2-2"><i class="fa fa-check"></i><b>5.4.2</b> Exercise (2)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html"><i class="fa fa-check"></i><b>6</b> Biogeography and Macroevolution in R</a>
<ul>
<li class="chapter" data-level="6.1" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#setting-up-the-environment-2"><i class="fa fa-check"></i><b>6.1</b> Setting up the environment</a></li>
<li class="chapter" data-level="6.2" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#ancestral-state-reconstructions"><i class="fa fa-check"></i><b>6.2</b> Ancestral State Reconstructions</a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#bonus-2"><i class="fa fa-check"></i><b>6.2.1</b> Bonus [2]</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#continuous-traits"><i class="fa fa-check"></i><b>6.3</b> Continuous traits</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#maximum-likelihood"><i class="fa fa-check"></i><b>6.3.1</b> Maximum likelihood</a></li>
<li class="chapter" data-level="6.3.2" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#phylogenetically-independent-contrasts"><i class="fa fa-check"></i><b>6.3.2</b> Phylogenetically independent contrasts</a></li>
<li class="chapter" data-level="6.3.3" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#plotting-ancestral-states"><i class="fa fa-check"></i><b>6.3.3</b> Plotting ancestral states</a></li>
<li class="chapter" data-level="6.3.4" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#exercise-1-3"><i class="fa fa-check"></i><b>6.3.4</b> Exercise (1)</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#discrete-traits"><i class="fa fa-check"></i><b>6.4</b> Discrete traits</a>
<ul>
<li class="chapter" data-level="6.4.1" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#exercise-2-3"><i class="fa fa-check"></i><b>6.4.1</b> Exercise (2)</a></li>
<li class="chapter" data-level="6.4.2" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#maximum-likelihood-1"><i class="fa fa-check"></i><b>6.4.2</b> Maximum likelihood</a></li>
<li class="chapter" data-level="6.4.3" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#choosing-models"><i class="fa fa-check"></i><b>6.4.3</b> Choosing models</a></li>
</ul></li>
<li class="chapter" data-level="6.5" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#saving-multiple-plots-together"><i class="fa fa-check"></i><b>6.5</b> Saving multiple plots together</a>
<ul>
<li class="chapter" data-level="6.5.1" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#exercise-3-2"><i class="fa fa-check"></i><b>6.5.1</b> Exercise (3)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="phylogenetic-diversity-in-r.html"><a href="phylogenetic-diversity-in-r.html"><i class="fa fa-check"></i><b>7</b> Phylogenetic Diversity in R</a>
<ul>
<li class="chapter" data-level="7.1" data-path="phylogenetic-diversity-in-r.html"><a href="phylogenetic-diversity-in-r.html#setting-up-the-environment-3"><i class="fa fa-check"></i><b>7.1</b> Setting up the environment</a></li>
<li class="chapter" data-level="7.2" data-path="phylogenetic-diversity-in-r.html"><a href="phylogenetic-diversity-in-r.html#phylogenetic-diversity"><i class="fa fa-check"></i><b>7.2</b> Phylogenetic Diversity</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="phylogenetic-diversity-in-r.html"><a href="phylogenetic-diversity-in-r.html#preprocessing"><i class="fa fa-check"></i><b>7.2.1</b> Preprocessing</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="phylogenetic-diversity-in-r.html"><a href="phylogenetic-diversity-in-r.html#estimating-phylogenetic-diversity-metrics"><i class="fa fa-check"></i><b>7.3</b> Estimating phylogenetic diversity metrics</a>
<ul>
<li class="chapter" data-level="7.3.1" data-path="phylogenetic-diversity-in-r.html"><a href="phylogenetic-diversity-in-r.html#exercise-1-4"><i class="fa fa-check"></i><b>7.3.1</b> Exercise (1)</a></li>
<li class="chapter" data-level="7.3.2" data-path="phylogenetic-diversity-in-r.html"><a href="phylogenetic-diversity-in-r.html#exercise-2-4"><i class="fa fa-check"></i><b>7.3.2</b> Exercise (2)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="community-phylogenetics.html"><a href="community-phylogenetics.html"><i class="fa fa-check"></i><b>8</b> Community Phylogenetics</a>
<ul>
<li class="chapter" data-level="8.1" data-path="community-phylogenetics.html"><a href="community-phylogenetics.html#setting-up-the-environment-4"><i class="fa fa-check"></i><b>8.1</b> Setting up the environment</a></li>
<li class="chapter" data-level="8.2" data-path="community-phylogenetics.html"><a href="community-phylogenetics.html#definition-of-community-phylogenetics"><i class="fa fa-check"></i><b>8.2</b> Definition of Community Phylogenetics</a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="community-phylogenetics.html"><a href="community-phylogenetics.html#alpha-diversity"><i class="fa fa-check"></i><b>8.2.1</b> Alpha diversity</a></li>
<li class="chapter" data-level="8.2.2" data-path="community-phylogenetics.html"><a href="community-phylogenetics.html#beta-diversity"><i class="fa fa-check"></i><b>8.2.2</b> Beta diversity</a></li>
<li class="chapter" data-level="8.2.3" data-path="community-phylogenetics.html"><a href="community-phylogenetics.html#exercise-1-5"><i class="fa fa-check"></i><b>8.2.3</b> Exercise (1)</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="community-phylogenetics.html"><a href="community-phylogenetics.html#standardized-effect-sizes"><i class="fa fa-check"></i><b>8.3</b> Standardized effect sizes</a>
<ul>
<li class="chapter" data-level="8.3.1" data-path="community-phylogenetics.html"><a href="community-phylogenetics.html#exercise-2-5"><i class="fa fa-check"></i><b>8.3.1</b> Exercise (2)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="coevolution-and-phylogenetic-signals-in-r.html"><a href="coevolution-and-phylogenetic-signals-in-r.html"><i class="fa fa-check"></i><b>9</b> Coevolution and Phylogenetic Signals in R</a>
<ul>
<li class="chapter" data-level="9.1" data-path="coevolution-and-phylogenetic-signals-in-r.html"><a href="coevolution-and-phylogenetic-signals-in-r.html#setting-up-the-environment-5"><i class="fa fa-check"></i><b>9.1</b> Setting up the environment</a></li>
<li class="chapter" data-level="9.2" data-path="coevolution-and-phylogenetic-signals-in-r.html"><a href="coevolution-and-phylogenetic-signals-in-r.html#phylogenetic-signal"><i class="fa fa-check"></i><b>9.2</b> Phylogenetic Signal</a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="coevolution-and-phylogenetic-signals-in-r.html"><a href="coevolution-and-phylogenetic-signals-in-r.html#blombergs-k"><i class="fa fa-check"></i><b>9.2.1</b> Blomberg’s K</a></li>
<li class="chapter" data-level="9.2.2" data-path="coevolution-and-phylogenetic-signals-in-r.html"><a href="coevolution-and-phylogenetic-signals-in-r.html#pagels-lambda-λ"><i class="fa fa-check"></i><b>9.2.2</b> Pagel’s lambda (λ)</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="coevolution-and-phylogenetic-signals-in-r.html"><a href="coevolution-and-phylogenetic-signals-in-r.html#phylogenetically-independent-contrasts-1"><i class="fa fa-check"></i><b>9.3</b> Phylogenetically Independent Contrasts</a></li>
<li class="chapter" data-level="9.4" data-path="coevolution-and-phylogenetic-signals-in-r.html"><a href="coevolution-and-phylogenetic-signals-in-r.html#coevolution"><i class="fa fa-check"></i><b>9.4</b> Coevolution</a>
<ul>
<li class="chapter" data-level="9.4.1" data-path="coevolution-and-phylogenetic-signals-in-r.html"><a href="coevolution-and-phylogenetic-signals-in-r.html#plotting-tanglegrams"><i class="fa fa-check"></i><b>9.4.1</b> Plotting tanglegrams</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="phylogenetic-visualization-in-r.html"><a href="phylogenetic-visualization-in-r.html"><i class="fa fa-check"></i><b>10</b> Phylogenetic Visualization in R</a>
<ul>
<li class="chapter" data-level="10.1" data-path="phylogenetic-visualization-in-r.html"><a href="phylogenetic-visualization-in-r.html#setting-up-the-environment-6"><i class="fa fa-check"></i><b>10.1</b> Setting up the environment</a></li>
<li class="chapter" data-level="10.2" data-path="phylogenetic-visualization-in-r.html"><a href="phylogenetic-visualization-in-r.html#using-phytools"><i class="fa fa-check"></i><b>10.2</b> Using <code>phytools</code></a>
<ul>
<li class="chapter" data-level="10.2.1" data-path="phylogenetic-visualization-in-r.html"><a href="phylogenetic-visualization-in-r.html#phenograms"><i class="fa fa-check"></i><b>10.2.1</b> Phenograms</a></li>
<li class="chapter" data-level="10.2.2" data-path="phylogenetic-visualization-in-r.html"><a href="phylogenetic-visualization-in-r.html#mapping-continuous-characters-on-branches"><i class="fa fa-check"></i><b>10.2.2</b> Mapping continuous characters on branches</a></li>
<li class="chapter" data-level="10.2.3" data-path="phylogenetic-visualization-in-r.html"><a href="phylogenetic-visualization-in-r.html#plotting-images-on-phylogenies"><i class="fa fa-check"></i><b>10.2.3</b> Plotting images on phylogenies</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="phylogenetic-visualization-in-r.html"><a href="phylogenetic-visualization-in-r.html#using-ggtree"><i class="fa fa-check"></i><b>10.3</b> Using <code>ggtree</code></a>
<ul>
<li class="chapter" data-level="10.3.1" data-path="phylogenetic-visualization-in-r.html"><a href="phylogenetic-visualization-in-r.html#basic-plotting"><i class="fa fa-check"></i><b>10.3.1</b> Basic plotting</a></li>
<li class="chapter" data-level="10.3.2" data-path="phylogenetic-visualization-in-r.html"><a href="phylogenetic-visualization-in-r.html#labeling-tips-nodes-and-clades"><i class="fa fa-check"></i><b>10.3.2</b> Labeling tips, nodes, and clades</a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="phylogenetic-visualization-in-r.html"><a href="phylogenetic-visualization-in-r.html#plotting-maps"><i class="fa fa-check"></i><b>10.4</b> Plotting maps</a>
<ul>
<li class="chapter" data-level="10.4.1" data-path="phylogenetic-visualization-in-r.html"><a href="phylogenetic-visualization-in-r.html#choropleth-maps"><i class="fa fa-check"></i><b>10.4.1</b> Choropleth maps</a></li>
<li class="chapter" data-level="10.4.2" data-path="phylogenetic-visualization-in-r.html"><a href="phylogenetic-visualization-in-r.html#working-with-rasters"><i class="fa fa-check"></i><b>10.4.2</b> Working with rasters</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html"><i class="fa fa-check"></i><b>11</b> Gene tree vs Species tree</a>
<ul>
<li class="chapter" data-level="11.1" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#learning-goals"><i class="fa fa-check"></i><b>11.1</b> Learning goals:</a></li>
<li class="chapter" data-level="11.2" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#tracing-ancestry-within-one-population"><i class="fa fa-check"></i><b>11.2</b> Tracing ancestry within one population</a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#vary-pop-size-and-generation"><i class="fa fa-check"></i><b>11.2.1</b> Vary pop size and generation</a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#speciation-and-coalescence"><i class="fa fa-check"></i><b>11.3</b> Speciation and coalescence</a>
<ul>
<li class="chapter" data-level="11.3.1" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#two-species-coalescent-history"><i class="fa fa-check"></i><b>11.3.1</b> Two-species coalescent history</a></li>
</ul></li>
<li class="chapter" data-level="11.4" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#exercise-i"><i class="fa fa-check"></i><b>11.4</b> Exercise I</a></li>
<li class="chapter" data-level="11.5" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#gene-tree-locus-tree-and-species-tree"><i class="fa fa-check"></i><b>11.5</b> Gene tree, Locus tree and Species tree</a>
<ul>
<li class="chapter" data-level="11.5.1" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#loading-data-and-packages"><i class="fa fa-check"></i><b>11.5.1</b> loading data and packages</a></li>
</ul></li>
<li class="chapter" data-level="11.6" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#effects-of-incomplete-lineage-sorting-ils-alone"><i class="fa fa-check"></i><b>11.6</b> Effects of Incomplete Lineage Sorting (ILS) alone</a>
<ul>
<li class="chapter" data-level="11.6.1" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#generate-species-tree-with-no-extinction"><i class="fa fa-check"></i><b>11.6.1</b> generate species tree with no extinction</a></li>
<li class="chapter" data-level="11.6.2" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#simulate-gene-trees-using-the-multispecies-coalescent-process"><i class="fa fa-check"></i><b>11.6.2</b> simulate gene trees using the multispecies coalescent process</a></li>
</ul></li>
<li class="chapter" data-level="11.7" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#calculate-tree-distances"><i class="fa fa-check"></i><b>11.7</b> Calculate tree distances</a>
<ul>
<li class="chapter" data-level="11.7.1" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#rf-distances"><i class="fa fa-check"></i><b>11.7.1</b> RF distances</a></li>
<li class="chapter" data-level="11.7.2" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#inspect-pairwise-rf-distances-among-all-the-gene-trees-generated-under-one-guiding-species-tree"><i class="fa fa-check"></i><b>11.7.2</b> Inspect pairwise RF distances among all the gene trees generated under one guiding species tree</a></li>
</ul></li>
<li class="chapter" data-level="11.8" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#inspect-node-support-i.e.-subtrees-defined-by-a-split"><i class="fa fa-check"></i><b>11.8</b> Inspect node support (i.e., subtrees defined by a split)</a></li>
<li class="chapter" data-level="11.9" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#combined-effects-of-gene-duplication-loss-and-horizontal-gene-transfer-events"><i class="fa fa-check"></i><b>11.9</b> Combined effects of gene duplication, loss and horizontal gene transfer events</a></li>
<li class="chapter" data-level="11.10" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#exercise-ii"><i class="fa fa-check"></i><b>11.10</b> Exercise II</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="overview-of-phylogenetic-inferences.html"><a href="overview-of-phylogenetic-inferences.html"><i class="fa fa-check"></i><b>12</b> Overview of Phylogenetic Inferences</a>
<ul>
<li class="chapter" data-level="12.1" data-path="overview-of-phylogenetic-inferences.html"><a href="overview-of-phylogenetic-inferences.html#mega"><i class="fa fa-check"></i><b>12.1</b> <code>MEGA</code></a></li>
<li class="chapter" data-level="12.2" data-path="overview-of-phylogenetic-inferences.html"><a href="overview-of-phylogenetic-inferences.html#view-and-align-sequences"><i class="fa fa-check"></i><b>12.2</b> View and Align Sequences</a></li>
<li class="chapter" data-level="12.3" data-path="overview-of-phylogenetic-inferences.html"><a href="overview-of-phylogenetic-inferences.html#dnaprotein-evolution-model"><i class="fa fa-check"></i><b>12.3</b> DNA/protein evolution model</a></li>
<li class="chapter" data-level="12.4" data-path="overview-of-phylogenetic-inferences.html"><a href="overview-of-phylogenetic-inferences.html#phylogenetic-inference"><i class="fa fa-check"></i><b>12.4</b> Phylogenetic inference</a></li>
<li class="chapter" data-level="12.5" data-path="overview-of-phylogenetic-inferences.html"><a href="overview-of-phylogenetic-inferences.html#estimate-divergence-times"><i class="fa fa-check"></i><b>12.5</b> Estimate divergence times</a></li>
<li class="chapter" data-level="12.6" data-path="overview-of-phylogenetic-inferences.html"><a href="overview-of-phylogenetic-inferences.html#exercises"><i class="fa fa-check"></i><b>12.6</b> Exercises</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html"><i class="fa fa-check"></i><b>13</b> Resources and Guidelines for the Term Project</a>
<ul>
<li class="chapter" data-level="13.1" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#bioinformatics-databases"><i class="fa fa-check"></i><b>13.1</b> Bioinformatics Databases</a>
<ul>
<li class="chapter" data-level="13.1.1" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#ncbi-national-center-for-biotechnology-information"><i class="fa fa-check"></i><b>13.1.1</b> NCBI (National Center for Biotechnology Information)</a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#retrieving-genetic-resources-for-your-taxa"><i class="fa fa-check"></i><b>13.2</b> Retrieving genetic resources for your taxa</a>
<ul>
<li class="chapter" data-level="13.2.1" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#ncbi-taxonomy-database"><i class="fa fa-check"></i><b>13.2.1</b> NCBI taxonomy database</a></li>
<li class="chapter" data-level="13.2.2" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#individual-genes-from-the-whole-genome-data"><i class="fa fa-check"></i><b>13.2.2</b> Individual genes from the whole genome data</a></li>
<li class="chapter" data-level="13.2.3" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#get-homologous-sequences-from-related-species"><i class="fa fa-check"></i><b>13.2.3</b> Get homologous sequences from related species</a></li>
<li class="chapter" data-level="13.2.4" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#european-bioinformatics-institute-ebi"><i class="fa fa-check"></i><b>13.2.4</b> European Bioinformatics Institute (EBI)</a></li>
<li class="chapter" data-level="13.2.5" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#protein-data-bank"><i class="fa fa-check"></i><b>13.2.5</b> Protein Data Bank</a></li>
</ul></li>
<li class="chapter" data-level="13.3" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#biogeographic-databases"><i class="fa fa-check"></i><b>13.3</b> Biogeographic Databases</a>
<ul>
<li class="chapter" data-level="13.3.1" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#climate-data"><i class="fa fa-check"></i><b>13.3.1</b> Climate Data</a></li>
<li class="chapter" data-level="13.3.2" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#occurrence-data"><i class="fa fa-check"></i><b>13.3.2</b> Occurrence Data</a></li>
<li class="chapter" data-level="13.3.3" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#topographic-data"><i class="fa fa-check"></i><b>13.3.3</b> TOPOGRAPHIC DATA</a></li>
<li class="chapter" data-level="13.3.4" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#trait-data"><i class="fa fa-check"></i><b>13.3.4</b> TRAIT DATA</a></li>
</ul></li>
<li class="chapter" data-level="13.4" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#publication-databases"><i class="fa fa-check"></i><b>13.4</b> Publication Databases</a></li>
<li class="chapter" data-level="13.5" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#example-research-questions"><i class="fa fa-check"></i><b>13.5</b> Example research questions:</a>
<ul>
<li class="chapter" data-level="13.5.1" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#biogeography-and-phylogenetic-diversity"><i class="fa fa-check"></i><b>13.5.1</b> Biogeography and Phylogenetic Diversity:</a></li>
<li class="chapter" data-level="13.5.2" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#community-ecology"><i class="fa fa-check"></i><b>13.5.2</b> Community Ecology:</a></li>
<li class="chapter" data-level="13.5.3" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#coevolution-1"><i class="fa fa-check"></i><b>13.5.3</b> Coevolution:</a></li>
<li class="chapter" data-level="13.5.4" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#species-tree"><i class="fa fa-check"></i><b>13.5.4</b> Species tree:</a></li>
</ul></li>
<li class="chapter" data-level="13.6" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#how-to-submit-jobs-on-scholar-cluster-and-run-remotely"><i class="fa fa-check"></i><b>13.6</b> How to submit jobs on Scholar Cluster and run remotely</a></li>
<li class="chapter" data-level="13.7" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#term-project-proposal"><i class="fa fa-check"></i><b>13.7</b> Term project proposal</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><i class="fa fa-check"></i><b>14</b> Systematic Retrieval of orthologous DNA sequences form GenBank</a>
<ul>
<li class="chapter" data-level="14.1" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#introduction"><i class="fa fa-check"></i><b>14.1</b> Introduction</a></li>
<li class="chapter" data-level="14.2" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#installing-tools"><i class="fa fa-check"></i><b>14.2</b> Installing Tools</a>
<ul>
<li class="chapter" data-level="14.2.1" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#windows-1"><i class="fa fa-check"></i><b>14.2.1</b> Windows</a></li>
<li class="chapter" data-level="14.2.2" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#mac"><i class="fa fa-check"></i><b>14.2.2</b> Mac</a></li>
<li class="chapter" data-level="14.2.3" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#ncbi-blast-tools"><i class="fa fa-check"></i><b>14.2.3</b> NCBI BLAST+ Tools</a></li>
</ul></li>
<li class="chapter" data-level="14.3" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#using-phylotar"><i class="fa fa-check"></i><b>14.3</b> Using <code>phylotaR</code></a>
<ul>
<li class="chapter" data-level="14.3.1" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#pipeline"><i class="fa fa-check"></i><b>14.3.1</b> Pipeline</a></li>
<li class="chapter" data-level="14.3.2" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#running"><i class="fa fa-check"></i><b>14.3.2</b> Running</a></li>
<li class="chapter" data-level="14.3.3" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#restarting"><i class="fa fa-check"></i><b>14.3.3</b> Restarting</a></li>
<li class="chapter" data-level="14.3.4" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#cluster-i.e.-orthologous-genes-selection"><i class="fa fa-check"></i><b>14.3.4</b> Cluster (i.e., orthologous genes) selection</a></li>
</ul></li>
<li class="chapter" data-level="14.4" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#exercise-i-1"><i class="fa fa-check"></i><b>14.4</b> Exercise I</a></li>
<li class="chapter" data-level="14.5" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#using-annotationbustr"><i class="fa fa-check"></i><b>14.5</b> Using <code>AnnotationBustR</code></a>
<ul>
<li class="chapter" data-level="14.5.1" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#introduction-1"><i class="fa fa-check"></i><b>14.5.1</b> Introduction</a></li>
<li class="chapter" data-level="14.5.2" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#install-the-tools"><i class="fa fa-check"></i><b>14.5.2</b> Install the tools</a></li>
<li class="chapter" data-level="14.5.3" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#specify-search-terms-of-genomes"><i class="fa fa-check"></i><b>14.5.3</b> Specify search terms of genomes</a></li>
<li class="chapter" data-level="14.5.4" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#load-gene-search-terms-of-interest"><i class="fa fa-check"></i><b>14.5.4</b> Load Gene Search Terms of Interest</a></li>
<li class="chapter" data-level="14.5.5" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#extract-sequences-with-annotationbust"><i class="fa fa-check"></i><b>14.5.5</b> Extract sequences with AnnotationBust</a></li>
</ul></li>
<li class="chapter" data-level="14.6" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#exercise-ii-1"><i class="fa fa-check"></i><b>14.6</b> Exercise II</a></li>
<li class="chapter" data-level="14.7" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#combine-gene-information-from-the-two-analyses"><i class="fa fa-check"></i><b>14.7</b> Combine gene information from the two analyses</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="sequence-alignment.html"><a href="sequence-alignment.html"><i class="fa fa-check"></i><b>15</b> Sequence alignment</a>
<ul>
<li class="chapter" data-level="15.1" data-path="sequence-alignment.html"><a href="sequence-alignment.html#load-libraries"><i class="fa fa-check"></i><b>15.1</b> Load libraries</a></li>
<li class="chapter" data-level="15.2" data-path="sequence-alignment.html"><a href="sequence-alignment.html#perform-alignments"><i class="fa fa-check"></i><b>15.2</b> Perform Alignments</a></li>
<li class="chapter" data-level="15.3" data-path="sequence-alignment.html"><a href="sequence-alignment.html#processing-multiple-alignments"><i class="fa fa-check"></i><b>15.3</b> Processing Multiple Alignments</a></li>
<li class="chapter" data-level="15.4" data-path="sequence-alignment.html"><a href="sequence-alignment.html#build-your-first-nj-tree"><i class="fa fa-check"></i><b>15.4</b> Build your first NJ tree</a></li>
<li class="chapter" data-level="15.5" data-path="sequence-alignment.html"><a href="sequence-alignment.html#exercise"><i class="fa fa-check"></i><b>15.5</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="maximum-likelihood-model-tests-tree-and-branch-tests.html"><a href="maximum-likelihood-model-tests-tree-and-branch-tests.html"><i class="fa fa-check"></i><b>16</b> Maximum likelihood, model tests, tree and branch tests</a>
<ul>
<li class="chapter" data-level="16.1" data-path="maximum-likelihood-model-tests-tree-and-branch-tests.html"><a href="maximum-likelihood-model-tests-tree-and-branch-tests.html#learning-goals-1"><i class="fa fa-check"></i><b>16.1</b> Learning Goals</a></li>
<li class="chapter" data-level="16.2" data-path="maximum-likelihood-model-tests-tree-and-branch-tests.html"><a href="maximum-likelihood-model-tests-tree-and-branch-tests.html#iqtree"><i class="fa fa-check"></i><b>16.2</b> IQtree</a>
<ul>
<li class="chapter" data-level="16.2.1" data-path="maximum-likelihood-model-tests-tree-and-branch-tests.html"><a href="maximum-likelihood-model-tests-tree-and-branch-tests.html#version"><i class="fa fa-check"></i><b>16.2.1</b> Version</a></li>
<li class="chapter" data-level="16.2.2" data-path="maximum-likelihood-model-tests-tree-and-branch-tests.html"><a href="maximum-likelihood-model-tests-tree-and-branch-tests.html#find-the-best-explanatory-sequence-evolution-model"><i class="fa fa-check"></i><b>16.2.2</b> find the best explanatory sequence evolution model</a></li>
<li class="chapter" data-level="16.2.3" data-path="maximum-likelihood-model-tests-tree-and-branch-tests.html"><a href="maximum-likelihood-model-tests-tree-and-branch-tests.html#using-the-best-sequence-evolution-model"><i class="fa fa-check"></i><b>16.2.3</b> using the best sequence evolution model</a></li>
<li class="chapter" data-level="16.2.4" data-path="maximum-likelihood-model-tests-tree-and-branch-tests.html"><a href="maximum-likelihood-model-tests-tree-and-branch-tests.html#different-partitions-for-the-data-for-multiple-genes-or-different-codons"><i class="fa fa-check"></i><b>16.2.4</b> Different partitions for the data for multiple genes, or different codons</a></li>
</ul></li>
<li class="chapter" data-level="16.3" data-path="maximum-likelihood-model-tests-tree-and-branch-tests.html"><a href="maximum-likelihood-model-tests-tree-and-branch-tests.html#run-an-ml-tree-with-multiple-gene-files-in-a-folder"><i class="fa fa-check"></i><b>16.3</b> Run an ML tree with multiple gene files in a folder</a></li>
<li class="chapter" data-level="16.4" data-path="maximum-likelihood-model-tests-tree-and-branch-tests.html"><a href="maximum-likelihood-model-tests-tree-and-branch-tests.html#gene-concordance-factor"><i class="fa fa-check"></i><b>16.4</b> Gene concordance factor</a></li>
<li class="chapter" data-level="16.5" data-path="maximum-likelihood-model-tests-tree-and-branch-tests.html"><a href="maximum-likelihood-model-tests-tree-and-branch-tests.html#tree-topology-tests"><i class="fa fa-check"></i><b>16.5</b> Tree topology tests</a></li>
<li class="chapter" data-level="16.6" data-path="maximum-likelihood-model-tests-tree-and-branch-tests.html"><a href="maximum-likelihood-model-tests-tree-and-branch-tests.html#tree-branches-tests"><i class="fa fa-check"></i><b>16.6</b> Tree branches tests</a></li>
<li class="chapter" data-level="16.7" data-path="maximum-likelihood-model-tests-tree-and-branch-tests.html"><a href="maximum-likelihood-model-tests-tree-and-branch-tests.html#assignments"><i class="fa fa-check"></i><b>16.7</b> Assignments</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="molecular-clock-and-phylogenetic-dating.html"><a href="molecular-clock-and-phylogenetic-dating.html"><i class="fa fa-check"></i><b>17</b> Molecular Clock and Phylogenetic Dating</a>
<ul>
<li class="chapter" data-level="17.1" data-path="molecular-clock-and-phylogenetic-dating.html"><a href="molecular-clock-and-phylogenetic-dating.html#inferring-time-tree-with-tip-dates"><i class="fa fa-check"></i><b>17.1</b> Inferring time tree with tip dates</a></li>
<li class="chapter" data-level="17.2" data-path="molecular-clock-and-phylogenetic-dating.html"><a href="molecular-clock-and-phylogenetic-dating.html#calibrating-tree-using-ancestral-dates"><i class="fa fa-check"></i><b>17.2</b> Calibrating tree using ancestral dates</a>
<ul>
<li class="chapter" data-level="17.2.1" data-path="molecular-clock-and-phylogenetic-dating.html"><a href="molecular-clock-and-phylogenetic-dating.html#search-for-an-ml-tree"><i class="fa fa-check"></i><b>17.2.1</b> search for an ML tree</a></li>
<li class="chapter" data-level="17.2.2" data-path="molecular-clock-and-phylogenetic-dating.html"><a href="molecular-clock-and-phylogenetic-dating.html#root-calibration"><i class="fa fa-check"></i><b>17.2.2</b> Root calibration</a></li>
<li class="chapter" data-level="17.2.3" data-path="molecular-clock-and-phylogenetic-dating.html"><a href="molecular-clock-and-phylogenetic-dating.html#internal-nodes"><i class="fa fa-check"></i><b>17.2.3</b> Internal nodes</a></li>
<li class="chapter" data-level="17.2.4" data-path="molecular-clock-and-phylogenetic-dating.html"><a href="molecular-clock-and-phylogenetic-dating.html#confidence-interval"><i class="fa fa-check"></i><b>17.2.4</b> Confidence interval</a></li>
</ul></li>
<li class="chapter" data-level="17.3" data-path="molecular-clock-and-phylogenetic-dating.html"><a href="molecular-clock-and-phylogenetic-dating.html#exercise-4"><i class="fa fa-check"></i><b>17.3</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="character-evolution-and-ancestral-state-reconstruction.html"><a href="character-evolution-and-ancestral-state-reconstruction.html"><i class="fa fa-check"></i><b>18</b> Character evolution and Ancestral state reconstruction</a>
<ul>
<li class="chapter" data-level="18.1" data-path="character-evolution-and-ancestral-state-reconstruction.html"><a href="character-evolution-and-ancestral-state-reconstruction.html#continuous-characters"><i class="fa fa-check"></i><b>18.1</b> Continuous characters</a>
<ul>
<li class="chapter" data-level="18.1.1" data-path="character-evolution-and-ancestral-state-reconstruction.html"><a href="character-evolution-and-ancestral-state-reconstruction.html#load-libraries-1"><i class="fa fa-check"></i><b>18.1.1</b> load libraries</a></li>
<li class="chapter" data-level="18.1.2" data-path="character-evolution-and-ancestral-state-reconstruction.html"><a href="character-evolution-and-ancestral-state-reconstruction.html#tree-file-and-the-character-file"><i class="fa fa-check"></i><b>18.1.2</b> Tree file and the character file</a></li>
<li class="chapter" data-level="18.1.3" data-path="character-evolution-and-ancestral-state-reconstruction.html"><a href="character-evolution-and-ancestral-state-reconstruction.html#visualization-of-the-reconstructed-nodes"><i class="fa fa-check"></i><b>18.1.3</b> Visualization of the reconstructed nodes</a></li>
<li class="chapter" data-level="18.1.4" data-path="character-evolution-and-ancestral-state-reconstruction.html"><a href="character-evolution-and-ancestral-state-reconstruction.html#visualization-of-phenotypic-evolution"><i class="fa fa-check"></i><b>18.1.4</b> Visualization of phenotypic evolution</a></li>
</ul></li>
<li class="chapter" data-level="18.2" data-path="character-evolution-and-ancestral-state-reconstruction.html"><a href="character-evolution-and-ancestral-state-reconstruction.html#brownian-motion-vs-ornstein-uhlenbeck-process"><i class="fa fa-check"></i><b>18.2</b> Brownian Motion vs Ornstein-Uhlenbeck Process</a></li>
<li class="chapter" data-level="18.3" data-path="character-evolution-and-ancestral-state-reconstruction.html"><a href="character-evolution-and-ancestral-state-reconstruction.html#brownian-motion"><i class="fa fa-check"></i><b>18.3</b> Brownian Motion</a>
<ul>
<li class="chapter" data-level="18.3.1" data-path="character-evolution-and-ancestral-state-reconstruction.html"><a href="character-evolution-and-ancestral-state-reconstruction.html#ornsteinuhlenbeck-process"><i class="fa fa-check"></i><b>18.3.1</b> Ornstein–Uhlenbeck process</a></li>
<li class="chapter" data-level="18.3.2" data-path="character-evolution-and-ancestral-state-reconstruction.html"><a href="character-evolution-and-ancestral-state-reconstruction.html#incorporate-prior-knowledge-using-bayesian-estimates"><i class="fa fa-check"></i><b>18.3.2</b> Incorporate prior knowledge using Bayesian estimates</a></li>
</ul></li>
<li class="chapter" data-level="18.4" data-path="character-evolution-and-ancestral-state-reconstruction.html"><a href="character-evolution-and-ancestral-state-reconstruction.html#discrete-characters"><i class="fa fa-check"></i><b>18.4</b> Discrete characters</a></li>
<li class="chapter" data-level="18.5" data-path="character-evolution-and-ancestral-state-reconstruction.html"><a href="character-evolution-and-ancestral-state-reconstruction.html#exercises-1"><i class="fa fa-check"></i><b>18.5</b> Exercises:</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="model-selection-of-character-evolution.html"><a href="model-selection-of-character-evolution.html"><i class="fa fa-check"></i><b>19</b> Model Selection of Character Evolution</a>
<ul>
<li class="chapter" data-level="19.1" data-path="model-selection-of-character-evolution.html"><a href="model-selection-of-character-evolution.html#load-libraries-2"><i class="fa fa-check"></i><b>19.1</b> Load libraries</a></li>
<li class="chapter" data-level="19.2" data-path="model-selection-of-character-evolution.html"><a href="model-selection-of-character-evolution.html#discrete-traits-model-fitting"><i class="fa fa-check"></i><b>19.2</b> Discrete traits model fitting</a>
<ul>
<li class="chapter" data-level="19.2.1" data-path="model-selection-of-character-evolution.html"><a href="model-selection-of-character-evolution.html#choosing-a-character-model"><i class="fa fa-check"></i><b>19.2.1</b> Choosing a character model</a></li>
<li class="chapter" data-level="19.2.2" data-path="model-selection-of-character-evolution.html"><a href="model-selection-of-character-evolution.html#marginal-ancestral-state-reconstruction"><i class="fa fa-check"></i><b>19.2.2</b> Marginal ancestral state reconstruction</a></li>
<li class="chapter" data-level="19.2.3" data-path="model-selection-of-character-evolution.html"><a href="model-selection-of-character-evolution.html#more-than-two-states"><i class="fa fa-check"></i><b>19.2.3</b> More than two states</a></li>
</ul></li>
<li class="chapter" data-level="19.3" data-path="model-selection-of-character-evolution.html"><a href="model-selection-of-character-evolution.html#exercise-i-2"><i class="fa fa-check"></i><b>19.3</b> Exercise I</a></li>
<li class="chapter" data-level="19.4" data-path="model-selection-of-character-evolution.html"><a href="model-selection-of-character-evolution.html#continuous-traits-model-fitting"><i class="fa fa-check"></i><b>19.4</b> Continuous traits model fitting</a>
<ul>
<li class="chapter" data-level="19.4.1" data-path="model-selection-of-character-evolution.html"><a href="model-selection-of-character-evolution.html#model-selection"><i class="fa fa-check"></i><b>19.4.1</b> Model selection</a></li>
<li class="chapter" data-level="19.4.2" data-path="model-selection-of-character-evolution.html"><a href="model-selection-of-character-evolution.html#ancestral-states"><i class="fa fa-check"></i><b>19.4.2</b> Ancestral states</a></li>
</ul></li>
<li class="chapter" data-level="19.5" data-path="model-selection-of-character-evolution.html"><a href="model-selection-of-character-evolution.html#exercise-ii-2"><i class="fa fa-check"></i><b>19.5</b> Exercise II</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">BIO595 The Tree of Life</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="phylogenetic-visualization-in-r" class="section level1 hasAnchor" number="10">
<h1><span class="header-section-number">Chapter 10</span> Phylogenetic Visualization in R<a href="phylogenetic-visualization-in-r.html#phylogenetic-visualization-in-r" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="setting-up-the-environment-6" class="section level2 hasAnchor" number="10.1">
<h2><span class="header-section-number">10.1</span> Setting up the environment<a href="phylogenetic-visualization-in-r.html#setting-up-the-environment-6" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Set your working directory and load the necessary data.
We will be using a package that is not on the <code>CRAN</code> repository - follow the instructions below to install it.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="phylogenetic-visualization-in-r.html#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">require</span>(<span class="st">&quot;BiocManager&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>))</span>
<span id="cb1-2"><a href="phylogenetic-visualization-in-r.html#cb1-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">install.packages</span>(<span class="st">&quot;BiocManager&quot;</span>)</span>
<span id="cb1-3"><a href="phylogenetic-visualization-in-r.html#cb1-3" aria-hidden="true" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;ggtree&quot;</span>)</span>
<span id="cb1-4"><a href="phylogenetic-visualization-in-r.html#cb1-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-5"><a href="phylogenetic-visualization-in-r.html#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;phytools&quot;</span>)</span>
<span id="cb1-6"><a href="phylogenetic-visualization-in-r.html#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;ggtree&quot;</span>)</span>
<span id="cb1-7"><a href="phylogenetic-visualization-in-r.html#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;ggplot2&quot;</span>)</span>
<span id="cb1-8"><a href="phylogenetic-visualization-in-r.html#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;png&quot;</span>)</span>
<span id="cb1-9"><a href="phylogenetic-visualization-in-r.html#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;rgdal&quot;</span>)</span>
<span id="cb1-10"><a href="phylogenetic-visualization-in-r.html#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;raster&quot;</span>)</span>
<span id="cb1-11"><a href="phylogenetic-visualization-in-r.html#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;rgeos&quot;</span>)</span>
<span id="cb1-12"><a href="phylogenetic-visualization-in-r.html#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;maptools&quot;</span>)</span>
<span id="cb1-13"><a href="phylogenetic-visualization-in-r.html#cb1-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-14"><a href="phylogenetic-visualization-in-r.html#cb1-14" aria-hidden="true" tabindex="-1"></a><span class="co"># set working directory</span></span>
<span id="cb1-15"><a href="phylogenetic-visualization-in-r.html#cb1-15" aria-hidden="true" tabindex="-1"></a>wd <span class="ot">&lt;-</span> <span class="st">&quot;/Users/Bioinstructor/Downloads/lab11/&quot;</span> <span class="co"># absolute address of the folder where you have the data you need and plan to work in.</span></span>
<span id="cb1-16"><a href="phylogenetic-visualization-in-r.html#cb1-16" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(wd)</span></code></pre></div>
<p><br></p>
</div>
<div id="using-phytools" class="section level2 hasAnchor" number="10.2">
<h2><span class="header-section-number">10.2</span> Using <code>phytools</code><a href="phylogenetic-visualization-in-r.html#using-phytools" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The <code>phytools</code> package comprises many useful functions for visualizing phylogenies and other associated data.
You have already seen a few examples in previous labs.
Let’s try a few more.</p>
<div id="phenograms" class="section level3 hasAnchor" number="10.2.1">
<h3><span class="header-section-number">10.2.1</span> Phenograms<a href="phylogenetic-visualization-in-r.html#phenograms" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Phenograms are projections of a phylogenetic tree in a space defined by phenotype (on the y axis) and time (on the x - assuming your tree is a chronogram).
See Evans et al. (<a href="https://www.journals.uchicago.edu/doi/full/10.1086/595757">2009</a>) for more information.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="phylogenetic-visualization-in-r.html#cb2-1" aria-hidden="true" tabindex="-1"></a>traits <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">fastBM</span>(tree, <span class="at">nsim =</span> <span class="dv">5</span>)) <span class="co"># simulate 5 continuous traits via BM</span></span>
<span id="cb2-2"><a href="phylogenetic-visualization-in-r.html#cb2-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-3"><a href="phylogenetic-visualization-in-r.html#cb2-3" aria-hidden="true" tabindex="-1"></a>t3 <span class="ot">&lt;-</span> <span class="fu">abs</span>(traits<span class="sc">$</span>X3)</span>
<span id="cb2-4"><a href="phylogenetic-visualization-in-r.html#cb2-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-5"><a href="phylogenetic-visualization-in-r.html#cb2-5" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(t3) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(traits) <span class="co"># traits need to be named numbers</span></span>
<span id="cb2-6"><a href="phylogenetic-visualization-in-r.html#cb2-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-7"><a href="phylogenetic-visualization-in-r.html#cb2-7" aria-hidden="true" tabindex="-1"></a><span class="fu">phenogram</span>(tree, t3, <span class="at">spread.labels =</span> <span class="cn">TRUE</span>, <span class="at">spread.cost =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">0</span>))</span>
<span id="cb2-8"><a href="phylogenetic-visualization-in-r.html#cb2-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-9"><a href="phylogenetic-visualization-in-r.html#cb2-9" aria-hidden="true" tabindex="-1"></a><span class="fu">fancyTree</span>(tree, <span class="at">type =</span> <span class="st">&quot;phenogram95&quot;</span>, <span class="at">x =</span> t3, <span class="at">spread.cost =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">0</span>)) <span class="co">#visualizing the uncertainty around the reconstructed values at internal nodes</span></span>
<span id="cb2-10"><a href="phylogenetic-visualization-in-r.html#cb2-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-11"><a href="phylogenetic-visualization-in-r.html#cb2-11" aria-hidden="true" tabindex="-1"></a><span class="fu">fancyTree</span>(tree, <span class="at">type =</span> <span class="st">&quot;phenogram95&quot;</span>, <span class="at">x =</span> t3, <span class="at">spread.cost =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">0</span>), <span class="at">fsize =</span> <span class="dv">1</span>, <span class="at">ftype =</span> <span class="st">&quot;i&quot;</span>, <span class="at">lty =</span> <span class="st">&quot;dashed&quot;</span>, <span class="at">offset =</span> <span class="dv">1</span>)</span></code></pre></div>
</div>
<div id="mapping-continuous-characters-on-branches" class="section level3 hasAnchor" number="10.2.2">
<h3><span class="header-section-number">10.2.2</span> Mapping continuous characters on branches<a href="phylogenetic-visualization-in-r.html#mapping-continuous-characters-on-branches" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Previously we mapped character states on the tips of a tree.
We can also map the evolution of continuous characters on the branches, but to do so, we need to know the ancestral states of them.
Again, we previously estimated (ancestral) character states at internal nodes with maximum likelihood approaches.
The <code>contMap</code> function puts these two together.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="phylogenetic-visualization-in-r.html#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="do">## trait evolution as color gradients:</span></span>
<span id="cb3-2"><a href="phylogenetic-visualization-in-r.html#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="fu">contMap</span>(tree, t3)</span>
<span id="cb3-3"><a href="phylogenetic-visualization-in-r.html#cb3-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-4"><a href="phylogenetic-visualization-in-r.html#cb3-4" aria-hidden="true" tabindex="-1"></a><span class="fu">contMap</span>(tree, t3, <span class="at">fsize =</span> <span class="dv">1</span>, <span class="at">ftype =</span> <span class="st">&quot;i&quot;</span>, <span class="at">lwd =</span> <span class="dv">3</span>, <span class="at">offset =</span> .<span class="dv">5</span>)</span>
<span id="cb3-5"><a href="phylogenetic-visualization-in-r.html#cb3-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-6"><a href="phylogenetic-visualization-in-r.html#cb3-6" aria-hidden="true" tabindex="-1"></a><span class="co"># you can also pass arguments to the plot function and add plot elements</span></span>
<span id="cb3-7"><a href="phylogenetic-visualization-in-r.html#cb3-7" aria-hidden="true" tabindex="-1"></a>obj <span class="ot">&lt;-</span> <span class="fu">contMap</span>(tree, t3, <span class="at">plot =</span> <span class="cn">FALSE</span>)</span>
<span id="cb3-8"><a href="phylogenetic-visualization-in-r.html#cb3-8" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(obj, <span class="at">lwd =</span> <span class="dv">3</span>, <span class="at">fsize =</span> <span class="dv">1</span>, <span class="at">offset =</span> .<span class="dv">5</span>, <span class="at">ftype =</span> <span class="st">&quot;i&quot;</span>)</span>
<span id="cb3-9"><a href="phylogenetic-visualization-in-r.html#cb3-9" aria-hidden="true" tabindex="-1"></a><span class="fu">errorbar.contMap</span>(obj)</span>
<span id="cb3-10"><a href="phylogenetic-visualization-in-r.html#cb3-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-11"><a href="phylogenetic-visualization-in-r.html#cb3-11" aria-hidden="true" tabindex="-1"></a><span class="do">## 3(b)plotting images with phytools and png</span></span>
<span id="cb3-12"><a href="phylogenetic-visualization-in-r.html#cb3-12" aria-hidden="true" tabindex="-1"></a>anole.tree <span class="ot">&lt;-</span> <span class="fu">read.tree</span>(<span class="st">&quot;http://phytools.org/Cordoba2017/data/Anolis.tre&quot;</span>)</span>
<span id="cb3-13"><a href="phylogenetic-visualization-in-r.html#cb3-13" aria-hidden="true" tabindex="-1"></a>svl <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;http://phytools.org/Cordoba2017/data/svl.csv&quot;</span>, <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="dv">1</span>)</span></code></pre></div>
<p></br></p>
</div>
<div id="plotting-images-on-phylogenies" class="section level3 hasAnchor" number="10.2.3">
<h3><span class="header-section-number">10.2.3</span> Plotting images on phylogenies<a href="phylogenetic-visualization-in-r.html#plotting-images-on-phylogenies" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="phylogenetic-visualization-in-r.html#cb4-1" aria-hidden="true" tabindex="-1"></a>anole.tree <span class="ot">&lt;-</span> <span class="fu">read.tree</span>(<span class="st">&quot;http://phytools.org/Cordoba2017/data/Anolis.tre&quot;</span>)</span>
<span id="cb4-2"><a href="phylogenetic-visualization-in-r.html#cb4-2" aria-hidden="true" tabindex="-1"></a>svl <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;http://phytools.org/Cordoba2017/data/svl.csv&quot;</span>, <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb4-3"><a href="phylogenetic-visualization-in-r.html#cb4-3" aria-hidden="true" tabindex="-1"></a>svl <span class="ot">&lt;-</span> <span class="fu">setNames</span>(svl[,<span class="dv">1</span>], <span class="fu">rownames</span>(svl)) <span class="co">#transforming dataframe to a named number</span></span>
<span id="cb4-4"><a href="phylogenetic-visualization-in-r.html#cb4-4" aria-hidden="true" tabindex="-1"></a>obj <span class="ot">&lt;-</span> <span class="fu">contMap</span>(anole.tree,svl, <span class="at">plot =</span> <span class="cn">FALSE</span>)</span>
<span id="cb4-5"><a href="phylogenetic-visualization-in-r.html#cb4-5" aria-hidden="true" tabindex="-1"></a>obj <span class="ot">&lt;-</span> <span class="fu">setMap</span>(obj, <span class="at">invert =</span> <span class="cn">TRUE</span>) <span class="co">#sets color scheme - in this case, we&#39;re just inverting it.</span></span>
<span id="cb4-6"><a href="phylogenetic-visualization-in-r.html#cb4-6" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(obj, <span class="at">fsize =</span> <span class="fu">c</span>(<span class="fl">0.4</span>, <span class="dv">1</span>), <span class="at">outline =</span> <span class="cn">FALSE</span>, <span class="at">lwd=</span><span class="fu">c</span>(<span class="dv">3</span>, <span class="dv">7</span>), <span class="at">leg.txt =</span> <span class="st">&quot;log(SVL)&quot;</span>) <span class="co">#the numbers for fsize and lwd correspond to the main plot and legend, in order.</span></span>
<span id="cb4-7"><a href="phylogenetic-visualization-in-r.html#cb4-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-8"><a href="phylogenetic-visualization-in-r.html#cb4-8" aria-hidden="true" tabindex="-1"></a><span class="do">## images</span></span>
<span id="cb4-9"><a href="phylogenetic-visualization-in-r.html#cb4-9" aria-hidden="true" tabindex="-1"></a><span class="do">## http://phytools.org/Cordoba2017/data/barbaratus.png</span></span>
<span id="cb4-10"><a href="phylogenetic-visualization-in-r.html#cb4-10" aria-hidden="true" tabindex="-1"></a><span class="do">## http://phytools.org/Cordoba2017/data/porcatus.png</span></span>
<span id="cb4-11"><a href="phylogenetic-visualization-in-r.html#cb4-11" aria-hidden="true" tabindex="-1"></a><span class="do">## http://phytools.org/Cordoba2017/data/cristatellus.png</span></span>
<span id="cb4-12"><a href="phylogenetic-visualization-in-r.html#cb4-12" aria-hidden="true" tabindex="-1"></a><span class="do">## http://phytools.org/Cordoba2017/data/equestris.png</span></span>
<span id="cb4-13"><a href="phylogenetic-visualization-in-r.html#cb4-13" aria-hidden="true" tabindex="-1"></a><span class="do">## http://phytools.org/Cordoba2017/data/sagrei.png</span></span>
<span id="cb4-14"><a href="phylogenetic-visualization-in-r.html#cb4-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-15"><a href="phylogenetic-visualization-in-r.html#cb4-15" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(obj, <span class="at">type =</span> <span class="st">&quot;fan&quot;</span>, <span class="at">outline =</span> <span class="cn">FALSE</span>, <span class="at">fsize =</span> <span class="fu">c</span>(<span class="fl">0.6</span>, <span class="dv">1</span>),<span class="at">legend =</span> <span class="dv">1</span>,</span>
<span id="cb4-16"><a href="phylogenetic-visualization-in-r.html#cb4-16" aria-hidden="true" tabindex="-1"></a>     <span class="at">lwd =</span> <span class="fu">c</span>(<span class="dv">4</span>, <span class="dv">8</span>), <span class="at">xlim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="fl">1.8</span>, <span class="fl">1.8</span>), <span class="at">leg.txt =</span> <span class="st">&quot;log(SVL)&quot;</span>)</span>
<span id="cb4-17"><a href="phylogenetic-visualization-in-r.html#cb4-17" aria-hidden="true" tabindex="-1"></a>spp <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;barbatus&quot;</span>, <span class="st">&quot;porcatus&quot;</span>, <span class="st">&quot;cristatellus&quot;</span>, <span class="st">&quot;equestris&quot;</span>, <span class="st">&quot;sagrei&quot;</span>)</span>
<span id="cb4-18"><a href="phylogenetic-visualization-in-r.html#cb4-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-19"><a href="phylogenetic-visualization-in-r.html#cb4-19" aria-hidden="true" tabindex="-1"></a>w <span class="ot">&lt;-</span> <span class="fl">0.5</span></span>
<span id="cb4-20"><a href="phylogenetic-visualization-in-r.html#cb4-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-21"><a href="phylogenetic-visualization-in-r.html#cb4-21" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(spp)){</span>
<span id="cb4-22"><a href="phylogenetic-visualization-in-r.html#cb4-22" aria-hidden="true" tabindex="-1"></a>  xy <span class="ot">&lt;-</span> <span class="fu">add.arrow</span>(obj,spp[i],<span class="at">col=</span><span class="st">&quot;transparent&quot;</span>,<span class="at">arrl=</span><span class="fl">0.45</span>,<span class="at">lwd=</span><span class="dv">3</span>,<span class="at">hedl=</span><span class="fl">0.1</span>)</span>
<span id="cb4-23"><a href="phylogenetic-visualization-in-r.html#cb4-23" aria-hidden="true" tabindex="-1"></a>  arrl <span class="ot">&lt;-</span> <span class="cf">if</span>(spp[i] <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;sagrei&quot;</span>, <span class="st">&quot;porcatus&quot;</span>)){</span>
<span id="cb4-24"><a href="phylogenetic-visualization-in-r.html#cb4-24" aria-hidden="true" tabindex="-1"></a>    <span class="fl">0.24</span></span>
<span id="cb4-25"><a href="phylogenetic-visualization-in-r.html#cb4-25" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span>(spp[i]<span class="sc">==</span><span class="st">&quot;barbatus&quot;</span>){</span>
<span id="cb4-26"><a href="phylogenetic-visualization-in-r.html#cb4-26" aria-hidden="true" tabindex="-1"></a>      <span class="fl">0.2</span></span>
<span id="cb4-27"><a href="phylogenetic-visualization-in-r.html#cb4-27" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb4-28"><a href="phylogenetic-visualization-in-r.html#cb4-28" aria-hidden="true" tabindex="-1"></a>    <span class="fl">0.34</span></span>
<span id="cb4-29"><a href="phylogenetic-visualization-in-r.html#cb4-29" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb4-30"><a href="phylogenetic-visualization-in-r.html#cb4-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">add.arrow</span>(obj, spp[i], <span class="at">col =</span> <span class="st">&quot;blue&quot;</span>, <span class="at">arrl =</span> arrl, <span class="at">lwd =</span> <span class="dv">3</span>, <span class="at">hedl =</span> <span class="fl">0.05</span>)</span>
<span id="cb4-31"><a href="phylogenetic-visualization-in-r.html#cb4-31" aria-hidden="true" tabindex="-1"></a>  link <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">&quot;http://phytools.org/Cordoba2017/data/&quot;</span>, spp[i],<span class="st">&quot;.png&quot;</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb4-32"><a href="phylogenetic-visualization-in-r.html#cb4-32" aria-hidden="true" tabindex="-1"></a>  z <span class="ot">&lt;-</span> <span class="fu">tempfile</span>() <span class="co">#create temporary file</span></span>
<span id="cb4-33"><a href="phylogenetic-visualization-in-r.html#cb4-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">download.file</span>(link, z, <span class="at">mode =</span> <span class="st">&quot;wb&quot;</span>) <span class="co">#download image to temporary file</span></span>
<span id="cb4-34"><a href="phylogenetic-visualization-in-r.html#cb4-34" aria-hidden="true" tabindex="-1"></a>  img <span class="ot">&lt;-</span><span class="fu">readPNG</span>(z) <span class="co">#read in image</span></span>
<span id="cb4-35"><a href="phylogenetic-visualization-in-r.html#cb4-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">file.remove</span>(z) <span class="co">#remove temporary file</span></span>
<span id="cb4-36"><a href="phylogenetic-visualization-in-r.html#cb4-36" aria-hidden="true" tabindex="-1"></a>  asp <span class="ot">&lt;-</span><span class="fu">dim</span>(img)[<span class="dv">1</span>]<span class="sc">/</span><span class="fu">dim</span>(img)[<span class="dv">2</span>]</span>
<span id="cb4-37"><a href="phylogenetic-visualization-in-r.html#cb4-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rasterImage</span>(img, xy<span class="sc">$</span>x0 <span class="sc">-</span> w<span class="sc">/</span><span class="dv">2</span>, xy<span class="sc">$</span>y0 <span class="sc">-</span> w<span class="sc">/</span><span class="dv">2</span><span class="sc">*</span>asp, xy<span class="sc">$</span>x0 <span class="sc">+</span> w<span class="sc">/</span><span class="dv">2</span>, xy<span class="sc">$</span>y0 <span class="sc">+</span> w<span class="sc">/</span><span class="dv">2</span><span class="sc">*</span>asp)</span>
<span id="cb4-38"><a href="phylogenetic-visualization-in-r.html#cb4-38" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
<p>See the following link for more in-depth examples of <code>phytools</code> tree visualizations: <a href="http://phytools.org/Cordoba2017/ex/15/Plotting-methods.html" class="uri">http://phytools.org/Cordoba2017/ex/15/Plotting-methods.html</a></p>
<p></br></p>
</div>
</div>
<div id="using-ggtree" class="section level2 hasAnchor" number="10.3">
<h2><span class="header-section-number">10.3</span> Using <code>ggtree</code><a href="phylogenetic-visualization-in-r.html#using-ggtree" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The <code>ggtree</code> package on <code>Bioconductor</code> is an extension of <code>ggplot2</code>, a very popular package for data visualization in <code>R</code> (and <code>Python</code>), for visualizing and annotating phylogenies.
It thus works in conjunction with <code>ggplot2</code> and responds to the same arguments and grammar.
This makes the package quite flexible and useful when you want to combine different types of plots and plot elements, especially if you are fluent with <code>ggplot</code>.
However the package can be a bit finicky at times and may not work properly in all environments.
You should be able to install the package on the <code>scholar</code> cluster.
For a tutorial on ggplot, see: <a href="http://r-statistics.co/Complete-Ggplot2-Tutorial-Part1-With-R-Code.html" class="uri">http://r-statistics.co/Complete-Ggplot2-Tutorial-Part1-With-R-Code.html</a>.</p>
<div id="basic-plotting" class="section level3 hasAnchor" number="10.3.1">
<h3><span class="header-section-number">10.3.1</span> Basic plotting<a href="phylogenetic-visualization-in-r.html#basic-plotting" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Let’s get familiar with <code>ggtree</code>.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="phylogenetic-visualization-in-r.html#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co"># load data</span></span>
<span id="cb5-2"><a href="phylogenetic-visualization-in-r.html#cb5-2" aria-hidden="true" tabindex="-1"></a>tree <span class="ot">&lt;-</span> <span class="fu">read.nexus</span>(<span class="st">&quot;https://raw.githubusercontent.com/rgriff23/Dissertation/master/Chapter_2/data/tree.nex&quot;</span>) <span class="co"># example primate phylogeny from GitHub (this tree originally came from 10kTrees)</span></span>
<span id="cb5-3"><a href="phylogenetic-visualization-in-r.html#cb5-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-4"><a href="phylogenetic-visualization-in-r.html#cb5-4" aria-hidden="true" tabindex="-1"></a><span class="do">## Basic base plot with ggtree</span></span>
<span id="cb5-5"><a href="phylogenetic-visualization-in-r.html#cb5-5" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">ggtree</span>(tree)</span>
<span id="cb5-6"><a href="phylogenetic-visualization-in-r.html#cb5-6" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(p1)</span>
<span id="cb5-7"><a href="phylogenetic-visualization-in-r.html#cb5-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-8"><a href="phylogenetic-visualization-in-r.html#cb5-8" aria-hidden="true" tabindex="-1"></a><span class="do">## Making a similar looking plot with ape</span></span>
<span id="cb5-9"><a href="phylogenetic-visualization-in-r.html#cb5-9" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(tree)</span>
<span id="cb5-10"><a href="phylogenetic-visualization-in-r.html#cb5-10" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(tree, <span class="at">show.tip.label =</span> F)</span>
<span id="cb5-11"><a href="phylogenetic-visualization-in-r.html#cb5-11" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">ladderize</span>(tree, <span class="at">right =</span> <span class="cn">FALSE</span>), <span class="at">show.tip.label =</span> F, <span class="at">edge.width =</span> <span class="fl">1.5</span>)</span>
<span id="cb5-12"><a href="phylogenetic-visualization-in-r.html#cb5-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-13"><a href="phylogenetic-visualization-in-r.html#cb5-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-14"><a href="phylogenetic-visualization-in-r.html#cb5-14" aria-hidden="true" tabindex="-1"></a><span class="do">## The layouts available in ggtree are basically the same as ape’s plot.phylo function</span></span>
<span id="cb5-15"><a href="phylogenetic-visualization-in-r.html#cb5-15" aria-hidden="true" tabindex="-1"></a><span class="do">## The names are different though, and defaults look slightly nicer.</span></span>
<span id="cb5-16"><a href="phylogenetic-visualization-in-r.html#cb5-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-17"><a href="phylogenetic-visualization-in-r.html#cb5-17" aria-hidden="true" tabindex="-1"></a>p2a <span class="ot">&lt;-</span> <span class="fu">ggtree</span>(tree, <span class="at">layout=</span><span class="st">&quot;rectangular&quot;</span>) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">&quot;rectangular&quot;</span>)</span>
<span id="cb5-18"><a href="phylogenetic-visualization-in-r.html#cb5-18" aria-hidden="true" tabindex="-1"></a>p2b <span class="ot">&lt;-</span> <span class="fu">ggtree</span>(tree, <span class="at">layout=</span><span class="st">&quot;slanted&quot;</span>) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">&quot;slanted&quot;</span>)</span>
<span id="cb5-19"><a href="phylogenetic-visualization-in-r.html#cb5-19" aria-hidden="true" tabindex="-1"></a>p2c <span class="ot">&lt;-</span> <span class="fu">ggtree</span>(tree, <span class="at">layout=</span><span class="st">&quot;circular&quot;</span>) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">&quot;circular&quot;</span>)</span>
<span id="cb5-20"><a href="phylogenetic-visualization-in-r.html#cb5-20" aria-hidden="true" tabindex="-1"></a>p2d <span class="ot">&lt;-</span> <span class="fu">ggtree</span>(tree, <span class="at">layout=</span><span class="st">&quot;radial&quot;</span>) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">&quot;radial&quot;</span>)</span>
<span id="cb5-21"><a href="phylogenetic-visualization-in-r.html#cb5-21" aria-hidden="true" tabindex="-1"></a>p2e <span class="ot">&lt;-</span> <span class="fu">ggtree</span>(tree, <span class="at">layout=</span><span class="st">&quot;unrooted&quot;</span>) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">&quot;unrooted&quot;</span>)</span>
<span id="cb5-22"><a href="phylogenetic-visualization-in-r.html#cb5-22" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(p2b)</span>
<span id="cb5-23"><a href="phylogenetic-visualization-in-r.html#cb5-23" aria-hidden="true" tabindex="-1"></a><span class="fu">multiplot</span>(p2a, p2b, p2c, p2d, p2e, <span class="at">ncol =</span> <span class="dv">3</span>)</span>
<span id="cb5-24"><a href="phylogenetic-visualization-in-r.html#cb5-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-25"><a href="phylogenetic-visualization-in-r.html#cb5-25" aria-hidden="true" tabindex="-1"></a><span class="do">## In ape:</span></span>
<span id="cb5-26"><a href="phylogenetic-visualization-in-r.html#cb5-26" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfcol =</span> <span class="fu">c</span>(<span class="dv">2</span>, <span class="dv">3</span>))</span>
<span id="cb5-27"><a href="phylogenetic-visualization-in-r.html#cb5-27" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(tree, <span class="at">type =</span> <span class="st">&quot;phylogram&quot;</span>, <span class="at">main =</span> <span class="st">&quot;phylogram&quot;</span>, <span class="at">show.tip.label =</span> F)</span>
<span id="cb5-28"><a href="phylogenetic-visualization-in-r.html#cb5-28" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(tree, <span class="at">type =</span> <span class="st">&quot;cladogram&quot;</span>, <span class="at">main =</span> <span class="st">&quot;cladogram&quot;</span>, <span class="at">show.tip.label =</span> F)</span>
<span id="cb5-29"><a href="phylogenetic-visualization-in-r.html#cb5-29" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(tree, <span class="at">type =</span> <span class="st">&quot;fan&quot;</span>, <span class="at">main =</span> <span class="st">&quot;fan&quot;</span>, <span class="at">show.tip.label =</span> F)</span>
<span id="cb5-30"><a href="phylogenetic-visualization-in-r.html#cb5-30" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(tree, <span class="at">type =</span> <span class="st">&quot;radial&quot;</span>, <span class="at">main =</span> <span class="st">&quot;radial&quot;</span>, <span class="at">show.tip.label =</span> F)</span>
<span id="cb5-31"><a href="phylogenetic-visualization-in-r.html#cb5-31" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(tree, <span class="at">type =</span> <span class="st">&quot;unrooted&quot;</span>, <span class="at">main =</span> <span class="st">&quot;unrooted&quot;</span>, <span class="at">show.tip.label =</span> F)</span>
<span id="cb5-32"><a href="phylogenetic-visualization-in-r.html#cb5-32" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfcol =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">1</span>)) <span class="co">#back to default - one plot in window</span></span>
<span id="cb5-33"><a href="phylogenetic-visualization-in-r.html#cb5-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-34"><a href="phylogenetic-visualization-in-r.html#cb5-34" aria-hidden="true" tabindex="-1"></a><span class="do">## also try:</span></span>
<span id="cb5-35"><a href="phylogenetic-visualization-in-r.html#cb5-35" aria-hidden="true" tabindex="-1"></a><span class="fu">ggtree</span>(tree, <span class="at">layout=</span><span class="st">&quot;roundrect&quot;</span>)</span>
<span id="cb5-36"><a href="phylogenetic-visualization-in-r.html#cb5-36" aria-hidden="true" tabindex="-1"></a><span class="fu">ggtree</span>(tree, <span class="at">layout=</span><span class="st">&quot;slanted&quot;</span>)</span>
<span id="cb5-37"><a href="phylogenetic-visualization-in-r.html#cb5-37" aria-hidden="true" tabindex="-1"></a><span class="fu">ggtree</span>(tree, <span class="at">layout=</span><span class="st">&quot;ellipse&quot;</span>)</span>
<span id="cb5-38"><a href="phylogenetic-visualization-in-r.html#cb5-38" aria-hidden="true" tabindex="-1"></a><span class="fu">ggtree</span>(tree, <span class="at">layout=</span><span class="st">&quot;circular&quot;</span>)</span>
<span id="cb5-39"><a href="phylogenetic-visualization-in-r.html#cb5-39" aria-hidden="true" tabindex="-1"></a><span class="fu">ggtree</span>(tree, <span class="at">layout=</span><span class="st">&quot;fan&quot;</span>, <span class="at">open.angle =</span> <span class="dv">120</span>)</span>
<span id="cb5-40"><a href="phylogenetic-visualization-in-r.html#cb5-40" aria-hidden="true" tabindex="-1"></a><span class="fu">ggtree</span>(tree, <span class="at">layout=</span><span class="st">&quot;equal_angle&quot;</span>)</span>
<span id="cb5-41"><a href="phylogenetic-visualization-in-r.html#cb5-41" aria-hidden="true" tabindex="-1"></a><span class="fu">ggtree</span>(tree, <span class="at">layout=</span><span class="st">&quot;daylight&quot;</span>)</span>
<span id="cb5-42"><a href="phylogenetic-visualization-in-r.html#cb5-42" aria-hidden="true" tabindex="-1"></a><span class="fu">ggtree</span>(tree, <span class="at">branch.length=</span><span class="st">&#39;none&#39;</span>)</span>
<span id="cb5-43"><a href="phylogenetic-visualization-in-r.html#cb5-43" aria-hidden="true" tabindex="-1"></a><span class="fu">ggtree</span>(tree, <span class="at">layout=</span><span class="st">&quot;ellipse&quot;</span>, <span class="at">branch.length=</span><span class="st">&quot;none&quot;</span>)</span>
<span id="cb5-44"><a href="phylogenetic-visualization-in-r.html#cb5-44" aria-hidden="true" tabindex="-1"></a><span class="fu">ggtree</span>(tree, <span class="at">branch.length=</span><span class="st">&#39;none&#39;</span>, <span class="at">layout=</span><span class="st">&#39;circular&#39;</span>)</span>
<span id="cb5-45"><a href="phylogenetic-visualization-in-r.html#cb5-45" aria-hidden="true" tabindex="-1"></a><span class="fu">ggtree</span>(tree, <span class="at">layout=</span><span class="st">&quot;daylight&quot;</span>, <span class="at">branch.length =</span> <span class="st">&#39;none&#39;</span>)</span>
<span id="cb5-46"><a href="phylogenetic-visualization-in-r.html#cb5-46" aria-hidden="true" tabindex="-1"></a><span class="fu">ggtree</span>(tree) <span class="sc">+</span> <span class="fu">scale_x_reverse</span>()</span>
<span id="cb5-47"><a href="phylogenetic-visualization-in-r.html#cb5-47" aria-hidden="true" tabindex="-1"></a><span class="fu">ggtree</span>(tree) <span class="sc">+</span> <span class="fu">coord_flip</span>()</span>
<span id="cb5-48"><a href="phylogenetic-visualization-in-r.html#cb5-48" aria-hidden="true" tabindex="-1"></a><span class="fu">ggtree</span>(tree) <span class="sc">+</span> <span class="fu">layout_dendrogram</span>()</span>
<span id="cb5-49"><a href="phylogenetic-visualization-in-r.html#cb5-49" aria-hidden="true" tabindex="-1"></a><span class="fu">ggtree</span>(tree, <span class="at">layout=</span><span class="st">&#39;slanted&#39;</span>) <span class="sc">+</span> <span class="fu">coord_flip</span>()</span>
<span id="cb5-50"><a href="phylogenetic-visualization-in-r.html#cb5-50" aria-hidden="true" tabindex="-1"></a><span class="fu">ggtree</span>(tree, <span class="at">layout=</span><span class="st">&#39;slanted&#39;</span>, <span class="at">branch.length=</span><span class="st">&#39;none&#39;</span>) <span class="sc">+</span> <span class="fu">layout_dendrogram</span>()</span>
<span id="cb5-51"><a href="phylogenetic-visualization-in-r.html#cb5-51" aria-hidden="true" tabindex="-1"></a><span class="fu">ggtree</span>(tree, <span class="at">layout=</span><span class="st">&#39;circular&#39;</span>) <span class="sc">+</span> <span class="fu">xlim</span>(<span class="sc">-</span><span class="dv">10</span>, <span class="cn">NA</span>)</span>
<span id="cb5-52"><a href="phylogenetic-visualization-in-r.html#cb5-52" aria-hidden="true" tabindex="-1"></a><span class="fu">ggtree</span>(tree) <span class="sc">+</span> <span class="fu">layout_inward_circular</span>()</span>
<span id="cb5-53"><a href="phylogenetic-visualization-in-r.html#cb5-53" aria-hidden="true" tabindex="-1"></a><span class="fu">ggtree</span>(tree) <span class="sc">+</span> <span class="fu">layout_inward_circular</span>(<span class="at">xlim=</span><span class="dv">15</span>)</span>
<span id="cb5-54"><a href="phylogenetic-visualization-in-r.html#cb5-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-55"><a href="phylogenetic-visualization-in-r.html#cb5-55" aria-hidden="true" tabindex="-1"></a><span class="do">## adding branch length scale:</span></span>
<span id="cb5-56"><a href="phylogenetic-visualization-in-r.html#cb5-56" aria-hidden="true" tabindex="-1"></a><span class="fu">ggtree</span>(tree) <span class="sc">+</span></span>
<span id="cb5-57"><a href="phylogenetic-visualization-in-r.html#cb5-57" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_treescale</span>()</span>
<span id="cb5-58"><a href="phylogenetic-visualization-in-r.html#cb5-58" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-59"><a href="phylogenetic-visualization-in-r.html#cb5-59" aria-hidden="true" tabindex="-1"></a><span class="fu">ggtree</span>(tree) <span class="sc">+</span> </span>
<span id="cb5-60"><a href="phylogenetic-visualization-in-r.html#cb5-60" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_treescale</span>(<span class="at">fontsize =</span> <span class="dv">6</span>, <span class="at">linesize =</span> <span class="dv">2</span>, <span class="at">offset =</span> <span class="dv">1</span>)</span>
<span id="cb5-61"><a href="phylogenetic-visualization-in-r.html#cb5-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-62"><a href="phylogenetic-visualization-in-r.html#cb5-62" aria-hidden="true" tabindex="-1"></a><span class="fu">ggtree</span>(tree) <span class="sc">+</span> </span>
<span id="cb5-63"><a href="phylogenetic-visualization-in-r.html#cb5-63" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_treescale</span>(<span class="at">x =</span> <span class="dv">0</span>, <span class="at">y =</span> <span class="dv">45</span>, <span class="at">width =</span> <span class="dv">10</span>, <span class="at">color=</span><span class="st">&#39;red&#39;</span>)</span>
<span id="cb5-64"><a href="phylogenetic-visualization-in-r.html#cb5-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-65"><a href="phylogenetic-visualization-in-r.html#cb5-65" aria-hidden="true" tabindex="-1"></a><span class="fu">ggtree</span>(tree) <span class="sc">+</span></span>
<span id="cb5-66"><a href="phylogenetic-visualization-in-r.html#cb5-66" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_tree2</span>() <span class="co"># this should work well in general.</span></span>
<span id="cb5-67"><a href="phylogenetic-visualization-in-r.html#cb5-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-68"><a href="phylogenetic-visualization-in-r.html#cb5-68" aria-hidden="true" tabindex="-1"></a><span class="do">## displaying nodes and/or tips</span></span>
<span id="cb5-69"><a href="phylogenetic-visualization-in-r.html#cb5-69" aria-hidden="true" tabindex="-1"></a><span class="fu">ggtree</span>(tree) <span class="sc">+</span> </span>
<span id="cb5-70"><a href="phylogenetic-visualization-in-r.html#cb5-70" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tiplab</span>(<span class="at">size =</span> <span class="dv">2</span>, <span class="at">color =</span> <span class="st">&quot;firebrick&quot;</span>)</span>
<span id="cb5-71"><a href="phylogenetic-visualization-in-r.html#cb5-71" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-72"><a href="phylogenetic-visualization-in-r.html#cb5-72" aria-hidden="true" tabindex="-1"></a><span class="fu">ggtree</span>(tree) <span class="sc">+</span> </span>
<span id="cb5-73"><a href="phylogenetic-visualization-in-r.html#cb5-73" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tiplab</span>(<span class="at">as_ylab =</span> <span class="cn">TRUE</span>, <span class="at">color=</span><span class="st">&#39;firebrick&#39;</span>) <span class="co"># this will align the tip labels regardless of branch lengths  </span></span>
<span id="cb5-74"><a href="phylogenetic-visualization-in-r.html#cb5-74" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-75"><a href="phylogenetic-visualization-in-r.html#cb5-75" aria-hidden="true" tabindex="-1"></a><span class="fu">ggtree</span>(tree) <span class="sc">+</span> </span>
<span id="cb5-76"><a href="phylogenetic-visualization-in-r.html#cb5-76" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">shape =</span> isTip, <span class="at">color =</span> isTip), <span class="at">size =</span> <span class="dv">3</span>)</span>
<span id="cb5-77"><a href="phylogenetic-visualization-in-r.html#cb5-77" aria-hidden="true" tabindex="-1"></a><span class="fu">ggtree</span>(tree) <span class="sc">+</span> </span>
<span id="cb5-78"><a href="phylogenetic-visualization-in-r.html#cb5-78" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_nodepoint</span>(<span class="at">color =</span> <span class="st">&quot;#b5e521&quot;</span>, <span class="at">alpha =</span> <span class="dv">1</span><span class="sc">/</span><span class="dv">4</span>, <span class="at">size =</span> <span class="dv">10</span>) <span class="sc">+</span></span>
<span id="cb5-79"><a href="phylogenetic-visualization-in-r.html#cb5-79" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tippoint</span>(<span class="at">color=</span><span class="st">&quot;#FDAC4F&quot;</span>, <span class="at">shape =</span> <span class="dv">8</span>, <span class="at">size =</span> <span class="dv">3</span>) <span class="co"># play around with different shapes</span></span>
<span id="cb5-80"><a href="phylogenetic-visualization-in-r.html#cb5-80" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-81"><a href="phylogenetic-visualization-in-r.html#cb5-81" aria-hidden="true" tabindex="-1"></a><span class="do">## As in ggplot2, you can control the aesthetics of the plot and add information</span></span>
<span id="cb5-82"><a href="phylogenetic-visualization-in-r.html#cb5-82" aria-hidden="true" tabindex="-1"></a><span class="do">## by adding layers to the basic plot.</span></span>
<span id="cb5-83"><a href="phylogenetic-visualization-in-r.html#cb5-83" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-84"><a href="phylogenetic-visualization-in-r.html#cb5-84" aria-hidden="true" tabindex="-1"></a>p3 <span class="ot">&lt;-</span> <span class="fu">ggtree</span>(tree, <span class="fu">aes</span>(<span class="at">color=</span>branch.length)) <span class="sc">+</span></span>
<span id="cb5-85"><a href="phylogenetic-visualization-in-r.html#cb5-85" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlim</span>(<span class="dv">0</span>, <span class="dv">90</span>) <span class="sc">+</span> </span>
<span id="cb5-86"><a href="phylogenetic-visualization-in-r.html#cb5-86" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tiplab</span>(<span class="at">size =</span> <span class="dv">2</span>, <span class="at">color =</span> <span class="st">&quot;plum1&quot;</span>) <span class="sc">+</span></span>
<span id="cb5-87"><a href="phylogenetic-visualization-in-r.html#cb5-87" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_label2</span>(<span class="fu">aes</span>(<span class="at">subset =</span> <span class="sc">!</span>isTip, <span class="at">label =</span> node), <span class="at">size =</span> <span class="dv">2</span>, <span class="at">color =</span> <span class="st">&quot;darkred&quot;</span>, <span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span> <span class="co">#alpha makes things transparent - experiment with different values!</span></span>
<span id="cb5-88"><a href="phylogenetic-visualization-in-r.html#cb5-88" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_tree2</span>(<span class="st">&quot;black&quot;</span>) <span class="sc">+</span> </span>
<span id="cb5-89"><a href="phylogenetic-visualization-in-r.html#cb5-89" aria-hidden="true" tabindex="-1"></a>  <span class="co"># there are many themes in ggplot2 - look them up or define your own!</span></span>
<span id="cb5-90"><a href="phylogenetic-visualization-in-r.html#cb5-90" aria-hidden="true" tabindex="-1"></a>  <span class="co"># https://www.google.com/search?client=firefox-b-1-e&amp;q=datanovia+ggplot+themes</span></span>
<span id="cb5-91"><a href="phylogenetic-visualization-in-r.html#cb5-91" aria-hidden="true" tabindex="-1"></a>  <span class="co"># a good sminimalist theme to start with:</span></span>
<span id="cb5-92"><a href="phylogenetic-visualization-in-r.html#cb5-92" aria-hidden="true" tabindex="-1"></a>  <span class="co"># theme_bw()</span></span>
<span id="cb5-93"><a href="phylogenetic-visualization-in-r.html#cb5-93" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_continuous</span>(<span class="at">low =</span> <span class="st">&#39;white&#39;</span>, <span class="at">high =</span> <span class="st">&#39;hotpink&#39;</span>, <span class="at">name =</span> <span class="st">&quot;Branch length (my)&quot;</span>) <span class="sc">+</span></span>
<span id="cb5-94"><a href="phylogenetic-visualization-in-r.html#cb5-94" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;bottom&quot;</span>)</span>
<span id="cb5-95"><a href="phylogenetic-visualization-in-r.html#cb5-95" aria-hidden="true" tabindex="-1"></a>p3</span>
<span id="cb5-96"><a href="phylogenetic-visualization-in-r.html#cb5-96" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-97"><a href="phylogenetic-visualization-in-r.html#cb5-97" aria-hidden="true" tabindex="-1"></a><span class="do">## What you see in your console is not necessarily what you will see when saving your plot</span></span>
<span id="cb5-98"><a href="phylogenetic-visualization-in-r.html#cb5-98" aria-hidden="true" tabindex="-1"></a><span class="do">## adjust the size of your text, symbols, and lines accordingly</span></span>
<span id="cb5-99"><a href="phylogenetic-visualization-in-r.html#cb5-99" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">plot =</span> p3, <span class="st">&quot;/home/park1330/lab11_R/small_phylo.pdf&quot;</span>, <span class="at">height =</span> <span class="dv">3</span>, <span class="at">width =</span> <span class="dv">2</span>)</span>
<span id="cb5-100"><a href="phylogenetic-visualization-in-r.html#cb5-100" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">plot =</span> p3, <span class="st">&quot;/home/park1330/lab11_R/mid_phylo.pdf&quot;</span>, <span class="at">height =</span> <span class="dv">5</span>, <span class="at">width =</span> <span class="dv">4</span>)</span>
<span id="cb5-101"><a href="phylogenetic-visualization-in-r.html#cb5-101" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">plot =</span> p3, <span class="st">&quot;/home/park1330/lab11_R/large_phylo.pdf&quot;</span>, <span class="at">height =</span> <span class="dv">15</span>, <span class="at">width =</span> <span class="dv">10</span>)</span></code></pre></div>
</div>
<div id="labeling-tips-nodes-and-clades" class="section level3 hasAnchor" number="10.3.2">
<h3><span class="header-section-number">10.3.2</span> Labeling tips, nodes, and clades<a href="phylogenetic-visualization-in-r.html#labeling-tips-nodes-and-clades" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Below are the internal node numbers corresponding to the primate superfamilies.
Let’s label the nodes and the clades that they give rise to.</p>
<ul>
<li><p>Node 124: Galagoidea</p></li>
<li><p>Node 113: Lemuroidea</p></li>
<li><p>Node 110: Tarsioidea</p></li>
<li><p>Node 96: Ceboidea</p></li>
<li><p>Node 89: Hominoidea</p></li>
<li><p>Node 70: Cercopithecoidea</p></li>
</ul>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="phylogenetic-visualization-in-r.html#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Highlighting clades</span></span>
<span id="cb6-2"><a href="phylogenetic-visualization-in-r.html#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="co"># geom_hilight draws rectangular box around clades </span></span>
<span id="cb6-3"><a href="phylogenetic-visualization-in-r.html#cb6-3" aria-hidden="true" tabindex="-1"></a><span class="fu">ggtree</span>(tree) <span class="sc">+</span> </span>
<span id="cb6-4"><a href="phylogenetic-visualization-in-r.html#cb6-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hilight</span>(<span class="at">node =</span> <span class="dv">124</span>, <span class="at">fill =</span> <span class="st">&quot;steelblue&quot;</span>, <span class="at">alpha =</span> .<span class="dv">6</span>) <span class="sc">+</span></span>
<span id="cb6-5"><a href="phylogenetic-visualization-in-r.html#cb6-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hilight</span>(<span class="at">node =</span> <span class="dv">70</span>, <span class="at">fill =</span> <span class="st">&quot;darkgreen&quot;</span>, <span class="at">alpha =</span> .<span class="dv">6</span>) </span>
<span id="cb6-6"><a href="phylogenetic-visualization-in-r.html#cb6-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-7"><a href="phylogenetic-visualization-in-r.html#cb6-7" aria-hidden="true" tabindex="-1"></a><span class="fu">ggtree</span>(tree) <span class="sc">+</span> </span>
<span id="cb6-8"><a href="phylogenetic-visualization-in-r.html#cb6-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hilight</span>(<span class="at">node =</span> <span class="dv">124</span>, <span class="at">fill =</span> <span class="st">&quot;steelblue&quot;</span>, <span class="at">alpha =</span> .<span class="dv">6</span>, <span class="at">type  =</span>  <span class="st">&quot;roundrect&quot;</span>) <span class="sc">+</span></span>
<span id="cb6-9"><a href="phylogenetic-visualization-in-r.html#cb6-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hilight</span>(<span class="at">node =</span> <span class="dv">70</span>, <span class="at">fill =</span> <span class="st">&quot;darkgreen&quot;</span>, <span class="at">alpha =</span> .<span class="dv">6</span>, <span class="at">type  =</span>  <span class="st">&quot;encircle&quot;</span>) </span>
<span id="cb6-10"><a href="phylogenetic-visualization-in-r.html#cb6-10" aria-hidden="true" tabindex="-1"></a><span class="do">## geom_balance draws rectangular box around clades and divides immediate subclades </span></span>
<span id="cb6-11"><a href="phylogenetic-visualization-in-r.html#cb6-11" aria-hidden="true" tabindex="-1"></a><span class="fu">ggtree</span>(tree) <span class="sc">+</span></span>
<span id="cb6-12"><a href="phylogenetic-visualization-in-r.html#cb6-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_balance</span>(<span class="at">node =</span> <span class="dv">124</span>, <span class="at">fill =</span> <span class="st">&#39;steelblue&#39;</span>, <span class="at">color =</span> <span class="st">&#39;white&#39;</span>, <span class="at">alpha =</span> <span class="fl">0.6</span>, <span class="at">extend =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb6-13"><a href="phylogenetic-visualization-in-r.html#cb6-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_balance</span>(<span class="at">node =</span> <span class="dv">70</span>, <span class="at">fill =</span> <span class="st">&#39;darkgreen&#39;</span>, <span class="at">color =</span> <span class="st">&#39;white&#39;</span>, <span class="at">alpha =</span> <span class="fl">0.6</span>, <span class="at">extend =</span> <span class="dv">1</span>) </span>
<span id="cb6-14"><a href="phylogenetic-visualization-in-r.html#cb6-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-15"><a href="phylogenetic-visualization-in-r.html#cb6-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Try this with different tree layouts: e.g., &quot;circular&quot;</span></span>
<span id="cb6-16"><a href="phylogenetic-visualization-in-r.html#cb6-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-17"><a href="phylogenetic-visualization-in-r.html#cb6-17" aria-hidden="true" tabindex="-1"></a>p4 <span class="ot">&lt;-</span> <span class="fu">ggtree</span>(tree) <span class="sc">+</span> </span>
<span id="cb6-18"><a href="phylogenetic-visualization-in-r.html#cb6-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlim</span>(<span class="dv">0</span>, <span class="dv">125</span>) <span class="sc">+</span></span>
<span id="cb6-19"><a href="phylogenetic-visualization-in-r.html#cb6-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tiplab</span>(<span class="at">size =</span> <span class="dv">2</span>, <span class="at">offset =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb6-20"><a href="phylogenetic-visualization-in-r.html#cb6-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hilight</span>(<span class="at">node =</span> <span class="dv">124</span>, <span class="at">fill =</span> <span class="st">&quot;steelblue&quot;</span>, <span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb6-21"><a href="phylogenetic-visualization-in-r.html#cb6-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hilight</span>(<span class="at">node =</span> <span class="dv">113</span>, <span class="at">fill =</span> <span class="st">&quot;darkgreen&quot;</span>, <span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb6-22"><a href="phylogenetic-visualization-in-r.html#cb6-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hilight</span>(<span class="at">node =</span> <span class="dv">110</span>, <span class="at">fill =</span> <span class="st">&quot;gray&quot;</span>, <span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb6-23"><a href="phylogenetic-visualization-in-r.html#cb6-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hilight</span>(<span class="at">node =</span> <span class="dv">96</span>, <span class="at">fill =</span> <span class="st">&quot;pink&quot;</span>, <span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb6-24"><a href="phylogenetic-visualization-in-r.html#cb6-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hilight</span>(<span class="at">node =</span> <span class="dv">89</span>, <span class="at">fill =</span> <span class="st">&quot;beige&quot;</span>, <span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb6-25"><a href="phylogenetic-visualization-in-r.html#cb6-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hilight</span>(<span class="at">node =</span> <span class="dv">70</span>, <span class="at">fill =</span> <span class="st">&quot;yellow&quot;</span>, <span class="at">alpha =</span> <span class="fl">0.5</span>) </span>
<span id="cb6-26"><a href="phylogenetic-visualization-in-r.html#cb6-26" aria-hidden="true" tabindex="-1"></a>p4</span>
<span id="cb6-27"><a href="phylogenetic-visualization-in-r.html#cb6-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-28"><a href="phylogenetic-visualization-in-r.html#cb6-28" aria-hidden="true" tabindex="-1"></a><span class="co"># Adding labeled bars across tips showing the extent of each clade </span></span>
<span id="cb6-29"><a href="phylogenetic-visualization-in-r.html#cb6-29" aria-hidden="true" tabindex="-1"></a>p5 <span class="ot">&lt;-</span> p4 <span class="sc">+</span> </span>
<span id="cb6-30"><a href="phylogenetic-visualization-in-r.html#cb6-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_cladelabel</span>(<span class="dv">124</span>, <span class="st">&quot;Galagoidea&quot;</span>, <span class="at">offset =</span> <span class="dv">25</span>, <span class="at">barsize =</span> <span class="dv">2</span>, <span class="at">angle =</span> <span class="dv">90</span>, <span class="at">offset.text =</span> <span class="fl">1.5</span>, <span class="at">hjust =</span> <span class="fl">0.5</span>, <span class="at">fontsize =</span> <span class="dv">3</span>) <span class="sc">+</span> </span>
<span id="cb6-31"><a href="phylogenetic-visualization-in-r.html#cb6-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_cladelabel</span>(<span class="dv">113</span>, <span class="st">&quot;Lemuroidea&quot;</span>, <span class="at">offset =</span> <span class="dv">25</span>, <span class="at">barsize =</span> <span class="dv">2</span>, <span class="at">angle =</span> <span class="dv">90</span>, <span class="at">offset.text =</span> <span class="fl">1.5</span>, <span class="at">hjust =</span> <span class="fl">0.5</span>, <span class="at">fontsize =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb6-32"><a href="phylogenetic-visualization-in-r.html#cb6-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_cladelabel</span>(<span class="dv">110</span>, <span class="st">&quot;Tarsioidea&quot;</span>, <span class="at">offset =</span> <span class="dv">25</span>, <span class="at">barsize =</span> <span class="dv">2</span>, <span class="at">angle =</span> <span class="dv">0</span>, <span class="at">offset.text =</span> <span class="fl">2.5</span>, <span class="at">hjust =</span> <span class="fl">0.2</span>, <span class="at">fontsize =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb6-33"><a href="phylogenetic-visualization-in-r.html#cb6-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_cladelabel</span>(<span class="dv">96</span>, <span class="st">&quot;Ceboidea&quot;</span>, <span class="at">offset =</span> <span class="dv">25</span>, <span class="at">barsize =</span> <span class="dv">2</span>, <span class="at">angle =</span> <span class="dv">90</span>, <span class="at">offset.text =</span> <span class="fl">1.5</span>, <span class="at">hjust =</span> <span class="fl">0.5</span>, <span class="at">fontsize =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb6-34"><a href="phylogenetic-visualization-in-r.html#cb6-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_cladelabel</span>(<span class="dv">89</span>, <span class="st">&quot;Hominoidea&quot;</span>, <span class="at">offset =</span> <span class="dv">25</span>, <span class="at">barsize =</span> <span class="dv">2</span>, <span class="at">angle =</span> <span class="dv">90</span>, <span class="at">offset.text =</span> <span class="fl">1.5</span>, <span class="at">hjust =</span> <span class="fl">0.5</span>, <span class="at">fontsize =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb6-35"><a href="phylogenetic-visualization-in-r.html#cb6-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_cladelabel</span>(<span class="dv">70</span>, <span class="st">&quot;Cercopithecoidea&quot;</span>, <span class="at">offset =</span> <span class="dv">25</span>, <span class="at">barsize =</span> <span class="dv">2</span>, <span class="at">angle =</span> <span class="dv">90</span>, <span class="at">offset.text =</span> <span class="fl">1.5</span>, <span class="at">hjust =</span> <span class="fl">0.5</span>, <span class="at">fontsize =</span> <span class="dv">3</span>)</span>
<span id="cb6-36"><a href="phylogenetic-visualization-in-r.html#cb6-36" aria-hidden="true" tabindex="-1"></a>p5</span>
<span id="cb6-37"><a href="phylogenetic-visualization-in-r.html#cb6-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-38"><a href="phylogenetic-visualization-in-r.html#cb6-38" aria-hidden="true" tabindex="-1"></a><span class="co"># Adding symbols to the tips</span></span>
<span id="cb6-39"><a href="phylogenetic-visualization-in-r.html#cb6-39" aria-hidden="true" tabindex="-1"></a>tipdata <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">cbind</span>(<span class="at">taxa =</span> tree<span class="sc">$</span>tip.label, <span class="at">shape =</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">&quot;triangle&quot;</span>, <span class="dv">35</span>), <span class="fu">rep</span>(<span class="st">&quot;circle&quot;</span>, <span class="dv">30</span>)))) <span class="co"># generating example categorical trait data</span></span>
<span id="cb6-40"><a href="phylogenetic-visualization-in-r.html#cb6-40" aria-hidden="true" tabindex="-1"></a>p5 <span class="sc">%&lt;+%</span> tipdata <span class="sc">+</span> <span class="fu">geom_tippoint</span>(<span class="fu">aes</span>(<span class="at">pch =</span> shape), <span class="at">size =</span> <span class="dv">3</span>)</span>
<span id="cb6-41"><a href="phylogenetic-visualization-in-r.html#cb6-41" aria-hidden="true" tabindex="-1"></a>p5 <span class="sc">%&lt;+%</span> tipdata <span class="sc">+</span> <span class="fu">geom_tippoint</span>(<span class="fu">aes</span>(<span class="at">color =</span> shape), <span class="at">size =</span> <span class="dv">3</span>)</span>
<span id="cb6-42"><a href="phylogenetic-visualization-in-r.html#cb6-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-43"><a href="phylogenetic-visualization-in-r.html#cb6-43" aria-hidden="true" tabindex="-1"></a><span class="co"># Linking tips</span></span>
<span id="cb6-44"><a href="phylogenetic-visualization-in-r.html#cb6-44" aria-hidden="true" tabindex="-1"></a><span class="fu">ggtree</span>(tree) <span class="sc">+</span> </span>
<span id="cb6-45"><a href="phylogenetic-visualization-in-r.html#cb6-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tiplab</span>() <span class="sc">+</span> </span>
<span id="cb6-46"><a href="phylogenetic-visualization-in-r.html#cb6-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_taxalink</span>(<span class="at">taxa1 =</span> <span class="st">&#39;Allenopithecus_nigroviridis&#39;</span>, <span class="at">taxa2=</span><span class="st">&#39;Chlorocebus_sabaeus&#39;</span>) <span class="sc">+</span> </span>
<span id="cb6-47"><a href="phylogenetic-visualization-in-r.html#cb6-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_taxalink</span>(<span class="at">taxa1=</span><span class="st">&#39;Nomascus_concolor&#39;</span>, <span class="at">taxa2=</span><span class="st">&#39;Carlito_syrichta&#39;</span>, <span class="at">color=</span><span class="st">&#39;red&#39;</span>, <span class="at">linetype =</span> <span class="st">&#39;dashed&#39;</span>,</span>
<span id="cb6-48"><a href="phylogenetic-visualization-in-r.html#cb6-48" aria-hidden="true" tabindex="-1"></a>                <span class="at">arrow=</span><span class="fu">arrow</span>(<span class="at">length=</span><span class="fu">unit</span>(<span class="fl">0.02</span>, <span class="st">&quot;npc&quot;</span>))) <span class="sc">+</span></span>
<span id="cb6-49"><a href="phylogenetic-visualization-in-r.html#cb6-49" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_taxalink</span>(<span class="at">taxa1=</span><span class="st">&quot;Otolemur_crassicaudatus&quot;</span>, <span class="at">taxa2=</span><span class="st">&quot;Avahi_laniger&quot;</span>, <span class="at">color=</span><span class="st">&quot;blue&quot;</span>, <span class="at">alpha=</span><span class="fl">0.5</span>, </span>
<span id="cb6-50"><a href="phylogenetic-visualization-in-r.html#cb6-50" aria-hidden="true" tabindex="-1"></a>                <span class="at">offset=</span><span class="fl">0.1</span>, <span class="at">outward=</span><span class="cn">FALSE</span>,</span>
<span id="cb6-51"><a href="phylogenetic-visualization-in-r.html#cb6-51" aria-hidden="true" tabindex="-1"></a>                <span class="at">arrow=</span><span class="fu">arrow</span>(<span class="at">length=</span><span class="fu">unit</span>(<span class="fl">0.01</span>, <span class="st">&quot;npc&quot;</span>)))</span>
<span id="cb6-52"><a href="phylogenetic-visualization-in-r.html#cb6-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-53"><a href="phylogenetic-visualization-in-r.html#cb6-53" aria-hidden="true" tabindex="-1"></a><span class="fu">ggtree</span>(tree, <span class="at">layout=</span><span class="st">&quot;inward_circular&quot;</span>, <span class="at">xlim=</span><span class="fu">c</span>(<span class="dv">150</span>, <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb6-54"><a href="phylogenetic-visualization-in-r.html#cb6-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_taxalink</span>(<span class="at">taxa1=</span><span class="st">&quot;Otolemur_crassicaudatus&quot;</span>, <span class="at">taxa2=</span><span class="st">&quot;Avahi_laniger&quot;</span>, </span>
<span id="cb6-55"><a href="phylogenetic-visualization-in-r.html#cb6-55" aria-hidden="true" tabindex="-1"></a>                <span class="at">color=</span><span class="st">&quot;blue&quot;</span>, </span>
<span id="cb6-56"><a href="phylogenetic-visualization-in-r.html#cb6-56" aria-hidden="true" tabindex="-1"></a>                <span class="at">size =</span> <span class="dv">3</span>,</span>
<span id="cb6-57"><a href="phylogenetic-visualization-in-r.html#cb6-57" aria-hidden="true" tabindex="-1"></a>                <span class="at">alpha=</span><span class="fl">0.5</span>, </span>
<span id="cb6-58"><a href="phylogenetic-visualization-in-r.html#cb6-58" aria-hidden="true" tabindex="-1"></a>                <span class="at">ncp=</span><span class="dv">10</span>,</span>
<span id="cb6-59"><a href="phylogenetic-visualization-in-r.html#cb6-59" aria-hidden="true" tabindex="-1"></a>                <span class="at">offset=</span><span class="fl">0.15</span>) <span class="sc">+</span> </span>
<span id="cb6-60"><a href="phylogenetic-visualization-in-r.html#cb6-60" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_taxalink</span>(<span class="at">taxa1=</span><span class="st">&quot;Nomascus_concolor&quot;</span>, <span class="at">taxa2=</span><span class="st">&quot;Carlito_syrichta&quot;</span>, </span>
<span id="cb6-61"><a href="phylogenetic-visualization-in-r.html#cb6-61" aria-hidden="true" tabindex="-1"></a>                <span class="at">color=</span><span class="st">&quot;red&quot;</span>, </span>
<span id="cb6-62"><a href="phylogenetic-visualization-in-r.html#cb6-62" aria-hidden="true" tabindex="-1"></a>                <span class="at">linetype =</span> <span class="st">&#39;dotted&#39;</span>,</span>
<span id="cb6-63"><a href="phylogenetic-visualization-in-r.html#cb6-63" aria-hidden="true" tabindex="-1"></a>                <span class="at">size =</span> <span class="fl">1.5</span>,</span>
<span id="cb6-64"><a href="phylogenetic-visualization-in-r.html#cb6-64" aria-hidden="true" tabindex="-1"></a>                <span class="at">alpha=</span><span class="fl">0.9</span>, </span>
<span id="cb6-65"><a href="phylogenetic-visualization-in-r.html#cb6-65" aria-hidden="true" tabindex="-1"></a>                <span class="at">ncp=</span><span class="dv">10</span>,</span>
<span id="cb6-66"><a href="phylogenetic-visualization-in-r.html#cb6-66" aria-hidden="true" tabindex="-1"></a>                <span class="at">offset=</span><span class="fl">0.15</span>) <span class="sc">+</span> </span>
<span id="cb6-67"><a href="phylogenetic-visualization-in-r.html#cb6-67" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_size_continuous</span>(<span class="at">range=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">3</span>))</span>
<span id="cb6-68"><a href="phylogenetic-visualization-in-r.html#cb6-68" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-69"><a href="phylogenetic-visualization-in-r.html#cb6-69" aria-hidden="true" tabindex="-1"></a><span class="co"># Plotting contiuous trait values for each tip</span></span>
<span id="cb6-70"><a href="phylogenetic-visualization-in-r.html#cb6-70" aria-hidden="true" tabindex="-1"></a>traits <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">fastBM</span>(tree, <span class="at">nsim=</span><span class="dv">5</span>)) <span class="co"># simulate 5 continuous traits via BM</span></span>
<span id="cb6-71"><a href="phylogenetic-visualization-in-r.html#cb6-71" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-72"><a href="phylogenetic-visualization-in-r.html#cb6-72" aria-hidden="true" tabindex="-1"></a>p8 <span class="ot">&lt;-</span> <span class="fu">ggtree</span>(tree) <span class="sc">+</span> </span>
<span id="cb6-73"><a href="phylogenetic-visualization-in-r.html#cb6-73" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlim</span>(<span class="dv">0</span>, <span class="dv">125</span>) <span class="sc">+</span></span>
<span id="cb6-74"><a href="phylogenetic-visualization-in-r.html#cb6-74" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tiplab</span>(<span class="at">size=</span><span class="dv">2</span>, <span class="at">offset=</span><span class="dv">17</span>) </span>
<span id="cb6-75"><a href="phylogenetic-visualization-in-r.html#cb6-75" aria-hidden="true" tabindex="-1"></a>p8</span>
<span id="cb6-76"><a href="phylogenetic-visualization-in-r.html#cb6-76" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-77"><a href="phylogenetic-visualization-in-r.html#cb6-77" aria-hidden="true" tabindex="-1"></a>p9 <span class="ot">&lt;-</span>  <span class="fu">gheatmap</span>(p8, traits, <span class="at">offset=</span><span class="fl">0.2</span>, <span class="at">width=</span><span class="fl">0.2</span>, <span class="at">low=</span><span class="st">&quot;white&quot;</span>, <span class="at">high=</span><span class="st">&quot;black&quot;</span>, <span class="at">colnames_position =</span> <span class="st">&quot;top&quot;</span>, <span class="at">font.size=</span><span class="dv">2</span>) <span class="co"># add heatmap</span></span>
<span id="cb6-78"><a href="phylogenetic-visualization-in-r.html#cb6-78" aria-hidden="true" tabindex="-1"></a>p9</span>
<span id="cb6-79"><a href="phylogenetic-visualization-in-r.html#cb6-79" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb6-80"><a href="phylogenetic-visualization-in-r.html#cb6-80" aria-hidden="true" tabindex="-1"></a><span class="co"># Plotting discrete traits (e.g., presence-absence)</span></span>
<span id="cb6-81"><a href="phylogenetic-visualization-in-r.html#cb6-81" aria-hidden="true" tabindex="-1"></a>traits.binary <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">matrix</span>(<span class="fu">ifelse</span>(<span class="fu">stack</span>(traits)<span class="sc">$</span>values <span class="sc">&gt;</span> <span class="fu">mean</span>(<span class="fu">stack</span>(traits)<span class="sc">$</span>values), <span class="st">&quot;present&quot;</span>, <span class="st">&quot;absent&quot;</span>), <span class="dv">65</span>, <span class="dv">5</span>))</span>
<span id="cb6-82"><a href="phylogenetic-visualization-in-r.html#cb6-82" aria-hidden="true" tabindex="-1"></a><span class="fu">row.names</span>(traits.binary) <span class="ot">&lt;-</span> <span class="fu">row.names</span>(traits)</span>
<span id="cb6-83"><a href="phylogenetic-visualization-in-r.html#cb6-83" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-84"><a href="phylogenetic-visualization-in-r.html#cb6-84" aria-hidden="true" tabindex="-1"></a>p10 <span class="ot">&lt;-</span>  <span class="fu">gheatmap</span>(p8, traits.binary, <span class="at">offset=</span><span class="fl">0.2</span>, <span class="at">width=</span><span class="fl">0.2</span>, <span class="at">low=</span><span class="st">&quot;white&quot;</span>, <span class="at">high=</span><span class="st">&quot;black&quot;</span>, <span class="at">colnames_position =</span> <span class="st">&quot;top&quot;</span>, <span class="at">font.size=</span><span class="dv">2</span>, <span class="at">color=</span><span class="st">&quot;black&quot;</span>) <span class="sc">+</span> </span>
<span id="cb6-85"><a href="phylogenetic-visualization-in-r.html#cb6-85" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">&quot;white&quot;</span>, <span class="st">&quot;black&quot;</span>))</span>
<span id="cb6-86"><a href="phylogenetic-visualization-in-r.html#cb6-86" aria-hidden="true" tabindex="-1"></a>p10</span>
<span id="cb6-87"><a href="phylogenetic-visualization-in-r.html#cb6-87" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb6-88"><a href="phylogenetic-visualization-in-r.html#cb6-88" aria-hidden="true" tabindex="-1"></a><span class="co"># Plotting tree with multiple types of data</span></span>
<span id="cb6-89"><a href="phylogenetic-visualization-in-r.html#cb6-89" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb6-90"><a href="phylogenetic-visualization-in-r.html#cb6-90" aria-hidden="true" tabindex="-1"></a><span class="co"># generate some random values for each tip label in the data</span></span>
<span id="cb6-91"><a href="phylogenetic-visualization-in-r.html#cb6-91" aria-hidden="true" tabindex="-1"></a>t1 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">id =</span> tree<span class="sc">$</span>tip.label, <span class="at">val =</span> <span class="fu">rnorm</span>(<span class="fu">length</span>(tree<span class="sc">$</span>tip.label), <span class="at">sd =</span> <span class="dv">3</span>))</span>
<span id="cb6-92"><a href="phylogenetic-visualization-in-r.html#cb6-92" aria-hidden="true" tabindex="-1"></a>t2 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">id =</span> tree<span class="sc">$</span>tip.label, <span class="at">val =</span> <span class="fu">abs</span>(<span class="fu">rnorm</span>(<span class="fu">length</span>(tree<span class="sc">$</span>tip.label), <span class="at">mean =</span> <span class="dv">100</span>, <span class="at">sd =</span> <span class="dv">50</span>)))</span>
<span id="cb6-93"><a href="phylogenetic-visualization-in-r.html#cb6-93" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb6-94"><a href="phylogenetic-visualization-in-r.html#cb6-94" aria-hidden="true" tabindex="-1"></a><span class="co"># Make a second plot with the original, naming the new plot &quot;dot&quot;, using the data you just created, with a point geom.</span></span>
<span id="cb6-95"><a href="phylogenetic-visualization-in-r.html#cb6-95" aria-hidden="true" tabindex="-1"></a>pp <span class="ot">&lt;-</span> <span class="fu">ggtree</span>(tree)</span>
<span id="cb6-96"><a href="phylogenetic-visualization-in-r.html#cb6-96" aria-hidden="true" tabindex="-1"></a>pp2 <span class="ot">&lt;-</span> <span class="fu">facet_plot</span>(pp, <span class="at">panel =</span> <span class="st">&quot;trait 1 (dot)&quot;</span>, <span class="at">data =</span> t1, <span class="at">geom =</span> geom_point, <span class="fu">aes</span>(<span class="at">x =</span> val), <span class="at">color =</span> <span class="st">&#39;red3&#39;</span>)</span>
<span id="cb6-97"><a href="phylogenetic-visualization-in-r.html#cb6-97" aria-hidden="true" tabindex="-1"></a>pp2</span>
<span id="cb6-98"><a href="phylogenetic-visualization-in-r.html#cb6-98" aria-hidden="true" tabindex="-1"></a>pp3 <span class="ot">&lt;-</span> <span class="fu">facet_plot</span>(pp2, <span class="at">panel =</span> <span class="st">&#39;trait 2 (bar)&#39;</span>, <span class="at">data =</span> t2, <span class="at">geom =</span> geom_segment, </span>
<span id="cb6-99"><a href="phylogenetic-visualization-in-r.html#cb6-99" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">aes</span>(<span class="at">x =</span> <span class="dv">0</span>, <span class="at">xend =</span> val, <span class="at">y =</span> y, <span class="at">yend =</span> y), <span class="at">size =</span> <span class="dv">3</span>, <span class="at">color =</span> <span class="st">&#39;blue4&#39;</span>) </span>
<span id="cb6-100"><a href="phylogenetic-visualization-in-r.html#cb6-100" aria-hidden="true" tabindex="-1"></a>pp3</span>
<span id="cb6-101"><a href="phylogenetic-visualization-in-r.html#cb6-101" aria-hidden="true" tabindex="-1"></a>pp3 <span class="sc">+</span></span>
<span id="cb6-102"><a href="phylogenetic-visualization-in-r.html#cb6-102" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_tree2</span>() <span class="co"># adds scale bars</span></span></code></pre></div>
<p>See the following link for more in-depth examples of ggtree tree visualizations: <a href="https://yulab-smu.top/treedata-book/" class="uri">https://yulab-smu.top/treedata-book/</a></p>
<p></br></p>
<div id="exercise-1-7" class="section level4 hasAnchor" number="10.3.2.1">
<h4><span class="header-section-number">10.3.2.1</span> Exercise (1)<a href="phylogenetic-visualization-in-r.html#exercise-1-7" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Download the tree and data files with your first name (e.g., “Daniel_tree.nex” and “Daniel_data.csv”) from BrightSpace (lab6).
Using these data:</p>
<ul>
<li><p>Use the <code>facet_plot</code> function in <code>ggtree</code> to plot a tree and traits in two different ways (dot/bar).
Plot weight in both panels.</p></li>
<li><p>Add different colored symbols at the tips of the tree to denote native vs introduced status.</p></li>
</ul>
<p>Save resulting plot with <code>ggsave</code> function as pdf file named “Firstname_Lastname.weight.pdf”.
Write your code and answers below:</p>
<hr />
<hr />
<hr />
<hr />
</div>
</div>
</div>
<div id="plotting-maps" class="section level2 hasAnchor" number="10.4">
<h2><span class="header-section-number">10.4</span> Plotting maps<a href="phylogenetic-visualization-in-r.html#plotting-maps" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Above we mapped different types of information on phylogenies.
Let’s try mapping phylogenetic information across space (i.e., maps).</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="phylogenetic-visualization-in-r.html#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="do">### let&#39;s download a shapefile from the web</span></span>
<span id="cb7-2"><a href="phylogenetic-visualization-in-r.html#cb7-2" aria-hidden="true" tabindex="-1"></a>USA <span class="ot">&lt;-</span> raster<span class="sc">::</span><span class="fu">getData</span>(<span class="at">name =</span> <span class="st">&quot;GADM&quot;</span>, <span class="at">country =</span> <span class="st">&quot;USA&quot;</span>, <span class="at">level =</span> <span class="dv">2</span>)</span>
<span id="cb7-3"><a href="phylogenetic-visualization-in-r.html#cb7-3" aria-hidden="true" tabindex="-1"></a>indiana <span class="ot">&lt;-</span> USA[<span class="fu">which</span>(USA<span class="sc">$</span>NAME_1 <span class="sc">==</span> <span class="st">&quot;Indiana&quot;</span>), ]</span>
<span id="cb7-4"><a href="phylogenetic-visualization-in-r.html#cb7-4" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(indiana)</span>
<span id="cb7-5"><a href="phylogenetic-visualization-in-r.html#cb7-5" aria-hidden="true" tabindex="-1"></a>indiana</span>
<span id="cb7-6"><a href="phylogenetic-visualization-in-r.html#cb7-6" aria-hidden="true" tabindex="-1"></a><span class="fu">View</span>(indiana<span class="sc">@</span>data)</span>
<span id="cb7-7"><a href="phylogenetic-visualization-in-r.html#cb7-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-8"><a href="phylogenetic-visualization-in-r.html#cb7-8" aria-hidden="true" tabindex="-1"></a><span class="fu">proj4string</span>(USA) <span class="co">#checking the coordinate system for our map of the USA</span></span>
<span id="cb7-9"><a href="phylogenetic-visualization-in-r.html#cb7-9" aria-hidden="true" tabindex="-1"></a><span class="fu">proj4string</span>(indiana) <span class="co">#checking the coordinate system for our map of Indiana</span></span></code></pre></div>
<p>The Earth isn’t flat, but small parts of it are close enough, and there are many ways to ‘flatten’ the globe onto a 2-dimensional surface.
A coordinate reference system (CRS) defines, with the help of coordinates, how the two-dimensional, projected map is related to real locations on the earth.
Many coordinate systems exist and any spatial object (e.g., shapefile, raster) can have one.
The important thing is to ensure that the CRS of all interacting spatial objects match - else things will not be plotted/located properly.</p>
<div id="choropleth-maps" class="section level3 hasAnchor" number="10.4.1">
<h3><span class="header-section-number">10.4.1</span> Choropleth maps<a href="phylogenetic-visualization-in-r.html#choropleth-maps" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Choropleth maps use differences in shading or coloring within predefined areas to visualize how a variable varies across a geographic area.
First let’s try plotting the area of counties in Indiana.
The function we will use to calculate area, <code>gArea</code> in the <code>rgeos</code> package, requires the input to be in a planar coordinate system.
Hence we will need to reproject our downloaded shapefile of the state.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="phylogenetic-visualization-in-r.html#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co"># you can transform and reproject spatial objects with different coordinate systems</span></span>
<span id="cb8-2"><a href="phylogenetic-visualization-in-r.html#cb8-2" aria-hidden="true" tabindex="-1"></a><span class="co"># US National Atlas Equal Area (Lambert azimuthal equal-area projection)</span></span>
<span id="cb8-3"><a href="phylogenetic-visualization-in-r.html#cb8-3" aria-hidden="true" tabindex="-1"></a>indiana2 <span class="ot">&lt;-</span> <span class="fu">spTransform</span>(indiana, <span class="fu">CRS</span>(<span class="st">&quot;+proj=laea +lat_0=45 +lon_0=-100 +x_0=0 +y_0=0 +a=6370997 +b=6370997 +units=m +no_defs&quot;</span>)) </span>
<span id="cb8-4"><a href="phylogenetic-visualization-in-r.html#cb8-4" aria-hidden="true" tabindex="-1"></a><span class="fu">proj4string</span>(indiana2)</span>
<span id="cb8-5"><a href="phylogenetic-visualization-in-r.html#cb8-5" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(indiana2)</span>
<span id="cb8-6"><a href="phylogenetic-visualization-in-r.html#cb8-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-7"><a href="phylogenetic-visualization-in-r.html#cb8-7" aria-hidden="true" tabindex="-1"></a><span class="fu">gArea</span>(indiana2)</span>
<span id="cb8-8"><a href="phylogenetic-visualization-in-r.html#cb8-8" aria-hidden="true" tabindex="-1"></a><span class="fu">gArea</span>(indiana2, <span class="at">byid =</span> <span class="cn">TRUE</span>) </span>
<span id="cb8-9"><a href="phylogenetic-visualization-in-r.html#cb8-9" aria-hidden="true" tabindex="-1"></a>indiana<span class="sc">$</span>area <span class="ot">&lt;-</span> <span class="fu">gArea</span>(indiana2, <span class="at">byid =</span> <span class="cn">TRUE</span>)  <span class="co">#gArea requires a planar coordinate system (not lonlat)</span></span>
<span id="cb8-10"><a href="phylogenetic-visualization-in-r.html#cb8-10" aria-hidden="true" tabindex="-1"></a><span class="fu">View</span>(indiana<span class="sc">@</span>data)</span>
<span id="cb8-11"><a href="phylogenetic-visualization-in-r.html#cb8-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-12"><a href="phylogenetic-visualization-in-r.html#cb8-12" aria-hidden="true" tabindex="-1"></a><span class="do">## add the ID to the dataframe itself for easier indexing</span></span>
<span id="cb8-13"><a href="phylogenetic-visualization-in-r.html#cb8-13" aria-hidden="true" tabindex="-1"></a>indiana<span class="sc">$</span>id <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">rownames</span>(indiana2<span class="sc">@</span>data))</span>
<span id="cb8-14"><a href="phylogenetic-visualization-in-r.html#cb8-14" aria-hidden="true" tabindex="-1"></a><span class="do">## create fortified version for plotting with ggplot()</span></span>
<span id="cb8-15"><a href="phylogenetic-visualization-in-r.html#cb8-15" aria-hidden="true" tabindex="-1"></a>indiana.fortified <span class="ot">&lt;-</span> <span class="fu">fortify</span>(indiana, <span class="at">region =</span> <span class="st">&quot;id&quot;</span>)</span>
<span id="cb8-16"><a href="phylogenetic-visualization-in-r.html#cb8-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-17"><a href="phylogenetic-visualization-in-r.html#cb8-17" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(indiana<span class="sc">@</span>data, <span class="fu">aes</span>(<span class="at">map_id =</span> id)) <span class="sc">+</span></span>
<span id="cb8-18"><a href="phylogenetic-visualization-in-r.html#cb8-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">expand_limits</span>(<span class="at">x =</span> indiana.fortified<span class="sc">$</span>long, <span class="at">y =</span> indiana.fortified<span class="sc">$</span>lat)<span class="sc">+</span></span>
<span id="cb8-19"><a href="phylogenetic-visualization-in-r.html#cb8-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_gradientn</span>(<span class="at">colours =</span> <span class="fu">c</span>(<span class="st">&quot;grey&quot;</span>,<span class="st">&quot;goldenrod&quot;</span>,<span class="st">&quot;darkgreen&quot;</span>,<span class="st">&quot;green&quot;</span>))<span class="sc">+</span> <span class="co">#experiment with different colors!</span></span>
<span id="cb8-20"><a href="phylogenetic-visualization-in-r.html#cb8-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_map</span>()<span class="sc">+</span></span>
<span id="cb8-21"><a href="phylogenetic-visualization-in-r.html#cb8-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_map</span>(<span class="fu">aes</span>(<span class="at">fill =</span> area), <span class="at">map =</span> indiana.fortified)</span></code></pre></div>
<p>Let’s try plotting phylogenetic diversity.
We will start by creating a community matrix like we used in previous labs.
Phylogenetic diversity is often calculated across spatial units.
Here we will treat each county as a community.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="phylogenetic-visualization-in-r.html#cb9-1" aria-hidden="true" tabindex="-1"></a>comm.indiana <span class="ot">&lt;-</span> <span class="cn">NULL</span> <span class="co">#creating empty object</span></span>
<span id="cb9-2"><a href="phylogenetic-visualization-in-r.html#cb9-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-3"><a href="phylogenetic-visualization-in-r.html#cb9-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(indiana<span class="sc">@</span>data)){</span>
<span id="cb9-4"><a href="phylogenetic-visualization-in-r.html#cb9-4" aria-hidden="true" tabindex="-1"></a>  pa <span class="ot">&lt;-</span> <span class="fu">sample</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>), <span class="at">replace =</span> <span class="cn">TRUE</span>, <span class="at">size =</span> <span class="fu">Ntip</span>(tree))</span>
<span id="cb9-5"><a href="phylogenetic-visualization-in-r.html#cb9-5" aria-hidden="true" tabindex="-1"></a>  comm.indiana <span class="ot">&lt;-</span> <span class="fu">rbind</span>(comm.indiana, pa)</span>
<span id="cb9-6"><a href="phylogenetic-visualization-in-r.html#cb9-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(i)</span>
<span id="cb9-7"><a href="phylogenetic-visualization-in-r.html#cb9-7" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb9-8"><a href="phylogenetic-visualization-in-r.html#cb9-8" aria-hidden="true" tabindex="-1"></a><span class="fu">View</span>(comm.indiana)</span>
<span id="cb9-9"><a href="phylogenetic-visualization-in-r.html#cb9-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-10"><a href="phylogenetic-visualization-in-r.html#cb9-10" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(comm.indiana) <span class="ot">&lt;-</span> tree<span class="sc">$</span>tip.label </span>
<span id="cb9-11"><a href="phylogenetic-visualization-in-r.html#cb9-11" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(comm.indiana) <span class="ot">&lt;-</span> indiana<span class="sc">$</span>NAME_2</span>
<span id="cb9-12"><a href="phylogenetic-visualization-in-r.html#cb9-12" aria-hidden="true" tabindex="-1"></a><span class="fu">View</span>(comm.indiana)</span>
<span id="cb9-13"><a href="phylogenetic-visualization-in-r.html#cb9-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-14"><a href="phylogenetic-visualization-in-r.html#cb9-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate PD</span></span>
<span id="cb9-15"><a href="phylogenetic-visualization-in-r.html#cb9-15" aria-hidden="true" tabindex="-1"></a>pd.indiana <span class="ot">&lt;-</span> picante<span class="sc">::</span><span class="fu">pd</span>(comm.indiana, tree)</span>
<span id="cb9-16"><a href="phylogenetic-visualization-in-r.html#cb9-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-17"><a href="phylogenetic-visualization-in-r.html#cb9-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Attach PD to Indiana shapefile data</span></span>
<span id="cb9-18"><a href="phylogenetic-visualization-in-r.html#cb9-18" aria-hidden="true" tabindex="-1"></a>indiana<span class="sc">$</span>PD <span class="ot">&lt;-</span> pd.indiana<span class="sc">$</span>PD</span>
<span id="cb9-19"><a href="phylogenetic-visualization-in-r.html#cb9-19" aria-hidden="true" tabindex="-1"></a><span class="fu">View</span>(indiana<span class="sc">@</span>data)</span>
<span id="cb9-20"><a href="phylogenetic-visualization-in-r.html#cb9-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-21"><a href="phylogenetic-visualization-in-r.html#cb9-21" aria-hidden="true" tabindex="-1"></a><span class="co"># Create fortified version for plotting with ggplot()</span></span>
<span id="cb9-22"><a href="phylogenetic-visualization-in-r.html#cb9-22" aria-hidden="true" tabindex="-1"></a>indiana.fortified <span class="ot">&lt;-</span> <span class="fu">fortify</span>(indiana, <span class="at">region =</span> <span class="st">&quot;id&quot;</span>)</span>
<span id="cb9-23"><a href="phylogenetic-visualization-in-r.html#cb9-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-24"><a href="phylogenetic-visualization-in-r.html#cb9-24" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(indiana<span class="sc">@</span>data, <span class="fu">aes</span>(<span class="at">map_id =</span> id)) <span class="sc">+</span></span>
<span id="cb9-25"><a href="phylogenetic-visualization-in-r.html#cb9-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">expand_limits</span>(<span class="at">x =</span> indiana.fortified<span class="sc">$</span>long, <span class="at">y =</span> indiana.fortified<span class="sc">$</span>lat)<span class="sc">+</span></span>
<span id="cb9-26"><a href="phylogenetic-visualization-in-r.html#cb9-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_map</span>()<span class="sc">+</span></span>
<span id="cb9-27"><a href="phylogenetic-visualization-in-r.html#cb9-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_map</span>(<span class="fu">aes</span>(<span class="at">fill =</span> PD), <span class="at">map =</span> indiana.fortified)</span>
<span id="cb9-28"><a href="phylogenetic-visualization-in-r.html#cb9-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-29"><a href="phylogenetic-visualization-in-r.html#cb9-29" aria-hidden="true" tabindex="-1"></a><span class="co"># You can save plots made with ggplot2 in a different way</span></span>
<span id="cb9-30"><a href="phylogenetic-visualization-in-r.html#cb9-30" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(indiana<span class="sc">@</span>data, <span class="fu">aes</span>(<span class="at">map_id =</span> id)) <span class="sc">+</span></span>
<span id="cb9-31"><a href="phylogenetic-visualization-in-r.html#cb9-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">expand_limits</span>(<span class="at">x =</span> indiana.fortified<span class="sc">$</span>long, <span class="at">y =</span> indiana.fortified<span class="sc">$</span>lat) <span class="sc">+</span></span>
<span id="cb9-32"><a href="phylogenetic-visualization-in-r.html#cb9-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_map</span>() <span class="sc">+</span></span>
<span id="cb9-33"><a href="phylogenetic-visualization-in-r.html#cb9-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_map</span>(<span class="fu">aes</span>(<span class="at">fill =</span> PD), <span class="at">map =</span> indiana.fortified) <span class="sc">+</span></span>
<span id="cb9-34"><a href="phylogenetic-visualization-in-r.html#cb9-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span>
<span id="cb9-35"><a href="phylogenetic-visualization-in-r.html#cb9-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-36"><a href="phylogenetic-visualization-in-r.html#cb9-36" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;Indiana_PD.pdf&quot;</span>, <span class="at">plot =</span> p, <span class="at">width =</span> <span class="dv">4</span>, <span class="at">height =</span> <span class="dv">6</span>)</span></code></pre></div>
</div>
<div id="working-with-rasters" class="section level3 hasAnchor" number="10.4.2">
<h3><span class="header-section-number">10.4.2</span> Working with rasters<a href="phylogenetic-visualization-in-r.html#working-with-rasters" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Rasters refer to spatial data structures dividing a region (e.g., a grid) into rectangles (e.g., cells, pixels) storing one or more values each.
Raster data files can be very large, especially when cells are at high resolution.
It is thus important to think about how much RAM is required to work with a raster.
As we learned previously, many types of spatially distributed data (e.g., climate variables) are stored as rasters.
Chances are, if you want to associate an organism’s occurrences with the environmental conditions of the localities, you will need to extract that information from rasters.
Let’s start by examining some rasters from <code>WorldClim</code>.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="phylogenetic-visualization-in-r.html#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Let&#39;s download a raster from WorldClim</span></span>
<span id="cb10-2"><a href="phylogenetic-visualization-in-r.html#cb10-2" aria-hidden="true" tabindex="-1"></a>bio <span class="ot">&lt;-</span> raster<span class="sc">::</span><span class="fu">getData</span>(<span class="st">&#39;worldclim&#39;</span>, <span class="at">var =</span> <span class="st">&#39;bio&#39;</span>, <span class="at">res =</span> <span class="dv">10</span>) <span class="co"># 19 bioclimatic variables</span></span>
<span id="cb10-3"><a href="phylogenetic-visualization-in-r.html#cb10-3" aria-hidden="true" tabindex="-1"></a>bio</span>
<span id="cb10-4"><a href="phylogenetic-visualization-in-r.html#cb10-4" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(bio[[<span class="dv">1</span>]])</span>
<span id="cb10-5"><a href="phylogenetic-visualization-in-r.html#cb10-5" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(bio[[<span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>]])</span>
<span id="cb10-6"><a href="phylogenetic-visualization-in-r.html#cb10-6" aria-hidden="true" tabindex="-1"></a><span class="fu">gain</span>(bio) <span class="ot">&lt;-</span> <span class="fl">0.1</span> <span class="co"># bioclimatic variables are in mm (precipitation variables) and 10x Celsius (temperature variables)</span></span>
<span id="cb10-7"><a href="phylogenetic-visualization-in-r.html#cb10-7" aria-hidden="true" tabindex="-1"></a>bio</span>
<span id="cb10-8"><a href="phylogenetic-visualization-in-r.html#cb10-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-9"><a href="phylogenetic-visualization-in-r.html#cb10-9" aria-hidden="true" tabindex="-1"></a><span class="co"># cropping rasters</span></span>
<span id="cb10-10"><a href="phylogenetic-visualization-in-r.html#cb10-10" aria-hidden="true" tabindex="-1"></a>bio1 <span class="ot">&lt;-</span> bio[[<span class="dv">1</span>]]</span>
<span id="cb10-11"><a href="phylogenetic-visualization-in-r.html#cb10-11" aria-hidden="true" tabindex="-1"></a>bio1.cropped <span class="ot">&lt;-</span> <span class="fu">crop</span>(bio1, <span class="fu">extent</span>(<span class="sc">-</span><span class="dv">86</span>,<span class="sc">-</span><span class="dv">85</span>, <span class="dv">40</span>, <span class="dv">41</span>)) <span class="co"># cropping with bounding box</span></span>
<span id="cb10-12"><a href="phylogenetic-visualization-in-r.html#cb10-12" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(bio1.cropped)</span>
<span id="cb10-13"><a href="phylogenetic-visualization-in-r.html#cb10-13" aria-hidden="true" tabindex="-1"></a><span class="fu">cellStats</span>(bio1.cropped, range)</span>
<span id="cb10-14"><a href="phylogenetic-visualization-in-r.html#cb10-14" aria-hidden="true" tabindex="-1"></a>below10 <span class="ot">&lt;-</span> bio1.cropped <span class="sc">&lt;</span> <span class="dv">10</span></span>
<span id="cb10-15"><a href="phylogenetic-visualization-in-r.html#cb10-15" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(below10)</span>
<span id="cb10-16"><a href="phylogenetic-visualization-in-r.html#cb10-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-17"><a href="phylogenetic-visualization-in-r.html#cb10-17" aria-hidden="true" tabindex="-1"></a>bio1.indiana <span class="ot">&lt;-</span> <span class="fu">crop</span>(bio1, <span class="fu">extent</span>(indiana)) <span class="co"># cropping with box bounding Indiana</span></span>
<span id="cb10-18"><a href="phylogenetic-visualization-in-r.html#cb10-18" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(bio1.indiana)</span>
<span id="cb10-19"><a href="phylogenetic-visualization-in-r.html#cb10-19" aria-hidden="true" tabindex="-1"></a>bio1.indiana <span class="ot">&lt;-</span> <span class="fu">mask</span>(bio1.indiana, indiana) <span class="co"># masking non-Indiana areas in the box</span></span>
<span id="cb10-20"><a href="phylogenetic-visualization-in-r.html#cb10-20" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(bio1.indiana)</span></code></pre></div>
<p>Now let’s try to extract values from rasters using coordinates.
Remember, all interacting spatial objects need to have the same CRS.
Along those lines,you cannot use a non-spatial string of numbers to extract information from a spatial object.
Nor can you plot a non-spatial string of numbers that represent coordinates on a spatial object.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="phylogenetic-visualization-in-r.html#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Getting centroid coordinates</span></span>
<span id="cb11-2"><a href="phylogenetic-visualization-in-r.html#cb11-2" aria-hidden="true" tabindex="-1"></a>centroids <span class="ot">&lt;-</span> <span class="fu">gCentroid</span>(indiana)</span>
<span id="cb11-3"><a href="phylogenetic-visualization-in-r.html#cb11-3" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(indiana)</span>
<span id="cb11-4"><a href="phylogenetic-visualization-in-r.html#cb11-4" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(centroids, <span class="at">col =</span> <span class="st">&quot;blue&quot;</span>, <span class="at">add =</span> <span class="cn">TRUE</span>)</span>
<span id="cb11-5"><a href="phylogenetic-visualization-in-r.html#cb11-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-6"><a href="phylogenetic-visualization-in-r.html#cb11-6" aria-hidden="true" tabindex="-1"></a>centroids <span class="ot">&lt;-</span> <span class="fu">gCentroid</span>(indiana, <span class="at">byid =</span> <span class="cn">TRUE</span>)</span>
<span id="cb11-7"><a href="phylogenetic-visualization-in-r.html#cb11-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-8"><a href="phylogenetic-visualization-in-r.html#cb11-8" aria-hidden="true" tabindex="-1"></a><span class="co"># if the projections of the objects are different, use spTranform to standardize them</span></span>
<span id="cb11-9"><a href="phylogenetic-visualization-in-r.html#cb11-9" aria-hidden="true" tabindex="-1"></a><span class="fu">proj4string</span>(centroids)</span>
<span id="cb11-10"><a href="phylogenetic-visualization-in-r.html#cb11-10" aria-hidden="true" tabindex="-1"></a><span class="fu">proj4string</span>(indiana)</span>
<span id="cb11-11"><a href="phylogenetic-visualization-in-r.html#cb11-11" aria-hidden="true" tabindex="-1"></a><span class="fu">proj4string</span>(bio1)</span>
<span id="cb11-12"><a href="phylogenetic-visualization-in-r.html#cb11-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-13"><a href="phylogenetic-visualization-in-r.html#cb11-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Extracting climate values of county centroid coordinates</span></span>
<span id="cb11-14"><a href="phylogenetic-visualization-in-r.html#cb11-14" aria-hidden="true" tabindex="-1"></a>bio1.centroids <span class="ot">&lt;-</span> raster<span class="sc">::</span><span class="fu">extract</span>(bio1, centroids)</span>
<span id="cb11-15"><a href="phylogenetic-visualization-in-r.html#cb11-15" aria-hidden="true" tabindex="-1"></a>bio1.centroids</span>
<span id="cb11-16"><a href="phylogenetic-visualization-in-r.html#cb11-16" aria-hidden="true" tabindex="-1"></a>bio1.centroids <span class="ot">&lt;-</span> <span class="fu">cbind</span>(centroids<span class="sc">@</span>coords, bio1.centroids) <span class="co"># combine with coordinates</span></span>
<span id="cb11-17"><a href="phylogenetic-visualization-in-r.html#cb11-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-18"><a href="phylogenetic-visualization-in-r.html#cb11-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Extracting climate values of polygons (an area)</span></span>
<span id="cb11-19"><a href="phylogenetic-visualization-in-r.html#cb11-19" aria-hidden="true" tabindex="-1"></a>bio1.indiana <span class="ot">&lt;-</span> raster<span class="sc">::</span><span class="fu">extract</span>(bio1, indiana) <span class="co"># raw values - may take a while</span></span>
<span id="cb11-20"><a href="phylogenetic-visualization-in-r.html#cb11-20" aria-hidden="true" tabindex="-1"></a>bio1.indiana.mean <span class="ot">&lt;-</span> raster<span class="sc">::</span><span class="fu">extract</span>(bio1, indiana, <span class="at">fun =</span> mean) <span class="co"># aggregate to county means - may take a while</span></span>
<span id="cb11-21"><a href="phylogenetic-visualization-in-r.html#cb11-21" aria-hidden="true" tabindex="-1"></a>bio1.indiana <span class="ot">&lt;-</span> <span class="fu">cbind</span>(indiana<span class="sc">@</span>data, <span class="at">mean_annual_temp =</span> bio1.indiana.mean) <span class="co"># this wont work unless you&#39;re using bio1 (and not bio1.cropped)</span></span></code></pre></div>
<p></br></p>
<div id="exercise-2-7" class="section level4 hasAnchor" number="10.4.2.1">
<h4><span class="header-section-number">10.4.2.1</span> Exercise (2)<a href="phylogenetic-visualization-in-r.html#exercise-2-7" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Download the tree and data files with your first name (e.g., “Daniel_tree.nex” and “Daniel_data.csv”) from BrightSpace (lab6).
Using these data:</p>
<ul>
<li><p>Plot a choropleth map of (alpha) MPD across Indiana counties using randomly generated communities and your tree.</p></li>
<li><p>Plot county (alpha) MPD (x-axis) against county mean annual precipitation (bio12; y-axis) across Indiana counties as a scatterplot.
Add a regression line (abline) illustrating their relationship.</p></li>
</ul>
<p>Feel free to experiment with plot parameters (e.g., colors, pch).
Save your plots separately as pdfs named “Firstname_Lastname.choropleth.pdf” and “Firstname_Lastname.scatter.pdf” respectively.</p>
<p>Write your code and answers below:</p>
<hr />
<hr />
<hr />
<hr />
<p></br></p>
<p><strong>Assignment [11]: Upload the code you wrote for the two exercises and their results to BrightSpace (lab11).</strong></p>
<p></br></p>

</div>
</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="coevolution-and-phylogenetic-signals-in-r.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="gene-tree-vs-species-tree.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/09_R_lab8_phyloViz.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["bookdown-demo.pdf", "bookdown-demo.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
