<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 4 R Basics II | BIO595 The Tree of Life</title>
  <meta name="description" content="Test writing the book for the course bookdown::gitbook." />
  <meta name="generator" content="bookdown 0.28 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 4 R Basics II | BIO595 The Tree of Life" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="Test writing the book for the course bookdown::gitbook." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 4 R Basics II | BIO595 The Tree of Life" />
  
  <meta name="twitter:description" content="Test writing the book for the course bookdown::gitbook." />
  

<meta name="author" content="Qixin He, Daniel Park" />


<meta name="date" content="2023-04-10" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="r-basics-i.html"/>
<link rel="next" href="tree-and-traits-in-r.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Tree of Life: Phylogenetic inferences and applications</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Overview</a></li>
<li class="chapter" data-level="2" data-path="unix.html"><a href="unix.html"><i class="fa fa-check"></i><b>2</b> Unix</a>
<ul>
<li class="chapter" data-level="2.1" data-path="unix.html"><a href="unix.html#what-is-unix-and-why-do-we-use-it"><i class="fa fa-check"></i><b>2.1</b> What is Unix and why do we use it?</a></li>
<li class="chapter" data-level="2.2" data-path="unix.html"><a href="unix.html#installation"><i class="fa fa-check"></i><b>2.2</b> Installation</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="unix.html"><a href="unix.html#mac-os-and-linux"><i class="fa fa-check"></i><b>2.2.1</b> Mac OS and Linux</a></li>
<li class="chapter" data-level="2.2.2" data-path="unix.html"><a href="unix.html#windows"><i class="fa fa-check"></i><b>2.2.2</b> Windows</a></li>
<li class="chapter" data-level="2.2.3" data-path="unix.html"><a href="unix.html#use-scholar-remote-desktop-during-the-class"><i class="fa fa-check"></i><b>2.2.3</b> Use Scholar Remote Desktop During the Class</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="unix.html"><a href="unix.html#operation"><i class="fa fa-check"></i><b>2.3</b> Operation</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="unix.html"><a href="unix.html#starting-the-shell"><i class="fa fa-check"></i><b>2.3.1</b> Starting the Shell</a></li>
<li class="chapter" data-level="2.3.2" data-path="unix.html"><a href="unix.html#directoryfolder-sturcture"><i class="fa fa-check"></i><b>2.3.2</b> Directory/folder sturcture</a></li>
<li class="chapter" data-level="2.3.3" data-path="unix.html"><a href="unix.html#absolute-and-relative-directory"><i class="fa fa-check"></i><b>2.3.3</b> Absolute and Relative directory</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="unix.html"><a href="unix.html#handling-directories-and-files"><i class="fa fa-check"></i><b>2.4</b> Handling directories and files</a></li>
<li class="chapter" data-level="2.5" data-path="unix.html"><a href="unix.html#excercises-i"><i class="fa fa-check"></i><b>2.5</b> Excercises I</a></li>
<li class="chapter" data-level="2.6" data-path="unix.html"><a href="unix.html#viewing-and-processing-text-files"><i class="fa fa-check"></i><b>2.6</b> Viewing and processing text files</a></li>
<li class="chapter" data-level="2.7" data-path="unix.html"><a href="unix.html#redirecting-printouts-and-piping"><i class="fa fa-check"></i><b>2.7</b> Redirecting printouts and piping</a></li>
<li class="chapter" data-level="2.8" data-path="unix.html"><a href="unix.html#exercises-ii"><i class="fa fa-check"></i><b>2.8</b> Exercises II</a></li>
<li class="chapter" data-level="2.9" data-path="unix.html"><a href="unix.html#useful-wildcards"><i class="fa fa-check"></i><b>2.9</b> Useful wildcards</a></li>
<li class="chapter" data-level="2.10" data-path="unix.html"><a href="unix.html#exercises-iii"><i class="fa fa-check"></i><b>2.10</b> Exercises III</a></li>
<li class="chapter" data-level="2.11" data-path="unix.html"><a href="unix.html#search-and-select-lines-using-grep"><i class="fa fa-check"></i><b>2.11</b> Search and select lines using <em>grep</em></a></li>
<li class="chapter" data-level="2.12" data-path="unix.html"><a href="unix.html#substituting-characters-using-tr"><i class="fa fa-check"></i><b>2.12</b> Substituting characters using <em>tr</em></a></li>
<li class="chapter" data-level="2.13" data-path="unix.html"><a href="unix.html#cut-sort-and-unique"><i class="fa fa-check"></i><b>2.13</b> cut, sort and unique</a></li>
<li class="chapter" data-level="2.14" data-path="unix.html"><a href="unix.html#exercises-iv"><i class="fa fa-check"></i><b>2.14</b> Exercises IV</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="r-basics-i.html"><a href="r-basics-i.html"><i class="fa fa-check"></i><b>3</b> R Basics I</a>
<ul>
<li class="chapter" data-level="3.1" data-path="r-basics-i.html"><a href="r-basics-i.html#what-is-programming"><i class="fa fa-check"></i><b>3.1</b> What is programming?</a></li>
<li class="chapter" data-level="3.2" data-path="r-basics-i.html"><a href="r-basics-i.html#r"><i class="fa fa-check"></i><b>3.2</b> <code>R</code></a></li>
<li class="chapter" data-level="3.3" data-path="r-basics-i.html"><a href="r-basics-i.html#installation-1"><i class="fa fa-check"></i><b>3.3</b> Installation</a></li>
<li class="chapter" data-level="3.4" data-path="r-basics-i.html"><a href="r-basics-i.html#variables"><i class="fa fa-check"></i><b>3.4</b> Variables</a>
<ul>
<li class="chapter" data-level="3.4.1" data-path="r-basics-i.html"><a href="r-basics-i.html#exercise-1"><i class="fa fa-check"></i><b>3.4.1</b> Exercise (1):</a></li>
<li class="chapter" data-level="3.4.2" data-path="r-basics-i.html"><a href="r-basics-i.html#variable-naming-conventions"><i class="fa fa-check"></i><b>3.4.2</b> Variable naming conventions</a></li>
<li class="chapter" data-level="3.4.3" data-path="r-basics-i.html"><a href="r-basics-i.html#arithmetic-operators"><i class="fa fa-check"></i><b>3.4.3</b> Arithmetic Operators</a></li>
<li class="chapter" data-level="3.4.4" data-path="r-basics-i.html"><a href="r-basics-i.html#logical-operators"><i class="fa fa-check"></i><b>3.4.4</b> Logical Operators</a></li>
<li class="chapter" data-level="3.4.5" data-path="r-basics-i.html"><a href="r-basics-i.html#basic-built-in-mathematical-functions"><i class="fa fa-check"></i><b>3.4.5</b> Basic built-in mathematical functions</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="r-basics-i.html"><a href="r-basics-i.html#subsetting-data"><i class="fa fa-check"></i><b>3.5</b> Subsetting data</a></li>
<li class="chapter" data-level="3.6" data-path="r-basics-i.html"><a href="r-basics-i.html#logical-values"><i class="fa fa-check"></i><b>3.6</b> Logical values</a>
<ul>
<li class="chapter" data-level="3.6.1" data-path="r-basics-i.html"><a href="r-basics-i.html#missing-dealing-with-na-values"><i class="fa fa-check"></i><b>3.6.1</b> Missing: dealing with <code>NA</code> values</a></li>
</ul></li>
<li class="chapter" data-level="3.7" data-path="r-basics-i.html"><a href="r-basics-i.html#generating-data"><i class="fa fa-check"></i><b>3.7</b> Generating Data</a></li>
<li class="chapter" data-level="3.8" data-path="r-basics-i.html"><a href="r-basics-i.html#data-structures-and-types"><i class="fa fa-check"></i><b>3.8</b> Data Structures and Types</a>
<ul>
<li class="chapter" data-level="3.8.1" data-path="r-basics-i.html"><a href="r-basics-i.html#vectors"><i class="fa fa-check"></i><b>3.8.1</b> Vectors</a></li>
</ul></li>
<li class="chapter" data-level="3.9" data-path="r-basics-i.html"><a href="r-basics-i.html#matrices"><i class="fa fa-check"></i><b>3.9</b> Matrices</a>
<ul>
<li class="chapter" data-level="3.9.1" data-path="r-basics-i.html"><a href="r-basics-i.html#exercise-2"><i class="fa fa-check"></i><b>3.9.1</b> Exercise (2):</a></li>
</ul></li>
<li class="chapter" data-level="3.10" data-path="r-basics-i.html"><a href="r-basics-i.html#data-frames"><i class="fa fa-check"></i><b>3.10</b> Data Frames</a>
<ul>
<li class="chapter" data-level="3.10.1" data-path="r-basics-i.html"><a href="r-basics-i.html#exercise-3"><i class="fa fa-check"></i><b>3.10.1</b> Exercise (3):</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="r-basics-ii.html"><a href="r-basics-ii.html"><i class="fa fa-check"></i><b>4</b> R Basics II</a>
<ul>
<li class="chapter" data-level="4.1" data-path="r-basics-ii.html"><a href="r-basics-ii.html#setting-up-the-environment"><i class="fa fa-check"></i><b>4.1</b> Setting up the environment</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="r-basics-ii.html"><a href="r-basics-ii.html#exercise-1-1"><i class="fa fa-check"></i><b>4.1.1</b> Exercise (1)</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="r-basics-ii.html"><a href="r-basics-ii.html#reading-in-data-from-files"><i class="fa fa-check"></i><b>4.2</b> Reading in data from files</a></li>
<li class="chapter" data-level="4.3" data-path="r-basics-ii.html"><a href="r-basics-ii.html#wrangling-data"><i class="fa fa-check"></i><b>4.3</b> Wrangling Data</a></li>
<li class="chapter" data-level="4.4" data-path="r-basics-ii.html"><a href="r-basics-ii.html#subsetting-by-columns-and-rows"><i class="fa fa-check"></i><b>4.4</b> Subsetting by columns and rows</a></li>
<li class="chapter" data-level="4.5" data-path="r-basics-ii.html"><a href="r-basics-ii.html#subsetting-with-conditions"><i class="fa fa-check"></i><b>4.5</b> Subsetting with conditions</a>
<ul>
<li class="chapter" data-level="4.5.1" data-path="r-basics-ii.html"><a href="r-basics-ii.html#exercise-2-1"><i class="fa fa-check"></i><b>4.5.1</b> Exercise (2)</a></li>
<li class="chapter" data-level="4.5.2" data-path="r-basics-ii.html"><a href="r-basics-ii.html#bonus-1"><i class="fa fa-check"></i><b>4.5.2</b> Bonus [1]</a></li>
</ul></li>
<li class="chapter" data-level="4.6" data-path="r-basics-ii.html"><a href="r-basics-ii.html#merging-data"><i class="fa fa-check"></i><b>4.6</b> Merging data</a></li>
<li class="chapter" data-level="4.7" data-path="r-basics-ii.html"><a href="r-basics-ii.html#saving-files-and-sessions"><i class="fa fa-check"></i><b>4.7</b> Saving files and sessions</a>
<ul>
<li class="chapter" data-level="4.7.1" data-path="r-basics-ii.html"><a href="r-basics-ii.html#exercise-3-1"><i class="fa fa-check"></i><b>4.7.1</b> Exercise (3)</a></li>
</ul></li>
<li class="chapter" data-level="4.8" data-path="r-basics-ii.html"><a href="r-basics-ii.html#the-for-loop"><i class="fa fa-check"></i><b>4.8</b> The <span class="smallcaps">for</span>-loop</a></li>
<li class="chapter" data-level="4.9" data-path="r-basics-ii.html"><a href="r-basics-ii.html#lists"><i class="fa fa-check"></i><b>4.9</b> Lists</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="tree-and-traits-in-r.html"><a href="tree-and-traits-in-r.html"><i class="fa fa-check"></i><b>5</b> Tree and Traits in R</a>
<ul>
<li class="chapter" data-level="5.1" data-path="tree-and-traits-in-r.html"><a href="tree-and-traits-in-r.html#setting-up-the-environment-1"><i class="fa fa-check"></i><b>5.1</b> Setting up the environment</a></li>
<li class="chapter" data-level="5.2" data-path="tree-and-traits-in-r.html"><a href="tree-and-traits-in-r.html#examining-phylogenetic-trees"><i class="fa fa-check"></i><b>5.2</b> Examining phylogenetic trees</a></li>
<li class="chapter" data-level="5.3" data-path="tree-and-traits-in-r.html"><a href="tree-and-traits-in-r.html#plotting-phylogenetic-trees"><i class="fa fa-check"></i><b>5.3</b> Plotting phylogenetic trees</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="tree-and-traits-in-r.html"><a href="tree-and-traits-in-r.html#exercise-1-2"><i class="fa fa-check"></i><b>5.3.1</b> Exercise (1)</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="tree-and-traits-in-r.html"><a href="tree-and-traits-in-r.html#associating-phylogenies-and-traits"><i class="fa fa-check"></i><b>5.4</b> Associating phylogenies and traits</a>
<ul>
<li class="chapter" data-level="5.4.1" data-path="tree-and-traits-in-r.html"><a href="tree-and-traits-in-r.html#saving-plots"><i class="fa fa-check"></i><b>5.4.1</b> Saving plots</a></li>
<li class="chapter" data-level="5.4.2" data-path="tree-and-traits-in-r.html"><a href="tree-and-traits-in-r.html#exercise-2-2"><i class="fa fa-check"></i><b>5.4.2</b> Exercise (2)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html"><i class="fa fa-check"></i><b>6</b> Biogeography and Macroevolution in R</a>
<ul>
<li class="chapter" data-level="6.1" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#setting-up-the-environment-2"><i class="fa fa-check"></i><b>6.1</b> Setting up the environment</a></li>
<li class="chapter" data-level="6.2" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#ancestral-state-reconstructions"><i class="fa fa-check"></i><b>6.2</b> Ancestral State Reconstructions</a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#bonus-2"><i class="fa fa-check"></i><b>6.2.1</b> Bonus [2]</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#continuous-traits"><i class="fa fa-check"></i><b>6.3</b> Continuous traits</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#maximum-likelihood"><i class="fa fa-check"></i><b>6.3.1</b> Maximum likelihood</a></li>
<li class="chapter" data-level="6.3.2" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#phylogenetically-independent-contrasts"><i class="fa fa-check"></i><b>6.3.2</b> Phylogenetically independent contrasts</a></li>
<li class="chapter" data-level="6.3.3" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#plotting-ancestral-states"><i class="fa fa-check"></i><b>6.3.3</b> Plotting ancestral states</a></li>
<li class="chapter" data-level="6.3.4" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#exercise-1-3"><i class="fa fa-check"></i><b>6.3.4</b> Exercise (1)</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#discrete-traits"><i class="fa fa-check"></i><b>6.4</b> Discrete traits</a>
<ul>
<li class="chapter" data-level="6.4.1" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#exercise-2-3"><i class="fa fa-check"></i><b>6.4.1</b> Exercise (2)</a></li>
<li class="chapter" data-level="6.4.2" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#maximum-likelihood-1"><i class="fa fa-check"></i><b>6.4.2</b> Maximum likelihood</a></li>
<li class="chapter" data-level="6.4.3" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#choosing-models"><i class="fa fa-check"></i><b>6.4.3</b> Choosing models</a></li>
</ul></li>
<li class="chapter" data-level="6.5" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#saving-multiple-plots-together"><i class="fa fa-check"></i><b>6.5</b> Saving multiple plots together</a>
<ul>
<li class="chapter" data-level="6.5.1" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#exercise-3-2"><i class="fa fa-check"></i><b>6.5.1</b> Exercise (3)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="phylogenetic-diversity-in-r.html"><a href="phylogenetic-diversity-in-r.html"><i class="fa fa-check"></i><b>7</b> Phylogenetic Diversity in R</a>
<ul>
<li class="chapter" data-level="7.1" data-path="phylogenetic-diversity-in-r.html"><a href="phylogenetic-diversity-in-r.html#setting-up-the-environment-3"><i class="fa fa-check"></i><b>7.1</b> Setting up the environment</a></li>
<li class="chapter" data-level="7.2" data-path="phylogenetic-diversity-in-r.html"><a href="phylogenetic-diversity-in-r.html#phylogenetic-diversity"><i class="fa fa-check"></i><b>7.2</b> Phylogenetic Diversity</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="phylogenetic-diversity-in-r.html"><a href="phylogenetic-diversity-in-r.html#preprocessing"><i class="fa fa-check"></i><b>7.2.1</b> Preprocessing</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="phylogenetic-diversity-in-r.html"><a href="phylogenetic-diversity-in-r.html#estimating-phylogenetic-diversity-metrics"><i class="fa fa-check"></i><b>7.3</b> Estimating phylogenetic diversity metrics</a>
<ul>
<li class="chapter" data-level="7.3.1" data-path="phylogenetic-diversity-in-r.html"><a href="phylogenetic-diversity-in-r.html#exercise-1-4"><i class="fa fa-check"></i><b>7.3.1</b> Exercise (1)</a></li>
<li class="chapter" data-level="7.3.2" data-path="phylogenetic-diversity-in-r.html"><a href="phylogenetic-diversity-in-r.html#exercise-2-4"><i class="fa fa-check"></i><b>7.3.2</b> Exercise (2)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="community-phylogenetics.html"><a href="community-phylogenetics.html"><i class="fa fa-check"></i><b>8</b> Community Phylogenetics</a>
<ul>
<li class="chapter" data-level="8.1" data-path="community-phylogenetics.html"><a href="community-phylogenetics.html#setting-up-the-environment-4"><i class="fa fa-check"></i><b>8.1</b> Setting up the environment</a></li>
<li class="chapter" data-level="8.2" data-path="community-phylogenetics.html"><a href="community-phylogenetics.html#definition-of-community-phylogenetics"><i class="fa fa-check"></i><b>8.2</b> Definition of Community Phylogenetics</a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="community-phylogenetics.html"><a href="community-phylogenetics.html#alpha-diversity"><i class="fa fa-check"></i><b>8.2.1</b> Alpha diversity</a></li>
<li class="chapter" data-level="8.2.2" data-path="community-phylogenetics.html"><a href="community-phylogenetics.html#beta-diversity"><i class="fa fa-check"></i><b>8.2.2</b> Beta diversity</a></li>
<li class="chapter" data-level="8.2.3" data-path="community-phylogenetics.html"><a href="community-phylogenetics.html#exercise-1-5"><i class="fa fa-check"></i><b>8.2.3</b> Exercise (1)</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="community-phylogenetics.html"><a href="community-phylogenetics.html#standardized-effect-sizes"><i class="fa fa-check"></i><b>8.3</b> Standardized effect sizes</a>
<ul>
<li class="chapter" data-level="8.3.1" data-path="community-phylogenetics.html"><a href="community-phylogenetics.html#exercise-2-5"><i class="fa fa-check"></i><b>8.3.1</b> Exercise (2)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="coevolution-and-phylogenetic-signals-in-r.html"><a href="coevolution-and-phylogenetic-signals-in-r.html"><i class="fa fa-check"></i><b>9</b> Coevolution and Phylogenetic Signals in R</a>
<ul>
<li class="chapter" data-level="9.1" data-path="coevolution-and-phylogenetic-signals-in-r.html"><a href="coevolution-and-phylogenetic-signals-in-r.html#setting-up-the-environment-5"><i class="fa fa-check"></i><b>9.1</b> Setting up the environment</a></li>
<li class="chapter" data-level="9.2" data-path="coevolution-and-phylogenetic-signals-in-r.html"><a href="coevolution-and-phylogenetic-signals-in-r.html#phylogenetic-signal"><i class="fa fa-check"></i><b>9.2</b> Phylogenetic Signal</a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="coevolution-and-phylogenetic-signals-in-r.html"><a href="coevolution-and-phylogenetic-signals-in-r.html#blombergs-k"><i class="fa fa-check"></i><b>9.2.1</b> Blomberg’s K</a></li>
<li class="chapter" data-level="9.2.2" data-path="coevolution-and-phylogenetic-signals-in-r.html"><a href="coevolution-and-phylogenetic-signals-in-r.html#pagels-lambda-λ"><i class="fa fa-check"></i><b>9.2.2</b> Pagel’s lambda (λ)</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="coevolution-and-phylogenetic-signals-in-r.html"><a href="coevolution-and-phylogenetic-signals-in-r.html#phylogenetically-independent-contrasts-1"><i class="fa fa-check"></i><b>9.3</b> Phylogenetically Independent Contrasts</a></li>
<li class="chapter" data-level="9.4" data-path="coevolution-and-phylogenetic-signals-in-r.html"><a href="coevolution-and-phylogenetic-signals-in-r.html#coevolution"><i class="fa fa-check"></i><b>9.4</b> Coevolution</a>
<ul>
<li class="chapter" data-level="9.4.1" data-path="coevolution-and-phylogenetic-signals-in-r.html"><a href="coevolution-and-phylogenetic-signals-in-r.html#plotting-tanglegrams"><i class="fa fa-check"></i><b>9.4.1</b> Plotting tanglegrams</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="phylogenetic-visualization-in-r.html"><a href="phylogenetic-visualization-in-r.html"><i class="fa fa-check"></i><b>10</b> Phylogenetic Visualization in R</a>
<ul>
<li class="chapter" data-level="10.1" data-path="phylogenetic-visualization-in-r.html"><a href="phylogenetic-visualization-in-r.html#setting-up-the-environment-6"><i class="fa fa-check"></i><b>10.1</b> Setting up the environment</a></li>
<li class="chapter" data-level="10.2" data-path="phylogenetic-visualization-in-r.html"><a href="phylogenetic-visualization-in-r.html#using-phytools"><i class="fa fa-check"></i><b>10.2</b> Using <code>phytools</code></a>
<ul>
<li class="chapter" data-level="10.2.1" data-path="phylogenetic-visualization-in-r.html"><a href="phylogenetic-visualization-in-r.html#phenograms"><i class="fa fa-check"></i><b>10.2.1</b> Phenograms</a></li>
<li class="chapter" data-level="10.2.2" data-path="phylogenetic-visualization-in-r.html"><a href="phylogenetic-visualization-in-r.html#mapping-continuous-characters-on-branches"><i class="fa fa-check"></i><b>10.2.2</b> Mapping continuous characters on branches</a></li>
<li class="chapter" data-level="10.2.3" data-path="phylogenetic-visualization-in-r.html"><a href="phylogenetic-visualization-in-r.html#plotting-images-on-phylogenies"><i class="fa fa-check"></i><b>10.2.3</b> Plotting images on phylogenies</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="phylogenetic-visualization-in-r.html"><a href="phylogenetic-visualization-in-r.html#using-ggtree"><i class="fa fa-check"></i><b>10.3</b> Using <code>ggtree</code></a>
<ul>
<li class="chapter" data-level="10.3.1" data-path="phylogenetic-visualization-in-r.html"><a href="phylogenetic-visualization-in-r.html#basic-plotting"><i class="fa fa-check"></i><b>10.3.1</b> Basic plotting</a></li>
<li class="chapter" data-level="10.3.2" data-path="phylogenetic-visualization-in-r.html"><a href="phylogenetic-visualization-in-r.html#labeling-tips-nodes-and-clades"><i class="fa fa-check"></i><b>10.3.2</b> Labeling tips, nodes, and clades</a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="phylogenetic-visualization-in-r.html"><a href="phylogenetic-visualization-in-r.html#plotting-maps"><i class="fa fa-check"></i><b>10.4</b> Plotting maps</a>
<ul>
<li class="chapter" data-level="10.4.1" data-path="phylogenetic-visualization-in-r.html"><a href="phylogenetic-visualization-in-r.html#choropleth-maps"><i class="fa fa-check"></i><b>10.4.1</b> Choropleth maps</a></li>
<li class="chapter" data-level="10.4.2" data-path="phylogenetic-visualization-in-r.html"><a href="phylogenetic-visualization-in-r.html#working-with-rasters"><i class="fa fa-check"></i><b>10.4.2</b> Working with rasters</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html"><i class="fa fa-check"></i><b>11</b> Gene tree vs Species tree</a>
<ul>
<li class="chapter" data-level="11.1" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#learning-goals"><i class="fa fa-check"></i><b>11.1</b> Learning goals:</a></li>
<li class="chapter" data-level="11.2" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#tracing-ancestry-within-one-population"><i class="fa fa-check"></i><b>11.2</b> Tracing ancestry within one population</a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#vary-pop-size-and-generation"><i class="fa fa-check"></i><b>11.2.1</b> Vary pop size and generation</a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#speciation-and-coalescence"><i class="fa fa-check"></i><b>11.3</b> Speciation and coalescence</a>
<ul>
<li class="chapter" data-level="11.3.1" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#two-species-coalescent-history"><i class="fa fa-check"></i><b>11.3.1</b> Two-species coalescent history</a></li>
</ul></li>
<li class="chapter" data-level="11.4" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#exercise-i"><i class="fa fa-check"></i><b>11.4</b> Exercise I</a></li>
<li class="chapter" data-level="11.5" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#gene-tree-locus-tree-and-species-tree"><i class="fa fa-check"></i><b>11.5</b> Gene tree, Locus tree and Species tree</a>
<ul>
<li class="chapter" data-level="11.5.1" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#loading-data-and-packages"><i class="fa fa-check"></i><b>11.5.1</b> loading data and packages</a></li>
</ul></li>
<li class="chapter" data-level="11.6" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#effects-of-incomplete-lineage-sorting-ils-alone"><i class="fa fa-check"></i><b>11.6</b> Effects of Incomplete Lineage Sorting (ILS) alone</a>
<ul>
<li class="chapter" data-level="11.6.1" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#generate-species-tree-with-no-extinction"><i class="fa fa-check"></i><b>11.6.1</b> generate species tree with no extinction</a></li>
<li class="chapter" data-level="11.6.2" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#simulate-gene-trees-using-the-multispecies-coalescent-process"><i class="fa fa-check"></i><b>11.6.2</b> simulate gene trees using the multispecies coalescent process</a></li>
</ul></li>
<li class="chapter" data-level="11.7" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#calculate-tree-distances"><i class="fa fa-check"></i><b>11.7</b> Calculate tree distances</a>
<ul>
<li class="chapter" data-level="11.7.1" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#rf-distances"><i class="fa fa-check"></i><b>11.7.1</b> RF distances</a></li>
<li class="chapter" data-level="11.7.2" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#inspect-pairwise-rf-distances-among-all-the-gene-trees-generated-under-one-guiding-species-tree"><i class="fa fa-check"></i><b>11.7.2</b> Inspect pairwise RF distances among all the gene trees generated under one guiding species tree</a></li>
</ul></li>
<li class="chapter" data-level="11.8" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#inspect-node-support-i.e.-subtrees-defined-by-a-split"><i class="fa fa-check"></i><b>11.8</b> Inspect node support (i.e., subtrees defined by a split)</a></li>
<li class="chapter" data-level="11.9" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#combined-effects-of-gene-duplication-loss-and-horizontal-gene-transfer-events"><i class="fa fa-check"></i><b>11.9</b> Combined effects of gene duplication, loss and horizontal gene transfer events</a></li>
<li class="chapter" data-level="11.10" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#exercise-ii"><i class="fa fa-check"></i><b>11.10</b> Exercise II</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="overview-of-phylogenetic-inferences.html"><a href="overview-of-phylogenetic-inferences.html"><i class="fa fa-check"></i><b>12</b> Overview of Phylogenetic Inferences</a>
<ul>
<li class="chapter" data-level="12.1" data-path="overview-of-phylogenetic-inferences.html"><a href="overview-of-phylogenetic-inferences.html#mega"><i class="fa fa-check"></i><b>12.1</b> <code>MEGA</code></a></li>
<li class="chapter" data-level="12.2" data-path="overview-of-phylogenetic-inferences.html"><a href="overview-of-phylogenetic-inferences.html#view-and-align-sequences"><i class="fa fa-check"></i><b>12.2</b> View and Align Sequences</a></li>
<li class="chapter" data-level="12.3" data-path="overview-of-phylogenetic-inferences.html"><a href="overview-of-phylogenetic-inferences.html#dnaprotein-evolution-model"><i class="fa fa-check"></i><b>12.3</b> DNA/protein evolution model</a></li>
<li class="chapter" data-level="12.4" data-path="overview-of-phylogenetic-inferences.html"><a href="overview-of-phylogenetic-inferences.html#phylogenetic-inference"><i class="fa fa-check"></i><b>12.4</b> Phylogenetic inference</a></li>
<li class="chapter" data-level="12.5" data-path="overview-of-phylogenetic-inferences.html"><a href="overview-of-phylogenetic-inferences.html#estimate-divergence-times"><i class="fa fa-check"></i><b>12.5</b> Estimate divergence times</a></li>
<li class="chapter" data-level="12.6" data-path="overview-of-phylogenetic-inferences.html"><a href="overview-of-phylogenetic-inferences.html#exercises"><i class="fa fa-check"></i><b>12.6</b> Exercises</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html"><i class="fa fa-check"></i><b>13</b> Resources and Guidelines for the Term Project</a>
<ul>
<li class="chapter" data-level="13.1" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#bioinformatics-databases"><i class="fa fa-check"></i><b>13.1</b> Bioinformatics Databases</a>
<ul>
<li class="chapter" data-level="13.1.1" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#ncbi-national-center-for-biotechnology-information"><i class="fa fa-check"></i><b>13.1.1</b> NCBI (National Center for Biotechnology Information)</a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#retrieving-genetic-resources-for-your-taxa"><i class="fa fa-check"></i><b>13.2</b> Retrieving genetic resources for your taxa</a>
<ul>
<li class="chapter" data-level="13.2.1" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#ncbi-taxonomy-database"><i class="fa fa-check"></i><b>13.2.1</b> NCBI taxonomy database</a></li>
<li class="chapter" data-level="13.2.2" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#individual-genes-from-the-whole-genome-data"><i class="fa fa-check"></i><b>13.2.2</b> Individual genes from the whole genome data</a></li>
<li class="chapter" data-level="13.2.3" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#get-homologous-sequences-from-related-species"><i class="fa fa-check"></i><b>13.2.3</b> Get homologous sequences from related species</a></li>
<li class="chapter" data-level="13.2.4" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#european-bioinformatics-institute-ebi"><i class="fa fa-check"></i><b>13.2.4</b> European Bioinformatics Institute (EBI)</a></li>
<li class="chapter" data-level="13.2.5" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#protein-data-bank"><i class="fa fa-check"></i><b>13.2.5</b> Protein Data Bank</a></li>
</ul></li>
<li class="chapter" data-level="13.3" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#biogeographic-databases"><i class="fa fa-check"></i><b>13.3</b> Biogeographic Databases</a>
<ul>
<li class="chapter" data-level="13.3.1" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#climate-data"><i class="fa fa-check"></i><b>13.3.1</b> Climate Data</a></li>
<li class="chapter" data-level="13.3.2" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#occurrence-data"><i class="fa fa-check"></i><b>13.3.2</b> Occurrence Data</a></li>
<li class="chapter" data-level="13.3.3" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#topographic-data"><i class="fa fa-check"></i><b>13.3.3</b> TOPOGRAPHIC DATA</a></li>
<li class="chapter" data-level="13.3.4" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#trait-data"><i class="fa fa-check"></i><b>13.3.4</b> TRAIT DATA</a></li>
</ul></li>
<li class="chapter" data-level="13.4" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#publication-databases"><i class="fa fa-check"></i><b>13.4</b> Publication Databases</a></li>
<li class="chapter" data-level="13.5" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#example-research-questions"><i class="fa fa-check"></i><b>13.5</b> Example research questions:</a>
<ul>
<li class="chapter" data-level="13.5.1" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#biogeography-and-phylogenetic-diversity"><i class="fa fa-check"></i><b>13.5.1</b> Biogeography and Phylogenetic Diversity:</a></li>
<li class="chapter" data-level="13.5.2" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#community-ecology"><i class="fa fa-check"></i><b>13.5.2</b> Community Ecology:</a></li>
<li class="chapter" data-level="13.5.3" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#coevolution-1"><i class="fa fa-check"></i><b>13.5.3</b> Coevolution:</a></li>
<li class="chapter" data-level="13.5.4" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#species-tree"><i class="fa fa-check"></i><b>13.5.4</b> Species tree:</a></li>
</ul></li>
<li class="chapter" data-level="13.6" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#how-to-submit-jobs-on-scholar-cluster-and-run-remotely"><i class="fa fa-check"></i><b>13.6</b> How to submit jobs on Scholar Cluster and run remotely</a></li>
<li class="chapter" data-level="13.7" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#term-project-proposal"><i class="fa fa-check"></i><b>13.7</b> Term project proposal</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><i class="fa fa-check"></i><b>14</b> Systematic Retrieval of orthologous DNA sequences form GenBank</a>
<ul>
<li class="chapter" data-level="14.1" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#introduction"><i class="fa fa-check"></i><b>14.1</b> Introduction</a></li>
<li class="chapter" data-level="14.2" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#installing-tools"><i class="fa fa-check"></i><b>14.2</b> Installing Tools</a>
<ul>
<li class="chapter" data-level="14.2.1" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#windows-1"><i class="fa fa-check"></i><b>14.2.1</b> Windows</a></li>
<li class="chapter" data-level="14.2.2" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#mac"><i class="fa fa-check"></i><b>14.2.2</b> Mac</a></li>
<li class="chapter" data-level="14.2.3" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#ncbi-blast-tools"><i class="fa fa-check"></i><b>14.2.3</b> NCBI BLAST+ Tools</a></li>
</ul></li>
<li class="chapter" data-level="14.3" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#using-phylotar"><i class="fa fa-check"></i><b>14.3</b> Using <code>phylotaR</code></a>
<ul>
<li class="chapter" data-level="14.3.1" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#pipeline"><i class="fa fa-check"></i><b>14.3.1</b> Pipeline</a></li>
<li class="chapter" data-level="14.3.2" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#running"><i class="fa fa-check"></i><b>14.3.2</b> Running</a></li>
<li class="chapter" data-level="14.3.3" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#restarting"><i class="fa fa-check"></i><b>14.3.3</b> Restarting</a></li>
<li class="chapter" data-level="14.3.4" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#cluster-i.e.-orthologous-genes-selection"><i class="fa fa-check"></i><b>14.3.4</b> Cluster (i.e., orthologous genes) selection</a></li>
</ul></li>
<li class="chapter" data-level="14.4" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#exercise-i-1"><i class="fa fa-check"></i><b>14.4</b> Exercise I</a></li>
<li class="chapter" data-level="14.5" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#using-annotationbustr"><i class="fa fa-check"></i><b>14.5</b> Using <code>AnnotationBustR</code></a>
<ul>
<li class="chapter" data-level="14.5.1" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#introduction-1"><i class="fa fa-check"></i><b>14.5.1</b> Introduction</a></li>
<li class="chapter" data-level="14.5.2" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#install-the-tools"><i class="fa fa-check"></i><b>14.5.2</b> Install the tools</a></li>
<li class="chapter" data-level="14.5.3" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#specify-search-terms-of-genomes"><i class="fa fa-check"></i><b>14.5.3</b> Specify search terms of genomes</a></li>
<li class="chapter" data-level="14.5.4" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#load-gene-search-terms-of-interest"><i class="fa fa-check"></i><b>14.5.4</b> Load Gene Search Terms of Interest</a></li>
<li class="chapter" data-level="14.5.5" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#extract-sequences-with-annotationbust"><i class="fa fa-check"></i><b>14.5.5</b> Extract sequences with AnnotationBust</a></li>
</ul></li>
<li class="chapter" data-level="14.6" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#exercise-ii-1"><i class="fa fa-check"></i><b>14.6</b> Exercise II</a></li>
<li class="chapter" data-level="14.7" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#combine-gene-information-from-the-two-analyses"><i class="fa fa-check"></i><b>14.7</b> Combine gene information from the two analyses</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="sequence-alignment.html"><a href="sequence-alignment.html"><i class="fa fa-check"></i><b>15</b> Sequence alignment</a>
<ul>
<li class="chapter" data-level="15.1" data-path="sequence-alignment.html"><a href="sequence-alignment.html#load-libraries"><i class="fa fa-check"></i><b>15.1</b> Load libraries</a></li>
<li class="chapter" data-level="15.2" data-path="sequence-alignment.html"><a href="sequence-alignment.html#perform-alignments"><i class="fa fa-check"></i><b>15.2</b> Perform Alignments</a></li>
<li class="chapter" data-level="15.3" data-path="sequence-alignment.html"><a href="sequence-alignment.html#processing-multiple-alignments"><i class="fa fa-check"></i><b>15.3</b> Processing Multiple Alignments</a></li>
<li class="chapter" data-level="15.4" data-path="sequence-alignment.html"><a href="sequence-alignment.html#build-your-first-nj-tree"><i class="fa fa-check"></i><b>15.4</b> Build your first NJ tree</a></li>
<li class="chapter" data-level="15.5" data-path="sequence-alignment.html"><a href="sequence-alignment.html#exercise"><i class="fa fa-check"></i><b>15.5</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="maximum-likelihood-model-tests-tree-and-branch-tests.html"><a href="maximum-likelihood-model-tests-tree-and-branch-tests.html"><i class="fa fa-check"></i><b>16</b> Maximum likelihood, model tests, tree and branch tests</a>
<ul>
<li class="chapter" data-level="16.1" data-path="maximum-likelihood-model-tests-tree-and-branch-tests.html"><a href="maximum-likelihood-model-tests-tree-and-branch-tests.html#learning-goals-1"><i class="fa fa-check"></i><b>16.1</b> Learning Goals</a></li>
<li class="chapter" data-level="16.2" data-path="maximum-likelihood-model-tests-tree-and-branch-tests.html"><a href="maximum-likelihood-model-tests-tree-and-branch-tests.html#iqtree"><i class="fa fa-check"></i><b>16.2</b> IQtree</a>
<ul>
<li class="chapter" data-level="16.2.1" data-path="maximum-likelihood-model-tests-tree-and-branch-tests.html"><a href="maximum-likelihood-model-tests-tree-and-branch-tests.html#version"><i class="fa fa-check"></i><b>16.2.1</b> Version</a></li>
<li class="chapter" data-level="16.2.2" data-path="maximum-likelihood-model-tests-tree-and-branch-tests.html"><a href="maximum-likelihood-model-tests-tree-and-branch-tests.html#find-the-best-explanatory-sequence-evolution-model"><i class="fa fa-check"></i><b>16.2.2</b> find the best explanatory sequence evolution model</a></li>
<li class="chapter" data-level="16.2.3" data-path="maximum-likelihood-model-tests-tree-and-branch-tests.html"><a href="maximum-likelihood-model-tests-tree-and-branch-tests.html#using-the-best-sequence-evolution-model"><i class="fa fa-check"></i><b>16.2.3</b> using the best sequence evolution model</a></li>
<li class="chapter" data-level="16.2.4" data-path="maximum-likelihood-model-tests-tree-and-branch-tests.html"><a href="maximum-likelihood-model-tests-tree-and-branch-tests.html#different-partitions-for-the-data-for-multiple-genes-or-different-codons"><i class="fa fa-check"></i><b>16.2.4</b> Different partitions for the data for multiple genes, or different codons</a></li>
</ul></li>
<li class="chapter" data-level="16.3" data-path="maximum-likelihood-model-tests-tree-and-branch-tests.html"><a href="maximum-likelihood-model-tests-tree-and-branch-tests.html#run-an-ml-tree-with-multiple-gene-files-in-a-folder"><i class="fa fa-check"></i><b>16.3</b> Run an ML tree with multiple gene files in a folder</a></li>
<li class="chapter" data-level="16.4" data-path="maximum-likelihood-model-tests-tree-and-branch-tests.html"><a href="maximum-likelihood-model-tests-tree-and-branch-tests.html#gene-concordance-factor"><i class="fa fa-check"></i><b>16.4</b> Gene concordance factor</a></li>
<li class="chapter" data-level="16.5" data-path="maximum-likelihood-model-tests-tree-and-branch-tests.html"><a href="maximum-likelihood-model-tests-tree-and-branch-tests.html#tree-topology-tests"><i class="fa fa-check"></i><b>16.5</b> Tree topology tests</a></li>
<li class="chapter" data-level="16.6" data-path="maximum-likelihood-model-tests-tree-and-branch-tests.html"><a href="maximum-likelihood-model-tests-tree-and-branch-tests.html#tree-branches-tests"><i class="fa fa-check"></i><b>16.6</b> Tree branches tests</a></li>
<li class="chapter" data-level="16.7" data-path="maximum-likelihood-model-tests-tree-and-branch-tests.html"><a href="maximum-likelihood-model-tests-tree-and-branch-tests.html#assignments"><i class="fa fa-check"></i><b>16.7</b> Assignments</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="molecular-clock-and-phylogenetic-dating.html"><a href="molecular-clock-and-phylogenetic-dating.html"><i class="fa fa-check"></i><b>17</b> Molecular Clock and Phylogenetic Dating</a>
<ul>
<li class="chapter" data-level="17.1" data-path="molecular-clock-and-phylogenetic-dating.html"><a href="molecular-clock-and-phylogenetic-dating.html#inferring-time-tree-with-tip-dates"><i class="fa fa-check"></i><b>17.1</b> Inferring time tree with tip dates</a></li>
<li class="chapter" data-level="17.2" data-path="molecular-clock-and-phylogenetic-dating.html"><a href="molecular-clock-and-phylogenetic-dating.html#calibrating-tree-using-ancestral-dates"><i class="fa fa-check"></i><b>17.2</b> Calibrating tree using ancestral dates</a>
<ul>
<li class="chapter" data-level="17.2.1" data-path="molecular-clock-and-phylogenetic-dating.html"><a href="molecular-clock-and-phylogenetic-dating.html#search-for-an-ml-tree"><i class="fa fa-check"></i><b>17.2.1</b> search for an ML tree</a></li>
<li class="chapter" data-level="17.2.2" data-path="molecular-clock-and-phylogenetic-dating.html"><a href="molecular-clock-and-phylogenetic-dating.html#root-calibration"><i class="fa fa-check"></i><b>17.2.2</b> Root calibration</a></li>
<li class="chapter" data-level="17.2.3" data-path="molecular-clock-and-phylogenetic-dating.html"><a href="molecular-clock-and-phylogenetic-dating.html#internal-nodes"><i class="fa fa-check"></i><b>17.2.3</b> Internal nodes</a></li>
<li class="chapter" data-level="17.2.4" data-path="molecular-clock-and-phylogenetic-dating.html"><a href="molecular-clock-and-phylogenetic-dating.html#confidence-interval"><i class="fa fa-check"></i><b>17.2.4</b> Confidence interval</a></li>
</ul></li>
<li class="chapter" data-level="17.3" data-path="molecular-clock-and-phylogenetic-dating.html"><a href="molecular-clock-and-phylogenetic-dating.html#exercise-4"><i class="fa fa-check"></i><b>17.3</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="character-evolution-and-ancestral-state-reconstruction.html"><a href="character-evolution-and-ancestral-state-reconstruction.html"><i class="fa fa-check"></i><b>18</b> Character evolution and Ancestral state reconstruction</a>
<ul>
<li class="chapter" data-level="18.1" data-path="character-evolution-and-ancestral-state-reconstruction.html"><a href="character-evolution-and-ancestral-state-reconstruction.html#continuous-characters"><i class="fa fa-check"></i><b>18.1</b> Continuous characters</a>
<ul>
<li class="chapter" data-level="18.1.1" data-path="character-evolution-and-ancestral-state-reconstruction.html"><a href="character-evolution-and-ancestral-state-reconstruction.html#load-libraries-1"><i class="fa fa-check"></i><b>18.1.1</b> load libraries</a></li>
<li class="chapter" data-level="18.1.2" data-path="character-evolution-and-ancestral-state-reconstruction.html"><a href="character-evolution-and-ancestral-state-reconstruction.html#tree-file-and-the-character-file"><i class="fa fa-check"></i><b>18.1.2</b> Tree file and the character file</a></li>
<li class="chapter" data-level="18.1.3" data-path="character-evolution-and-ancestral-state-reconstruction.html"><a href="character-evolution-and-ancestral-state-reconstruction.html#visualization-of-the-reconstructed-nodes"><i class="fa fa-check"></i><b>18.1.3</b> Visualization of the reconstructed nodes</a></li>
<li class="chapter" data-level="18.1.4" data-path="character-evolution-and-ancestral-state-reconstruction.html"><a href="character-evolution-and-ancestral-state-reconstruction.html#visualization-of-phenotypic-evolution"><i class="fa fa-check"></i><b>18.1.4</b> Visualization of phenotypic evolution</a></li>
</ul></li>
<li class="chapter" data-level="18.2" data-path="character-evolution-and-ancestral-state-reconstruction.html"><a href="character-evolution-and-ancestral-state-reconstruction.html#brownian-motion-vs-ornstein-uhlenbeck-process"><i class="fa fa-check"></i><b>18.2</b> Brownian Motion vs Ornstein-Uhlenbeck Process</a></li>
<li class="chapter" data-level="18.3" data-path="character-evolution-and-ancestral-state-reconstruction.html"><a href="character-evolution-and-ancestral-state-reconstruction.html#brownian-motion"><i class="fa fa-check"></i><b>18.3</b> Brownian Motion</a>
<ul>
<li class="chapter" data-level="18.3.1" data-path="character-evolution-and-ancestral-state-reconstruction.html"><a href="character-evolution-and-ancestral-state-reconstruction.html#ornsteinuhlenbeck-process"><i class="fa fa-check"></i><b>18.3.1</b> Ornstein–Uhlenbeck process</a></li>
<li class="chapter" data-level="18.3.2" data-path="character-evolution-and-ancestral-state-reconstruction.html"><a href="character-evolution-and-ancestral-state-reconstruction.html#incorporate-prior-knowledge-using-bayesian-estimates"><i class="fa fa-check"></i><b>18.3.2</b> Incorporate prior knowledge using Bayesian estimates</a></li>
</ul></li>
<li class="chapter" data-level="18.4" data-path="character-evolution-and-ancestral-state-reconstruction.html"><a href="character-evolution-and-ancestral-state-reconstruction.html#discrete-characters"><i class="fa fa-check"></i><b>18.4</b> Discrete characters</a></li>
<li class="chapter" data-level="18.5" data-path="character-evolution-and-ancestral-state-reconstruction.html"><a href="character-evolution-and-ancestral-state-reconstruction.html#exercises-1"><i class="fa fa-check"></i><b>18.5</b> Exercises:</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="model-selection-of-character-evolution.html"><a href="model-selection-of-character-evolution.html"><i class="fa fa-check"></i><b>19</b> Model Selection of Character Evolution</a>
<ul>
<li class="chapter" data-level="19.1" data-path="model-selection-of-character-evolution.html"><a href="model-selection-of-character-evolution.html#load-libraries-2"><i class="fa fa-check"></i><b>19.1</b> Load libraries</a></li>
<li class="chapter" data-level="19.2" data-path="model-selection-of-character-evolution.html"><a href="model-selection-of-character-evolution.html#discrete-traits-model-fitting"><i class="fa fa-check"></i><b>19.2</b> Discrete traits model fitting</a>
<ul>
<li class="chapter" data-level="19.2.1" data-path="model-selection-of-character-evolution.html"><a href="model-selection-of-character-evolution.html#choosing-a-character-model"><i class="fa fa-check"></i><b>19.2.1</b> Choosing a character model</a></li>
<li class="chapter" data-level="19.2.2" data-path="model-selection-of-character-evolution.html"><a href="model-selection-of-character-evolution.html#marginal-ancestral-state-reconstruction"><i class="fa fa-check"></i><b>19.2.2</b> Marginal ancestral state reconstruction</a></li>
<li class="chapter" data-level="19.2.3" data-path="model-selection-of-character-evolution.html"><a href="model-selection-of-character-evolution.html#more-than-two-states"><i class="fa fa-check"></i><b>19.2.3</b> More than two states</a></li>
</ul></li>
<li class="chapter" data-level="19.3" data-path="model-selection-of-character-evolution.html"><a href="model-selection-of-character-evolution.html#exercise-i-2"><i class="fa fa-check"></i><b>19.3</b> Exercise I</a></li>
<li class="chapter" data-level="19.4" data-path="model-selection-of-character-evolution.html"><a href="model-selection-of-character-evolution.html#continuous-traits-model-fitting"><i class="fa fa-check"></i><b>19.4</b> Continuous traits model fitting</a>
<ul>
<li class="chapter" data-level="19.4.1" data-path="model-selection-of-character-evolution.html"><a href="model-selection-of-character-evolution.html#model-selection"><i class="fa fa-check"></i><b>19.4.1</b> Model selection</a></li>
<li class="chapter" data-level="19.4.2" data-path="model-selection-of-character-evolution.html"><a href="model-selection-of-character-evolution.html#ancestral-states"><i class="fa fa-check"></i><b>19.4.2</b> Ancestral states</a></li>
</ul></li>
<li class="chapter" data-level="19.5" data-path="model-selection-of-character-evolution.html"><a href="model-selection-of-character-evolution.html#exercise-ii-2"><i class="fa fa-check"></i><b>19.5</b> Exercise II</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">BIO595 The Tree of Life</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="r-basics-ii" class="section level1 hasAnchor" number="4">
<h1><span class="header-section-number">Chapter 4</span> R Basics II<a href="r-basics-ii.html#r-basics-ii" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="setting-up-the-environment" class="section level2 hasAnchor" number="4.1">
<h2><span class="header-section-number">4.1</span> Setting up the environment<a href="r-basics-ii.html#setting-up-the-environment" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>There are a few things you want to do before you start working with data in R.</p>
<ul>
<li><p>Install necessary packages</p></li>
<li><p>Load necessary packages</p></li>
<li><p>Set working directory</p></li>
</ul>
<p>Packages are the fundamental units of reproducible <code>R</code> code.
They include reusable <code>R</code> functions, the documentation that describes how to use them, and sample data.
You can install packages using <code>RStudio</code>’s interface - go to the <code>Tools</code> tab and select <code>install packages</code>.
This is the equivalent of running the <code>install.packages()</code> function in <code>R</code>.</p>
<p>For instance:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="r-basics-ii.html#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;ape&quot;</span>)</span>
<span id="cb1-2"><a href="r-basics-ii.html#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;phytools&quot;</span>)</span></code></pre></div>
<blockquote>
<p><code>R</code> may ask you to choose a <a href="https://cran.r-project.org/">CRAN</a> mirror.
<a href="https://cran.r-project.org/">CRAN</a> is the distributed network of servers that provides access to <code>R</code>’s software.
It doesn’t really matter which one you chose, but geographically closer ones may be slightly faster.</p>
</blockquote>
<p>Installed packages still need to be loaded for you to be able to access the functions and data within them.
You can load the packages that you need using the <code>library()</code> or <code>require()</code> functions.
The two functions are slightly different:</p>
<ul>
<li><p><code>require()</code>will output a warning if a package is not installed and then continue to execute the code.</p></li>
<li><p><code>library()</code> will output an error and stop the execution of the code.</p></li>
</ul>
<p>However, in this course they can be used interchangeably.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="r-basics-ii.html#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;ape&quot;</span>)</span>
<span id="cb2-2"><a href="r-basics-ii.html#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(<span class="st">&quot;phytools&quot;</span>)</span></code></pre></div>
<p></br></p>
<p>It is often useful to set up a <code>"working directory"</code> for your tasks.
A working directory is just a file path on your computer that sets the default location of any files you read into <code>R</code>, or save from <code>R</code> during your session.
Remember the difference between absolute paths and relative paths?
Setting a working directory and working within in it makes using relative paths easier so you don’t have to type (and make typos) as much.
The function <code>setwd()</code> can be used to set your working directory and <code>getwd()</code> can be used to see what directory you’re currently working out of.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="r-basics-ii.html#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">getwd</span>() <span class="co">#check where you are</span></span>
<span id="cb3-2"><a href="r-basics-ii.html#cb3-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3"><a href="r-basics-ii.html#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="co"># set working directory</span></span>
<span id="cb3-4"><a href="r-basics-ii.html#cb3-4" aria-hidden="true" tabindex="-1"></a>wd <span class="ot">&lt;-</span> <span class="st">&quot;~/Dropbox/teaching/bookdown-demo-main/data2023/&quot;</span> <span class="co"># absolute address of the folder where you have the data you need and plan to work in.</span></span>
<span id="cb3-5"><a href="r-basics-ii.html#cb3-5" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(wd)</span>
<span id="cb3-6"><a href="r-basics-ii.html#cb3-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-7"><a href="r-basics-ii.html#cb3-7" aria-hidden="true" tabindex="-1"></a><span class="fu">getwd</span>() <span class="co">#check where you are now</span></span></code></pre></div>
<p></br></p>
<div id="exercise-1-1" class="section level3 hasAnchor" number="4.1.1">
<h3><span class="header-section-number">4.1.1</span> Exercise (1)<a href="r-basics-ii.html#exercise-1-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Download data for lab 5 into a folder named <code>"lab5"</code> and set your working directory to that folder.
To easily get the absolute path of a file or folder:</p>
<ul>
<li><p><strong>MacOS</strong>: click on file/folder with two fingers (or right-click) to bring up the menu and press the <code>option</code> key - you should see an option that says <code>"Copy as Pathname"</code>.</p></li>
<li><p><strong>Windows</strong>: press <code>shift</code> and right click on the file/folder - you should see an option that says “Copy as path”.
Once you paste the path into R you will have to replace all the back slashes with front slashes.</p></li>
</ul>
<p>Write your code below:</p>
<hr />
<hr />
<hr />
<hr />
<p></br></p>
</div>
</div>
<div id="reading-in-data-from-files" class="section level2 hasAnchor" number="4.2">
<h2><span class="header-section-number">4.2</span> Reading in data from files<a href="r-basics-ii.html#reading-in-data-from-files" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In general, you will use specific <code>R</code> functions designed to recognize, parse, and load different types of data.
Data are commonly stored as text files, which are most often comma (<code>".csv"</code>) or tab delimited (<code>".txt"</code>/<code>".tsv"</code>).
Even phylogenetic data are almost always stored as text files despite having various extensions (e.g., <code>".nex"</code>, <code>".fasta"</code>,<code>".tre"</code>, <code>".phy"</code>).</p>
<p>Let’s read in the data you downloaded and placed in your working directory and examine the data.
The <code>read.delim</code> function requires you to specify what delimits your data - i.e., “what separates the columns”.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="r-basics-ii.html#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># reading in tab-delimited text files</span></span>
<span id="cb4-2"><a href="r-basics-ii.html#cb4-2" aria-hidden="true" tabindex="-1"></a>oc1 <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(<span class="st">&quot;occurrence.txt&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span>
<span id="cb4-3"><a href="r-basics-ii.html#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(oc1)</span>
<span id="cb4-4"><a href="r-basics-ii.html#cb4-4" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(oc1)</span>
<span id="cb4-5"><a href="r-basics-ii.html#cb4-5" aria-hidden="true" tabindex="-1"></a><span class="fu">View</span>(oc1)</span>
<span id="cb4-6"><a href="r-basics-ii.html#cb4-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-7"><a href="r-basics-ii.html#cb4-7" aria-hidden="true" tabindex="-1"></a><span class="co"># reading in comma separated files</span></span>
<span id="cb4-8"><a href="r-basics-ii.html#cb4-8" aria-hidden="true" tabindex="-1"></a>mm1 <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(<span class="st">&quot;multimedia.csv&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;,&quot;</span>)</span>
<span id="cb4-9"><a href="r-basics-ii.html#cb4-9" aria-hidden="true" tabindex="-1"></a><span class="co"># or:</span></span>
<span id="cb4-10"><a href="r-basics-ii.html#cb4-10" aria-hidden="true" tabindex="-1"></a>mm1 <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;multimedia.csv&quot;</span>)</span>
<span id="cb4-11"><a href="r-basics-ii.html#cb4-11" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(mm1)</span>
<span id="cb4-12"><a href="r-basics-ii.html#cb4-12" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(oc1)</span>
<span id="cb4-13"><a href="r-basics-ii.html#cb4-13" aria-hidden="true" tabindex="-1"></a><span class="fu">View</span>(mm1)</span></code></pre></div>
<p></br></p>
</div>
<div id="wrangling-data" class="section level2 hasAnchor" number="4.3">
<h2><span class="header-section-number">4.3</span> Wrangling Data<a href="r-basics-ii.html#wrangling-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Data are often not in the format we want them to be in.
They can also comprise information that we do not need and/or information that” is incorrect. Thus, it is often necessary to wrangle your data into a more useful state. Along these lines, removing, subsetting, and filtering data generally involve the same processes. Here we will work with data frames, one of the most common type of data storage objects you will encounter.</p>
<p></br></p>
</div>
<div id="subsetting-by-columns-and-rows" class="section level2 hasAnchor" number="4.4">
<h2><span class="header-section-number">4.4</span> Subsetting by columns and rows<a href="r-basics-ii.html#subsetting-by-columns-and-rows" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Data frames are two-dimensional arrays of data, and hence you can reference cells using columns and rows.
Think x, y coordinates.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="r-basics-ii.html#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co"># subsetting data with column numbers</span></span>
<span id="cb5-2"><a href="r-basics-ii.html#cb5-2" aria-hidden="true" tabindex="-1"></a>oc2 <span class="ot">&lt;-</span> oc1[, <span class="dv">100</span><span class="sc">:</span><span class="dv">115</span>]</span>
<span id="cb5-3"><a href="r-basics-ii.html#cb5-3" aria-hidden="true" tabindex="-1"></a>oc3 <span class="ot">&lt;-</span> oc1[, <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">23</span>, <span class="dv">110</span>, <span class="dv">240</span>, <span class="dv">241</span>)]</span>
<span id="cb5-4"><a href="r-basics-ii.html#cb5-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-5"><a href="r-basics-ii.html#cb5-5" aria-hidden="true" tabindex="-1"></a><span class="co"># subsetting data with row numbers</span></span>
<span id="cb5-6"><a href="r-basics-ii.html#cb5-6" aria-hidden="true" tabindex="-1"></a>oc4 <span class="ot">&lt;-</span> oc1[<span class="dv">10</span><span class="sc">:</span><span class="dv">80</span>, ]</span>
<span id="cb5-7"><a href="r-basics-ii.html#cb5-7" aria-hidden="true" tabindex="-1"></a>oc5 <span class="ot">&lt;-</span> oc1[<span class="fu">c</span>(<span class="dv">56</span>, <span class="dv">57</span>, <span class="dv">68</span>, <span class="dv">100</span>), ]</span>
<span id="cb5-8"><a href="r-basics-ii.html#cb5-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-9"><a href="r-basics-ii.html#cb5-9" aria-hidden="true" tabindex="-1"></a><span class="co"># subsetting data with coulmns numbers and row numbers</span></span>
<span id="cb5-10"><a href="r-basics-ii.html#cb5-10" aria-hidden="true" tabindex="-1"></a>oc6 <span class="ot">&lt;-</span> oc1[<span class="fu">c</span>(<span class="dv">56</span>, <span class="dv">57</span>, <span class="dv">68</span>, <span class="dv">100</span>), <span class="dv">200</span><span class="sc">:</span><span class="dv">215</span>]</span></code></pre></div>
<p>Instead of just using numbers, you can refer to columns names.
This is often the more convenient option.
You can also refer to row names, but R can be a bit erratic with row names so it is best to stick with numbers or different conditions (see section below) in the case of rows.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="r-basics-ii.html#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co"># subsetting data with column names</span></span>
<span id="cb6-2"><a href="r-basics-ii.html#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(oc1) <span class="co"># list column names</span></span>
<span id="cb6-3"><a href="r-basics-ii.html#cb6-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-4"><a href="r-basics-ii.html#cb6-4" aria-hidden="true" tabindex="-1"></a><span class="co"># directly list columns of interest</span></span>
<span id="cb6-5"><a href="r-basics-ii.html#cb6-5" aria-hidden="true" tabindex="-1"></a>oc7 <span class="ot">&lt;-</span> oc1[, <span class="fu">c</span>(<span class="st">&quot;gbifID&quot;</span>,<span class="st">&quot;stateProvince&quot;</span>, <span class="st">&quot;county&quot;</span>, <span class="st">&quot;decimalLongitude&quot;</span>, <span class="st">&quot;decimalLatitude&quot;</span>)]</span>
<span id="cb6-6"><a href="r-basics-ii.html#cb6-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-7"><a href="r-basics-ii.html#cb6-7" aria-hidden="true" tabindex="-1"></a><span class="co"># or designate columns of interest first</span></span>
<span id="cb6-8"><a href="r-basics-ii.html#cb6-8" aria-hidden="true" tabindex="-1"></a>cols <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;gbifID&quot;</span>,<span class="st">&quot;continent&quot;</span>, <span class="st">&quot;countryCode&quot;</span>, <span class="st">&quot;stateProvince&quot;</span>, <span class="st">&quot;county&quot;</span>, <span class="st">&quot;municipality&quot;</span>, <span class="st">&quot;locality&quot;</span>, <span class="st">&quot;decimalLongitude&quot;</span>, <span class="st">&quot;decimalLatitude&quot;</span>, <span class="st">&quot;coordinateUncertaintyInMeters&quot;</span>, <span class="st">&quot;georeferencedBy&quot;</span>, <span class="st">&quot;georeferencedDate&quot;</span>, <span class="st">&quot;georeferenceProtocol&quot;</span>, <span class="st">&quot;georeferenceSources&quot;</span>, <span class="st">&quot;georeferenceVerificationStatus&quot;</span>, <span class="st">&quot;georeferenceRemarks&quot;</span>, <span class="st">&quot;year&quot;</span>, <span class="st">&quot;month&quot;</span>, <span class="st">&quot;day&quot;</span>, <span class="st">&quot;scientificName&quot;</span>, <span class="st">&quot;family&quot;</span>, <span class="st">&quot;genus&quot;</span>, <span class="st">&quot;specificEpithet&quot;</span>, <span class="st">&quot;infraspecificEpithet&quot;</span>, <span class="st">&quot;verbatimScientificName&quot;</span>, <span class="st">&quot;level0Gid&quot;</span>, <span class="st">&quot;level0Name&quot;</span>, <span class="st">&quot;level1Gid&quot;</span>, <span class="st">&quot;level1Name&quot;</span>, <span class="st">&quot;level2Gid&quot;</span>, <span class="st">&quot;level2Name&quot;</span>, <span class="st">&quot;level3Gid&quot;</span>, <span class="st">&quot;level3Name&quot;</span>)</span>
<span id="cb6-9"><a href="r-basics-ii.html#cb6-9" aria-hidden="true" tabindex="-1"></a>oc8 <span class="ot">&lt;-</span> oc1[, cols]</span></code></pre></div>
<p></br></p>
</div>
<div id="subsetting-with-conditions" class="section level2 hasAnchor" number="4.5">
<h2><span class="header-section-number">4.5</span> Subsetting with conditions<a href="r-basics-ii.html#subsetting-with-conditions" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Often you will need to filter out data that do not meet certain conditions.
For instance, you may want to exclude data from certain times or places, or those that have incomplete fields.
Though you can pass specific conditions for both rows and columns, in general it is more common to do so with rows.</p>
<p>The function <code>which</code> is very useful for applying conditions.
First, let’s try to create a subset of data that was collected in the US.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="r-basics-ii.html#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(oc8<span class="sc">$</span>countryCode) <span class="co"># list unique countries</span></span>
<span id="cb7-2"><a href="r-basics-ii.html#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(oc8<span class="sc">$</span>countryCode) <span class="co"># list how often each country occurs in the dataset</span></span>
<span id="cb7-3"><a href="r-basics-ii.html#cb7-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-4"><a href="r-basics-ii.html#cb7-4" aria-hidden="true" tabindex="-1"></a><span class="co"># rows that have &quot;US&quot; in the &#39;countyCode&#39; column</span></span>
<span id="cb7-5"><a href="r-basics-ii.html#cb7-5" aria-hidden="true" tabindex="-1"></a>oc8.us <span class="ot">&lt;-</span> oc8[<span class="fu">which</span>(oc8<span class="sc">$</span>countryCode <span class="sc">==</span> <span class="st">&quot;US&quot;</span>), ]</span>
<span id="cb7-6"><a href="r-basics-ii.html#cb7-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-7"><a href="r-basics-ii.html#cb7-7" aria-hidden="true" tabindex="-1"></a><span class="co"># rows that have values in the &#39;countyCode&#39; column that match those listed </span></span>
<span id="cb7-8"><a href="r-basics-ii.html#cb7-8" aria-hidden="true" tabindex="-1"></a>oc8.us1 <span class="ot">&lt;-</span> oc8[<span class="fu">which</span>(oc8<span class="sc">$</span>countryCode <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;US&quot;</span>)), ]</span>
<span id="cb7-9"><a href="r-basics-ii.html#cb7-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-10"><a href="r-basics-ii.html#cb7-10" aria-hidden="true" tabindex="-1"></a><span class="co"># rows that do not have values in the &#39;countyCode&#39; column that match those listed </span></span>
<span id="cb7-11"><a href="r-basics-ii.html#cb7-11" aria-hidden="true" tabindex="-1"></a>oc8.us2 <span class="ot">&lt;-</span> oc8[<span class="fu">which</span>(<span class="sc">!</span>oc8<span class="sc">$</span>countryCode <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;CN&quot;</span>, <span class="st">&quot;CA&quot;</span>, <span class="st">&quot;&quot;</span>, <span class="st">&quot;SE&quot;</span>), ]</span>
<span id="cb7-12"><a href="r-basics-ii.html#cb7-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-13"><a href="r-basics-ii.html#cb7-13" aria-hidden="true" tabindex="-1"></a><span class="co"># rows that have values in the &#39;year&#39; column that are greater than 2000</span></span>
<span id="cb7-14"><a href="r-basics-ii.html#cb7-14" aria-hidden="true" tabindex="-1"></a>oc8<span class="fl">.20</span> <span class="ot">&lt;-</span> oc8[<span class="fu">which</span>(oc8<span class="sc">$</span>year <span class="sc">&gt;</span> <span class="dv">2000</span>), ]</span></code></pre></div>
<p></br></p>
<p>You can also pass multiple conditions at once.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="r-basics-ii.html#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co"># rows that have &quot;US&quot; in the &#39;countyCode&#39; column AND &quot;California&quot; in the &#39;stateProvince&#39; column</span></span>
<span id="cb8-2"><a href="r-basics-ii.html#cb8-2" aria-hidden="true" tabindex="-1"></a>oc8.ca <span class="ot">&lt;-</span> oc8[<span class="fu">which</span>(oc8<span class="sc">$</span>countryCode <span class="sc">==</span> <span class="st">&quot;US&quot;</span> <span class="sc">&amp;</span> oc8<span class="sc">$</span>stateProvince <span class="sc">==</span> <span class="st">&quot;California&quot;</span>), ]</span>
<span id="cb8-3"><a href="r-basics-ii.html#cb8-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-4"><a href="r-basics-ii.html#cb8-4" aria-hidden="true" tabindex="-1"></a><span class="co"># rows that have &quot;US&quot; OR &quot;CA&quot; in the &#39;countyCode&#39; column</span></span>
<span id="cb8-5"><a href="r-basics-ii.html#cb8-5" aria-hidden="true" tabindex="-1"></a>oc8.usca <span class="ot">&lt;-</span> oc8[<span class="fu">which</span>(oc8<span class="sc">$</span>countryCode <span class="sc">==</span> <span class="st">&quot;US&quot;</span> <span class="sc">|</span> oc8<span class="sc">$</span>countryCode <span class="sc">==</span> <span class="st">&quot;CA&quot;</span>), ]</span>
<span id="cb8-6"><a href="r-basics-ii.html#cb8-6" aria-hidden="true" tabindex="-1"></a>oc8.usca <span class="ot">&lt;-</span> oc8[<span class="fu">which</span>(oc8<span class="sc">$</span>countryCode <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;US&quot;</span>, <span class="st">&quot;CA&quot;</span>)), ]</span></code></pre></div>
<p></br></p>
<p>Let’s create a subset of data that has both latitudinal and longitudinal coordinates.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="r-basics-ii.html#cb9-1" aria-hidden="true" tabindex="-1"></a>oc9 <span class="ot">&lt;-</span> oc8[<span class="fu">which</span>(<span class="fu">is.na</span>(oc8<span class="sc">$</span>decimalLatitude) <span class="sc">==</span> <span class="cn">FALSE</span> <span class="sc">&amp;</span> <span class="fu">is.na</span>(oc8<span class="sc">$</span>decimalLongitude) <span class="sc">==</span> <span class="cn">FALSE</span>), ]</span></code></pre></div>
<p></br></p>
<div id="exercise-2-1" class="section level3 hasAnchor" number="4.5.1">
<h3><span class="header-section-number">4.5.1</span> Exercise (2)<a href="r-basics-ii.html#exercise-2-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The code below also creates subsets of data that have both latitudinal and longitudinal coordinate values.
However, it arrives there in a slightly different logical process.
Explain what these lines of code are instructing the computer to do as you would to a friend who has no knowledge of R.
Be specific and explain each argument/condition passed.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="r-basics-ii.html#cb10-1" aria-hidden="true" tabindex="-1"></a>oc8[<span class="fu">which</span>(<span class="fu">is.na</span>(oc8<span class="sc">$</span>decimalLatitude) <span class="sc">!=</span> <span class="cn">TRUE</span> <span class="sc">&amp;</span> <span class="fu">is.na</span>(oc8<span class="sc">$</span>decimalLongitude) <span class="sc">!=</span> <span class="cn">TRUE</span>), ]</span>
<span id="cb10-2"><a href="r-basics-ii.html#cb10-2" aria-hidden="true" tabindex="-1"></a>oc8[<span class="fu">which</span>(<span class="sc">!</span><span class="fu">is.na</span>(oc8<span class="sc">$</span>decimalLatitude) <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(oc8<span class="sc">$</span>decimalLongitude)), ]</span></code></pre></div>
<p>For instance: “We are creating a subset of a data frame named”oc8” which has… and does not have…”</p>
<p>Write your code below:</p>
<hr />
<hr />
<hr />
<hr />
<p></br></p>
</div>
<div id="bonus-1" class="section level3 hasAnchor" number="4.5.2">
<h3><span class="header-section-number">4.5.2</span> Bonus [1]<a href="r-basics-ii.html#bonus-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Create the following:</p>
<ol style="list-style-type: decimal">
<li><p>A subset of the data frame <code>"oc9"</code> named <code>"oc10"</code> which only includes records (i.e., rows) that have ‘gbifID’ values that are in common with the ‘gbifID’ values of the data frame <code>"mm1"</code>.</p></li>
<li><p>A subset of the data frame <code>"mm1"</code> named <code>"mm2"</code> which only includes records (i.e., rows) that have ‘gbifID’ values that are in common with the ‘gbifID’ values of the data frame <code>"oc9"</code>.</p></li>
</ol>
<p>Write your code below:</p>
<hr />
<hr />
<hr />
<hr />
<p></br></p>
</div>
</div>
<div id="merging-data" class="section level2 hasAnchor" number="4.6">
<h2><span class="header-section-number">4.6</span> Merging data<a href="r-basics-ii.html#merging-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>It is often necessary to combine data from different sources.
Let’s think about how this would be done.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="r-basics-ii.html#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="co"># combine vectors</span></span>
<span id="cb11-2"><a href="r-basics-ii.html#cb11-2" aria-hidden="true" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;a&quot;</span>, <span class="st">&quot;b&quot;</span>, <span class="st">&quot;c&quot;</span>, <span class="st">&quot;d&quot;</span>)</span>
<span id="cb11-3"><a href="r-basics-ii.html#cb11-3" aria-hidden="true" tabindex="-1"></a>y <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">10</span>, <span class="dv">12</span>, <span class="dv">10</span>, <span class="dv">11</span>)</span>
<span id="cb11-4"><a href="r-basics-ii.html#cb11-4" aria-hidden="true" tabindex="-1"></a>z <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;e&quot;</span>, <span class="dv">8</span>)</span>
<span id="cb11-5"><a href="r-basics-ii.html#cb11-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-6"><a href="r-basics-ii.html#cb11-6" aria-hidden="true" tabindex="-1"></a>xy <span class="ot">&lt;-</span> <span class="fu">cbind</span>(x, y)</span>
<span id="cb11-7"><a href="r-basics-ii.html#cb11-7" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(xy)</span>
<span id="cb11-8"><a href="r-basics-ii.html#cb11-8" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(xy) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;letters&quot;</span>, <span class="st">&quot;numbers&quot;</span>)</span>
<span id="cb11-9"><a href="r-basics-ii.html#cb11-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-10"><a href="r-basics-ii.html#cb11-10" aria-hidden="true" tabindex="-1"></a>xy2 <span class="ot">&lt;-</span> <span class="fu">cbind</span>(<span class="at">letters =</span> x, <span class="at">numbers =</span> y)</span>
<span id="cb11-11"><a href="r-basics-ii.html#cb11-11" aria-hidden="true" tabindex="-1"></a>xyz <span class="ot">&lt;-</span> <span class="fu">rbind</span>(xy, z)</span>
<span id="cb11-12"><a href="r-basics-ii.html#cb11-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-13"><a href="r-basics-ii.html#cb11-13" aria-hidden="true" tabindex="-1"></a><span class="co"># merge data frames</span></span>
<span id="cb11-14"><a href="r-basics-ii.html#cb11-14" aria-hidden="true" tabindex="-1"></a>d1 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">species =</span> <span class="fu">c</span>(<span class="st">&quot;sp2&quot;</span>, <span class="st">&quot;sp3&quot;</span>, <span class="st">&quot;sp1&quot;</span>), <span class="at">height =</span> <span class="fu">c</span>(<span class="dv">10</span>, <span class="dv">20</span>, <span class="dv">15</span>), <span class="at">color =</span> <span class="fu">c</span>(<span class="st">&quot;blue&quot;</span>, <span class="st">&quot;red&quot;</span>, <span class="st">&quot;green&quot;</span>))</span>
<span id="cb11-15"><a href="r-basics-ii.html#cb11-15" aria-hidden="true" tabindex="-1"></a>d2 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">species =</span> <span class="fu">c</span>(<span class="st">&quot;sp3&quot;</span>, <span class="st">&quot;sp1&quot;</span>, <span class="st">&quot;sp2&quot;</span>), <span class="at">weight =</span> <span class="fu">c</span>(<span class="dv">100</span>, <span class="dv">420</span>, <span class="dv">115</span>))</span>
<span id="cb11-16"><a href="r-basics-ii.html#cb11-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-17"><a href="r-basics-ii.html#cb11-17" aria-hidden="true" tabindex="-1"></a>d12 <span class="ot">&lt;-</span> <span class="fu">cbind</span>(d1, d2) <span class="co"># does this look right?</span></span>
<span id="cb11-18"><a href="r-basics-ii.html#cb11-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-19"><a href="r-basics-ii.html#cb11-19" aria-hidden="true" tabindex="-1"></a>d1 <span class="ot">&lt;-</span> d1[<span class="fu">order</span>(d1<span class="sc">$</span>species), ] <span class="co"># order d1 by the &#39;species&#39; column  </span></span>
<span id="cb11-20"><a href="r-basics-ii.html#cb11-20" aria-hidden="true" tabindex="-1"></a>d2 <span class="ot">&lt;-</span> d2[<span class="fu">order</span>(d2<span class="sc">$</span>species), ] <span class="co"># order d2 by the &#39;species&#39; column</span></span>
<span id="cb11-21"><a href="r-basics-ii.html#cb11-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-22"><a href="r-basics-ii.html#cb11-22" aria-hidden="true" tabindex="-1"></a>d3 <span class="ot">&lt;-</span> <span class="fu">cbind</span>(d1, d2<span class="sc">$</span>weight)</span></code></pre></div>
<p></br></p>
<p>Let’s try a different example.
The data frames <code>"oc10"</code> and <code>"mm2"</code> should have the same ‘gbifID’ values in common.
Try to add the information in the <code>"identifier"</code> column of <code>"mm2"</code> to the corresponding records in <code>"oc10"</code>.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="r-basics-ii.html#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="co"># check if gbifID values match</span></span>
<span id="cb12-2"><a href="r-basics-ii.html#cb12-2" aria-hidden="true" tabindex="-1"></a>oc10<span class="sc">$</span>gbifID <span class="sc">%in%</span> mm2<span class="sc">$</span>gbifID</span>
<span id="cb12-3"><a href="r-basics-ii.html#cb12-3" aria-hidden="true" tabindex="-1"></a><span class="fu">all</span>(oc10<span class="sc">$</span>gbifID <span class="sc">%in%</span> mm2<span class="sc">$</span>gbifID) <span class="co"># the &#39;all&#39; function asks if all values are TRUE</span></span>
<span id="cb12-4"><a href="r-basics-ii.html#cb12-4" aria-hidden="true" tabindex="-1"></a><span class="fu">all</span>(mm2<span class="sc">$</span>gbifID <span class="sc">%in%</span> oc10<span class="sc">$</span>gbifID) <span class="co"># the &#39;all&#39; function asks if all values are TRUE</span></span>
<span id="cb12-5"><a href="r-basics-ii.html#cb12-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-6"><a href="r-basics-ii.html#cb12-6" aria-hidden="true" tabindex="-1"></a><span class="co"># order data frame by gbifID</span></span>
<span id="cb12-7"><a href="r-basics-ii.html#cb12-7" aria-hidden="true" tabindex="-1"></a>oc10 <span class="ot">&lt;-</span> oc10[<span class="fu">order</span>(oc10<span class="sc">$</span>gbifID), ]</span>
<span id="cb12-8"><a href="r-basics-ii.html#cb12-8" aria-hidden="true" tabindex="-1"></a>mm2 <span class="ot">&lt;-</span> mm2[<span class="fu">order</span>(mm2<span class="sc">$</span>gbifID), ]</span>
<span id="cb12-9"><a href="r-basics-ii.html#cb12-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-10"><a href="r-basics-ii.html#cb12-10" aria-hidden="true" tabindex="-1"></a><span class="co"># combine the two data frames</span></span>
<span id="cb12-11"><a href="r-basics-ii.html#cb12-11" aria-hidden="true" tabindex="-1"></a>om <span class="ot">&lt;-</span> <span class="fu">cbind</span>(oc10, mm2)</span></code></pre></div>
<p>Did that work?
If not, why?</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="r-basics-ii.html#cb13-1" aria-hidden="true" tabindex="-1"></a>?merge</span>
<span id="cb13-2"><a href="r-basics-ii.html#cb13-2" aria-hidden="true" tabindex="-1"></a>om <span class="ot">&lt;-</span> <span class="fu">merge</span>(oc10, mm2, <span class="at">by =</span> <span class="st">&quot;gbifID&quot;</span>)</span>
<span id="cb13-3"><a href="r-basics-ii.html#cb13-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-4"><a href="r-basics-ii.html#cb13-4" aria-hidden="true" tabindex="-1"></a><span class="co"># what do the following lines of code do?</span></span>
<span id="cb13-5"><a href="r-basics-ii.html#cb13-5" aria-hidden="true" tabindex="-1"></a><span class="fu">merge</span>(oc9, mm1, <span class="at">by =</span> <span class="st">&quot;gbifID&quot;</span>, <span class="at">all =</span> <span class="cn">TRUE</span>)</span>
<span id="cb13-6"><a href="r-basics-ii.html#cb13-6" aria-hidden="true" tabindex="-1"></a><span class="fu">merge</span>(oc9, mm1, <span class="at">by =</span> <span class="st">&quot;gbifID&quot;</span>, <span class="at">all.x =</span> <span class="cn">TRUE</span>)</span>
<span id="cb13-7"><a href="r-basics-ii.html#cb13-7" aria-hidden="true" tabindex="-1"></a><span class="fu">merge</span>(oc9, mm1, <span class="at">by =</span> <span class="st">&quot;gbifID&quot;</span>, <span class="at">all.y =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p>You can use the <code>merge</code> function to merge data frames based on a common index column.
It automates the process we went through previously.</p>
<p></br></p>
</div>
<div id="saving-files-and-sessions" class="section level2 hasAnchor" number="4.7">
<h2><span class="header-section-number">4.7</span> Saving files and sessions<a href="r-basics-ii.html#saving-files-and-sessions" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Saving files is similar to reading them in.
You need to specify the object(s) you want to save and where you want to save it.
Different functions and/or arguments are used to save data in different file formats.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="r-basics-ii.html#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="co"># saving as a tab-delimited text file</span></span>
<span id="cb14-2"><a href="r-basics-ii.html#cb14-2" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(om, <span class="st">&quot;merged_file.txt&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span>
<span id="cb14-3"><a href="r-basics-ii.html#cb14-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-4"><a href="r-basics-ii.html#cb14-4" aria-hidden="true" tabindex="-1"></a><span class="co"># saving as a comma-separated value file</span></span>
<span id="cb14-5"><a href="r-basics-ii.html#cb14-5" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(om, <span class="st">&quot;merged_file1.csv&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;,&quot;</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb14-6"><a href="r-basics-ii.html#cb14-6" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(om, <span class="st">&quot;merged_file2.csv&quot;</span>)</span></code></pre></div>
<p>You can also save you R worksapce.
This creates an image of everything loaded in your current R session as a “” file.
You can load this file in R later and continue where you left off previously.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="r-basics-ii.html#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="co"># saving image of workspace</span></span>
<span id="cb15-2"><a href="r-basics-ii.html#cb15-2" aria-hidden="true" tabindex="-1"></a><span class="fu">save.image</span>(<span class="st">&quot;workspace_20230125.RData&quot;</span>)</span></code></pre></div>
<p></br></p>
<div id="exercise-3-1" class="section level3 hasAnchor" number="4.7.1">
<h3><span class="header-section-number">4.7.1</span> Exercise (3)<a href="r-basics-ii.html#exercise-3-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Create a new data frame that merges:</p>
<ol style="list-style-type: decimal">
<li><p>Records in <code>"oc9"</code> that were collected in Oregon before the year 2000 (look at the “stateProvince” and “year” columns);</p></li>
<li><p>and the “format” and “identifier” columns of <code>"mm1"</code></p></li>
</ol>
<p>Save the result as a <code>".csv"</code> file named “firstname_lastname_lab5.csv” without row names.</p>
<p>Write your code below:</p>
<hr />
<hr />
<hr />
<hr />
<p></br></p>
</div>
</div>
<div id="the-for-loop" class="section level2 hasAnchor" number="4.8">
<h2><span class="header-section-number">4.8</span> The <span class="smallcaps">for</span>-loop<a href="r-basics-ii.html#the-for-loop" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The <code>for</code> command allows you to loop through a number of variables and apply the same functions to each of them.
This is useful when you want to repeatedly run the same bit of code, say, for example, to download a bunch of images.
First, let’s make a folder to download images in your working directory.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="r-basics-ii.html#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="co"># create directory</span></span>
<span id="cb16-2"><a href="r-basics-ii.html#cb16-2" aria-hidden="true" tabindex="-1"></a><span class="fu">dir.create</span>(<span class="st">&quot;images&quot;</span>)</span>
<span id="cb16-3"><a href="r-basics-ii.html#cb16-3" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">&quot;images&quot;</span>)</span>
<span id="cb16-4"><a href="r-basics-ii.html#cb16-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-5"><a href="r-basics-ii.html#cb16-5" aria-hidden="true" tabindex="-1"></a><span class="co"># downloading images</span></span>
<span id="cb16-6"><a href="r-basics-ii.html#cb16-6" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span> <span class="fu">nrow</span>(om)){</span>
<span id="cb16-7"><a href="r-basics-ii.html#cb16-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-8"><a href="r-basics-ii.html#cb16-8" aria-hidden="true" tabindex="-1"></a>  x <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb16-9"><a href="r-basics-ii.html#cb16-9" aria-hidden="true" tabindex="-1"></a>  attempt <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb16-10"><a href="r-basics-ii.html#cb16-10" aria-hidden="true" tabindex="-1"></a>  <span class="cf">while</span>(<span class="fu">is.null</span>(x) <span class="sc">&amp;&amp;</span> attempt <span class="sc">&lt;=</span> <span class="dv">3</span>) { <span class="co">#trying 3 times for failed downloads</span></span>
<span id="cb16-11"><a href="r-basics-ii.html#cb16-11" aria-hidden="true" tabindex="-1"></a>    attempt <span class="ot">&lt;-</span> attempt <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb16-12"><a href="r-basics-ii.html#cb16-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">try</span>(</span>
<span id="cb16-13"><a href="r-basics-ii.html#cb16-13" aria-hidden="true" tabindex="-1"></a>      x <span class="ot">&lt;-</span> <span class="fu">download.file</span>(<span class="at">url =</span> <span class="fu">as.character</span>(om<span class="sc">$</span>identifier[[i]]),</span>
<span id="cb16-14"><a href="r-basics-ii.html#cb16-14" aria-hidden="true" tabindex="-1"></a>                         <span class="at">destfile =</span> <span class="fu">paste</span>(om<span class="sc">$</span>genus[[i]], om<span class="sc">$</span>specificEpithet[[i]],</span>
<span id="cb16-15"><a href="r-basics-ii.html#cb16-15" aria-hidden="true" tabindex="-1"></a>                                          <span class="fu">as.character</span>(om<span class="sc">$</span>gbifID[[i]]),</span>
<span id="cb16-16"><a href="r-basics-ii.html#cb16-16" aria-hidden="true" tabindex="-1"></a>                                          i, <span class="st">&quot;jpg&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;.&quot;</span>),</span>
<span id="cb16-17"><a href="r-basics-ii.html#cb16-17" aria-hidden="true" tabindex="-1"></a>                         <span class="at">mode =</span> <span class="st">&quot;wb&quot;</span>)</span>
<span id="cb16-18"><a href="r-basics-ii.html#cb16-18" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb16-19"><a href="r-basics-ii.html#cb16-19" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb16-20"><a href="r-basics-ii.html#cb16-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(i)</span>
<span id="cb16-21"><a href="r-basics-ii.html#cb16-21" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
<p></br></p>
</div>
<div id="lists" class="section level2 hasAnchor" number="4.9">
<h2><span class="header-section-number">4.9</span> Lists<a href="r-basics-ii.html#lists" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>A list in <code>R</code> is a generic object consisting of an ordered collection of objects.
Lists are one-dimensional, heterogeneous data structures.
The list can be a list of vectors, a list of matrices, a list of characters, a list of functions, combinations of all of the above and more.
Thus, while a list is considered one-dimensional, it can comprise higher-dimensional elements.
A list in R is created with the use of <code>list()</code> function.
<code>R</code> allows accessing elements of a list with the use of the index value.
In <code>R</code>, the indexing of a list starts with 1 instead of 0 like other programming languages.
Elements are accessed in the list using <code>[[]]</code> (double square brackets), or by the element label.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="r-basics-ii.html#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create list containing two vectors</span></span>
<span id="cb17-2"><a href="r-basics-ii.html#cb17-2" aria-hidden="true" tabindex="-1"></a>mylist <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">Names =</span> <span class="fu">c</span>(<span class="st">&quot;a&quot;</span>, <span class="st">&quot;b&quot;</span>, <span class="st">&quot;c&quot;</span>, <span class="st">&quot;d&quot;</span>),</span>
<span id="cb17-3"><a href="r-basics-ii.html#cb17-3" aria-hidden="true" tabindex="-1"></a>               <span class="at">values =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>))</span>
<span id="cb17-4"><a href="r-basics-ii.html#cb17-4" aria-hidden="true" tabindex="-1"></a>mylist</span></code></pre></div>
<pre><code>## $Names
## [1] &quot;a&quot; &quot;b&quot; &quot;c&quot; &quot;d&quot;
## 
## $values
## [1] 1 2 3</code></pre>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="r-basics-ii.html#cb19-1" aria-hidden="true" tabindex="-1"></a>mylist[[<span class="dv">1</span>]]</span></code></pre></div>
<pre><code>## [1] &quot;a&quot; &quot;b&quot; &quot;c&quot; &quot;d&quot;</code></pre>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="r-basics-ii.html#cb21-1" aria-hidden="true" tabindex="-1"></a>mylist[[<span class="dv">2</span>]]</span></code></pre></div>
<pre><code>## [1] 1 2 3</code></pre>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="r-basics-ii.html#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="co"># access by label</span></span>
<span id="cb23-2"><a href="r-basics-ii.html#cb23-2" aria-hidden="true" tabindex="-1"></a>mylist<span class="sc">$</span>Names</span></code></pre></div>
<pre><code>## [1] &quot;a&quot; &quot;b&quot; &quot;c&quot; &quot;d&quot;</code></pre>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="r-basics-ii.html#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="co"># another syntax</span></span>
<span id="cb25-2"><a href="r-basics-ii.html#cb25-2" aria-hidden="true" tabindex="-1"></a>mylist[[<span class="st">&quot;Names&quot;</span>]]</span></code></pre></div>
<pre><code>## [1] &quot;a&quot; &quot;b&quot; &quot;c&quot; &quot;d&quot;</code></pre>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="r-basics-ii.html#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="co"># the third element in the vector of Values</span></span>
<span id="cb27-2"><a href="r-basics-ii.html#cb27-2" aria-hidden="true" tabindex="-1"></a>mylist[[<span class="st">&quot;values&quot;</span>]][<span class="dv">3</span>]</span></code></pre></div>
<pre><code>## [1] 3</code></pre>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="r-basics-ii.html#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="co"># if variable name was accessed wrongly, the output will show NULL</span></span>
<span id="cb29-2"><a href="r-basics-ii.html#cb29-2" aria-hidden="true" tabindex="-1"></a>mylist[[<span class="st">&quot;Values&quot;</span>]][<span class="dv">3</span>]</span></code></pre></div>
<pre><code>## NULL</code></pre>
<p>Some of the more complex data objects, such as phylogenetic trees, are essentially treated as lists in <code>R</code>.
We will revisit this in the following weeks.</p>
<p></br></p>
<p></br></p>
<p><strong>Assignment [5]: Upload the code you wrote for the three exercises and their results to BrightSpace (lab5).</strong></p>
<p></br></p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="r-basics-i.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="tree-and-traits-in-r.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/03_R_lab2_basics.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["bookdown-demo.pdf", "bookdown-demo.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
