<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 2 Unix | BIO595 The Tree of Life</title>
  <meta name="description" content="Test writing the book for the course bookdown::gitbook." />
  <meta name="generator" content="bookdown 0.28 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 2 Unix | BIO595 The Tree of Life" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="Test writing the book for the course bookdown::gitbook." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 2 Unix | BIO595 The Tree of Life" />
  
  <meta name="twitter:description" content="Test writing the book for the course bookdown::gitbook." />
  

<meta name="author" content="Qixin He, Daniel Park" />


<meta name="date" content="2023-04-10" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="index.html"/>
<link rel="next" href="r-basics-i.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Tree of Life: Phylogenetic inferences and applications</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Overview</a></li>
<li class="chapter" data-level="2" data-path="unix.html"><a href="unix.html"><i class="fa fa-check"></i><b>2</b> Unix</a>
<ul>
<li class="chapter" data-level="2.1" data-path="unix.html"><a href="unix.html#what-is-unix-and-why-do-we-use-it"><i class="fa fa-check"></i><b>2.1</b> What is Unix and why do we use it?</a></li>
<li class="chapter" data-level="2.2" data-path="unix.html"><a href="unix.html#installation"><i class="fa fa-check"></i><b>2.2</b> Installation</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="unix.html"><a href="unix.html#mac-os-and-linux"><i class="fa fa-check"></i><b>2.2.1</b> Mac OS and Linux</a></li>
<li class="chapter" data-level="2.2.2" data-path="unix.html"><a href="unix.html#windows"><i class="fa fa-check"></i><b>2.2.2</b> Windows</a></li>
<li class="chapter" data-level="2.2.3" data-path="unix.html"><a href="unix.html#use-scholar-remote-desktop-during-the-class"><i class="fa fa-check"></i><b>2.2.3</b> Use Scholar Remote Desktop During the Class</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="unix.html"><a href="unix.html#operation"><i class="fa fa-check"></i><b>2.3</b> Operation</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="unix.html"><a href="unix.html#starting-the-shell"><i class="fa fa-check"></i><b>2.3.1</b> Starting the Shell</a></li>
<li class="chapter" data-level="2.3.2" data-path="unix.html"><a href="unix.html#directoryfolder-sturcture"><i class="fa fa-check"></i><b>2.3.2</b> Directory/folder sturcture</a></li>
<li class="chapter" data-level="2.3.3" data-path="unix.html"><a href="unix.html#absolute-and-relative-directory"><i class="fa fa-check"></i><b>2.3.3</b> Absolute and Relative directory</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="unix.html"><a href="unix.html#handling-directories-and-files"><i class="fa fa-check"></i><b>2.4</b> Handling directories and files</a></li>
<li class="chapter" data-level="2.5" data-path="unix.html"><a href="unix.html#excercises-i"><i class="fa fa-check"></i><b>2.5</b> Excercises I</a></li>
<li class="chapter" data-level="2.6" data-path="unix.html"><a href="unix.html#viewing-and-processing-text-files"><i class="fa fa-check"></i><b>2.6</b> Viewing and processing text files</a></li>
<li class="chapter" data-level="2.7" data-path="unix.html"><a href="unix.html#redirecting-printouts-and-piping"><i class="fa fa-check"></i><b>2.7</b> Redirecting printouts and piping</a></li>
<li class="chapter" data-level="2.8" data-path="unix.html"><a href="unix.html#exercises-ii"><i class="fa fa-check"></i><b>2.8</b> Exercises II</a></li>
<li class="chapter" data-level="2.9" data-path="unix.html"><a href="unix.html#useful-wildcards"><i class="fa fa-check"></i><b>2.9</b> Useful wildcards</a></li>
<li class="chapter" data-level="2.10" data-path="unix.html"><a href="unix.html#exercises-iii"><i class="fa fa-check"></i><b>2.10</b> Exercises III</a></li>
<li class="chapter" data-level="2.11" data-path="unix.html"><a href="unix.html#search-and-select-lines-using-grep"><i class="fa fa-check"></i><b>2.11</b> Search and select lines using <em>grep</em></a></li>
<li class="chapter" data-level="2.12" data-path="unix.html"><a href="unix.html#substituting-characters-using-tr"><i class="fa fa-check"></i><b>2.12</b> Substituting characters using <em>tr</em></a></li>
<li class="chapter" data-level="2.13" data-path="unix.html"><a href="unix.html#cut-sort-and-unique"><i class="fa fa-check"></i><b>2.13</b> cut, sort and unique</a></li>
<li class="chapter" data-level="2.14" data-path="unix.html"><a href="unix.html#exercises-iv"><i class="fa fa-check"></i><b>2.14</b> Exercises IV</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="r-basics-i.html"><a href="r-basics-i.html"><i class="fa fa-check"></i><b>3</b> R Basics I</a>
<ul>
<li class="chapter" data-level="3.1" data-path="r-basics-i.html"><a href="r-basics-i.html#what-is-programming"><i class="fa fa-check"></i><b>3.1</b> What is programming?</a></li>
<li class="chapter" data-level="3.2" data-path="r-basics-i.html"><a href="r-basics-i.html#r"><i class="fa fa-check"></i><b>3.2</b> <code>R</code></a></li>
<li class="chapter" data-level="3.3" data-path="r-basics-i.html"><a href="r-basics-i.html#installation-1"><i class="fa fa-check"></i><b>3.3</b> Installation</a></li>
<li class="chapter" data-level="3.4" data-path="r-basics-i.html"><a href="r-basics-i.html#variables"><i class="fa fa-check"></i><b>3.4</b> Variables</a>
<ul>
<li class="chapter" data-level="3.4.1" data-path="r-basics-i.html"><a href="r-basics-i.html#exercise-1"><i class="fa fa-check"></i><b>3.4.1</b> Exercise (1):</a></li>
<li class="chapter" data-level="3.4.2" data-path="r-basics-i.html"><a href="r-basics-i.html#variable-naming-conventions"><i class="fa fa-check"></i><b>3.4.2</b> Variable naming conventions</a></li>
<li class="chapter" data-level="3.4.3" data-path="r-basics-i.html"><a href="r-basics-i.html#arithmetic-operators"><i class="fa fa-check"></i><b>3.4.3</b> Arithmetic Operators</a></li>
<li class="chapter" data-level="3.4.4" data-path="r-basics-i.html"><a href="r-basics-i.html#logical-operators"><i class="fa fa-check"></i><b>3.4.4</b> Logical Operators</a></li>
<li class="chapter" data-level="3.4.5" data-path="r-basics-i.html"><a href="r-basics-i.html#basic-built-in-mathematical-functions"><i class="fa fa-check"></i><b>3.4.5</b> Basic built-in mathematical functions</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="r-basics-i.html"><a href="r-basics-i.html#subsetting-data"><i class="fa fa-check"></i><b>3.5</b> Subsetting data</a></li>
<li class="chapter" data-level="3.6" data-path="r-basics-i.html"><a href="r-basics-i.html#logical-values"><i class="fa fa-check"></i><b>3.6</b> Logical values</a>
<ul>
<li class="chapter" data-level="3.6.1" data-path="r-basics-i.html"><a href="r-basics-i.html#missing-dealing-with-na-values"><i class="fa fa-check"></i><b>3.6.1</b> Missing: dealing with <code>NA</code> values</a></li>
</ul></li>
<li class="chapter" data-level="3.7" data-path="r-basics-i.html"><a href="r-basics-i.html#generating-data"><i class="fa fa-check"></i><b>3.7</b> Generating Data</a></li>
<li class="chapter" data-level="3.8" data-path="r-basics-i.html"><a href="r-basics-i.html#data-structures-and-types"><i class="fa fa-check"></i><b>3.8</b> Data Structures and Types</a>
<ul>
<li class="chapter" data-level="3.8.1" data-path="r-basics-i.html"><a href="r-basics-i.html#vectors"><i class="fa fa-check"></i><b>3.8.1</b> Vectors</a></li>
</ul></li>
<li class="chapter" data-level="3.9" data-path="r-basics-i.html"><a href="r-basics-i.html#matrices"><i class="fa fa-check"></i><b>3.9</b> Matrices</a>
<ul>
<li class="chapter" data-level="3.9.1" data-path="r-basics-i.html"><a href="r-basics-i.html#exercise-2"><i class="fa fa-check"></i><b>3.9.1</b> Exercise (2):</a></li>
</ul></li>
<li class="chapter" data-level="3.10" data-path="r-basics-i.html"><a href="r-basics-i.html#data-frames"><i class="fa fa-check"></i><b>3.10</b> Data Frames</a>
<ul>
<li class="chapter" data-level="3.10.1" data-path="r-basics-i.html"><a href="r-basics-i.html#exercise-3"><i class="fa fa-check"></i><b>3.10.1</b> Exercise (3):</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="r-basics-ii.html"><a href="r-basics-ii.html"><i class="fa fa-check"></i><b>4</b> R Basics II</a>
<ul>
<li class="chapter" data-level="4.1" data-path="r-basics-ii.html"><a href="r-basics-ii.html#setting-up-the-environment"><i class="fa fa-check"></i><b>4.1</b> Setting up the environment</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="r-basics-ii.html"><a href="r-basics-ii.html#exercise-1-1"><i class="fa fa-check"></i><b>4.1.1</b> Exercise (1)</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="r-basics-ii.html"><a href="r-basics-ii.html#reading-in-data-from-files"><i class="fa fa-check"></i><b>4.2</b> Reading in data from files</a></li>
<li class="chapter" data-level="4.3" data-path="r-basics-ii.html"><a href="r-basics-ii.html#wrangling-data"><i class="fa fa-check"></i><b>4.3</b> Wrangling Data</a></li>
<li class="chapter" data-level="4.4" data-path="r-basics-ii.html"><a href="r-basics-ii.html#subsetting-by-columns-and-rows"><i class="fa fa-check"></i><b>4.4</b> Subsetting by columns and rows</a></li>
<li class="chapter" data-level="4.5" data-path="r-basics-ii.html"><a href="r-basics-ii.html#subsetting-with-conditions"><i class="fa fa-check"></i><b>4.5</b> Subsetting with conditions</a>
<ul>
<li class="chapter" data-level="4.5.1" data-path="r-basics-ii.html"><a href="r-basics-ii.html#exercise-2-1"><i class="fa fa-check"></i><b>4.5.1</b> Exercise (2)</a></li>
<li class="chapter" data-level="4.5.2" data-path="r-basics-ii.html"><a href="r-basics-ii.html#bonus-1"><i class="fa fa-check"></i><b>4.5.2</b> Bonus [1]</a></li>
</ul></li>
<li class="chapter" data-level="4.6" data-path="r-basics-ii.html"><a href="r-basics-ii.html#merging-data"><i class="fa fa-check"></i><b>4.6</b> Merging data</a></li>
<li class="chapter" data-level="4.7" data-path="r-basics-ii.html"><a href="r-basics-ii.html#saving-files-and-sessions"><i class="fa fa-check"></i><b>4.7</b> Saving files and sessions</a>
<ul>
<li class="chapter" data-level="4.7.1" data-path="r-basics-ii.html"><a href="r-basics-ii.html#exercise-3-1"><i class="fa fa-check"></i><b>4.7.1</b> Exercise (3)</a></li>
</ul></li>
<li class="chapter" data-level="4.8" data-path="r-basics-ii.html"><a href="r-basics-ii.html#the-for-loop"><i class="fa fa-check"></i><b>4.8</b> The <span class="smallcaps">for</span>-loop</a></li>
<li class="chapter" data-level="4.9" data-path="r-basics-ii.html"><a href="r-basics-ii.html#lists"><i class="fa fa-check"></i><b>4.9</b> Lists</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="tree-and-traits-in-r.html"><a href="tree-and-traits-in-r.html"><i class="fa fa-check"></i><b>5</b> Tree and Traits in R</a>
<ul>
<li class="chapter" data-level="5.1" data-path="tree-and-traits-in-r.html"><a href="tree-and-traits-in-r.html#setting-up-the-environment-1"><i class="fa fa-check"></i><b>5.1</b> Setting up the environment</a></li>
<li class="chapter" data-level="5.2" data-path="tree-and-traits-in-r.html"><a href="tree-and-traits-in-r.html#examining-phylogenetic-trees"><i class="fa fa-check"></i><b>5.2</b> Examining phylogenetic trees</a></li>
<li class="chapter" data-level="5.3" data-path="tree-and-traits-in-r.html"><a href="tree-and-traits-in-r.html#plotting-phylogenetic-trees"><i class="fa fa-check"></i><b>5.3</b> Plotting phylogenetic trees</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="tree-and-traits-in-r.html"><a href="tree-and-traits-in-r.html#exercise-1-2"><i class="fa fa-check"></i><b>5.3.1</b> Exercise (1)</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="tree-and-traits-in-r.html"><a href="tree-and-traits-in-r.html#associating-phylogenies-and-traits"><i class="fa fa-check"></i><b>5.4</b> Associating phylogenies and traits</a>
<ul>
<li class="chapter" data-level="5.4.1" data-path="tree-and-traits-in-r.html"><a href="tree-and-traits-in-r.html#saving-plots"><i class="fa fa-check"></i><b>5.4.1</b> Saving plots</a></li>
<li class="chapter" data-level="5.4.2" data-path="tree-and-traits-in-r.html"><a href="tree-and-traits-in-r.html#exercise-2-2"><i class="fa fa-check"></i><b>5.4.2</b> Exercise (2)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html"><i class="fa fa-check"></i><b>6</b> Biogeography and Macroevolution in R</a>
<ul>
<li class="chapter" data-level="6.1" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#setting-up-the-environment-2"><i class="fa fa-check"></i><b>6.1</b> Setting up the environment</a></li>
<li class="chapter" data-level="6.2" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#ancestral-state-reconstructions"><i class="fa fa-check"></i><b>6.2</b> Ancestral State Reconstructions</a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#bonus-2"><i class="fa fa-check"></i><b>6.2.1</b> Bonus [2]</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#continuous-traits"><i class="fa fa-check"></i><b>6.3</b> Continuous traits</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#maximum-likelihood"><i class="fa fa-check"></i><b>6.3.1</b> Maximum likelihood</a></li>
<li class="chapter" data-level="6.3.2" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#phylogenetically-independent-contrasts"><i class="fa fa-check"></i><b>6.3.2</b> Phylogenetically independent contrasts</a></li>
<li class="chapter" data-level="6.3.3" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#plotting-ancestral-states"><i class="fa fa-check"></i><b>6.3.3</b> Plotting ancestral states</a></li>
<li class="chapter" data-level="6.3.4" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#exercise-1-3"><i class="fa fa-check"></i><b>6.3.4</b> Exercise (1)</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#discrete-traits"><i class="fa fa-check"></i><b>6.4</b> Discrete traits</a>
<ul>
<li class="chapter" data-level="6.4.1" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#exercise-2-3"><i class="fa fa-check"></i><b>6.4.1</b> Exercise (2)</a></li>
<li class="chapter" data-level="6.4.2" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#maximum-likelihood-1"><i class="fa fa-check"></i><b>6.4.2</b> Maximum likelihood</a></li>
<li class="chapter" data-level="6.4.3" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#choosing-models"><i class="fa fa-check"></i><b>6.4.3</b> Choosing models</a></li>
</ul></li>
<li class="chapter" data-level="6.5" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#saving-multiple-plots-together"><i class="fa fa-check"></i><b>6.5</b> Saving multiple plots together</a>
<ul>
<li class="chapter" data-level="6.5.1" data-path="biogeography-and-macroevolution-in-r.html"><a href="biogeography-and-macroevolution-in-r.html#exercise-3-2"><i class="fa fa-check"></i><b>6.5.1</b> Exercise (3)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="phylogenetic-diversity-in-r.html"><a href="phylogenetic-diversity-in-r.html"><i class="fa fa-check"></i><b>7</b> Phylogenetic Diversity in R</a>
<ul>
<li class="chapter" data-level="7.1" data-path="phylogenetic-diversity-in-r.html"><a href="phylogenetic-diversity-in-r.html#setting-up-the-environment-3"><i class="fa fa-check"></i><b>7.1</b> Setting up the environment</a></li>
<li class="chapter" data-level="7.2" data-path="phylogenetic-diversity-in-r.html"><a href="phylogenetic-diversity-in-r.html#phylogenetic-diversity"><i class="fa fa-check"></i><b>7.2</b> Phylogenetic Diversity</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="phylogenetic-diversity-in-r.html"><a href="phylogenetic-diversity-in-r.html#preprocessing"><i class="fa fa-check"></i><b>7.2.1</b> Preprocessing</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="phylogenetic-diversity-in-r.html"><a href="phylogenetic-diversity-in-r.html#estimating-phylogenetic-diversity-metrics"><i class="fa fa-check"></i><b>7.3</b> Estimating phylogenetic diversity metrics</a>
<ul>
<li class="chapter" data-level="7.3.1" data-path="phylogenetic-diversity-in-r.html"><a href="phylogenetic-diversity-in-r.html#exercise-1-4"><i class="fa fa-check"></i><b>7.3.1</b> Exercise (1)</a></li>
<li class="chapter" data-level="7.3.2" data-path="phylogenetic-diversity-in-r.html"><a href="phylogenetic-diversity-in-r.html#exercise-2-4"><i class="fa fa-check"></i><b>7.3.2</b> Exercise (2)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="community-phylogenetics.html"><a href="community-phylogenetics.html"><i class="fa fa-check"></i><b>8</b> Community Phylogenetics</a>
<ul>
<li class="chapter" data-level="8.1" data-path="community-phylogenetics.html"><a href="community-phylogenetics.html#setting-up-the-environment-4"><i class="fa fa-check"></i><b>8.1</b> Setting up the environment</a></li>
<li class="chapter" data-level="8.2" data-path="community-phylogenetics.html"><a href="community-phylogenetics.html#definition-of-community-phylogenetics"><i class="fa fa-check"></i><b>8.2</b> Definition of Community Phylogenetics</a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="community-phylogenetics.html"><a href="community-phylogenetics.html#alpha-diversity"><i class="fa fa-check"></i><b>8.2.1</b> Alpha diversity</a></li>
<li class="chapter" data-level="8.2.2" data-path="community-phylogenetics.html"><a href="community-phylogenetics.html#beta-diversity"><i class="fa fa-check"></i><b>8.2.2</b> Beta diversity</a></li>
<li class="chapter" data-level="8.2.3" data-path="community-phylogenetics.html"><a href="community-phylogenetics.html#exercise-1-5"><i class="fa fa-check"></i><b>8.2.3</b> Exercise (1)</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="community-phylogenetics.html"><a href="community-phylogenetics.html#standardized-effect-sizes"><i class="fa fa-check"></i><b>8.3</b> Standardized effect sizes</a>
<ul>
<li class="chapter" data-level="8.3.1" data-path="community-phylogenetics.html"><a href="community-phylogenetics.html#exercise-2-5"><i class="fa fa-check"></i><b>8.3.1</b> Exercise (2)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="coevolution-and-phylogenetic-signals-in-r.html"><a href="coevolution-and-phylogenetic-signals-in-r.html"><i class="fa fa-check"></i><b>9</b> Coevolution and Phylogenetic Signals in R</a>
<ul>
<li class="chapter" data-level="9.1" data-path="coevolution-and-phylogenetic-signals-in-r.html"><a href="coevolution-and-phylogenetic-signals-in-r.html#setting-up-the-environment-5"><i class="fa fa-check"></i><b>9.1</b> Setting up the environment</a></li>
<li class="chapter" data-level="9.2" data-path="coevolution-and-phylogenetic-signals-in-r.html"><a href="coevolution-and-phylogenetic-signals-in-r.html#phylogenetic-signal"><i class="fa fa-check"></i><b>9.2</b> Phylogenetic Signal</a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="coevolution-and-phylogenetic-signals-in-r.html"><a href="coevolution-and-phylogenetic-signals-in-r.html#blombergs-k"><i class="fa fa-check"></i><b>9.2.1</b> Blomberg’s K</a></li>
<li class="chapter" data-level="9.2.2" data-path="coevolution-and-phylogenetic-signals-in-r.html"><a href="coevolution-and-phylogenetic-signals-in-r.html#pagels-lambda-λ"><i class="fa fa-check"></i><b>9.2.2</b> Pagel’s lambda (λ)</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="coevolution-and-phylogenetic-signals-in-r.html"><a href="coevolution-and-phylogenetic-signals-in-r.html#phylogenetically-independent-contrasts-1"><i class="fa fa-check"></i><b>9.3</b> Phylogenetically Independent Contrasts</a></li>
<li class="chapter" data-level="9.4" data-path="coevolution-and-phylogenetic-signals-in-r.html"><a href="coevolution-and-phylogenetic-signals-in-r.html#coevolution"><i class="fa fa-check"></i><b>9.4</b> Coevolution</a>
<ul>
<li class="chapter" data-level="9.4.1" data-path="coevolution-and-phylogenetic-signals-in-r.html"><a href="coevolution-and-phylogenetic-signals-in-r.html#plotting-tanglegrams"><i class="fa fa-check"></i><b>9.4.1</b> Plotting tanglegrams</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="phylogenetic-visualization-in-r.html"><a href="phylogenetic-visualization-in-r.html"><i class="fa fa-check"></i><b>10</b> Phylogenetic Visualization in R</a>
<ul>
<li class="chapter" data-level="10.1" data-path="phylogenetic-visualization-in-r.html"><a href="phylogenetic-visualization-in-r.html#setting-up-the-environment-6"><i class="fa fa-check"></i><b>10.1</b> Setting up the environment</a></li>
<li class="chapter" data-level="10.2" data-path="phylogenetic-visualization-in-r.html"><a href="phylogenetic-visualization-in-r.html#using-phytools"><i class="fa fa-check"></i><b>10.2</b> Using <code>phytools</code></a>
<ul>
<li class="chapter" data-level="10.2.1" data-path="phylogenetic-visualization-in-r.html"><a href="phylogenetic-visualization-in-r.html#phenograms"><i class="fa fa-check"></i><b>10.2.1</b> Phenograms</a></li>
<li class="chapter" data-level="10.2.2" data-path="phylogenetic-visualization-in-r.html"><a href="phylogenetic-visualization-in-r.html#mapping-continuous-characters-on-branches"><i class="fa fa-check"></i><b>10.2.2</b> Mapping continuous characters on branches</a></li>
<li class="chapter" data-level="10.2.3" data-path="phylogenetic-visualization-in-r.html"><a href="phylogenetic-visualization-in-r.html#plotting-images-on-phylogenies"><i class="fa fa-check"></i><b>10.2.3</b> Plotting images on phylogenies</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="phylogenetic-visualization-in-r.html"><a href="phylogenetic-visualization-in-r.html#using-ggtree"><i class="fa fa-check"></i><b>10.3</b> Using <code>ggtree</code></a>
<ul>
<li class="chapter" data-level="10.3.1" data-path="phylogenetic-visualization-in-r.html"><a href="phylogenetic-visualization-in-r.html#basic-plotting"><i class="fa fa-check"></i><b>10.3.1</b> Basic plotting</a></li>
<li class="chapter" data-level="10.3.2" data-path="phylogenetic-visualization-in-r.html"><a href="phylogenetic-visualization-in-r.html#labeling-tips-nodes-and-clades"><i class="fa fa-check"></i><b>10.3.2</b> Labeling tips, nodes, and clades</a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="phylogenetic-visualization-in-r.html"><a href="phylogenetic-visualization-in-r.html#plotting-maps"><i class="fa fa-check"></i><b>10.4</b> Plotting maps</a>
<ul>
<li class="chapter" data-level="10.4.1" data-path="phylogenetic-visualization-in-r.html"><a href="phylogenetic-visualization-in-r.html#choropleth-maps"><i class="fa fa-check"></i><b>10.4.1</b> Choropleth maps</a></li>
<li class="chapter" data-level="10.4.2" data-path="phylogenetic-visualization-in-r.html"><a href="phylogenetic-visualization-in-r.html#working-with-rasters"><i class="fa fa-check"></i><b>10.4.2</b> Working with rasters</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html"><i class="fa fa-check"></i><b>11</b> Gene tree vs Species tree</a>
<ul>
<li class="chapter" data-level="11.1" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#learning-goals"><i class="fa fa-check"></i><b>11.1</b> Learning goals:</a></li>
<li class="chapter" data-level="11.2" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#tracing-ancestry-within-one-population"><i class="fa fa-check"></i><b>11.2</b> Tracing ancestry within one population</a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#vary-pop-size-and-generation"><i class="fa fa-check"></i><b>11.2.1</b> Vary pop size and generation</a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#speciation-and-coalescence"><i class="fa fa-check"></i><b>11.3</b> Speciation and coalescence</a>
<ul>
<li class="chapter" data-level="11.3.1" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#two-species-coalescent-history"><i class="fa fa-check"></i><b>11.3.1</b> Two-species coalescent history</a></li>
</ul></li>
<li class="chapter" data-level="11.4" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#exercise-i"><i class="fa fa-check"></i><b>11.4</b> Exercise I</a></li>
<li class="chapter" data-level="11.5" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#gene-tree-locus-tree-and-species-tree"><i class="fa fa-check"></i><b>11.5</b> Gene tree, Locus tree and Species tree</a>
<ul>
<li class="chapter" data-level="11.5.1" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#loading-data-and-packages"><i class="fa fa-check"></i><b>11.5.1</b> loading data and packages</a></li>
</ul></li>
<li class="chapter" data-level="11.6" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#effects-of-incomplete-lineage-sorting-ils-alone"><i class="fa fa-check"></i><b>11.6</b> Effects of Incomplete Lineage Sorting (ILS) alone</a>
<ul>
<li class="chapter" data-level="11.6.1" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#generate-species-tree-with-no-extinction"><i class="fa fa-check"></i><b>11.6.1</b> generate species tree with no extinction</a></li>
<li class="chapter" data-level="11.6.2" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#simulate-gene-trees-using-the-multispecies-coalescent-process"><i class="fa fa-check"></i><b>11.6.2</b> simulate gene trees using the multispecies coalescent process</a></li>
</ul></li>
<li class="chapter" data-level="11.7" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#calculate-tree-distances"><i class="fa fa-check"></i><b>11.7</b> Calculate tree distances</a>
<ul>
<li class="chapter" data-level="11.7.1" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#rf-distances"><i class="fa fa-check"></i><b>11.7.1</b> RF distances</a></li>
<li class="chapter" data-level="11.7.2" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#inspect-pairwise-rf-distances-among-all-the-gene-trees-generated-under-one-guiding-species-tree"><i class="fa fa-check"></i><b>11.7.2</b> Inspect pairwise RF distances among all the gene trees generated under one guiding species tree</a></li>
</ul></li>
<li class="chapter" data-level="11.8" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#inspect-node-support-i.e.-subtrees-defined-by-a-split"><i class="fa fa-check"></i><b>11.8</b> Inspect node support (i.e., subtrees defined by a split)</a></li>
<li class="chapter" data-level="11.9" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#combined-effects-of-gene-duplication-loss-and-horizontal-gene-transfer-events"><i class="fa fa-check"></i><b>11.9</b> Combined effects of gene duplication, loss and horizontal gene transfer events</a></li>
<li class="chapter" data-level="11.10" data-path="gene-tree-vs-species-tree.html"><a href="gene-tree-vs-species-tree.html#exercise-ii"><i class="fa fa-check"></i><b>11.10</b> Exercise II</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="overview-of-phylogenetic-inferences.html"><a href="overview-of-phylogenetic-inferences.html"><i class="fa fa-check"></i><b>12</b> Overview of Phylogenetic Inferences</a>
<ul>
<li class="chapter" data-level="12.1" data-path="overview-of-phylogenetic-inferences.html"><a href="overview-of-phylogenetic-inferences.html#mega"><i class="fa fa-check"></i><b>12.1</b> <code>MEGA</code></a></li>
<li class="chapter" data-level="12.2" data-path="overview-of-phylogenetic-inferences.html"><a href="overview-of-phylogenetic-inferences.html#view-and-align-sequences"><i class="fa fa-check"></i><b>12.2</b> View and Align Sequences</a></li>
<li class="chapter" data-level="12.3" data-path="overview-of-phylogenetic-inferences.html"><a href="overview-of-phylogenetic-inferences.html#dnaprotein-evolution-model"><i class="fa fa-check"></i><b>12.3</b> DNA/protein evolution model</a></li>
<li class="chapter" data-level="12.4" data-path="overview-of-phylogenetic-inferences.html"><a href="overview-of-phylogenetic-inferences.html#phylogenetic-inference"><i class="fa fa-check"></i><b>12.4</b> Phylogenetic inference</a></li>
<li class="chapter" data-level="12.5" data-path="overview-of-phylogenetic-inferences.html"><a href="overview-of-phylogenetic-inferences.html#estimate-divergence-times"><i class="fa fa-check"></i><b>12.5</b> Estimate divergence times</a></li>
<li class="chapter" data-level="12.6" data-path="overview-of-phylogenetic-inferences.html"><a href="overview-of-phylogenetic-inferences.html#exercises"><i class="fa fa-check"></i><b>12.6</b> Exercises</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html"><i class="fa fa-check"></i><b>13</b> Resources and Guidelines for the Term Project</a>
<ul>
<li class="chapter" data-level="13.1" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#bioinformatics-databases"><i class="fa fa-check"></i><b>13.1</b> Bioinformatics Databases</a>
<ul>
<li class="chapter" data-level="13.1.1" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#ncbi-national-center-for-biotechnology-information"><i class="fa fa-check"></i><b>13.1.1</b> NCBI (National Center for Biotechnology Information)</a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#retrieving-genetic-resources-for-your-taxa"><i class="fa fa-check"></i><b>13.2</b> Retrieving genetic resources for your taxa</a>
<ul>
<li class="chapter" data-level="13.2.1" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#ncbi-taxonomy-database"><i class="fa fa-check"></i><b>13.2.1</b> NCBI taxonomy database</a></li>
<li class="chapter" data-level="13.2.2" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#individual-genes-from-the-whole-genome-data"><i class="fa fa-check"></i><b>13.2.2</b> Individual genes from the whole genome data</a></li>
<li class="chapter" data-level="13.2.3" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#get-homologous-sequences-from-related-species"><i class="fa fa-check"></i><b>13.2.3</b> Get homologous sequences from related species</a></li>
<li class="chapter" data-level="13.2.4" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#european-bioinformatics-institute-ebi"><i class="fa fa-check"></i><b>13.2.4</b> European Bioinformatics Institute (EBI)</a></li>
<li class="chapter" data-level="13.2.5" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#protein-data-bank"><i class="fa fa-check"></i><b>13.2.5</b> Protein Data Bank</a></li>
</ul></li>
<li class="chapter" data-level="13.3" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#biogeographic-databases"><i class="fa fa-check"></i><b>13.3</b> Biogeographic Databases</a>
<ul>
<li class="chapter" data-level="13.3.1" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#climate-data"><i class="fa fa-check"></i><b>13.3.1</b> Climate Data</a></li>
<li class="chapter" data-level="13.3.2" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#occurrence-data"><i class="fa fa-check"></i><b>13.3.2</b> Occurrence Data</a></li>
<li class="chapter" data-level="13.3.3" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#topographic-data"><i class="fa fa-check"></i><b>13.3.3</b> TOPOGRAPHIC DATA</a></li>
<li class="chapter" data-level="13.3.4" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#trait-data"><i class="fa fa-check"></i><b>13.3.4</b> TRAIT DATA</a></li>
</ul></li>
<li class="chapter" data-level="13.4" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#publication-databases"><i class="fa fa-check"></i><b>13.4</b> Publication Databases</a></li>
<li class="chapter" data-level="13.5" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#example-research-questions"><i class="fa fa-check"></i><b>13.5</b> Example research questions:</a>
<ul>
<li class="chapter" data-level="13.5.1" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#biogeography-and-phylogenetic-diversity"><i class="fa fa-check"></i><b>13.5.1</b> Biogeography and Phylogenetic Diversity:</a></li>
<li class="chapter" data-level="13.5.2" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#community-ecology"><i class="fa fa-check"></i><b>13.5.2</b> Community Ecology:</a></li>
<li class="chapter" data-level="13.5.3" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#coevolution-1"><i class="fa fa-check"></i><b>13.5.3</b> Coevolution:</a></li>
<li class="chapter" data-level="13.5.4" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#species-tree"><i class="fa fa-check"></i><b>13.5.4</b> Species tree:</a></li>
</ul></li>
<li class="chapter" data-level="13.6" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#how-to-submit-jobs-on-scholar-cluster-and-run-remotely"><i class="fa fa-check"></i><b>13.6</b> How to submit jobs on Scholar Cluster and run remotely</a></li>
<li class="chapter" data-level="13.7" data-path="resources-and-guidelines-for-the-term-project.html"><a href="resources-and-guidelines-for-the-term-project.html#term-project-proposal"><i class="fa fa-check"></i><b>13.7</b> Term project proposal</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><i class="fa fa-check"></i><b>14</b> Systematic Retrieval of orthologous DNA sequences form GenBank</a>
<ul>
<li class="chapter" data-level="14.1" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#introduction"><i class="fa fa-check"></i><b>14.1</b> Introduction</a></li>
<li class="chapter" data-level="14.2" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#installing-tools"><i class="fa fa-check"></i><b>14.2</b> Installing Tools</a>
<ul>
<li class="chapter" data-level="14.2.1" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#windows-1"><i class="fa fa-check"></i><b>14.2.1</b> Windows</a></li>
<li class="chapter" data-level="14.2.2" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#mac"><i class="fa fa-check"></i><b>14.2.2</b> Mac</a></li>
<li class="chapter" data-level="14.2.3" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#ncbi-blast-tools"><i class="fa fa-check"></i><b>14.2.3</b> NCBI BLAST+ Tools</a></li>
</ul></li>
<li class="chapter" data-level="14.3" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#using-phylotar"><i class="fa fa-check"></i><b>14.3</b> Using <code>phylotaR</code></a>
<ul>
<li class="chapter" data-level="14.3.1" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#pipeline"><i class="fa fa-check"></i><b>14.3.1</b> Pipeline</a></li>
<li class="chapter" data-level="14.3.2" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#running"><i class="fa fa-check"></i><b>14.3.2</b> Running</a></li>
<li class="chapter" data-level="14.3.3" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#restarting"><i class="fa fa-check"></i><b>14.3.3</b> Restarting</a></li>
<li class="chapter" data-level="14.3.4" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#cluster-i.e.-orthologous-genes-selection"><i class="fa fa-check"></i><b>14.3.4</b> Cluster (i.e., orthologous genes) selection</a></li>
</ul></li>
<li class="chapter" data-level="14.4" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#exercise-i-1"><i class="fa fa-check"></i><b>14.4</b> Exercise I</a></li>
<li class="chapter" data-level="14.5" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#using-annotationbustr"><i class="fa fa-check"></i><b>14.5</b> Using <code>AnnotationBustR</code></a>
<ul>
<li class="chapter" data-level="14.5.1" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#introduction-1"><i class="fa fa-check"></i><b>14.5.1</b> Introduction</a></li>
<li class="chapter" data-level="14.5.2" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#install-the-tools"><i class="fa fa-check"></i><b>14.5.2</b> Install the tools</a></li>
<li class="chapter" data-level="14.5.3" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#specify-search-terms-of-genomes"><i class="fa fa-check"></i><b>14.5.3</b> Specify search terms of genomes</a></li>
<li class="chapter" data-level="14.5.4" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#load-gene-search-terms-of-interest"><i class="fa fa-check"></i><b>14.5.4</b> Load Gene Search Terms of Interest</a></li>
<li class="chapter" data-level="14.5.5" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#extract-sequences-with-annotationbust"><i class="fa fa-check"></i><b>14.5.5</b> Extract sequences with AnnotationBust</a></li>
</ul></li>
<li class="chapter" data-level="14.6" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#exercise-ii-1"><i class="fa fa-check"></i><b>14.6</b> Exercise II</a></li>
<li class="chapter" data-level="14.7" data-path="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html"><a href="systematic-retrieval-of-orthologous-dna-sequences-form-genbank.html#combine-gene-information-from-the-two-analyses"><i class="fa fa-check"></i><b>14.7</b> Combine gene information from the two analyses</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="sequence-alignment.html"><a href="sequence-alignment.html"><i class="fa fa-check"></i><b>15</b> Sequence alignment</a>
<ul>
<li class="chapter" data-level="15.1" data-path="sequence-alignment.html"><a href="sequence-alignment.html#load-libraries"><i class="fa fa-check"></i><b>15.1</b> Load libraries</a></li>
<li class="chapter" data-level="15.2" data-path="sequence-alignment.html"><a href="sequence-alignment.html#perform-alignments"><i class="fa fa-check"></i><b>15.2</b> Perform Alignments</a></li>
<li class="chapter" data-level="15.3" data-path="sequence-alignment.html"><a href="sequence-alignment.html#processing-multiple-alignments"><i class="fa fa-check"></i><b>15.3</b> Processing Multiple Alignments</a></li>
<li class="chapter" data-level="15.4" data-path="sequence-alignment.html"><a href="sequence-alignment.html#build-your-first-nj-tree"><i class="fa fa-check"></i><b>15.4</b> Build your first NJ tree</a></li>
<li class="chapter" data-level="15.5" data-path="sequence-alignment.html"><a href="sequence-alignment.html#exercise"><i class="fa fa-check"></i><b>15.5</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="maximum-likelihood-model-tests-tree-and-branch-tests.html"><a href="maximum-likelihood-model-tests-tree-and-branch-tests.html"><i class="fa fa-check"></i><b>16</b> Maximum likelihood, model tests, tree and branch tests</a>
<ul>
<li class="chapter" data-level="16.1" data-path="maximum-likelihood-model-tests-tree-and-branch-tests.html"><a href="maximum-likelihood-model-tests-tree-and-branch-tests.html#learning-goals-1"><i class="fa fa-check"></i><b>16.1</b> Learning Goals</a></li>
<li class="chapter" data-level="16.2" data-path="maximum-likelihood-model-tests-tree-and-branch-tests.html"><a href="maximum-likelihood-model-tests-tree-and-branch-tests.html#iqtree"><i class="fa fa-check"></i><b>16.2</b> IQtree</a>
<ul>
<li class="chapter" data-level="16.2.1" data-path="maximum-likelihood-model-tests-tree-and-branch-tests.html"><a href="maximum-likelihood-model-tests-tree-and-branch-tests.html#version"><i class="fa fa-check"></i><b>16.2.1</b> Version</a></li>
<li class="chapter" data-level="16.2.2" data-path="maximum-likelihood-model-tests-tree-and-branch-tests.html"><a href="maximum-likelihood-model-tests-tree-and-branch-tests.html#find-the-best-explanatory-sequence-evolution-model"><i class="fa fa-check"></i><b>16.2.2</b> find the best explanatory sequence evolution model</a></li>
<li class="chapter" data-level="16.2.3" data-path="maximum-likelihood-model-tests-tree-and-branch-tests.html"><a href="maximum-likelihood-model-tests-tree-and-branch-tests.html#using-the-best-sequence-evolution-model"><i class="fa fa-check"></i><b>16.2.3</b> using the best sequence evolution model</a></li>
<li class="chapter" data-level="16.2.4" data-path="maximum-likelihood-model-tests-tree-and-branch-tests.html"><a href="maximum-likelihood-model-tests-tree-and-branch-tests.html#different-partitions-for-the-data-for-multiple-genes-or-different-codons"><i class="fa fa-check"></i><b>16.2.4</b> Different partitions for the data for multiple genes, or different codons</a></li>
</ul></li>
<li class="chapter" data-level="16.3" data-path="maximum-likelihood-model-tests-tree-and-branch-tests.html"><a href="maximum-likelihood-model-tests-tree-and-branch-tests.html#run-an-ml-tree-with-multiple-gene-files-in-a-folder"><i class="fa fa-check"></i><b>16.3</b> Run an ML tree with multiple gene files in a folder</a></li>
<li class="chapter" data-level="16.4" data-path="maximum-likelihood-model-tests-tree-and-branch-tests.html"><a href="maximum-likelihood-model-tests-tree-and-branch-tests.html#gene-concordance-factor"><i class="fa fa-check"></i><b>16.4</b> Gene concordance factor</a></li>
<li class="chapter" data-level="16.5" data-path="maximum-likelihood-model-tests-tree-and-branch-tests.html"><a href="maximum-likelihood-model-tests-tree-and-branch-tests.html#tree-topology-tests"><i class="fa fa-check"></i><b>16.5</b> Tree topology tests</a></li>
<li class="chapter" data-level="16.6" data-path="maximum-likelihood-model-tests-tree-and-branch-tests.html"><a href="maximum-likelihood-model-tests-tree-and-branch-tests.html#tree-branches-tests"><i class="fa fa-check"></i><b>16.6</b> Tree branches tests</a></li>
<li class="chapter" data-level="16.7" data-path="maximum-likelihood-model-tests-tree-and-branch-tests.html"><a href="maximum-likelihood-model-tests-tree-and-branch-tests.html#assignments"><i class="fa fa-check"></i><b>16.7</b> Assignments</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="molecular-clock-and-phylogenetic-dating.html"><a href="molecular-clock-and-phylogenetic-dating.html"><i class="fa fa-check"></i><b>17</b> Molecular Clock and Phylogenetic Dating</a>
<ul>
<li class="chapter" data-level="17.1" data-path="molecular-clock-and-phylogenetic-dating.html"><a href="molecular-clock-and-phylogenetic-dating.html#inferring-time-tree-with-tip-dates"><i class="fa fa-check"></i><b>17.1</b> Inferring time tree with tip dates</a></li>
<li class="chapter" data-level="17.2" data-path="molecular-clock-and-phylogenetic-dating.html"><a href="molecular-clock-and-phylogenetic-dating.html#calibrating-tree-using-ancestral-dates"><i class="fa fa-check"></i><b>17.2</b> Calibrating tree using ancestral dates</a>
<ul>
<li class="chapter" data-level="17.2.1" data-path="molecular-clock-and-phylogenetic-dating.html"><a href="molecular-clock-and-phylogenetic-dating.html#search-for-an-ml-tree"><i class="fa fa-check"></i><b>17.2.1</b> search for an ML tree</a></li>
<li class="chapter" data-level="17.2.2" data-path="molecular-clock-and-phylogenetic-dating.html"><a href="molecular-clock-and-phylogenetic-dating.html#root-calibration"><i class="fa fa-check"></i><b>17.2.2</b> Root calibration</a></li>
<li class="chapter" data-level="17.2.3" data-path="molecular-clock-and-phylogenetic-dating.html"><a href="molecular-clock-and-phylogenetic-dating.html#internal-nodes"><i class="fa fa-check"></i><b>17.2.3</b> Internal nodes</a></li>
<li class="chapter" data-level="17.2.4" data-path="molecular-clock-and-phylogenetic-dating.html"><a href="molecular-clock-and-phylogenetic-dating.html#confidence-interval"><i class="fa fa-check"></i><b>17.2.4</b> Confidence interval</a></li>
</ul></li>
<li class="chapter" data-level="17.3" data-path="molecular-clock-and-phylogenetic-dating.html"><a href="molecular-clock-and-phylogenetic-dating.html#exercise-4"><i class="fa fa-check"></i><b>17.3</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="character-evolution-and-ancestral-state-reconstruction.html"><a href="character-evolution-and-ancestral-state-reconstruction.html"><i class="fa fa-check"></i><b>18</b> Character evolution and Ancestral state reconstruction</a>
<ul>
<li class="chapter" data-level="18.1" data-path="character-evolution-and-ancestral-state-reconstruction.html"><a href="character-evolution-and-ancestral-state-reconstruction.html#continuous-characters"><i class="fa fa-check"></i><b>18.1</b> Continuous characters</a>
<ul>
<li class="chapter" data-level="18.1.1" data-path="character-evolution-and-ancestral-state-reconstruction.html"><a href="character-evolution-and-ancestral-state-reconstruction.html#load-libraries-1"><i class="fa fa-check"></i><b>18.1.1</b> load libraries</a></li>
<li class="chapter" data-level="18.1.2" data-path="character-evolution-and-ancestral-state-reconstruction.html"><a href="character-evolution-and-ancestral-state-reconstruction.html#tree-file-and-the-character-file"><i class="fa fa-check"></i><b>18.1.2</b> Tree file and the character file</a></li>
<li class="chapter" data-level="18.1.3" data-path="character-evolution-and-ancestral-state-reconstruction.html"><a href="character-evolution-and-ancestral-state-reconstruction.html#visualization-of-the-reconstructed-nodes"><i class="fa fa-check"></i><b>18.1.3</b> Visualization of the reconstructed nodes</a></li>
<li class="chapter" data-level="18.1.4" data-path="character-evolution-and-ancestral-state-reconstruction.html"><a href="character-evolution-and-ancestral-state-reconstruction.html#visualization-of-phenotypic-evolution"><i class="fa fa-check"></i><b>18.1.4</b> Visualization of phenotypic evolution</a></li>
</ul></li>
<li class="chapter" data-level="18.2" data-path="character-evolution-and-ancestral-state-reconstruction.html"><a href="character-evolution-and-ancestral-state-reconstruction.html#brownian-motion-vs-ornstein-uhlenbeck-process"><i class="fa fa-check"></i><b>18.2</b> Brownian Motion vs Ornstein-Uhlenbeck Process</a></li>
<li class="chapter" data-level="18.3" data-path="character-evolution-and-ancestral-state-reconstruction.html"><a href="character-evolution-and-ancestral-state-reconstruction.html#brownian-motion"><i class="fa fa-check"></i><b>18.3</b> Brownian Motion</a>
<ul>
<li class="chapter" data-level="18.3.1" data-path="character-evolution-and-ancestral-state-reconstruction.html"><a href="character-evolution-and-ancestral-state-reconstruction.html#ornsteinuhlenbeck-process"><i class="fa fa-check"></i><b>18.3.1</b> Ornstein–Uhlenbeck process</a></li>
<li class="chapter" data-level="18.3.2" data-path="character-evolution-and-ancestral-state-reconstruction.html"><a href="character-evolution-and-ancestral-state-reconstruction.html#incorporate-prior-knowledge-using-bayesian-estimates"><i class="fa fa-check"></i><b>18.3.2</b> Incorporate prior knowledge using Bayesian estimates</a></li>
</ul></li>
<li class="chapter" data-level="18.4" data-path="character-evolution-and-ancestral-state-reconstruction.html"><a href="character-evolution-and-ancestral-state-reconstruction.html#discrete-characters"><i class="fa fa-check"></i><b>18.4</b> Discrete characters</a></li>
<li class="chapter" data-level="18.5" data-path="character-evolution-and-ancestral-state-reconstruction.html"><a href="character-evolution-and-ancestral-state-reconstruction.html#exercises-1"><i class="fa fa-check"></i><b>18.5</b> Exercises:</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="model-selection-of-character-evolution.html"><a href="model-selection-of-character-evolution.html"><i class="fa fa-check"></i><b>19</b> Model Selection of Character Evolution</a>
<ul>
<li class="chapter" data-level="19.1" data-path="model-selection-of-character-evolution.html"><a href="model-selection-of-character-evolution.html#load-libraries-2"><i class="fa fa-check"></i><b>19.1</b> Load libraries</a></li>
<li class="chapter" data-level="19.2" data-path="model-selection-of-character-evolution.html"><a href="model-selection-of-character-evolution.html#discrete-traits-model-fitting"><i class="fa fa-check"></i><b>19.2</b> Discrete traits model fitting</a>
<ul>
<li class="chapter" data-level="19.2.1" data-path="model-selection-of-character-evolution.html"><a href="model-selection-of-character-evolution.html#choosing-a-character-model"><i class="fa fa-check"></i><b>19.2.1</b> Choosing a character model</a></li>
<li class="chapter" data-level="19.2.2" data-path="model-selection-of-character-evolution.html"><a href="model-selection-of-character-evolution.html#marginal-ancestral-state-reconstruction"><i class="fa fa-check"></i><b>19.2.2</b> Marginal ancestral state reconstruction</a></li>
<li class="chapter" data-level="19.2.3" data-path="model-selection-of-character-evolution.html"><a href="model-selection-of-character-evolution.html#more-than-two-states"><i class="fa fa-check"></i><b>19.2.3</b> More than two states</a></li>
</ul></li>
<li class="chapter" data-level="19.3" data-path="model-selection-of-character-evolution.html"><a href="model-selection-of-character-evolution.html#exercise-i-2"><i class="fa fa-check"></i><b>19.3</b> Exercise I</a></li>
<li class="chapter" data-level="19.4" data-path="model-selection-of-character-evolution.html"><a href="model-selection-of-character-evolution.html#continuous-traits-model-fitting"><i class="fa fa-check"></i><b>19.4</b> Continuous traits model fitting</a>
<ul>
<li class="chapter" data-level="19.4.1" data-path="model-selection-of-character-evolution.html"><a href="model-selection-of-character-evolution.html#model-selection"><i class="fa fa-check"></i><b>19.4.1</b> Model selection</a></li>
<li class="chapter" data-level="19.4.2" data-path="model-selection-of-character-evolution.html"><a href="model-selection-of-character-evolution.html#ancestral-states"><i class="fa fa-check"></i><b>19.4.2</b> Ancestral states</a></li>
</ul></li>
<li class="chapter" data-level="19.5" data-path="model-selection-of-character-evolution.html"><a href="model-selection-of-character-evolution.html#exercise-ii-2"><i class="fa fa-check"></i><b>19.5</b> Exercise II</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">BIO595 The Tree of Life</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="unix" class="section level1 hasAnchor" number="2">
<h1><span class="header-section-number">Chapter 2</span> Unix<a href="unix.html#unix" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Adapted from <em>Computing Skills for Biologists</em>, Chapter 1</p>
<div id="what-is-unix-and-why-do-we-use-it" class="section level2 hasAnchor" number="2.1">
<h2><span class="header-section-number">2.1</span> What is Unix and why do we use it?<a href="unix.html#what-is-unix-and-why-do-we-use-it" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Unix is an operating system, which is the software that lets you interface with the computer. Apple’s OS, Linux are among the popular variants of Unix used today. In this lab, we’ll focus the use of Unix shell, the interface that is used to communicate with the core of the operating system. Using the shell is quite a different experience than the usual graphical user interfaces (GUIs). Within the shell, we directly type commands to ask certain tasks to be done, instead of interacting with buttons and windows.</p>
<p>Learning basic command lines is very important for biologists, especially researchers because many small programs are available or can be developed to perform simple tasks, scaling up the analyses on batches of files, and integrate different programs into a analyzing pipeline. Such scripts make your work 100% reproducible, easy to debug, and avoid random mistakes introduced because you clicked wrong buttons on a GUI interface.</p>
<p>Lastly, biological research replies heavily on large text files of genetic and/or character sequences. Unix is designed around text viewing and editing. Phylogenetic analyses usually involves large amount of calculation, and requires high-performance computing platforms, which are unix-based. Mastering the skills of command-line operations will allow you to work with large files effortlessly.</p>
</div>
<div id="installation" class="section level2 hasAnchor" number="2.2">
<h2><span class="header-section-number">2.2</span> Installation<a href="unix.html#installation" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="mac-os-and-linux" class="section level3 hasAnchor" number="2.2.1">
<h3><span class="header-section-number">2.2.1</span> Mac OS and Linux<a href="unix.html#mac-os-and-linux" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>These systems are members of unix-based operating systems. No installation is required for unix support.</p>
</div>
<div id="windows" class="section level3 hasAnchor" number="2.2.2">
<h3><span class="header-section-number">2.2.2</span> Windows<a href="unix.html#windows" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>To install <code>Git</code> and <code>Git Bash</code> in Windows, first download the file at [<a href="https://git-scm.com/download/win" class="uri">https://git-scm.com/download/win</a>], then install it. We recommend to use all the default settings for the installation, with the following two exceptions: In <em>Adjusting Your PATH Environment</em>, select <em>Use Git from Git Bash Only</em>. In the prompt <em>Configuring the Line Ending Conversions</em>, choose <em>Checkout as-is, commit as-is</em>. You can then access all <code>Git</code> functionalities from the application <code>Git Bash</code>.</p>
</div>
<div id="use-scholar-remote-desktop-during-the-class" class="section level3 hasAnchor" number="2.2.3">
<h3><span class="header-section-number">2.2.3</span> Use Scholar Remote Desktop During the Class<a href="unix.html#use-scholar-remote-desktop-during-the-class" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li>Go the website: [<a href="https://www.rcac.purdue.edu/compute/scholar" class="uri">https://www.rcac.purdue.edu/compute/scholar</a>]</li>
<li>Click Login on the upper right corner.</li>
<li>Click “Remote Desktop/Launch” to launch a remote linux desktop. Use your 2Factor BoilerKey to login.</li>
</ol>
</div>
</div>
<div id="operation" class="section level2 hasAnchor" number="2.3">
<h2><span class="header-section-number">2.3</span> Operation<a href="unix.html#operation" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="starting-the-shell" class="section level3 hasAnchor" number="2.3.1">
<h3><span class="header-section-number">2.3.1</span> Starting the Shell<a href="unix.html#starting-the-shell" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>In Linux, click <em>Application Finder</em> and type <em>Terminal</em>. In OS X, you can find the <em>Terminal</em> in <em>Applications/Utilities</em>. Alternatively, you can type <em>Terminal</em> in <em>Spotlight</em>. In Windows, you can launch <em>Git Bash</em> to use the shell.</p>
</div>
<div id="directoryfolder-sturcture" class="section level3 hasAnchor" number="2.3.2">
<h3><span class="header-section-number">2.3.2</span> Directory/folder sturcture<a href="unix.html#directoryfolder-sturcture" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="figure">
<img src="pics/directory-tree.png" alt="" />
<p class="caption">Fig 2.1, A typical unix directory</p>
</div>
<blockquote>
<p>Q: What’s the full path of <code>install.md</code>?</p>
</blockquote>
<div class="figure">
<img src="pics/windows-directory.png" alt="" />
<p class="caption">Fig 2.2, Compared with Windows directory</p>
</div>
<p><code>/bin</code> contains several basic programs</p>
<p><code>/dev</code> contains the files connecting to devices</p>
<p><code>/etc</code> contains configuration files</p>
<p><code>/tmp</code> contains temporary files</p>
<p>When a new shell starts, it always opens in your home directory. The tilde (~) symbol is shorthand for the home directory in unix. The exact path to the home directory varies among different operating systems. To see its actual location, type:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb1-1"><a href="unix.html#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="va">$HOME</span></span>
<span id="cb1-2"><a href="unix.html#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="co"># copy all the lab1 material into the current folder</span></span>
<span id="cb1-3"><a href="unix.html#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="co"># meaning of this command will be explained later</span></span>
<span id="cb1-4"><a href="unix.html#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">cp</span> <span class="at">-r</span> /class/heqixin/data/labs/lab1_unix ./</span></code></pre></div>
<p>In OS X, you should see <strong>/Users/&lt;username&gt;</strong>. In linux, you should see <strong>/home/&lt;username&gt;</strong>. In windows, it would be <strong>/c/Users/&lt;username&gt;</strong>. When specifying a path, if it doesn’t start with “/”, then it refers to subfolders or files under the current directory. If the path starts with “/”, it denotes directories from the topmost “root” directory.</p>
<p>Next, let’s explore the bsic commands to navigate the hierarchical directory system.</p>
<p><code>pwd</code> <strong>P</strong>rint the path of the current <strong>w</strong>orking <strong>d</strong>irectory. This command prints your current location of the shell.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb2-1"><a href="unix.html#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="bu">pwd</span></span></code></pre></div>
<p>Since we didn’t change directory, the <code>pwd</code> command shows that the shell currently operates in our home directory. We can <strong>c</strong>hange <strong>d</strong>irectories by the command <code>cd</code>.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb3-1"><a href="unix.html#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co">#assuming you saved the folder lab1_data in your home directory</span></span>
<span id="cb3-2"><a href="unix.html#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="co">#navigate to our data folder</span></span>
<span id="cb3-3"><a href="unix.html#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> ~/lab1_unix</span>
<span id="cb3-4"><a href="unix.html#cb3-4" aria-hidden="true" tabindex="-1"></a><span class="co">#now the working directory is changed to lab1_data folder</span></span>
<span id="cb3-5"><a href="unix.html#cb3-5" aria-hidden="true" tabindex="-1"></a><span class="bu">pwd</span></span></code></pre></div>
<p><code>cd</code> command requires one argument (i.e., the path to the directory you want to change into). Here are some commonly used commands for navigation:</p>
<p><code>cd ..</code> Move one directory up.</p>
<p><code>cd /</code> Move to the root directory.</p>
<p><code>cd ~</code> Move to your home directory.</p>
<p><code>cd -</code> go back to the directory you visited previously.</p>
<p><code>ls</code> command <strong>L</strong>i<strong>s</strong>ts the files and subdirectories in the current directory.</p>
</div>
<div id="absolute-and-relative-directory" class="section level3 hasAnchor" number="2.3.3">
<h3><span class="header-section-number">2.3.3</span> Absolute and Relative directory<a href="unix.html#absolute-and-relative-directory" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>One can navigate through the directory hierarchy by providing either the <em>absolute</em> path (i.e., starts with “/”) or a <em>relative</em> path (i.e., utilize “..”, “-”, “./”, or directly type the subdirectory name in the current folder).</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb4-1"><a href="unix.html#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># absolute path</span></span>
<span id="cb4-2"><a href="unix.html#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> ~/lab1_unix</span>
<span id="cb4-3"><a href="unix.html#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="co"># see all the subdirectories and files in this directory</span></span>
<span id="cb4-4"><a href="unix.html#cb4-4" aria-hidden="true" tabindex="-1"></a><span class="fu">ls</span></span></code></pre></div>
<blockquote>
<p>Tip1: The <strong>Tab</strong> key provides options or autocomplete when you only remember part of the file/folder name. For example, after typing <code>cd u</code>, and press the <strong>Tab</strong> key, if the current directory only has one file/folder that starts with “u” (e.g., unix), then the command will autocomplete the command to be <code>cd unix</code>. If there are multiple ones that starts with “u”, the list of options will be shown, and you can keep typing the following letter until there’s only one unique option and press <strong>Tab</strong> again. This is very helpful when folder or file names are long or confusing.</p>
</blockquote>
<div class="sourceCode" id="cb5"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb5-1"><a href="unix.html#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co"># relatvie path</span></span>
<span id="cb5-2"><a href="unix.html#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="co">#cd lab1_unix</span></span>
<span id="cb5-3"><a href="unix.html#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> ./data</span>
<span id="cb5-4"><a href="unix.html#cb5-4" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> ../sandbox</span>
<span id="cb5-5"><a href="unix.html#cb5-5" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> <span class="at">-</span></span>
<span id="cb5-6"><a href="unix.html#cb5-6" aria-hidden="true" tabindex="-1"></a><span class="co"># ./ or . refers to the current directory</span></span>
<span id="cb5-7"><a href="unix.html#cb5-7" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> ./</span>
<span id="cb5-8"><a href="unix.html#cb5-8" aria-hidden="true" tabindex="-1"></a><span class="bu">pwd</span></span>
<span id="cb5-9"><a href="unix.html#cb5-9" aria-hidden="true" tabindex="-1"></a><span class="fu">ls</span></span>
<span id="cb5-10"><a href="unix.html#cb5-10" aria-hidden="true" tabindex="-1"></a><span class="co"># try the Tab key to test autocomplete.</span></span>
<span id="cb5-11"><a href="unix.html#cb5-11" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> ../../</span></code></pre></div>
<blockquote>
<p>Tip2: space in unix command indicates separation of arguments. Thus, spaces within a directory or file name becomes confusing. Try avoid using spaces in naming directories or files. When existing files/directories contain spaces, you can refer to it with quotation marks or a backslash (\) before a space:</p>
</blockquote>
<div class="sourceCode" id="cb6"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb6-1"><a href="unix.html#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co"># if we would like to open a directory</span></span>
<span id="cb6-2"><a href="unix.html#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="co"># called &quot;unix names should not contain spaces&quot;</span></span>
<span id="cb6-3"><a href="unix.html#cb6-3" aria-hidden="true" tabindex="-1"></a><span class="co"># this command does not work</span></span>
<span id="cb6-4"><a href="unix.html#cb6-4" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> lab1_unix/sandbox</span>
<span id="cb6-5"><a href="unix.html#cb6-5" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> unix names should not contain spaces</span>
<span id="cb6-6"><a href="unix.html#cb6-6" aria-hidden="true" tabindex="-1"></a><span class="co"># works</span></span>
<span id="cb6-7"><a href="unix.html#cb6-7" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> <span class="st">&quot;unix names should not contain spaces&quot;</span></span>
<span id="cb6-8"><a href="unix.html#cb6-8" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> ..</span>
<span id="cb6-9"><a href="unix.html#cb6-9" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> unix<span class="dt">\ </span>names<span class="dt">\ </span>should<span class="dt">\ </span>not<span class="dt">\ </span>contain<span class="dt">\ </span>spaces</span>
<span id="cb6-10"><a href="unix.html#cb6-10" aria-hidden="true" tabindex="-1"></a><span class="bu">pwd</span></span>
<span id="cb6-11"><a href="unix.html#cb6-11" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> ..</span></code></pre></div>
</div>
</div>
<div id="handling-directories-and-files" class="section level2 hasAnchor" number="2.4">
<h2><span class="header-section-number">2.4</span> Handling directories and files<a href="unix.html#handling-directories-and-files" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><code>cp &lt;original path to file/directory&gt; &lt;destination&gt;</code> <strong>C</strong>o<strong>p</strong>y a file or directory.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb7-1"><a href="unix.html#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co"># copy a file from data folder to sandbox</span></span>
<span id="cb7-2"><a href="unix.html#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> lab1_unix/sandbox</span>
<span id="cb7-3"><a href="unix.html#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="fu">cp</span> ../data/Marra2014_about.txt ./underscore_is_preferred</span>
<span id="cb7-4"><a href="unix.html#cb7-4" aria-hidden="true" tabindex="-1"></a><span class="co"># you can also rename the newly copied file during copying</span></span>
<span id="cb7-5"><a href="unix.html#cb7-5" aria-hidden="true" tabindex="-1"></a><span class="fu">cp</span> ../data/Marra2014_about.txt ./underscore_is_preferred/Marra2014_about_new.txt</span>
<span id="cb7-6"><a href="unix.html#cb7-6" aria-hidden="true" tabindex="-1"></a><span class="co"># if you would like to a whole directory, then add &quot;-r&quot; option</span></span>
<span id="cb7-7"><a href="unix.html#cb7-7" aria-hidden="true" tabindex="-1"></a><span class="fu">cp</span> <span class="at">-r</span> copyThisFolder ./underscore_is_preferred/</span></code></pre></div>
<p><code>mv &lt;original path to file/directory&gt; &lt;destination&gt;</code> can be used to move or rename a file or directory.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb8-1"><a href="unix.html#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co"># move a file to the fllder &quot;copyThisFolder&quot;</span></span>
<span id="cb8-2"><a href="unix.html#cb8-2" aria-hidden="true" tabindex="-1"></a><span class="fu">mv</span> underscore_is_preferred/Marra2014_about.txt copyThisFolder</span>
<span id="cb8-3"><a href="unix.html#cb8-3" aria-hidden="true" tabindex="-1"></a><span class="co"># you can also rename a file at the same location</span></span>
<span id="cb8-4"><a href="unix.html#cb8-4" aria-hidden="true" tabindex="-1"></a><span class="fu">mv</span> underscore_is_preferred/Marra2014_about.txt underscore_is_preferred/Marra2014_about_change_name.txt</span>
<span id="cb8-5"><a href="unix.html#cb8-5" aria-hidden="true" tabindex="-1"></a><span class="co"># move a whole directory</span></span>
<span id="cb8-6"><a href="unix.html#cb8-6" aria-hidden="true" tabindex="-1"></a><span class="fu">mv</span> underscore_is_preferred/copyThisFolder underscore_is_preferred/newLocation</span></code></pre></div>
<p><code>rm file</code> <strong>R</strong>e<strong>m</strong>oves a file. <code>rm -r</code> deletes all the contents of a directory (including all the files and the subdirectories). Use this with caution!!! or in conjunction with <code>-i</code> option, which prompts the user to confirm the action. Option <code>-f</code> will erase a write-protected file without a prompt. This is different from using trash bin, as you cannot recover the file!</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb9-1"><a href="unix.html#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co"># touch updates the date of last access of a file. </span></span>
<span id="cb9-2"><a href="unix.html#cb9-2" aria-hidden="true" tabindex="-1"></a><span class="co">#If the file doesn&#39;t exit, it creates an empty new one.</span></span>
<span id="cb9-3"><a href="unix.html#cb9-3" aria-hidden="true" tabindex="-1"></a><span class="fu">touch</span> new_file.txt</span>
<span id="cb9-4"><a href="unix.html#cb9-4" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span> <span class="at">-i</span> new_file.txt</span>
<span id="cb9-5"><a href="unix.html#cb9-5" aria-hidden="true" tabindex="-1"></a><span class="co"># type y or n to confirm or cancel the deletion.</span></span></code></pre></div>
<p><code>mkdir</code> <strong>M</strong>a<strong>k</strong>e a new <strong>dir</strong>ectory.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb10-1"><a href="unix.html#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> test123</span>
<span id="cb10-2"><a href="unix.html#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="co"># create multiple nested directories, use the option -p</span></span>
<span id="cb10-3"><a href="unix.html#cb10-3" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> d1/d2/d3</span>
<span id="cb10-4"><a href="unix.html#cb10-4" aria-hidden="true" tabindex="-1"></a><span class="co"># remove d1 and its subdirectories recursively</span></span>
<span id="cb10-5"><a href="unix.html#cb10-5" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span> <span class="at">-r</span> d1</span></code></pre></div>
</div>
<div id="excercises-i" class="section level2 hasAnchor" number="2.5">
<h2><span class="header-section-number">2.5</span> Excercises I<a href="unix.html#excercises-i" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ol style="list-style-type: decimal">
<li>If you are currently at the home directory, write down the command to go to <code>data</code> directory (one with absolute path, one with relative path)</li>
<li>write down the command to go to <em>miRNA</em> folder and return back.</li>
</ol>
</div>
<div id="viewing-and-processing-text-files" class="section level2 hasAnchor" number="2.6">
<h2><span class="header-section-number">2.6</span> Viewing and processing text files<a href="unix.html#viewing-and-processing-text-files" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>It is very easy to view, combine, cut, search and substitute text files using command lines.</p>
<p><code>less &lt;filename&gt;</code> prints a file progressively on the screen. You can take a peak of a very large file without opening the entire file (which is usually really large if it contains genomic information!) While in <code>less</code> environment, press <em>Ctrl+F</em> or <em>Ctrl+B</em> for next page or previous page, press <em>h</em> to see more options, press <em>q</em> to exit <code>less</code>.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb11-1"><a href="unix.html#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="co"># assume you are in lab1_unix/data folder</span></span>
<span id="cb11-2"><a href="unix.html#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="fu">less</span> Marra2014_data.fasta</span>
<span id="cb11-3"><a href="unix.html#cb11-3" aria-hidden="true" tabindex="-1"></a><span class="co"># you&#39;ll see</span></span>
<span id="cb11-4"><a href="unix.html#cb11-4" aria-hidden="true" tabindex="-1"></a><span class="co"># &gt;contig00001  length=527  numreads=2  gene=isogroup00001  status=it_thresh</span></span>
<span id="cb11-5"><a href="unix.html#cb11-5" aria-hidden="true" tabindex="-1"></a><span class="co"># ATCCTAGCTACTCTGGAGACTGAGGATTGAAGTTCAAAGTCAGCTCAAGCAAGAGATTTG</span></span>
<span id="cb11-6"><a href="unix.html#cb11-6" aria-hidden="true" tabindex="-1"></a><span class="co"># .....</span></span></code></pre></div>
<p><code>cat file1 file2 file3</code> Concatenate and print files. Supply with one file, you’ll see the printout of the whole file. Providing several files will concatenate them all and print to the screen.</p>
<p><code>cat file1 file2 file3 &gt; newfile</code> This is a useful command to concatenate files and save them to a new file.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb12-1"><a href="unix.html#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span> Marra2014_about.txt</span>
<span id="cb12-2"><a href="unix.html#cb12-2" aria-hidden="true" tabindex="-1"></a><span class="co"># concatenate files and save as new one</span></span>
<span id="cb12-3"><a href="unix.html#cb12-3" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span> Marra2014_about.txt miRNA/miRNA_about.txt <span class="op">&gt;</span> new_about.txt</span></code></pre></div>
<p><code>wc</code> word count of a file.</p>
<p><code>wc -l</code> line count of a file.</p>
<p><code>head -n &lt;number&gt; &lt;filename&gt;</code> This command prints the first <number> of lines of a file, specified by the option <code>-n</code>.</p>
<p><code>tail -n &lt;number&gt; &lt;filename&gt;</code> This command prints the tail (i.e., last few lines) of a file.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb13-1"><a href="unix.html#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> lab1_unix/data</span>
<span id="cb13-2"><a href="unix.html#cb13-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span> <span class="at">-n</span> 3 Marra2014_data.fasta</span>
<span id="cb13-3"><a href="unix.html#cb13-3" aria-hidden="true" tabindex="-1"></a><span class="fu">tail</span> <span class="at">-n</span> 2 Gesquiere2011_data.csv</span>
<span id="cb13-4"><a href="unix.html#cb13-4" aria-hidden="true" tabindex="-1"></a><span class="co"># sometimes we want to skip the first few lines of headers and see the contents</span></span>
<span id="cb13-5"><a href="unix.html#cb13-5" aria-hidden="true" tabindex="-1"></a><span class="co"># This command displays from line 2 onward.</span></span>
<span id="cb13-6"><a href="unix.html#cb13-6" aria-hidden="true" tabindex="-1"></a><span class="fu">tail</span> <span class="at">-n</span> +2 Gesquiere2011_about.txt</span></code></pre></div>
</div>
<div id="redirecting-printouts-and-piping" class="section level2 hasAnchor" number="2.7">
<h2><span class="header-section-number">2.7</span> Redirecting printouts and piping<a href="unix.html#redirecting-printouts-and-piping" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>So far, we have all the outputs from the command shown on screen. In fact, any printout to the screen can be redirected to be saved as a file</p>
<p>If you would like the printout to be saved as a new file, then</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb14-1"><a href="unix.html#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="co"># save the directory contents to a file</span></span>
<span id="cb14-2"><a href="unix.html#cb14-2" aria-hidden="true" tabindex="-1"></a><span class="co"># if the file exists, then the new content will replace the original content</span></span>
<span id="cb14-3"><a href="unix.html#cb14-3" aria-hidden="true" tabindex="-1"></a><span class="fu">ls</span> <span class="op">&gt;</span> directoryList.txt</span>
<span id="cb14-4"><a href="unix.html#cb14-4" aria-hidden="true" tabindex="-1"></a><span class="co"># If we only want to append new lines to the file, then we should use</span></span>
<span id="cb14-5"><a href="unix.html#cb14-5" aria-hidden="true" tabindex="-1"></a><span class="fu">ls</span> <span class="op">&gt;&gt;</span> directoryList.txt</span>
<span id="cb14-6"><a href="unix.html#cb14-6" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">&quot;end of the file&quot;</span> <span class="op">&gt;&gt;</span> directoryList.txt</span>
<span id="cb14-7"><a href="unix.html#cb14-7" aria-hidden="true" tabindex="-1"></a><span class="co"># To see how the file looks</span></span>
<span id="cb14-8"><a href="unix.html#cb14-8" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span> directoryList.txt</span>
<span id="cb14-9"><a href="unix.html#cb14-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Sometimes, the output of the first step is the input of the second step</span></span>
<span id="cb14-10"><a href="unix.html#cb14-10" aria-hidden="true" tabindex="-1"></a><span class="co"># for example, we would like to know how many files or subdirectories </span></span>
<span id="cb14-11"><a href="unix.html#cb14-11" aria-hidden="true" tabindex="-1"></a><span class="co"># there are in the current directory</span></span>
<span id="cb14-12"><a href="unix.html#cb14-12" aria-hidden="true" tabindex="-1"></a><span class="co"># We can skip the creation of an intermediate file by using &quot;|&quot;</span></span>
<span id="cb14-13"><a href="unix.html#cb14-13" aria-hidden="true" tabindex="-1"></a><span class="fu">ls</span> <span class="kw">|</span> <span class="fu">wc</span> <span class="at">-l</span></span>
<span id="cb14-14"><a href="unix.html#cb14-14" aria-hidden="true" tabindex="-1"></a><span class="co"># this command will direct the directory listing output </span></span>
<span id="cb14-15"><a href="unix.html#cb14-15" aria-hidden="true" tabindex="-1"></a><span class="co"># to the &quot;wc&quot; command input, and count the lines of the output. </span></span></code></pre></div>
</div>
<div id="exercises-ii" class="section level2 hasAnchor" number="2.8">
<h2><span class="header-section-number">2.8</span> Exercises II<a href="unix.html#exercises-ii" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ol style="list-style-type: decimal">
<li>command to create an empty file “createEmpty.txt” under the directory of <em>sandbox</em>.</li>
<li>list the contents in data/miRNA, and count the number of files.</li>
<li>list line counts of all the files under data/Saavedra2013</li>
</ol>
</div>
<div id="useful-wildcards" class="section level2 hasAnchor" number="2.9">
<h2><span class="header-section-number">2.9</span> Useful wildcards<a href="unix.html#useful-wildcards" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Wildcards are special symbols that work as placeholders for one or more
characters.</p>
<p><code>*</code> stands for any characters, length from 0 to more zero or more, with the exception of a leading dot. Unix uses a leading dot for hidden files, so this means that hidden files are ignored in a search using this wildcard (show hidden files using ls -a).</p>
<p><code>?</code> is a placeholder for any single character, again with the exception of a leading dot.</p>
<p><code>[]</code> The square brackets match any one of characters inside the brackets. For example, <em>[xyz]</em> will match any of “x”, “y”, or “z”. A range of characters in the alphabet can be matched using the hyphen. For example, <em>[a-z]</em> will match all the lower case alphabets, <em>[a-zA-Z]</em> will match all the alphabets, <em>[0-9]</em> will match any single numbers.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb15-1"><a href="unix.html#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="co"># change into the directory</span></span>
<span id="cb15-2"><a href="unix.html#cb15-2" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> lab1_unix/data/miRNA</span>
<span id="cb15-3"><a href="unix.html#cb15-3" aria-hidden="true" tabindex="-1"></a><span class="co"># count the numbers of lines in all the .fasta files</span></span>
<span id="cb15-4"><a href="unix.html#cb15-4" aria-hidden="true" tabindex="-1"></a><span class="fu">wc</span> <span class="at">-l</span> <span class="pp">*</span>.fasta</span>
<span id="cb15-5"><a href="unix.html#cb15-5" aria-hidden="true" tabindex="-1"></a><span class="co"># print the first two lines of each file</span></span>
<span id="cb15-6"><a href="unix.html#cb15-6" aria-hidden="true" tabindex="-1"></a><span class="co"># whose name starts with mi</span></span>
<span id="cb15-7"><a href="unix.html#cb15-7" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span> <span class="at">-n</span> 2 mi<span class="pp">*</span></span>
<span id="cb15-8"><a href="unix.html#cb15-8" aria-hidden="true" tabindex="-1"></a><span class="co"># print the last three lines of each file</span></span>
<span id="cb15-9"><a href="unix.html#cb15-9" aria-hidden="true" tabindex="-1"></a><span class="co"># whose name starts with p or s</span></span>
<span id="cb15-10"><a href="unix.html#cb15-10" aria-hidden="true" tabindex="-1"></a><span class="fu">tail</span> <span class="at">-n</span> 3 [ps]<span class="pp">*</span>.fasta</span></code></pre></div>
</div>
<div id="exercises-iii" class="section level2 hasAnchor" number="2.10">
<h2><span class="header-section-number">2.10</span> Exercises III<a href="unix.html#exercises-iii" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ol style="list-style-type: decimal">
<li>get the first two lines of all the files ending with .csv within <em>data/</em> directory, and save them to a single file called “headers.csv” under <em>sandbox/</em></li>
</ol>
</div>
<div id="search-and-select-lines-using-grep" class="section level2 hasAnchor" number="2.11">
<h2><span class="header-section-number">2.11</span> Search and select lines using <em>grep</em><a href="unix.html#search-and-select-lines-using-grep" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><code>grep &lt;pattern&gt; &lt;filename&gt;</code> will try to match all the lines in the file and output the lines with a match to the pattern specified.
This is very useful when we would like to pick out sequences of certain species or other specifications from a large sequence file.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb16-1"><a href="unix.html#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="co"># change into the directory</span></span>
<span id="cb16-2"><a href="unix.html#cb16-2" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> lab1_unix/data</span>
<span id="cb16-3"><a href="unix.html#cb16-3" aria-hidden="true" tabindex="-1"></a><span class="co"># inspect Marra2014_data.fast</span></span>
<span id="cb16-4"><a href="unix.html#cb16-4" aria-hidden="true" tabindex="-1"></a><span class="fu">less</span> Marra2014_data.fasta</span>
<span id="cb16-5"><a href="unix.html#cb16-5" aria-hidden="true" tabindex="-1"></a><span class="ex">q</span></span>
<span id="cb16-6"><a href="unix.html#cb16-6" aria-hidden="true" tabindex="-1"></a><span class="co"># count how many sequences there are for the gene isogroup00001</span></span>
<span id="cb16-7"><a href="unix.html#cb16-7" aria-hidden="true" tabindex="-1"></a><span class="fu">grep</span> isogroup00001 Marra2014_data.fasta <span class="kw">|</span> <span class="fu">wc</span> <span class="at">-l</span></span>
<span id="cb16-8"><a href="unix.html#cb16-8" aria-hidden="true" tabindex="-1"></a><span class="co"># search for gene isogroup00001 and show the first two</span></span>
<span id="cb16-9"><a href="unix.html#cb16-9" aria-hidden="true" tabindex="-1"></a><span class="fu">grep</span> isogroup00001 Marra2014_data.fasta <span class="kw">|</span> <span class="fu">head</span> <span class="at">-n</span> 2</span>
<span id="cb16-10"><a href="unix.html#cb16-10" aria-hidden="true" tabindex="-1"></a><span class="co"># alternatively, with &quot;-c&quot; option, it outputs the count of lines directly</span></span>
<span id="cb16-11"><a href="unix.html#cb16-11" aria-hidden="true" tabindex="-1"></a><span class="fu">grep</span> <span class="at">-c</span> isogroup00001 Marra2014_data.fasta</span>
<span id="cb16-12"><a href="unix.html#cb16-12" aria-hidden="true" tabindex="-1"></a><span class="co"># &quot;-w&quot; option matches whole words only, compare these two commands</span></span>
<span id="cb16-13"><a href="unix.html#cb16-13" aria-hidden="true" tabindex="-1"></a><span class="fu">grep</span> <span class="at">-c</span> <span class="at">-w</span> isogroup000 Marra2014_data.fasta</span>
<span id="cb16-14"><a href="unix.html#cb16-14" aria-hidden="true" tabindex="-1"></a><span class="fu">grep</span> <span class="at">-c</span> <span class="at">-w</span> isogroup00001 Marra2014_data.fasta</span>
<span id="cb16-15"><a href="unix.html#cb16-15" aria-hidden="true" tabindex="-1"></a><span class="co"># &quot;-i&quot; ignores case</span></span>
<span id="cb16-16"><a href="unix.html#cb16-16" aria-hidden="true" tabindex="-1"></a><span class="fu">grep</span> <span class="at">-c</span> <span class="at">-i</span> ISOgroup00001 Marra2014_data.fasta</span>
<span id="cb16-17"><a href="unix.html#cb16-17" aria-hidden="true" tabindex="-1"></a><span class="fu">grep</span> <span class="at">-c</span> ISOgroup00001 Marra2014_data.fasta</span>
<span id="cb16-18"><a href="unix.html#cb16-18" aria-hidden="true" tabindex="-1"></a><span class="co"># use wildcards within grep, we need to add &quot;\&quot; before them</span></span>
<span id="cb16-19"><a href="unix.html#cb16-19" aria-hidden="true" tabindex="-1"></a><span class="fu">grep</span> <span class="at">-c</span> isogroup0000<span class="dt">\[</span>1-9] Marra2014_data.fasta</span>
<span id="cb16-20"><a href="unix.html#cb16-20" aria-hidden="true" tabindex="-1"></a><span class="fu">grep</span> <span class="at">-c</span> isogroup<span class="dt">\*</span> Marra2014_data.fasta</span>
<span id="cb16-21"><a href="unix.html#cb16-21" aria-hidden="true" tabindex="-1"></a><span class="co"># To print all the lines that do not match a given pattern, use the option -v.</span></span>
<span id="cb16-22"><a href="unix.html#cb16-22" aria-hidden="true" tabindex="-1"></a><span class="fu">grep</span> <span class="at">-c</span> <span class="at">-v</span> isogroup<span class="dt">\*</span> Marra2014_data.fasta</span>
<span id="cb16-23"><a href="unix.html#cb16-23" aria-hidden="true" tabindex="-1"></a><span class="co"># grep can also search through many files at the same time</span></span>
<span id="cb16-24"><a href="unix.html#cb16-24" aria-hidden="true" tabindex="-1"></a><span class="fu">grep</span> <span class="at">-c</span> contig <span class="pp">*</span>.fasta</span></code></pre></div>
</div>
<div id="substituting-characters-using-tr" class="section level2 hasAnchor" number="2.12">
<h2><span class="header-section-number">2.12</span> Substituting characters using <em>tr</em><a href="unix.html#substituting-characters-using-tr" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><code>tr &lt;original characters&gt; &lt;substitution&gt;</code> will fine all the occurrences original characters and substitute with the second. If no specification, the result of substition will print out on the screen. You can redirect it to a new file.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb17-1"><a href="unix.html#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Substitute all characters a with b</span></span>
<span id="cb17-2"><a href="unix.html#cb17-2" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">&quot;aaaabbb&quot;</span> <span class="kw">|</span> <span class="fu">tr</span> <span class="st">&quot;a&quot;</span> <span class="st">&quot;b&quot;</span></span>
<span id="cb17-3"><a href="unix.html#cb17-3" aria-hidden="true" tabindex="-1"></a><span class="co">#Substitute every digit in the range 1 through 5 with 0:</span></span>
<span id="cb17-4"><a href="unix.html#cb17-4" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">&quot;123456789&quot;</span> <span class="kw">|</span> <span class="fu">tr</span> 1-5 0</span>
<span id="cb17-5"><a href="unix.html#cb17-5" aria-hidden="true" tabindex="-1"></a><span class="co"># special: translate all lower case letters to upper case</span></span>
<span id="cb17-6"><a href="unix.html#cb17-6" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">&quot;ACtGGcAaTT&quot;</span> <span class="kw">|</span> <span class="fu">tr</span> [:lower:] [:upper:]</span>
<span id="cb17-7"><a href="unix.html#cb17-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Delete all occurrences of a:</span></span>
<span id="cb17-8"><a href="unix.html#cb17-8" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">&quot;aaaaabbbb&quot;</span> <span class="kw">|</span> <span class="fu">tr</span> <span class="at">-d</span> a</span>
<span id="cb17-9"><a href="unix.html#cb17-9" aria-hidden="true" tabindex="-1"></a><span class="co"># substitute delimiters:</span></span>
<span id="cb17-10"><a href="unix.html#cb17-10" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">&quot;species1 species2&quot;</span> <span class="kw">|</span> <span class="fu">tr</span> <span class="st">&quot; &quot;</span> <span class="st">&quot;,&quot;</span></span>
<span id="cb17-11"><a href="unix.html#cb17-11" aria-hidden="true" tabindex="-1"></a><span class="co"># substitute space with tab</span></span>
<span id="cb17-12"><a href="unix.html#cb17-12" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">&quot;species1 species2&quot;</span> <span class="kw">|</span> <span class="fu">tr</span> <span class="st">&quot; &quot;</span> <span class="st">&quot;\t&quot;</span></span></code></pre></div>
</div>
<div id="cut-sort-and-unique" class="section level2 hasAnchor" number="2.13">
<h2><span class="header-section-number">2.13</span> cut, sort and unique<a href="unix.html#cut-sort-and-unique" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><code>sort</code> Sort the lines of a file and print the result to the screen. By default, it sorts the lines using alphabetical order. Use option <code>-n</code>
for numerical sorting and <code>-r</code> to reverse the order. The option <code>-k</code> is useful to sort a delimiter-separated (specified by <code>-t</code>) file by a specific column</p>
<p><code>cut</code> selects a specific column separated by a delimiter (usually ends with csv). <code>-d</code> specifies the delimiter, e.g., “,”, “;”, or ” “,”. <code>-f</code> specifies the column number or range, e.g., -f 1 or -f 2-4 selects the first or second to the fourth columns of a file.</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb18-1"><a href="unix.html#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="co"># go to the sandbox folder</span></span>
<span id="cb18-2"><a href="unix.html#cb18-2" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> lab1_unix/sandbox</span>
<span id="cb18-3"><a href="unix.html#cb18-3" aria-hidden="true" tabindex="-1"></a><span class="fu">less</span> test.csv</span>
<span id="cb18-4"><a href="unix.html#cb18-4" aria-hidden="true" tabindex="-1"></a><span class="ex">q</span></span>
<span id="cb18-5"><a href="unix.html#cb18-5" aria-hidden="true" tabindex="-1"></a><span class="co"># it&#39;s an &quot;;&quot; delimited file of characteristics of mammal species</span></span>
<span id="cb18-6"><a href="unix.html#cb18-6" aria-hidden="true" tabindex="-1"></a><span class="co"># sort the header by &quot;AdultBodyMass_g&quot;</span></span>
<span id="cb18-7"><a href="unix.html#cb18-7" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span> <span class="at">-n</span> 1 test.csv</span>
<span id="cb18-8"><a href="unix.html#cb18-8" aria-hidden="true" tabindex="-1"></a><span class="co"># AdultBodyMass_g is the 6th column, let&#39;s confirm first</span></span>
<span id="cb18-9"><a href="unix.html#cb18-9" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span> <span class="at">-n</span> 1 test.csv<span class="kw">|</span> <span class="fu">cut</span> <span class="at">-d</span> <span class="st">&quot;;&quot;</span> <span class="at">-f</span> 6</span>
<span id="cb18-10"><a href="unix.html#cb18-10" aria-hidden="true" tabindex="-1"></a><span class="co"># get species name and Adult body mass, and sort by body mass</span></span>
<span id="cb18-11"><a href="unix.html#cb18-11" aria-hidden="true" tabindex="-1"></a><span class="fu">cut</span> <span class="at">-d</span> <span class="st">&quot;;&quot;</span> <span class="at">-f</span> 5,6 test.csv<span class="kw">|</span> <span class="fu">sort</span> <span class="at">-n</span> <span class="at">-t</span> <span class="st">&quot;;&quot;</span> <span class="at">-k</span> 2</span>
<span id="cb18-12"><a href="unix.html#cb18-12" aria-hidden="true" tabindex="-1"></a><span class="co"># get unique list of genus number, remember to remove the header line</span></span>
<span id="cb18-13"><a href="unix.html#cb18-13" aria-hidden="true" tabindex="-1"></a><span class="fu">tail</span> <span class="at">-n</span> +2 test.csv <span class="kw">|</span> <span class="fu">cut</span> <span class="at">-d</span> <span class="st">&quot;;&quot;</span> <span class="at">-f</span> 4 <span class="kw">|</span> <span class="fu">sort</span> <span class="kw">|</span><span class="fu">uniq</span></span>
<span id="cb18-14"><a href="unix.html#cb18-14" aria-hidden="true" tabindex="-1"></a><span class="co"># get unique list of genus number, and count the total number</span></span>
<span id="cb18-15"><a href="unix.html#cb18-15" aria-hidden="true" tabindex="-1"></a><span class="fu">tail</span> <span class="at">-n</span> +2 test.csv <span class="kw">|</span> <span class="fu">cut</span> <span class="at">-d</span> <span class="st">&quot;;&quot;</span> <span class="at">-f</span> 4 <span class="kw">|</span> <span class="fu">sort</span> <span class="kw">|</span><span class="fu">uniq</span><span class="kw">|</span><span class="fu">wc</span> <span class="at">-l</span></span></code></pre></div>
</div>
<div id="exercises-iv" class="section level2 hasAnchor" number="2.14">
<h2><span class="header-section-number">2.14</span> Exercises IV<a href="unix.html#exercises-iv" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In this exercise we work with next generation sequencing (NGS) data. Unix is excellent at manipulating the huge FASTA files that are generated in NGS experiments. FASTA files contain sequence data in text format. Each sequence segment is preceded by a single-line description. The first character of the description line is a “greater than” sign (&gt;).</p>
<p>The NGS data set we will be working with was published by Marra and DeWoody (2014), who investigated the immunogenetic repertoire of rodents. You will find the sequence file Marra2014_data.fasta in the directory lab1_unix/data. The file contains sequence segments (contigs) of variable size. The description of each contig provides its length, the number of reads that contributed to the contig, its isogroup (representing the collection of alternative splice products of a possible gene), and the isotig status.</p>
<ol style="list-style-type: decimal">
<li>Open a shell, change working directory to lab1_unix/sandbox.</li>
<li>Create a copy of Marra2014_data.fasta in the sandbox and name it my_file.fasta.</li>
<li>How many lines are there in my_file.fasta?</li>
<li>How many sequences are there in my_file.fasta (hint: each new try of sequence starts with “&gt;contig”, when using grep remember to add “\” before “&gt;”, as “&gt;” means directing to a file without the backslash )</li>
<li>save all the lines start with “&gt;contig” to a new file.</li>
<li>How many contigs are classified as isogroup00030?</li>
<li>Replace “=” and double spaces with a comma. (hint: remove consecutive spaces, use “-s” option in <code>tr</code>)</li>
<li>How many unique isogroups are in the file?</li>
<li>Which contig has the highest number of reads (numreads)? How many
reads does it have?</li>
</ol>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="r-basics-i.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/01_unix.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["bookdown-demo.pdf", "bookdown-demo.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
